library(Seurat)
library(dplyr)
library(ggplot2)


ls1.data = Read10X('/media/chang/HDD-11/mgkeefe/220324_STICR_local_slice/transcriptome/GW18_merge/filtered_feature_bc_matrix/')
ls1 = CreateSeuratObject(counts = ls1.data, project = "GW18_0215_3S", min.cells = 10, min.features = 200)
ls1
#15115 features across 1160 samples within 1 assay 
ls1[["percent.mt"]] = PercentageFeatureSet(ls1, pattern="^MT-")
VlnPlot(ls1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls1 <- subset(ls1, subset = 
                nFeature_RNA > 50 & nFeature_RNA < 30000 &
                nFeature_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls1
#15115 features across 1097 samples within 1 assay 


ls3.data = Read10X('/media/chang/HDD-11/mgkeefe/220324_STICR_local_slice/transcriptome/GW24_merge/filtered_feature_bc_matrix/')
ls3 = CreateSeuratObject(counts = ls3.data, project = "GW24_0221_1S", min.cells = 10, min.features = 200)
ls3
#18336 features across 6843 samples within 1 assay 
ls3[["percent.mt"]] = PercentageFeatureSet(ls3, pattern="^MT-")
VlnPlot(ls3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls3 <- subset(ls3, subset = 
                nFeature_RNA > 500 & nFeature_RNA < 30000 &
                nFeature_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls3, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls3
#18336 features across 6294 samples within 1 assay 

ls11.data = Read10X('/media/chang/HDD-11/mgkeefe/220513_STICR_local_slice/transcriptome/GW21_0502_sub_FBS/filtered_feature_bc_matrix')
ls11 = CreateSeuratObject(counts = ls11.data, project = "GW21_0502_1S", min.cells = 10, min.features = 200)
ls11
#14271 features across 549 samples within 1 assay 
ls11[["percent.mt"]] = PercentageFeatureSet(ls11, pattern="^MT-")
VlnPlot(ls11, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls11 <- subset(ls11, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls11, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls11
#14271 features across 430 samples within 1 assay 


ls21.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/220929_NextSeq_transcriptomes/230206_remade_fastqs/cellranger_outs/GW23_0910_sub_FBS_S1_fixed-fastqs_cellranger/filtered_feature_bc_matrix.h5')
ls21 = CreateSeuratObject(counts = ls21.data, project = "GW23_0910_S1", min.cells = 10, min.features = 200)
ls21
#25555 features across 12136 samples within 1 assay 
ls21[["percent.mt"]] = PercentageFeatureSet(ls21, pattern="^MT-")
VlnPlot(ls21, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls21 <- subset(ls21, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls21, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls21
#25555 features across 11409 samples within 1 assay

ls22.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/220929_NextSeq_transcriptomes/230206_remade_fastqs/cellranger_outs/GW23_0910_sub_FBS_S2L2_fixed-fastqs_cellranger/filtered_feature_bc_matrix.h5')
ls22 = CreateSeuratObject(counts = ls22.data, project = "GW23_0910_S2", min.cells = 10, min.features = 200)
ls22
#21052 features across 3516 samples within 1 assay 
ls22[["percent.mt"]] = PercentageFeatureSet(ls22, pattern="^MT-")
VlnPlot(ls22, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls22 <- subset(ls22, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls22, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls22
#21052 features across 3367 samples within 1 assay 

ls23.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/220929_NextSeq_transcriptomes/230206_remade_fastqs/cellranger_outs/GW23_0910_sub_FBS_S4_fixed-fastqs_cellranger/filtered_feature_bc_matrix.h5')
ls23 = CreateSeuratObject(counts = ls23.data, project = "GW23_0910_S4", min.cells = 10, min.features = 200)
ls23
#23357 features across 5891 samples within 1 assay
ls23[["percent.mt"]] = PercentageFeatureSet(ls23, pattern="^MT-")
VlnPlot(ls23, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls23 <- subset(ls23, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls23, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls23
#23357 features across 5519 samples within 1 assay 

ls27.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/cellranger_outs/GW22_221118_sub_FBS_S1_cellranger/filtered_feature_bc_matrix.h5')
ls27 = CreateSeuratObject(counts = ls27.data, project = "GW22_1118_S1", min.cells = 10, min.features = 200)
ls27
#21080 features across 3633 samples within 1 assay
ls27[["percent.mt"]] = PercentageFeatureSet(ls27, pattern="^MT-")
VlnPlot(ls27, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls27 <- subset(ls27, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls27, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls27
#21080 features across 3418 samples within 1 assay

ls28.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/cellranger_outs/GW22_221118_sub_FBS_S2_cellranger/filtered_feature_bc_matrix.h5')
ls28 = CreateSeuratObject(counts = ls28.data, project = "GW22_1118_S2", min.cells = 10, min.features = 200)
ls28
#20776 features across 3280 samples within 1 assay
ls28[["percent.mt"]] = PercentageFeatureSet(ls28, pattern="^MT-")
VlnPlot(ls28, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls28 <- subset(ls28, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls28, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls28
#20776 features across 3142 samples within 1 assay  

ls29.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/cellranger_outs/GW22_221118_sub_FBS_S3_cellranger/filtered_feature_bc_matrix.h5')
ls29 = CreateSeuratObject(counts = ls29.data, project = "GW22_1118_S3", min.cells = 10, min.features = 200)
ls29
#19504 features across 2480 samples within 1 assay
ls29[["percent.mt"]] = PercentageFeatureSet(ls29, pattern="^MT-")
VlnPlot(ls29, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls29 <- subset(ls29, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls29, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls29
#19504 features across 2362 samples within 1 assay

ls30.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/cellranger_outs/GW16_1220_sub_FBS_S1_cellranger/filtered_feature_bc_matrix.h5')
ls30 = CreateSeuratObject(counts = ls30.data, project = "GW16_1220_S1", min.cells = 10, min.features = 200)
ls30
#24587 features across 11123 samples within 1 assay
ls30[["percent.mt"]] = PercentageFeatureSet(ls30, pattern="^MT-")
VlnPlot(ls30, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls30 <- subset(ls30, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 1500 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls30, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls30
#24587 features across 9816 samples within 1 assay

ls31.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/cellranger_outs/GW16_1220_sub_FBS_S2_cellranger/filtered_feature_bc_matrix.h5')
ls31 = CreateSeuratObject(counts = ls31.data, project = "GW16_1220_S2", min.cells = 10, min.features = 200)
ls31
#23581 features across 5577 samples within 1 assay
ls31[["percent.mt"]] = PercentageFeatureSet(ls31, pattern="^MT-")
VlnPlot(ls31, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls31 <- subset(ls31, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 3000 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls31, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls31
#23581 features across 4250 samples within 1 assay

ls32.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230303_local_STICR/cellranger_outs/GW19_0221_sub_FBS_S1_cellranger/filtered_feature_bc_matrix.h5')
ls32 = CreateSeuratObject(counts = ls32.data, project = "GW19_0221_S1", min.cells = 10, min.features = 200)
ls32
#24862 features across 9376 samples within 1 assay 
ls32[["percent.mt"]] = PercentageFeatureSet(ls32, pattern="^MT-")
VlnPlot(ls32, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls32 <- subset(ls32, subset = 
                 nFeature_RNA > 50 & nFeature_RNA < 30000 &
                 nFeature_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls32, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls32
#24862 features across 7854 samples within 1 assay  

ls33.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230303_local_STICR/cellranger_outs/GW19_0221_sub_FBS_S2_cellranger/filtered_feature_bc_matrix.h5')
ls33 = CreateSeuratObject(counts = ls33.data, project = "GW19_0221_S2", min.cells = 10, min.features = 200)
ls33
#22614 features across 4327 samples within 1 assay 
ls33[["percent.mt"]] = PercentageFeatureSet(ls33, pattern="^MT-")
VlnPlot(ls33, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls33 <- subset(ls33, subset = 
                 nFeature_RNA > 50 & nFeature_RNA < 30000 &
                 nFeature_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls33, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls33
#22614 features across 3211 samples within 1 assay 

ls34.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230317_local_STICR/transcriptome/cellranger_outs/GW20_230308_sub_FBS_V1_S1_cellranger/filtered_feature_bc_matrix.h5')
ls34 = CreateSeuratObject(counts = ls34.data, project = "GW20_0308_V1_S1", min.cells = 10, min.features = 200)
ls34
#25042 features across 14694 samples within 1 assay 
ls34[["percent.mt"]] = PercentageFeatureSet(ls34, pattern="^MT-")
VlnPlot(ls34, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls34 <- subset(ls34, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls34, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls34
#25042 features across 14276 samples within 1 assay 


ls35.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230317_local_STICR/transcriptome/cellranger_outs/GW20_230308_sub_FBS_V1_S2_cellranger/filtered_feature_bc_matrix.h5')
ls35 = CreateSeuratObject(counts = ls35.data, project = "GW20_0308_V1_S2", min.cells = 10, min.features = 200)
ls35
#25364 features across 17590 samples within 1 assay 
ls35[["percent.mt"]] = PercentageFeatureSet(ls35, pattern="^MT-")
VlnPlot(ls35, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls35 <- subset(ls35, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls35, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls35
#25364 features across 17387 samples within 1 assay 

ls36.data = Read10X_h5('/media/chang/HDD-11/mgkeefe/230317_local_STICR/transcriptome/cellranger_outs/GW20_230308_sub_FBS_PFC_2S_cellranger/filtered_feature_bc_matrix.h5')
ls36 = CreateSeuratObject(counts = ls36.data, project = "GW20_0308_PFC_2S", min.cells = 10, min.features = 200)
ls36
#25260 features across 11694 samples within 1 assay 
ls36[["percent.mt"]] = PercentageFeatureSet(ls36, pattern="^MT-")
VlnPlot(ls36, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls36 <- subset(ls36, subset = 
                 nFeature_RNA > 500 & nFeature_RNA < 30000 &
                 nCount_RNA > 800 & nCount_RNA < 30000 & percent.mt < 10)
VlnPlot(ls36, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
ls36
#25260 features across 11422 samples within 1 assay 


ls_synthesis_v3 = merge(ls1, c(ls3,
                               ls11,
                               ls21,
                               ls22,
                               ls23,
                               ls27,
                               ls28,
                               ls29,
                               ls30,
                               ls31,
                               ls32,
                               ls33,
                               ls34,
                               ls35,
                               ls36
),
project="local_STICR_synthesis_v3")

ls_synthesis_v3
#27581 features across 105254 samples within 1 assay 
table(ls_synthesis_v3@meta.data$orig.ident)
#GW16_1220_S1     GW16_1220_S2     GW18_0215_3S     GW19_0221_S1     GW19_0221_S2 GW20_0308_PFC_2S 
#        9816             4250             1097             7854             3211            11422 
#GW20_0308_V1_S1  GW20_0308_V1_S2     GW21_0502_1S     GW22_1118_S1     GW22_1118_S2     GW22_1118_S3 
#          14276            17387              430             3418             3142             2362 
#GW23_0910_S1     GW23_0910_S2     GW23_0910_S4     GW24_0221_1S 
#       11409             3367             5519             6294 

# normalize data
ls_synthesis_v3 <- NormalizeData(ls_synthesis_v3, normalization.method = "LogNormalize", scale.factor = 10000)
# scale data based on normalization
all.genes <- rownames(ls_synthesis_v3)
ls_synthesis_v3 <- ScaleData(ls_synthesis_v3, features = all.genes)
#ls_synthesis_v3 <- ScaleData(ls_synthesis_v3, features = all.genes, vars.to.regress = "nCount_RNA")
# find variably expressed genes
ls_synthesis_v3 <- FindVariableFeatures(ls_synthesis_v3, selection.method = "vst", nfeatures = 2000)
# perform dimensionality reduction
ls_synthesis_v3 <- RunPCA(ls_synthesis_v3, features = VariableFeatures(object = ls_synthesis_v3))
ElbowPlot(ls_synthesis_v3)
# cluster cells
ls_synthesis_v3 <- FindNeighbors(ls_synthesis_v3, dims = 1:15)
ls_synthesis_v3 <- FindClusters(ls_synthesis_v3, resolution = .4)
# make graph
ls_synthesis_v3 <- RunUMAP(ls_synthesis_v3, dims = 1:15)

setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal')
DimPlot(ls_synthesis_v3, reduction = "umap",label=T, raster=FALSE)
DimPlot(ls_synthesis_v3,group.by = "orig.ident", raster=FALSE)
FeaturePlot(ls_synthesis_v3, features=c("SATB2", "GAD2", "EOMES", "VIM"))
FeaturePlot(ls_synthesis_v3, features=c("TBR1", "BCL11B", "SATB2", "NEUROD2"))

FeaturePlot(ls_synthesis_v3, features=c("XIST"))
#FeaturePlot(ls_synthesis_v3, features=c("XIST"), split.by="orig.ident")
VlnPlot(ls_synthesis_v3, features=c("XIST"), group.by="orig.ident")

#add sex as metadata feature
dim(ls_synthesis_v3@meta.data)
head(ls_synthesis_v3@meta.data)
unique(ls_synthesis_v3@meta.data$orig.ident)
ls_synthesis_v3 = AddMetaData(ls_synthesis_v3, rep('XY', length(rownames(ls_synthesis_v3@meta.data))), 'sex')
head(ls_synthesis_v3@meta.data)
ls_synthesis_v3@meta.data$sex[ls_synthesis_v3@meta.data$orig.ident=="GW18_0215_3S"]="XX"
ls_synthesis_v3@meta.data$sex[ls_synthesis_v3@meta.data$orig.ident=="GW21_0502_1S"]="XX"
ls_synthesis_v3@meta.data$sex[ls_synthesis_v3@meta.data$orig.ident=="GW24_0221_1S"]="XX"

#add pseudoage as a metadata feature
ls_synthesis_v3 = AddMetaData(ls_synthesis_v3, rep('>GW20', length(rownames(ls_synthesis_v3@meta.data))), 'age_pseudo')
head(ls_synthesis_v3@meta.data)
ls_synthesis_v3@meta.data$age_pseudo[ls_synthesis_v3@meta.data$orig.ident=="GW18_0215_3S"]="<GW20"
ls_synthesis_v3@meta.data$age_pseudo[ls_synthesis_v3@meta.data$orig.ident=="GW16_1220_S1"]="<GW20"
ls_synthesis_v3@meta.data$age_pseudo[ls_synthesis_v3@meta.data$orig.ident=="GW16_1220_S2"]="<GW20"
ls_synthesis_v3@meta.data$age_pseudo[ls_synthesis_v3@meta.data$orig.ident=="GW19_0221_S1"]="<GW20"
ls_synthesis_v3@meta.data$age_pseudo[ls_synthesis_v3@meta.data$orig.ident=="GW19_0221_S2"]="<GW20"

DimPlot(ls_synthesis_v3,group.by = "sex", raster=F)
DimPlot(ls_synthesis_v3,group.by = "age_pseudo", raster=F)
table(ls_synthesis_v3@meta.data$age_pseudo)
#<GW20 >GW20 
#26228 79026  

## run harmony on orig.ident
ls_synthesis_v3_harmony_v2 <- RunHarmony(ls_synthesis_v3, "orig.ident", kmeans_init_iter_max=100, kmeans_init_nstart=50, plot_convergence = TRUE, project.dim = F)
ls_synthesis_v3_harmony_v2 <- FindNeighbors(ls_synthesis_v3_harmony_v2, reduction = "harmony", dims = 1:15)
ls_synthesis_v3_harmony_v2 <- FindClusters(ls_synthesis_v3_harmony_v2, resolution = .4)
ls_synthesis_v3_harmony_v2 <- RunUMAP(ls_synthesis_v3_harmony_v2, reduction = "harmony", dims=c(1:15))
DimPlot(ls_synthesis_v3_harmony_v2, reduction = "umap",label=T, raster=F)
## inverting the UMAP embeddings so it looks more like ls_synthesis_v3_harmony_v1
ls_synthesis_v3_harmony_v2[["umap"]]@cell.embeddings = -ls_synthesis_v3_harmony_v2[["umap"]]@cell.embeddings
DimPlot(ls_synthesis_v3_harmony_v2, reduction = "umap",label=T, raster=F)
DimPlot(ls_synthesis_v3_harmony_v2,group.by = "orig.ident", raster=F)
DimPlot(ls_synthesis_v3_harmony_v2,group.by = "sex", raster=F)
DimPlot(ls_synthesis_v3_harmony_v2,group.by = "age_pseudo", raster=F)

unique(ls_synthesis_v3_harmony_v2@meta.data$orig.ident)
#[1] "GW18_0215_3S"     "GW24_0221_1S"     "GW21_0502_1S"     "GW23_0910_S1"     "GW23_0910_S2"    
#[6] "GW23_0910_S4"     "GW22_1118_S1"     "GW22_1118_S2"     "GW22_1118_S3"     "GW16_1220_S1"    
#[11] "GW16_1220_S2"     "GW19_0221_S1"     "GW19_0221_S2"     "GW20_0308_V1_S1"  "GW20_0308_V1_S2" 
#[16] "GW20_0308_PFC_2S"

FeaturePlot(ls_synthesis_v3_harmony_v2, features=c("percent.mt"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v2, features=c("nFeature_RNA"), raster=F)

FeaturePlot(ls_synthesis_v3_harmony_v2, features=c("SATB2", "OLIG2", "GAD2", "MKI67"), raster=F)

#should cluster 10 be removed?
VlnPlot(ls_synthesis_v3_harmony_v2, features = c("nCount_RNA", "percent.mt"), ncol = 2, pt.size = 0)
table(ls_synthesis_v3_harmony_v2@meta.data$seurat_clusters)
#0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16 
#16134 15102 14824 11283  8814  6760  6594  4726  4382  4239  3514  2922  2530  1168   921   888   453 

## remove it and cluster 15 based on nCountRNA
subset(ls_synthesis_v3_harmony_v2, seurat_clusters %in% c(0,1,2,3,4,5,6,7,8,9,11,12,13,14,16))
saveRDS(ls_synthesis_v3_harmony_v2, '230402_ls_synthesis_v3_harmony_v2_obj.rds')
ls_synthesis_v3_harmony_v3= subset(ls_synthesis_v3_harmony_v2, seurat_clusters %in% c(0,1,2,3,4,5,6,7,8,9,11,12,13,14,16))
ls_synthesis_v3_harmony_v3

#rerun clustering after removing those cells
ls_synthesis_v3_harmony_v3 <- FindNeighbors(ls_synthesis_v3_harmony_v3, reduction = "harmony", dims = 1:15)
ls_synthesis_v3_harmony_v3 <- FindClusters(ls_synthesis_v3_harmony_v3, resolution = .4)
ls_synthesis_v3_harmony_v3 <- RunUMAP(ls_synthesis_v3_harmony_v3, reduction = "harmony", dims=c(1:15))
DimPlot(ls_synthesis_v3_harmony_v3, reduction = "umap",label=T, raster=F)
#need to invert just UMAP_1 to restore same positioning
ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings[,1] = -ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings[,1]
DimPlot(ls_synthesis_v3_harmony_v3, reduction = "umap",label=T, raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features = c("nCount_RNA", "percent.mt"), ncol = 2, pt.size = 0)
#now have to go in and delete cells from cluster 15... getting tiring
table(ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters)
#    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15 
#16449 14403 13755 10345  8120  6783  6618  6026  5558  4119  3103  2793  1120   921   458   281
ls_synthesis_v3_harmony_v3= subset(ls_synthesis_v3_harmony_v3, seurat_clusters %in% c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,14))
ls_synthesis_v3_harmony_v3 <- FindNeighbors(ls_synthesis_v3_harmony_v3, reduction = "harmony", dims = 1:15)
ls_synthesis_v3_harmony_v3 <- FindClusters(ls_synthesis_v3_harmony_v3, resolution = .4)
ls_synthesis_v3_harmony_v3 <- RunUMAP(ls_synthesis_v3_harmony_v3, reduction = "harmony", dims=c(1:15))
DimPlot(ls_synthesis_v3_harmony_v3, reduction = "umap",label=T, raster=F)
#need to invert just UMAP_2 to restore same positioning
ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings[,2] = -ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings[,2]
DimPlot(ls_synthesis_v3_harmony_v3, reduction = "umap",label=T, raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features = c("nCount_RNA", "percent.mt"), ncol = 2, pt.size = 0)

# find cluster markers
ls_synthesis_v3_harmony_v3.markers <- FindAllMarkers(ls_synthesis_v3_harmony_v3, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_synthesis_v3_harmony_v3.markers)
ls_synthesis_v3_harmony_v3_clustermarkers_unfilt = as.data.frame(ls_synthesis_v3_harmony_v3.markers %>% group_by(cluster))
write.csv(ls_synthesis_v3_harmony_v3_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v3_clustermarkers_v1_unfilt.csv',row.names=F)
ls_synthesis_v3_harmony_v3.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_synthesis_v3_harmony_v3_clustermarkers = as.data.frame(ls_synthesis_v3_harmony_v3.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_synthesis_v3_harmony_v3_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v3_clustermarkers_v1.csv',row.names=F)


#name clusters and assign metadata for easier future parsing
clust_idents = c("EN_0","EN_early","EN_2","EN_UL","OPC","DIV","RG","IPC","IN_local","IN_CGE","IN_MGE","EN_11","OLIG","RG_unk")
full_clust_list = ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters
full_clust_idents = rep("EN_DL", length(full_clust_list))
for (s in c(0:13)) {
  full_clust_idents[full_clust_list==s] = clust_idents[s+1]
}
full_clust_idents = factor(full_clust_idents, levels=c("EN_0","EN_early","EN_2","EN_UL","OPC","DIV","RG","IPC","IN_local","IN_CGE","IN_MGE","EN_11","OLIG","RG_unk"))
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, full_clust_idents, col.name = "clust_idents")
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="clust_idents")
DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)

## fix pseudoage to include GW20 as "<=GW20"
unique(ls_synthesis_v3_harmony_v3@meta.data$orig.ident)
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo = rep('>GW20', length(rownames(ls_synthesis_v3_harmony_v3@meta.data)))
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW18_0215_3S"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW16_1220_S1"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW16_1220_S2"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW19_0221_S1"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW19_0221_S2"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW20_0308_V1_S1"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW20_0308_V1_S2"]="<=GW20"
ls_synthesis_v3_harmony_v3@meta.data$age_pseudo[ls_synthesis_v3_harmony_v3@meta.data$orig.ident=="GW20_0308_PFC_2S"]="<=GW20"
table(ls_synthesis_v3_harmony_v3$age_pseudo)
#<=GW20  >GW20 
#65917  34654 

## add in STICR barcodes (now with read count data)
head(ls_synthesis_v3_harmony_v3@meta.data)
unique(ls_synthesis_v3_harmony_v3@meta.data$orig.ident)
seurat_meta = ls_synthesis_v3_harmony_v3@meta.data
seurat_meta$CBC = rownames(seurat_meta)
dim(seurat_meta)
#[1] 100571     10

#9/10 S1
sample_loadpath = '/media/chang/HDD-11/mgkeefe/221016_local_STICR_analysis/STICR_barcodes/220910_GW23/GW23_0910_sub_FBS_S1_scAR_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW23_0910_S1"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = sample_bcs
#9/10 S2
sample_loadpath = '/media/chang/HDD-11/mgkeefe/221016_local_STICR_analysis/STICR_barcodes/220910_GW23/GW23_0910_sub_FBS_S2_scAR_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW23_0910_S2"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#9/10 S4
sample_loadpath = '/media/chang/HDD-11/mgkeefe/221016_local_STICR_analysis/STICR_barcodes/220910_GW23/GW23_0910_sub_FBS_S4_scAR_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW23_0910_S4"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#5/02 sub FBS 
sample_loadpath = '/media/chang/HDD-11/mgkeefe/221016_local_STICR_analysis/STICR_barcodes/220502_GW21/GW21_0502_sub_FBS_scAR_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW21_0502_1S"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#2/21 GW24 merge
sample_loadpath = '/media/chang/HDD-11/mgkeefe/221016_local_STICR_analysis/STICR_barcodes/220221_GW24/GW24_0324_sub_FBS_S1_merge_scAR_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW24_0221_1S"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#12/20 GW16 S1
#note -- have to change the code here because there are two digits in the CBC suffix!
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/230103_STICR_BCs/STICR_outs_GW16_1220/GW16_221220_S1_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW16_1220_S1"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#12/20 GW16 S2
#note -- have to change the code here because there are two digits in the CBC suffix!
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/230103_STICR_BCs/STICR_outs_GW16_1220/GW16_221220_S2_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW16_1220_S2"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#11/18 GW22 S1
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/230103_STICR_BCs/STICR_outs_GW22_221118/GW22_221118_S1_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW22_1118_S1"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#11/18 GW22 S2
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/230103_STICR_BCs/STICR_outs_GW22_221118/GW22_221118_S2_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW22_1118_S2"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#11/18 GW22 S3
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230106_MKTN09_local_STICR/230103_STICR_BCs/STICR_outs_GW22_221118/GW22_221118_S3_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW22_1118_S3"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-3,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#2/21/23 GW19 S1
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230303_local_STICR/bc_outs/GW19_0221_S1/GW19_0221_S1_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW19_0221_S1"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#2/21/23 GW19 S2
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230303_local_STICR/bc_outs/GW19_0221_S2/GW19_0221_S2_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW19_0221_S2"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#GW20_0308_PFC_2S
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230317_local_STICR/barcodes/bc_outs_PFC_2S/GW20_0308_PFC_2S_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW20_0308_PFC_2S"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#GW20_0308_V1_S1
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230317_local_STICR/barcodes/bc_outs_V1_S1/GW20_0308_V1_S1_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW20_0308_V1_S1"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)
#GW20_0308_V1_S2
sample_loadpath = '/media/chang/HDD-11/mgkeefe/230317_local_STICR/barcodes/bc_outs_V1_S2/GW20_0308_V1_S2_bcs_final_metadata_UMIs_reads.csv'
sample_id = "GW20_0308_V1_S2"
sample_bcs = read.csv(sample_loadpath, row.names = 1)
head(sample_bcs)
char_len=nchar(seurat_meta$CBC[seurat_meta$orig.ident==sample_id])[1]
cbc_suffix=substr(seurat_meta$CBC[seurat_meta==sample_id][1],char_len-4,char_len)
sample_bcs$CBC = paste(sample_bcs$CBC, cbc_suffix, sep="")
head(sample_bcs)
master_sticr_umi_df = rbind(master_sticr_umi_df, sample_bcs)

seurat_meta_sticr_umi = left_join(seurat_meta, master_sticr_umi_df, by='CBC')
dim(seurat_meta_sticr_umi)
dim(ls_synthesis_v3_harmony_v3@meta.data)
rownames(seurat_meta_sticr_umi) = seurat_meta_sticr_umi$CBC
sum(is.na(seurat_meta_sticr_umi$Clone_barcodes))
table(seurat_meta_sticr_umi$orig.ident[is.na(seurat_meta_sticr_umi$Clone_barcodes)]) #check distribution of missing BCs
#GW16_1220_S1     GW16_1220_S2     GW18_0215_3S     GW19_0221_S1     GW19_0221_S2 GW20_0308_PFC_2S 
#.        325              102              955              891              547              771 
#GW20_0308_V1_S1  GW20_0308_V1_S2    GW21_0502_1S     GW22_1118_S1     GW22_1118_S2     GW22_1118_S3 
#.           676             830               63               99               86               90 
#GW23_0910_S1     GW23_0910_S2     GW23_0910_S4     GW24_0221_1S 
#        314              193              384              684 
colnames(seurat_meta_sticr_umi)
#[1] "orig.ident"      "nCount_RNA"      "nFeature_RNA"    "percent.mt"      "RNA_snn_res.0.4"
#[6] "seurat_clusters" "sex"             "age_pseudo"      "RNA_snn_res.0.6" "RNA_snn_res.0.5"
#[11] "CBC"             "Clone_barcodes"  "n_barcodes"      "Clone_size"      "Clone_IDs"      
#[16] "Clone_Index"     "UMI_Count"       "CBC_barcode"     "read_counts"     "thresh_pass" 
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$Clone_barcodes, col.name='Clone_barcodes')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$Clone_size, col.name='Clone_size')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$Clone_IDs, col.name='Clone_IDs')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$Clone_Index, col.name='Clone_Index')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$UMI_Count, col.name='Barcode_UMI_Count')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         seurat_meta_sticr_umi$read_counts, col.name='read_counts')

#fix read_thresh_pass to make it Boolean:
read_thresh_bool = rep(FALSE, length(seurat_meta_sticr_umi$thresh_pass))
read_thresh_bool[seurat_meta_sticr_umi$thresh_pass == "True"] = TRUE
sum(read_thresh_bool)
#1] 72658
length(read_thresh_bool)
#[1] 100571
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 
                                         read_thresh_bool, col.name='read_thresh_pass')

table(ls_synthesis_v3_harmony_v3$Barcode_UMI_Count)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18 
#27359  6228  4367  3785  3526  3387  3114  2902  2581  2366  2281  2102  2083  1896  1732  1549  1530  1380
hist(log10(ls_synthesis_v3_harmony_v3$Barcode_UMI_Count))
hist(log10(ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]))


colnames(ls_synthesis_v3_harmony_v3@meta.data)


ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='Clone_barcodes_filt')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='Clone_size_filt')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='Clone_IDs_filt')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='Clone_Index_filt')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='Barcode_UMI_Count_filt')
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, rep(NA, dim(ls_synthesis_v3_harmony_v3@meta.data)[1]), col.name='read_counts_filt')


ls_synthesis_v3_harmony_v3@meta.data$Clone_barcodes_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$Clone_barcodes[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]
ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$Clone_size[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]
ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]
ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$Clone_Index[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]
ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]
ls_synthesis_v3_harmony_v3@meta.data$read_counts_filt[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass] = ls_synthesis_v3_harmony_v3@meta.data$read_counts[ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass]

sum(ls_synthesis_v3_harmony_v3@meta.data$read_thresh_pass)
length(ls_synthesis_v3_harmony_v3@meta.data$Clone_Index)
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_Index))
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt))

table(ls_synthesis_v3_harmony_v3@meta.data$Clone_Index)
#Index3 IndexE 
#44995  48566
table(ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt)
#Index3 IndexE 
#34915  37743 

sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt))/length(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt)
#[1] 0.2775452
sum(!is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt))/length(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt)
#[1] 0.7224548

osvz_col = "#a45bfb" #OSVZ (Index 3) is a light purple
vz_col = "#00621e" #VZ (Index E) is a dark green
    
DimPlot(ls_synthesis_v3_harmony_v3, group.by="Clone_Index_filt", raster=FALSE)
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))

DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='IndexE'], cols.highlight=vz_col) +
  ggtitle('Index E') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='Index3'], cols.highlight=osvz_col) +
  ggtitle('Index 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, group.by = "Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=1) +
  ggtitle('STICR-labeled cells') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, split.by = "Clone_Index_filt")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt==1], cols.highlight='black') +
  ggtitle('Clone Size == 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>1], cols.highlight='#2dba3b') +
  ggtitle('Clone Size > 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>3], cols.highlight='#1d8027') +
  ggtitle('Clone Size > 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>5], cols.highlight='#0f4515') +
  ggtitle('Clone Size > 5') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>5], cols.highlight='#0f4515', split.by="Clone_Index_filt") +
  ggtitle('Large clones (>5) by index') +
  theme(plot.title = element_text(hjust = 0.5))

table(ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count[!is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_barcodes_filt)])
table(ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count_filt[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters == 10])


#clonal analysis
multicell_clones = unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>=3])
multicell_clones = multicell_clones[!is.na(multicell_clones)]
multicell_metadata = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>=3,]
multicell_metadata = multicell_metadata[!is.na(multicell_metadata$Clone_IDs),]
head(multicell_metadata)
length(multicell_clones)
#[1] 6841
dim(multicell_metadata)
#[1] 19059    23
multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],]
dim(multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],])[1]
multicell_metadata$Clone_size_corrected = 0
colnames(multicell_metadata)
for (mc in multicell_clones) {
  Clone_size_corrected = dim(multicell_metadata[multicell_metadata$Clone_IDs %in% mc,])[1]
  multicell_metadata[multicell_metadata$Clone_IDs_filt %in% mc,25] = Clone_size_corrected
}
head(multicell_metadata)
table(multicell_metadata$Clone_size)
#   3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   22   24 
#6618 3729 2498 1565 1182  891  583  432  223  344  200  161   80  103  132   35   46   29   17   59   44 
#26   27   29   30   45 
# 3   14   25   28   18 
table(multicell_metadata$Clone_size_corrected)
#   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21 
#2137 2892 4863 2864 1840 1332  756  480  450  260  308  264   78   70   75   80  119   36   19   40   21 
#24   25   26 
#24   25   26 

#re-filter to just include clones that are more legitimately size >=3
multicell_metadata = multicell_metadata[multicell_metadata$Clone_size_corrected>=3,]
dim(multicell_metadata)
#[1] 14030    24
length(unique(multicell_metadata$Clone_IDs))
#[1] 3258
multicell_clones = unique(multicell_metadata$Clone_IDs)

unique(multicell_metadata$clust_idents)
en_clusts = c("EN_0", "EN_early", "EN_2", "EN_UL", "EN_11", "IPC")
in_clusts = c("IN_local", "IN_CGE", "IN_MGE")
other_clusts = c("RG", "OPC", "RG_unk", "OLIG", "DIV")

clone_list = rep(multicell_clones, 3)
clust_ident_list = c(rep('EN', length(multicell_clones)),
                     rep('IN', length(multicell_clones)),
                     rep('OTHER', length(multicell_clones))
)
compiled_clust_df = data.frame(multicell_clones)
compiled_clust_df$EN = rep(0,length(multicell_clones))
compiled_clust_df$IN = rep(0,length(multicell_clones))
compiled_clust_df$OTHER = rep(0,length(multicell_clones))
compiled_clust_df$EN_pct = rep(0,length(multicell_clones))
compiled_clust_df$IN_pct = rep(0,length(multicell_clones))
compiled_clust_df$OTHER_pct = rep(0,length(multicell_clones))
for(clone in multicell_clones) {
  clone_size = multicell_metadata$Clone_size_corrected[multicell_metadata$Clone_IDs==clone][1]
  clone_clusts = table(multicell_metadata$clust_idents[multicell_metadata$Clone_IDs==clone])
  clone_row = which(compiled_clust_df$multicell_clones==clone)
  compiled_clust_df[clone_row,2] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% en_clusts])
  compiled_clust_df[clone_row,3] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% in_clusts])
  compiled_clust_df[clone_row,4] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% other_clusts])
  compiled_clust_df[clone_row,5] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% en_clusts])/clone_size
  compiled_clust_df[clone_row,6] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% in_clusts])/clone_size
  compiled_clust_df[clone_row,7] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% other_clusts])/clone_size
}
compiled_clust_df
clust_number_list = c(compiled_clust_df$EN, compiled_clust_df$IN, compiled_clust_df$OTHER)
clust_pct_list = c(compiled_clust_df$EN_pct, compiled_clust_df$IN_pct, compiled_clust_df$OTHER_pct)
clone_cluster_pivot = data.frame(clone_list, clust_ident_list, clust_number_list, clust_pct_list)

ordered_clone_ids_en_pct = compiled_clust_df[order(compiled_clust_df$EN_pct-compiled_clust_df$IN_pct),1]
ggplot(clone_cluster_pivot, aes(fill=clust_ident_list, x=clone_list, y=clust_number_list)) +
  scale_x_discrete(limits = ordered_clone_ids_en_pct) +
  geom_bar(position='fill', stat='identity', width=1) +
  scale_fill_manual(values = c('#e6394b', '#191b85', '#e6be39', '#444545')) +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
  ggtitle("All samples, all indices")

#do for individual samples:
multicell_metadata_unique = multicell_metadata[!duplicated(multicell_metadata$Clone_IDs),]
head(clone_cluster_pivot)
colnames(clone_cluster_pivot) = c("Clone_IDs","Identity","Cell_count", "Cell_percentage")
head(multicell_metadata_unique)
colnames(multicell_metadata_unique)
#[1] "orig.ident"             "nCount_RNA"             "nFeature_RNA"           "percent.mt"            
#[5] "RNA_snn_res.0.4"        "seurat_clusters"        "sex"                    "age_pseudo"            
#[9] "RNA_snn_res.0.6"        "RNA_snn_res.0.5"        "Clone_barcodes"         "Clone_size"            
#[13] "Clone_IDs"              "Clone_Index"            "Barcode_UMI_Count"      "read_counts"           
#[17] "read_thresh_pass"       "Clone_barcodes_filt"    "Clone_size_filt"        "Clone_IDs_filt"        
#[21] "Clone_Index_filt"       "Barcode_UMI_Count_filt" "read_counts_filt"       "clust_idents"          
#[25] "Clone_size_corrected" 

#add: orig.ident, Clone_IDs, Clone_Index, clust_idents, Barcode_UMI_count, Clone_size_corrected
head(left_join(clone_cluster_pivot,multicell_metadata_unique[,c(1,13,14,24,15,25)]))
clone_cluster_pivot_idents = left_join(clone_cluster_pivot,multicell_metadata_unique[,c(1,12,13,14,20)])
table(clone_cluster_pivot_idents$orig.ident)
#GW16_1220_S1     GW16_1220_S2     GW19_0221_S1     GW19_0221_S2 GW20_0308_PFC_2S  GW20_0308_V1_S1 
#         840              267              243               21             1182             1758 
#GW20_0308_V1_S2     GW21_0502_1S     GW22_1118_S1     GW22_1118_S2     GW22_1118_S3     GW23_0910_S1 
#           2196                3              390              429              264              795 
#GW23_0910_S2     GW23_0910_S4     GW24_0221_1S 
#          33              690              663 

unique(clone_cluster_pivot_idents$orig.ident)
#[1] "GW24_0221_1S"     "GW21_0502_1S"     "GW23_0910_S1"     "GW23_0910_S2"     "GW23_0910_S4"    
#[6] "GW22_1118_S1"     "GW22_1118_S2"     "GW22_1118_S3"     "GW16_1220_S1"     "GW16_1220_S2"    
#[11] "GW19_0221_S1"     "GW19_0221_S2"     "GW20_0308_V1_S1"  "GW20_0308_V1_S2"  "GW20_0308_PFC_2S"
dusty_red = "#e6394b"
gold = "#e6be39"
navy = "#191b85"
barchart_tricolor_palette = c(dusty_red, navy, gold)
clonal_comp_stacked_barchart_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EN"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="OTHER"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_tricolor_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EN"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="OTHER"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_tricolor_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func("GW16_1220_S1", "GW16_1220_S1")
clonal_comp_stacked_barchart_func("GW16_1220_S2", "GW16_1220_S2")
clonal_comp_stacked_barchart_func("GW19_0221_S1", "GW19_0221_S1")
clonal_comp_stacked_barchart_func("GW19_0221_S2", "GW19_0221_S2")
clonal_comp_stacked_barchart_func("GW21_0502_1S", "GW21_0502_1S")
clonal_comp_stacked_barchart_func("GW22_1118_S1", "GW22_1118_S1")
clonal_comp_stacked_barchart_func("GW22_1118_S2", "GW22_1118_S2")
clonal_comp_stacked_barchart_func("GW22_1118_S3", "GW22_1118_S3")
clonal_comp_stacked_barchart_func("GW23_0910_S1", "GW23_0910_S1")
clonal_comp_stacked_barchart_func("GW23_0910_S2", "GW23_0910_S2")
clonal_comp_stacked_barchart_func("GW23_0910_S4", "GW23_0910_S4")
clonal_comp_stacked_barchart_func("GW24_0221_1S", "GW24_0221_1S")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S1", "GW20_0308_V1_S1")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S2", "GW20_0308_V1_S2")
clonal_comp_stacked_barchart_func("GW20_0308_PFC_2S", "GW20_0308_PFC_2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1"), "GW16_1220_S1-S2")
clonal_comp_stacked_barchart_func(c("GW19_0221_S1", "GW19_0221_S2"), "GW19_0221_S1-S2")
clonal_comp_stacked_barchart_func(c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4"), "GW23_0910_S1-S2-S4")
clonal_comp_stacked_barchart_func(c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3"), "GW22_1118_S1-S2-S3")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2"), "GW20_0308_V1_S1-S2")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "GW20_0308_V1_S1-S2_PFC-2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                    "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_func(c('GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'
), "All samples >GW20")
clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")


head(clone_cluster_pivot_idents)
head(compiled_clust_df)
clone_cluster_pivot_idents = clone_cluster_pivot_idents[clone_cluster_pivot_idents$Clone_Index!="Index1",]

ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, colour=Clone_Index)) +
  geom_boxplot()+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Clonal composition by index')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, colour=Clone_Index)) +
  geom_violin()+
  geom_point(position=position_jitterdodge(), alpha=0.2)+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Clonal composition by index')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))


ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="EN",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('EN pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="IN",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('IN pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="OTHER",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('OTHER pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))

compiled_clust_df_idents = compiled_clust_df
colnames(compiled_clust_df) = c("Clone_IDs", "EN", "IN", "OTHER", "EN_pct", "IN_pct", "OTHER_pct")
colnames(multicell_metadata_unique[,c(13,14,24,15,25)])
head(left_join(compiled_clust_df, multicell_metadata_unique[,c(13,14,24,15,25)]))
compiled_clust_df_idents = left_join(compiled_clust_df, multicell_metadata_unique[,c(13,14,24,15,25)])
dim(compiled_clust_df_idents)
compiled_clust_df_idents$point_col = rep(osvz_col, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_col[compiled_clust_df_idents$Clone_Index == "IndexE"] = vz_col
compiled_clust_df_idents$point_pch = rep(2, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_pch[compiled_clust_df_idents$Clone_Index == "IndexE"] = 6
table(compiled_clust_df_idents$Clone_Index)
#Index3 IndexE 
#1278   1980 
table(compiled_clust_df_idents$point_col)
table(compiled_clust_df_idents$point_pch)
compiled_clust_df_idents = compiled_clust_df_idents[compiled_clust_df_idents$Clone_Index!='Index1',]
head(compiled_clust_df_idents)
##add sample age
colnames(multicell_metadata_unique[,c(1,8,13)])
#[1] "orig.ident" "age_pseudo" "Clone_IDs" 
compiled_clust_df_idents = left_join(compiled_clust_df_idents, multicell_metadata_unique[,c(1,8,13)])
all_sample_ids
#[1] "GW24_0221_1S"     "GW21_0502_1S"     "GW23_0910_S1"     "GW23_0910_S2"     "GW23_0910_S4"    
#[6] "GW22_1118_S1"     "GW22_1118_S2"     "GW22_1118_S3"     "GW16_1220_S1"     "GW16_1220_S2"    
#[11] "GW19_0221_S1"     "GW19_0221_S2"     "GW20_0308_V1_S1"  "GW20_0308_V1_S2"  "GW20_0308_PFC_2S"
compiled_clust_df_idents$sample_age = rep(20, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW16_1220_S2', 'GW16_1220_S1')] = 16
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW19_0221_S1', 'GW19_0221_S2')] = 19
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW21_0502_1S')] = 21
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 22
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 23
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW24_0221_1S')] = 24
## create jitter for nicer looking plots
compiled_clust_df_idents_jitter = compiled_clust_df_idents
jitter_amt = seq(-0.005, 0.005, by=0.001)
for(r in c(1:length(rownames(compiled_clust_df_idents_jitter)))){
  compiled_clust_df_idents_jitter[r,5] = compiled_clust_df_idents_jitter[r,5]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,6] = compiled_clust_df_idents_jitter[r,6]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,7] = compiled_clust_df_idents_jitter[r,7]+sample(jitter_amt, 1)
}

head(compiled_clust_df_idents_jitter)

## making some clean plots with proper colors and themes and removing GW21 (only 3 clones)
table(compiled_clust_df_idents_jitter$sample_age)
compiled_clust_df_idents_jitter = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age!=21,]

ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clone index')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EN_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=OTHER_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EN_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=OTHER_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=EN_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of EN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=IN_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=OTHER_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

## generate ternary plots for each sample
library(Ternary)
tri_ternary_plot = function(samp_data, samp_age) {
  par(mfrow = c(1, 3), mar = rep(0.2, 4))
  TernaryPlot(alab = "pct ENs", blab = "pct INs", clab = "pct Progenitors")
  TernaryPoints(samp_data[samp_data$Clone_Index=="IndexE", c("EN_pct", "IN_pct", "OTHER_pct")], 
                pch=samp_data$point_pch[samp_data$Clone_Index=="IndexE"],
                col=samp_data$point_col[samp_data$Clone_Index=="IndexE"])
  legend("topleft", 
         legend = c("Index E", "Index 3"),
         cex = 1.2, bty = "n", pch = c(6,2), pt.cex = 2,
         col = c(vz_col, osvz_col),
         title = "Clone Index"
  )
  #title(paste("GW",as.character(samp_age)), cex.main = 1.5)
  
  TernaryPlot(alab = "pct ENs", blab = "pct INs", clab = "pct Progenitors")
  TernaryPoints(samp_data[samp_data$Clone_Index=="Index3", c("EN_pct", "IN_pct", "OTHER_pct")], 
                pch=samp_data$point_pch[samp_data$Clone_Index=="Index3"],
                col=samp_data$point_col[samp_data$Clone_Index=="Index3"])
  #legend("topleft", 
  #       legend = c("Index E", "Index 3"),
  #       cex = 1.2, bty = "n", pch = c(6,2), pt.cex = 2,
  #       col = c(vz_col, osvz_col),
  #       title = "Clone Index"
  #)
  title(paste("GW",as.character(samp_age),"clone composition"), cex.main = 1.5)
  
  TernaryPlot(alab = "pct ENs", blab = "pct INs", clab = "pct Progenitors")
  TernaryPoints(samp_data[, c("EN_pct", "IN_pct", "OTHER_pct")], 
                pch=samp_data$point_pch,
                col=samp_data$point_col)
  #legend("topleft", 
  #       legend = c("Index E", "Index 3"),
  #       cex = 1.2, bty = "n", pch = c(6,2), pt.cex = 2,
  #       col = c(vz_col, osvz_col),
  #       title = "Clone Index"
  #)
  #title(paste("GW",as.character(samp_age)), cex.main = 1.5)
}

samp_age = 16
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = 19
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = 20
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = 22
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = 23
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = 24
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = c(16,19,20)
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = c(22,23,24)
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)

samp_age = c(16,18,19,20,22,23,24)
samp_data = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age %in% c(samp_age),]
tri_ternary_plot(samp_data, samp_age)


head(compiled_clust_df_idents)
compiled_clust_df_idents$clonal_restriction = rep("EN_only", dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN == 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$OTHER == 0] = "IN_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$OTHER != 0] = "OTHER_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$OTHER == 0] = "EN_IN_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN != 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$OTHER != 0] = "EN_OTHER_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN == 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$OTHER != 0] = "IN_OTHER_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EN != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$OTHER != 0] = "tripotent"
table(compiled_clust_df_idents$clonal_restriction)

compiled_clust_df_idents$clonal_restriction = factor(compiled_clust_df_idents$clonal_restriction, levels=(
  c("EN_only", "IN_only", "OTHER_only", "EN_IN_shared", "EN_OTHER_shared", "IN_OTHER_shared", "tripotent")
))
table(compiled_clust_df_idents$clonal_restriction)
#        EN_only         IN_only      OTHER_only    EN_IN_shared EN_OTHER_shared IN_OTHER_shared 
#            691             122             930              43             728             616 
#      tripotent 
#            128 

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=16
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
samp_age=19
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


## apply clone_size_corrected to the whole metadata object
seurat_meta = ls_synthesis_v3_harmony_v3@meta.data
head(seurat_meta)
head(multicell_metadata)
colnames(multicell_metadata)

seurat_meta$Clone_size_corrected = NA
for (mc in multicell_clones) {
  seurat_meta$Clone_size_corrected[seurat_meta$Clone_IDs_filt %in% mc] = multicell_metadata$Clone_size_corrected[multicell_metadata$Clone_IDs %in% mc][1]
  #Clone_size_corrected = dim(multicell_metadata[multicell_metadata$Clone_IDs %in% mc,])[1]
  #multicell_metadata[multicell_metadata$Clone_IDs %in% mc,20] = Clone_size_corrected
}
seurat_meta[seurat_meta$Clone_IDs %in% mc, ]
seurat_meta[seurat_meta$Clone_IDs %in% "LS36_Clone_30", ]


table(seurat_meta$Clone_size_corrected)
sum(is.na(seurat_meta$Clone_size_corrected))
sum(!is.na(seurat_meta$Clone_size_corrected))


ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt, col.name="Clone_size_corrected")
ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected[!is.na(seurat_meta$Clone_size_corrected)] = seurat_meta$Clone_size_corrected[!is.na(seurat_meta$Clone_size_corrected)]

DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='IndexE'], cols.highlight=vz_col) +
  ggtitle('Index E') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='Index3'], cols.highlight=osvz_col) +
  ggtitle('Index 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, group.by = "Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=1) +
  ggtitle('STICR-labeled cells') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, split.by = "Clone_Index_filt")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected==1], cols.highlight='black') +
  ggtitle('Clone Size == 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1], cols.highlight='#2dba3b') +
  ggtitle('Clone Size > 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3], cols.highlight='#1d8027') +
  ggtitle('Clone Size >= 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>5], cols.highlight='#0f4515') +
  ggtitle('Clone Size > 5') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>5], cols.highlight='#0f4550', split.by="Clone_Index_filt") +
  ggtitle('Large clones (>5) by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3], cols.highlight='#1d8027', split.by="age_pseudo") +
  ggtitle('Clone Size >= 3 by age') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% ">GW20"],
        cols.highlight='#0f6040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 3 in >GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% "<=GW20"],
        cols.highlight='#0f6040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 3 in <=GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% ">GW20"],
        cols.highlight='#0f3040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 2 in >GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))

## try to start subclustering to get more specific cell identities
ls_div = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% "DIV")
ls_div
#27581 features across 7240 samples within 1 assay 
ls_div <- ScaleData(ls_div, features = all.genes)
ls_div = FindVariableFeatures(ls_div, selection.method="vst", nfeatures=2000)
ls_div = RunPCA(ls_div, features=VariableFeatures(object=ls_div))
ElbowPlot(ls_div)
ls_div <- RunHarmony(ls_div, c("orig.ident"))
ls_div <- FindNeighbors(ls_div, reduction = "harmony", dims = 1:15)
ls_div <- FindClusters(ls_div, resolution = .5)
ls_div <- RunUMAP(ls_div, reduction = "harmony", dims=c(1:15))
DimPlot(ls_div, reduction = "umap",label=T)
DimPlot(ls_div, reduction = "umap",group.by="orig.ident")
DimPlot(ls_div, reduction = "umap",group.by="age_pseudo")
FeaturePlot(ls_div, features=c("VIM", "HOPX", "EOMES", "DLX2", "OLIG2", "NEUROD2", "ID3", "PPP1R17", "MKI67"))
VlnPlot(ls_div, features = c("nCount_RNA", "percent.mt"), ncol = 2)
VlnPlot(ls_div, features = c("DLX2", "EOMES", "VIM", "OLIG2"), ncol=4)
#subcluster assignments:
# 0, 6, 9: IPC_EX_DIV
# 1, 2, 8: GLIA_DIV
# 7: OPC_DIV
# 3, 4, 5: IPC_IN_DIV
ls_div$subclust_idents = NA
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(0,6,9)] = "IPC_EX_DIV"
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(1,2,8)] = "GLIA_DIV"
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(7)] = "OPC_DIV"
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(3,4,5)] = "IPC_IN_DIV"
table(ls_div$subclust_idents)
#GLIA_DIV IPC_EX_DIV IPC_IN_DIV    OPC_DIV
#    2303       2060       2334        543 
ls_div = SetIdent(ls_div, value="subclust_idents")
DimPlot(ls_div, label=T)
DimPlot(ls_div, group.by="Clone_Index_filt", pt.size=1.5)


ls_glia = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% c("RG", "OPC", "OLIG", "RG_unk"))
ls_glia
#27581 features across 15778 samples within 1 assay
ls_glia <- ScaleData(ls_glia, features = all.genes)
ls_glia = FindVariableFeatures(ls_glia, selection.method="vst", nfeatures=2000)
ls_glia = RunPCA(ls_glia, features=VariableFeatures(object=ls_glia))
ElbowPlot(ls_glia)
ls_glia <- RunHarmony(ls_glia, c("orig.ident"))
ls_glia <- FindNeighbors(ls_glia, reduction = "harmony", dims = 1:15)
ls_glia <- FindClusters(ls_glia, resolution = .5)
ls_glia <- RunUMAP(ls_glia, reduction = "harmony", dims=c(1:15))
DimPlot(ls_glia, reduction = "umap",label=T, pt.size=1)
VlnPlot(ls_glia, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_glia, reduction = "umap",group.by="orig.ident", pt.size=1)
DimPlot(ls_glia, reduction = "umap",group.by="age_pseudo", pt.size=1)
FeaturePlot(ls_glia, label=T, features=c("VIM", "HOPX", "ID3", "DLX5", "OLIG2", "MBP", "EOMES", "PPP1R17", "AQP4"))
FeaturePlot(ls_glia, label=T, features=c("VIM", "NEUROD2", "EOMES", "TBR1"))
FeaturePlot(ls_glia, label=T, features=c("nCount_RNA"))
VlnPlot(ls_glia, features = c("VIM", "PPP1R17", "OLIG2", "DLX5"), ncol=4)
#subcluster assignments
#0, 1, 2, 5, 7, 11, 12: GLIA
#3, 9: OPC
#8: OLIG
#6: IPC_IN
#10: IPC_EX
#4: RG_EX ##possibly doublets?
#13: NA
ls_glia$subclust_idents = NA
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(0,1,2,5,7,11,12)] = "GLIA"
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(3,9)] = "OPC"
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(8)] = "OLIG"
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(6)] = "IPC_IN"
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(10)] = "IPC_EX"
ls_glia$subclust_idents[ls_glia$seurat_clusters %in% c(4)] = "RG_EX"
table(ls_glia$subclust_idents)
#GLIA IPC_EX IPC_IN   OLIG    OPC  RG_EX 
#9344    709   1192    755   2249   1446 
ls_glia = SetIdent(ls_glia, value="subclust_idents")
DimPlot(ls_glia, label=T)


table(ls_synthesis_v3_harmony_v3$clust_idents)
ls_in = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% c("IN_CGE", "IN_MGE", "IN_local"))
ls_in
#27581 features across 12025 samples within 1 assay
ls_in <- ScaleData(ls_in, features = all.genes)
ls_in = FindVariableFeatures(ls_in, selection.method="vst", nfeatures=2000)
ls_in = RunPCA(ls_in, features=VariableFeatures(object=ls_in))
ElbowPlot(ls_in)
ls_in <- RunHarmony(ls_in, c("orig.ident"))
ls_in <- FindNeighbors(ls_in, reduction = "harmony", dims = 1:15)
ls_in <- FindClusters(ls_in, resolution = .5)
ls_in <- RunUMAP(ls_in, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in, reduction = "umap",label=T, pt.size=1)
VlnPlot(ls_in, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_in, reduction = "umap",group.by="orig.ident", pt.size=1)
DimPlot(ls_in, reduction = "umap",group.by="age_pseudo", pt.size=1)
FeaturePlot(ls_in, label=T, features=c("GAD2", "NEUROD2", "NR2F1", "PAX6", "MKI67", "NPY", "NR2F2", "ADARB2", "MAF"))
FeaturePlot(ls_in, label=T, features=c("nCount_RNA"))
DimPlot(ls_in, label=T, cells.highlight=rownames(ls_in@meta.data[ls_in@meta.data$"Clone_size_filt">=3,])) +ggtitle("Clone size >=3")
#subcluster assignments
# 0, 8: IN_CGE
# 2, 6: IN_MGE
# 7: IN_unk ##possibly doublets?
# 1, 3, 4, 5, 9: IN_local
# 10: IPC_IN_DIV
ls_in$subclust_idents = NA
ls_in$subclust_idents[ls_in$seurat_clusters %in% c(1, 3, 4, 5, 9)] = "IN_local"
ls_in$subclust_idents[ls_in$seurat_clusters %in% c(0, 8)] = "IN_CGE"
ls_in$subclust_idents[ls_in$seurat_clusters %in% c(2, 6)] = "IN_MGE"
ls_in$subclust_idents[ls_in$seurat_clusters %in% c(7)] = "IN_UNK"
ls_in$subclust_idents[ls_in$seurat_clusters %in% c(10)] = "IPC_IN_DIV"
ls_in = SetIdent(ls_in, value="subclust_idents")
DimPlot(ls_in, label=T)

table(ls_synthesis_v3_harmony_v3$clust_idents)
ls_ex = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% c("EN_0", "EN_early", "EN_UL", "IPC", "EN_11", "EN_2"))
ls_ex
#27581 features across 65528 samples within 1 assay 
ls_ex <- ScaleData(ls_ex, features = all.genes)
ls_ex = FindVariableFeatures(ls_ex, selection.method="vst", nfeatures=2000)
ls_ex = RunPCA(ls_ex, features=VariableFeatures(object=ls_ex))
ElbowPlot(ls_ex)
ls_ex <- RunHarmony(ls_ex, c("orig.ident"))
ls_ex <- FindNeighbors(ls_ex, reduction = "harmony", dims = 1:15)
ls_ex <- FindClusters(ls_ex, resolution = .5)
ls_ex <- RunUMAP(ls_ex, reduction = "harmony", dims=c(1:15))
DimPlot(ls_ex, reduction = "umap",label=T)
VlnPlot(ls_ex, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_ex, reduction = "umap",group.by="orig.ident", pt.size=0.5)
DimPlot(ls_ex, reduction = "umap",group.by="age_pseudo", pt.size=0.5)
FeaturePlot(ls_ex, label=T, features=c("EOMES", "NEUROD2", "MARCH1", "TBR1", "BCL11B", "SATB2", "CUX2", "CUX1", "TLE4"))
FeaturePlot(ls_ex, label=T, features=c("TLE4", "NR4A2", "TBR1", "OTX1", "BCL11B", "RORB", "SATB2", "POU3F2", "CUX2"))
FeaturePlot(ls_ex, label=T, features=c("VIM", "HOPX", "NEUROD1", "NEUROD2"))
DimPlot(ls_ex, reduction = "umap", group.by="Clone_Index_filt", pt.size=0.5)
DimPlot(ls_ex, reduction = "umap", split.by="Clone_Index_filt", pt.size=0.5)
#subcluster assignments
#0, 1, 2, 3, 4, 7, 9: EX
#5, 10: EX_early
#6, 8: IPC_EX
ls_ex$subclust_idents = NA
ls_ex$subclust_idents[ls_ex$seurat_clusters %in% c(0, 1, 2, 3, 4, 7, 9)] = "EX"
ls_ex$subclust_idents[ls_ex$seurat_clusters %in% c(5, 10)] = "EX_early"
ls_ex$subclust_idents[ls_ex$seurat_clusters %in% c(6, 8)] = "IPC_EX"
ls_ex = SetIdent(ls_ex, value="subclust_idents")
DimPlot(ls_ex, label=T)

ls_ex_unmerge = FindNeighbors(ls_ex, dims=1:15)
ls_ex_unmerge = FindClusters(ls_ex_unmerge, resolution=.5)
ls_ex_unmerge = RunUMAP(ls_ex_unmerge, dims=c(1:15))
DimPlot(ls_ex_unmerge, label=T)
DimPlot(ls_ex_unmerge, group.by="orig.ident")
DimPlot(ls_ex_unmerge, group.by="age_pseudo")
DimPlot(ls_ex_unmerge, group.by="sex")
DimPlot(ls_ex_unmerge, cells.highlight = rownames(ls_ex_unmerge@meta.data[ls_ex_unmerge@meta.data$Clone_size_corrected>=3,]))
FeaturePlot(ls_ex_unmerge, label=T, features=c("EOMES", "NEUROD2", "MARCH1", "TBR1", "BCL11B", "SATB2", "CUX2", "CUX1", "TLE4"))


## compile all subclust_idents and add to the main object
all_subclust_idents = c(ls_div$subclust_idents, ls_in$subclust_idents, ls_glia$subclust_idents, ls_ex$subclust_idents)
all_subclust_idents = data.frame(all_subclust_idents)
all_subclust_idents$CBC = rownames(all_subclust_idents)
colnames(all_subclust_idents) = c("subclust_idents", "CBC")
head(all_subclust_idents)
dim(all_subclust_idents)
dim(seurat_meta)
colnames(seurat_meta)
seurat_meta$CBC = rownames(seurat_meta)
head(left_join(seurat_meta, all_subclust_idents))
seurat_meta = left_join(seurat_meta, all_subclust_idents)
sum(is.na(seurat_meta$subclust_idents))
#[1] 83

ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, seurat_meta$subclust_idents, col.name = "subclust_idents")
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
#   EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#49657       7235       9344       2303       2860       5445       2621        828       9345 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#      2060       1192       2605        755       2249        543       1446

## idea for new binning of broad cell types:
## EX: EX, EX_early, IPC_EX, IPC_EX_DIV, RG_EX(???)
## IN: IN_CGE, IN_local, IN_MGE, IPC_IN, IPC_IN_DIV
## GLIA: GLIA, GLIA_DIV
## OLIG: OPC, OPC_DIV, OLIG
## excluding IN_UNK for now because those seem highly likely to be doublets
table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% "IN_UNK"])
#  1   2   3   4   5   6   7   8   9  10  12  13  16  17  18 
#422 122  64  20  17  12  16   3   7   4   4   1   2   5   1 
FeaturePlot(ls_in, label=T, features=c("GAD2", "SATB2", "ADARB2", "MAF"))

ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, NA, col.name = "broad_celltype")
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("EX", "EX_early", "IPC_EX", "IPC_EX_DIV", "RG_EX")
  ] = "EX"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("IN_CGE", "IN_local", "IN_MGE", "IPC_IN", "IPC_IN_DIV")
] = "IN"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("GLIA", "GLIA_DIV")
] = "GLIA"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("OPC", "OPC_DIV", "OLIG")
] = "OLIG"
table(ls_synthesis_v3_harmony_v3$broad_celltype)
#   EX  GLIA    IN  OLIG 
#69743 11647 14723  3547
DimPlot(ls_synthesis_v3_harmony_v3, group.by="broad_celltype", raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents", raster=F)


## re-doing clonal analysis with new cell type calls from subclustering
multicell_clones = unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3])
multicell_clones = multicell_clones[!is.na(multicell_clones)]
multicell_metadata = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3,]
multicell_metadata = multicell_metadata[!is.na(multicell_metadata$Clone_IDs),]
head(multicell_metadata)
length(multicell_clones)
#[1] 6841
dim(multicell_metadata)
#[1] 19059    27
table(multicell_metadata$Clone_size_corrected)
#   3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   24   25 
#8397 3757 2143 1456  819  517  473  280  316  277   80   71   80   80  119   37   19   40   21   24   25 
#26   30 
#26    2 
table(multicell_metadata$Clone_size_filt)
#   3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   22   24 
#6618 3729 2498 1565 1182  891  583  432  223  344  200  161   80  103  132   35   46   29   17   59   44 
#26   27   29   30   45 
# 3   14   25   28   18 

#take a quick look at some example clones?
multicell_metadata[multicell_metadata$Clone_size_corrected == 24, c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count",  "clust_idents", "subclust_idents", "broad_celltype")]
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS36_Clone_0",])) + ggtitle("LS36_Clone_0")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS36_Clone_1",])) + ggtitle("LS36_Clone_1")
multicell_metadata[multicell_metadata$Clone_IDs == "LS36_Clone_1", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count",  "clust_idents", "subclust_idents", "broad_celltype")]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% "LS36_Clone_1", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count", "read_counts", "clust_idents", "subclust_idents", "broad_celltype")]

#what's the deal with the clone size of 30 but only in 2 cells...?
multicell_metadata[multicell_metadata$Clone_size_corrected == 30, c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count",  "clust_idents", "subclust_idents", "broad_celltype")]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% "LS31_Clone_0", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count", "read_counts", "clust_idents", "subclust_idents", "broad_celltype")]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% "LS31_Clone_1", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count", "read_counts", "clust_idents", "subclust_idents", "broad_celltype")]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% "LS31_Clone_2", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count", "read_counts", "clust_idents", "subclust_idents", "broad_celltype")]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% "LS31_Clone_60", c("orig.ident", "Clone_IDs", "Clone_Index", "Clone_size", "Clone_size_filt", "Clone_size_corrected", "Barcode_UMI_Count", "read_counts", "clust_idents", "subclust_idents", "broad_celltype")]
## it seems like there's an overall issue, where the following happened:
## scAR found a lot of cells that were part of a clone, but many of those cells ended up being filtered out because they weren't real or they got removed from downstream processing
## in the filtering to multicell clones only, they were not included because there weren't enough of them
## clone_size_corrected only applied to those clones that actually had at least three cells, so when clone_size_corrected copied over clone_size_filt, it wasn't updated
## in general, should pause now, re-do clone_size_corrected for the entire dataframe, and then restart

for (sbc in ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs[ls_synthesis_v3_harmony_v3@meta.data$Clone_size!=1]) {
  size_corr = dim(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% sbc,])[1]
  ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% sbc] = rep(size_corr, size_corr)
}
ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected[is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt)] = NA
table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18 
#47851 21619  9386  4707  2971  1833  1353  1023   650   469   247   368   226   178    92   106   138    36 
#19    20    21    22    24    26    27    29    30    45 
#49    37    18    63    46     4    14    27    31    19
table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected)
#     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18 
# 49274  9354  4863  2864  1840  1332   756   480   450   260   308   264    78    70    75    80   119    36 
#19    20    21    24    25    26 
#19    40    21    24    25    26 

multicell_clones = unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3])
multicell_clones = multicell_clones[!is.na(multicell_clones)]
multicell_metadata = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3,]
multicell_metadata = multicell_metadata[!is.na(multicell_metadata$Clone_IDs),]
head(multicell_metadata)
length(multicell_clones)
#[1] 3258
dim(multicell_metadata)
#[1] 14030    27
multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],]
dim(multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],])[1]

clone_list = rep(multicell_clones, 4)
clust_ident_list = c(rep('EX', length(multicell_clones)),
                     rep('IN', length(multicell_clones)),
                     rep('GLIA', length(multicell_clones)),
                     rep('OLIG', length(multicell_clones))
)
compiled_clust_df = data.frame(multicell_clones)
compiled_clust_df$EX = rep(0,length(multicell_clones))
compiled_clust_df$IN = rep(0,length(multicell_clones))
compiled_clust_df$GLIA = rep(0,length(multicell_clones))
compiled_clust_df$OLIG = rep(0,length(multicell_clones))
compiled_clust_df$EX_pct = rep(0,length(multicell_clones))
compiled_clust_df$IN_pct = rep(0,length(multicell_clones))
compiled_clust_df$GLIA_pct = rep(0,length(multicell_clones))
compiled_clust_df$OLIG_pct = rep(0,length(multicell_clones))

for(clone in multicell_clones) {
  clone_size = multicell_metadata$Clone_size_corrected[multicell_metadata$Clone_IDs %in% clone][1]
  clone_clusts = table(multicell_metadata$broad_celltype[multicell_metadata$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df$multicell_clones %in% clone)
  compiled_clust_df[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df[clone_row,"GLIA"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])
  compiled_clust_df[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df[clone_row,"GLIA_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])/clone_size
  compiled_clust_df[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
}
compiled_clust_df
clust_number_list = c(compiled_clust_df$EX, compiled_clust_df$IN, compiled_clust_df$GLIA, compiled_clust_df$OLIG)
clust_pct_list = c(compiled_clust_df$EX_pct, compiled_clust_df$IN_pct, compiled_clust_df$GLIA_pct, compiled_clust_df$OLIG_pct)
clone_cluster_pivot = data.frame(clone_list, clust_ident_list, clust_number_list, clust_pct_list)

ordered_clone_ids_ex_pct = compiled_clust_df[order(compiled_clust_df$EX_pct-compiled_clust_df$IN_pct),1]
ggplot(clone_cluster_pivot, aes(fill=clust_ident_list, x=clone_list, y=clust_number_list)) +
  scale_x_discrete(limits = ordered_clone_ids_ex_pct) +
  geom_bar(position='fill', stat='identity', width=1) +
  scale_fill_manual(values = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')) +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
  ggtitle("All samples, all indices")

#do for individual samples:
multicell_metadata_unique = multicell_metadata[!duplicated(multicell_metadata$Clone_IDs),]
head(clone_cluster_pivot)
colnames(clone_cluster_pivot) = c("Clone_IDs","Identity","Cell_count", "Cell_percentage")
head(multicell_metadata_unique)
colnames(multicell_metadata_unique)
#add: orig.ident, Clone_Index, clust_idents, Barcode_UMI_count
head(left_join(clone_cluster_pivot,multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_count")]))
clone_cluster_pivot_idents = left_join(clone_cluster_pivot,multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_count")])
table(clone_cluster_pivot_idents$orig.ident)
#GW16_1220_S1     GW16_1220_S2     GW19_0221_S1     GW19_0221_S2 GW20_0308_PFC_2S  GW20_0308_V1_S1 
#         882              435              270               33             1137             1749 
#GW20_0308_V1_S2     GW21_0502_1S     GW22_1118_S1     GW22_1118_S2     GW22_1118_S3     GW23_0910_S1 
#           2070                3              405              432              252              843 
#GW23_0910_S2     GW23_0910_S4     GW24_0221_1S 
#          30              675              675
head(multicell_metadata_unique)
colnames(multicell_metadata_unique)


unique(clone_cluster_pivot_idents$orig.ident)
#[1] "GW24_0221_1S"     "GW21_0502_1S"     "GW23_0910_S1"     "GW23_0910_S2"     "GW23_0910_S4"    
#[6] "GW22_1118_S1"     "GW22_1118_S2"     "GW22_1118_S3"     "GW16_1220_S1"     "GW16_1220_S2"    
#[11] "GW19_0221_S1"     "GW19_0221_S2"     "GW20_0308_V1_S1"  "GW20_0308_V1_S2"  "GW20_0308_PFC_2S"

barchart_color_palette = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')
clonal_comp_stacked_barchart_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func("GW16_1220_S1", "GW16_1220_S1")
clonal_comp_stacked_barchart_func("GW16_1220_S2", "GW16_1220_S2")
clonal_comp_stacked_barchart_func("GW19_0221_S1", "GW19_0221_S1")
clonal_comp_stacked_barchart_func("GW19_0221_S2", "GW19_0221_S2")
clonal_comp_stacked_barchart_func("GW21_0502_1S", "GW21_0502_1S")
clonal_comp_stacked_barchart_func("GW22_1118_S1", "GW22_1118_S1")
clonal_comp_stacked_barchart_func("GW22_1118_S2", "GW22_1118_S2")
clonal_comp_stacked_barchart_func("GW22_1118_S3", "GW22_1118_S3")
clonal_comp_stacked_barchart_func("GW23_0910_S1", "GW23_0910_S1")
clonal_comp_stacked_barchart_func("GW23_0910_S2", "GW23_0910_S2")
clonal_comp_stacked_barchart_func("GW23_0910_S4", "GW23_0910_S4")
clonal_comp_stacked_barchart_func("GW24_0221_1S", "GW24_0221_1S")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S1", "GW20_0308_V1_S1")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S2", "GW20_0308_V1_S2")
clonal_comp_stacked_barchart_func("GW20_0308_PFC_2S", "GW20_0308_PFC_2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1"), "GW16_1220_S1-S2")
clonal_comp_stacked_barchart_func(c("GW19_0221_S1", "GW19_0221_S2"), "GW19_0221_S1-S2")
clonal_comp_stacked_barchart_func(c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4"), "GW23_0910_S1-S2-S4")
clonal_comp_stacked_barchart_func(c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3"), "GW22_1118_S1-S2-S3")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2"), "GW20_0308_V1_S1-S2")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "GW20_0308_V1_S1-S2_PFC-2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                    "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_func(c('GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'
), "All samples >GW20")
clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")


head(clone_cluster_pivot_idents)
head(compiled_clust_df)
clone_cluster_pivot_idents = clone_cluster_pivot_idents[clone_cluster_pivot_idents$Clone_Index!="Index1",]

ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="EX",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('EX pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="IN",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('IN pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="GLIA",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('GLIA pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))

colnames(compiled_clust_df) = c("Clone_IDs", "EX", "IN", "GLIA", "OLIG", "EX_pct", "IN_pct", "GLIA_pct", "OLIG_pct")
compiled_clust_df_idents = left_join(compiled_clust_df, multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_Count")])
dim(compiled_clust_df_idents)
compiled_clust_df_idents$point_col = rep(osvz_col, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_col[compiled_clust_df_idents$Clone_Index == "IndexE"] = vz_col
compiled_clust_df_idents$point_pch = rep(2, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_pch[compiled_clust_df_idents$Clone_Index == "IndexE"] = 6
table(compiled_clust_df_idents$Clone_Index)
table(compiled_clust_df_idents$point_col)
table(compiled_clust_df_idents$point_pch)
head(compiled_clust_df_idents)
compiled_clust_df_idents$sample_age = rep(20, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW16_1220_S2', 'GW16_1220_S1')] = 16
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW19_0221_S1', 'GW19_0221_S2')] = 19
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW21_0502_1S')] = 21
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 22
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 23
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW24_0221_1S')] = 24
## create jitter for nicer looking plots
compiled_clust_df_idents_jitter = compiled_clust_df_idents
jitter_amt = seq(-0.005, 0.005, by=0.001)
for(r in c(1:length(rownames(compiled_clust_df_idents_jitter)))){
  compiled_clust_df_idents_jitter[r,"EX_pct"] = compiled_clust_df_idents_jitter[r,"EX_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"IN_pct"] = compiled_clust_df_idents_jitter[r,"IN_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"GLIA_pct"] = compiled_clust_df_idents_jitter[r,"GLIA_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"OLIG_pct"] = compiled_clust_df_idents_jitter[r,"OLIG_pct"]+sample(jitter_amt, 1)
}
head(compiled_clust_df_idents_jitter)
## making some clean plots with proper colors and themes and removing GW21 (only 3 clones)
table(compiled_clust_df_idents_jitter$sample_age)
#16   19   20   21   22   23   24 
#369   88 1712    1  361  506  221 
compiled_clust_df_idents_jitter = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age!=21,]

ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clone index')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EX_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=GLIA_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=OLIG_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of OPCs/oligos in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EX_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=GLIA_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=EX_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=IN_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=GLIA_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=OLIG_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of OPCs/oligos in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')


apply(data.frame(compiled_clust_df_idents[1:5,c("GLIA","OLIG")]==0), 1, sum) ==0

colnames(compiled_clust_df_idents)
head(compiled_clust_df_idents)
compiled_clust_df_idents$clonal_restriction = rep("EX_only", dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "IN_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG == 0] = "GLIA_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG != 0] = "OLIG_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "EX_IN_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN == 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG != 0] = "quadpotent"
table(compiled_clust_df_idents$clonal_restriction)
#EX_GLIA/OLIG_shared EX_IN_GLIA/OLIG_shared           EX_IN_shared                EX_only 
#                564                    186                    188                   1156 
#GLIA_only    IN_GLIA/OLIG_shared                IN_only              OLIG_only 
#      172                    445                    411                     54 
#quadpotent 
#       82 

compiled_clust_df_idents$clonal_restriction = factor(compiled_clust_df_idents$clonal_restriction, levels=(
  c("EX_only", "IN_only", "GLIA_only", "OLIG_only", "EX_IN_shared", "EX_GLIA/OLIG_shared", "IN_GLIA/OLIG_shared", "EX_IN_GLIA/OLIG_shared", "quadpotent")
))
table(compiled_clust_df_idents$clonal_restriction)
#EX_only                IN_only              GLIA_only              OLIG_only 
#1156                    411                    172                     54 
#EX_IN_shared    EX_GLIA/OLIG_shared    IN_GLIA/OLIG_shared EX_IN_GLIA/OLIG_shared 
#188                    564                    445                    186 
#quadpotent 
#82 

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=16
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=19
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_age))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

compiled_clust_df_idents$ex_in_presence = rep("EX_only", dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0] = "IN_only"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0] = "no_EX_or_IN"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0] = "EX_IN_shared"
table(compiled_clust_df_idents$ex_in_presence)
compiled_clust_df_idents$ex_in_presence = factor(compiled_clust_df_idents$ex_in_presence, levels=(
  c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN")
))
table(compiled_clust_df_idents$ex_in_presence)
#     EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#        1583          856          456          363 
ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


#plot INs by age, but remove IN only clones
compiled_clust_df_idents = compiled_clust_df_idents[compiled_clust_df_idents$sample_age!=21,]
in_only_clones = compiled_clust_df_idents[apply(compiled_clust_df_idents[,c("EX", "GLIA", "OLIG")], 1, sum) == 0 & compiled_clust_df_idents_jitter$IN != 0, "Clone_IDs"]
in_only_clones
dim(compiled_clust_df_idents[!(compiled_clust_df_idents$Clone_IDs %in% in_only_clones),])
#[1] 2846   17

ggplot(compiled_clust_df_idents[!(compiled_clust_df_idents$Clone_IDs %in% in_only_clones),],
       aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age (IN_only clusters removed)') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')


#what is the percentage of clones from a given VZ that are EX only, IN only, etc?
samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=Clone_Index, fill=ex_in_presence)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=Clone_Index, fill=ex_in_presence)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


#is there a bias in direct vs. indirect neurogenesis from VZ/OSVZ?
dim(multicell_metadata)
table(multicell_metadata$subclust_idents)
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       710       1058       1859        978         12       1655         16        121       2949 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       778        607       1447         96       1010        287        437 

table(multicell_metadata$subclust_idents[multicell_metadata$Clone_Index %in% "IndexE"])
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       510        836        954        524          4        830          5         78       2493 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       605        274        663         47        465        131        274
table(multicell_metadata$subclust_idents[multicell_metadata$Clone_Index %in% "Index3"])
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       200        222        905        454          8        825         11         43        456 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       173        333        784         49        545        156        163 


ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')


ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents)) +
  geom_bar(position = "dodge") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

## NOTE: this is the code for how to get stacked barcharts to show proportions!!!!!!!!
## after_stat(prop)!!!!!!
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c("<=GW20", ">GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c("<=GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c(">GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

#revisit RG_EX...
DimPlot(ls_glia, label=T)
FeaturePlot(ls_glia, label=T, features=c("VIM", "HOPX", "NEUROD1", "NEUROD2", "MARCH1", "DLX5", "PTPRZ1", "FAM107A", "MKI67"))
FeaturePlot(ls_glia, features=c("NEUROD2"), split.by="Clone_Index", label=T)
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% "RG_EX",]))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% "IPC_IN_DIV",]))

FeaturePlot(ls_synthesis_v3_harmony_v3, raster=F, features=c("NEUROD2", "VIM"))
FeaturePlot(ls_glia, label=T, features=c("VIM", "HOPX", "DCX", "NEUROD2"))


## start of clonal analysis that includes RG_EX as GLIA
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% "RG_EX"] = "GLIA"
table(ls_synthesis_v3_harmony_v3@meta.data$broad_celltype)
#   EX  GLIA    IN  OLIG 
#68297 13093 14723  3547
multicell_clones = unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3])
multicell_clones = multicell_clones[!is.na(multicell_clones)]
multicell_metadata = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3,]
multicell_metadata = multicell_metadata[!is.na(multicell_metadata$Clone_IDs),]
head(multicell_metadata)
length(multicell_clones)
#[1] 3258
dim(multicell_metadata)
#[1] 14030    27
multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],]
dim(multicell_metadata[multicell_metadata$Clone_IDs %in% multicell_clones[2],])[1]

clone_list = rep(multicell_clones, 4)
clust_ident_list = c(rep('EX', length(multicell_clones)),
                     rep('IN', length(multicell_clones)),
                     rep('GLIA', length(multicell_clones)),
                     rep('OLIG', length(multicell_clones))
)
compiled_clust_df = data.frame(multicell_clones)
compiled_clust_df$EX = rep(0,length(multicell_clones))
compiled_clust_df$IN = rep(0,length(multicell_clones))
compiled_clust_df$GLIA = rep(0,length(multicell_clones))
compiled_clust_df$OLIG = rep(0,length(multicell_clones))
compiled_clust_df$EX_pct = rep(0,length(multicell_clones))
compiled_clust_df$IN_pct = rep(0,length(multicell_clones))
compiled_clust_df$GLIA_pct = rep(0,length(multicell_clones))
compiled_clust_df$OLIG_pct = rep(0,length(multicell_clones))

for(clone in multicell_clones) {
  clone_size = multicell_metadata$Clone_size_corrected[multicell_metadata$Clone_IDs %in% clone][1]
  clone_clusts = table(multicell_metadata$broad_celltype[multicell_metadata$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df$multicell_clones %in% clone)
  compiled_clust_df[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df[clone_row,"GLIA"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])
  compiled_clust_df[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df[clone_row,"GLIA_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])/clone_size
  compiled_clust_df[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
}
compiled_clust_df
clust_number_list = c(compiled_clust_df$EX, compiled_clust_df$IN, compiled_clust_df$GLIA, compiled_clust_df$OLIG)
clust_pct_list = c(compiled_clust_df$EX_pct, compiled_clust_df$IN_pct, compiled_clust_df$GLIA_pct, compiled_clust_df$OLIG_pct)
clone_cluster_pivot = data.frame(clone_list, clust_ident_list, clust_number_list, clust_pct_list)

ordered_clone_ids_ex_pct = compiled_clust_df[order(compiled_clust_df$EX_pct-compiled_clust_df$IN_pct),1]
ggplot(clone_cluster_pivot, aes(fill=clust_ident_list, x=clone_list, y=clust_number_list)) +
  scale_x_discrete(limits = ordered_clone_ids_ex_pct) +
  geom_bar(position='fill', stat='identity', width=1) +
  scale_fill_manual(values = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')) +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
  ggtitle("All samples, all indices")

#do for individual samples:
multicell_metadata_unique = multicell_metadata[!duplicated(multicell_metadata$Clone_IDs),]
head(clone_cluster_pivot)
colnames(clone_cluster_pivot) = c("Clone_IDs","Identity","Cell_count", "Cell_percentage")
head(multicell_metadata_unique)
colnames(multicell_metadata_unique)
#add: orig.ident, Clone_Index, clust_idents, Barcode_UMI_count
head(left_join(clone_cluster_pivot,multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_Count")]))
clone_cluster_pivot_idents = left_join(clone_cluster_pivot,multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_Count")])
table(clone_cluster_pivot_idents$orig.ident)
#GW16_1220_S1     GW16_1220_S2     GW19_0221_S1     GW19_0221_S2 GW20_0308_PFC_2S  GW20_0308_V1_S1 
#1120              356              324               28             1576             2344 
#GW20_0308_V1_S2     GW21_0502_1S     GW22_1118_S1     GW22_1118_S2     GW22_1118_S3     GW23_0910_S1 
#2928                4              520              572              352             1060 
#GW23_0910_S2     GW23_0910_S4     GW24_0221_1S 
#44              920              884 
head(multicell_metadata_unique)
colnames(multicell_metadata_unique)

unique(clone_cluster_pivot_idents$orig.ident)
#[1] "GW24_0221_1S"     "GW21_0502_1S"     "GW23_0910_S1"     "GW23_0910_S2"     "GW23_0910_S4"    
#[6] "GW22_1118_S1"     "GW22_1118_S2"     "GW22_1118_S3"     "GW16_1220_S1"     "GW16_1220_S2"    
#[11] "GW19_0221_S1"     "GW19_0221_S2"     "GW20_0308_V1_S1"  "GW20_0308_V1_S2"  "GW20_0308_PFC_2S"

barchart_color_palette = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')
clonal_comp_stacked_barchart_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func("GW16_1220_S1", "GW16_1220_S1")
clonal_comp_stacked_barchart_func("GW16_1220_S2", "GW16_1220_S2")
clonal_comp_stacked_barchart_func("GW19_0221_S1", "GW19_0221_S1")
clonal_comp_stacked_barchart_func("GW19_0221_S2", "GW19_0221_S2")
clonal_comp_stacked_barchart_func("GW21_0502_1S", "GW21_0502_1S")
clonal_comp_stacked_barchart_func("GW22_1118_S1", "GW22_1118_S1")
clonal_comp_stacked_barchart_func("GW22_1118_S2", "GW22_1118_S2")
clonal_comp_stacked_barchart_func("GW22_1118_S3", "GW22_1118_S3")
clonal_comp_stacked_barchart_func("GW23_0910_S1", "GW23_0910_S1")
clonal_comp_stacked_barchart_func("GW23_0910_S2", "GW23_0910_S2")
clonal_comp_stacked_barchart_func("GW23_0910_S4", "GW23_0910_S4")
clonal_comp_stacked_barchart_func("GW24_0221_1S", "GW24_0221_1S")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S1", "GW20_0308_V1_S1")
clonal_comp_stacked_barchart_func("GW20_0308_V1_S2", "GW20_0308_V1_S2")
clonal_comp_stacked_barchart_func("GW20_0308_PFC_2S", "GW20_0308_PFC_2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1"), "GW16_1220_S1-S2")
clonal_comp_stacked_barchart_func(c("GW19_0221_S1", "GW19_0221_S2"), "GW19_0221_S1-S2")
clonal_comp_stacked_barchart_func(c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4"), "GW23_0910_S1-S2-S4")
clonal_comp_stacked_barchart_func(c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3"), "GW22_1118_S1-S2-S3")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2"), "GW20_0308_V1_S1-S2")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "GW20_0308_V1_S1-S2_PFC-2S")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                    "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_func(c('GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'
), "All samples >GW20")
clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")


head(clone_cluster_pivot_idents)
head(compiled_clust_df)
clone_cluster_pivot_idents = clone_cluster_pivot_idents[clone_cluster_pivot_idents$Clone_Index!="Index1",]

ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="EX",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('EX pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="IN",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('IN pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(clone_cluster_pivot_idents[clone_cluster_pivot_idents$Identity=="GLIA",], aes(x=Cell_percentage, colour=Clone_Index)) +
  geom_freqpoly() +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('GLIA pct')+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5))

colnames(compiled_clust_df) = c("Clone_IDs", "EX", "IN", "GLIA", "OLIG", "EX_pct", "IN_pct", "GLIA_pct", "OLIG_pct")
compiled_clust_df_idents = left_join(compiled_clust_df, multicell_metadata_unique[,c("orig.ident","Clone_IDs","Clone_Index","Barcode_UMI_Count")])
dim(compiled_clust_df_idents)
compiled_clust_df_idents$point_col = rep(osvz_col, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_col[compiled_clust_df_idents$Clone_Index == "IndexE"] = vz_col
compiled_clust_df_idents$point_pch = rep(2, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$point_pch[compiled_clust_df_idents$Clone_Index == "IndexE"] = 6
table(compiled_clust_df_idents$Clone_Index)
table(compiled_clust_df_idents$point_col)
table(compiled_clust_df_idents$point_pch)
head(compiled_clust_df_idents)
compiled_clust_df_idents$sample_age = rep(20, dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW16_1220_S2', 'GW16_1220_S1')] = 16
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW19_0221_S1', 'GW19_0221_S2')] = 19
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW21_0502_1S')] = 21
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 22
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 23
compiled_clust_df_idents$sample_age[compiled_clust_df_idents$orig.ident %in% 
                                      c('GW24_0221_1S')] = 24
## create jitter for nicer looking plots
compiled_clust_df_idents_jitter = compiled_clust_df_idents
jitter_amt = seq(-0.005, 0.005, by=0.001)
for(r in c(1:length(rownames(compiled_clust_df_idents_jitter)))){
  compiled_clust_df_idents_jitter[r,"EX_pct"] = compiled_clust_df_idents_jitter[r,"EX_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"IN_pct"] = compiled_clust_df_idents_jitter[r,"IN_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"GLIA_pct"] = compiled_clust_df_idents_jitter[r,"GLIA_pct"]+sample(jitter_amt, 1)
  compiled_clust_df_idents_jitter[r,"OLIG_pct"] = compiled_clust_df_idents_jitter[r,"OLIG_pct"]+sample(jitter_amt, 1)
}
head(compiled_clust_df_idents_jitter)
## making some clean plots with proper colors and themes and removing GW21 (only 3 clones)
table(compiled_clust_df_idents_jitter$sample_age)
#16   19   20   21   22   23   24 
#369   88 1712    1  361  506  221 
compiled_clust_df_idents_jitter = compiled_clust_df_idents_jitter[compiled_clust_df_idents_jitter$sample_age!=21,]

ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clone index')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EX_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=GLIA_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=OLIG_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of OPCs/oligos in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EX_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=GLIA_pct, colour=Clone_Index)) +
  geom_violin() +
  geom_point(position=position_jitterdodge())+
  scale_colour_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=EX_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=IN_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=GLIA_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=sample_age, y=OLIG_pct, colour=Clone_Index)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_colour_manual(values=c(osvz_col, vz_col))+
  ggtitle('Percentage of OPCs/oligos in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')


apply(data.frame(compiled_clust_df_idents[1:5,c("GLIA","OLIG")]==0), 1, sum) ==0

colnames(compiled_clust_df_idents)
head(compiled_clust_df_idents)
compiled_clust_df_idents$clonal_restriction = rep("EX_only", dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "IN_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG == 0] = "GLIA_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG != 0] = "OLIG_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "EX_IN_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN == 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG != 0] = "quadpotent"
table(compiled_clust_df_idents$clonal_restriction)
#EX_GLIA/OLIG_shared EX_IN_GLIA/OLIG_shared           EX_IN_shared                EX_only 
#5               58                    160                    148                   1108 
#GLIA_only    IN_GLIA/OLIG_shared                IN_only              OLIG_only 
#      226                    539                    411                     54 
#quadpotent 
#        54 

compiled_clust_df_idents$clonal_restriction = factor(compiled_clust_df_idents$clonal_restriction, levels=(
  c("EX_only", "IN_only", "GLIA_only", "OLIG_only", "EX_IN_shared", "EX_GLIA/OLIG_shared", "IN_GLIA/OLIG_shared", "EX_IN_GLIA/OLIG_shared", "quadpotent")
))
table(compiled_clust_df_idents$clonal_restriction)
#     EX_only                IN_only              GLIA_only              OLIG_only 
#         1108                    411                    226                     54 
#EX_IN_shared    EX_GLIA/OLIG_shared    IN_GLIA/OLIG_shared EX_IN_GLIA/OLIG_shared 
#         148                    558                    539                    160 
#quadpotent 
#       54 

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

compiled_clust_df_idents$ex_in_presence = rep("EX_only", dim(compiled_clust_df_idents)[1])
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0] = "IN_only"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0] = "no_EX_or_IN"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0] = "EX_IN_shared"
table(compiled_clust_df_idents$ex_in_presence)
compiled_clust_df_idents$ex_in_presence = factor(compiled_clust_df_idents$ex_in_presence, levels=(
  c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN")
))
table(compiled_clust_df_idents$ex_in_presence)
#     EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#        1497          950          362          449 
ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

table(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),"ex_in_presence"])
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#59          680           80          269 
table(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "IndexE","ex_in_presence"])
#Index E count
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#23          213           28           52
table(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "IndexE","ex_in_presence"])/sum(compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "IndexE")
#Index E pct
#   EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#0.07278481   0.67405063   0.08860759   0.16455696 
table(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "Index3","ex_in_presence"])
#Index 3 count
# EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#36          467           52          217 
table(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "Index3","ex_in_presence"])/sum(compiled_clust_df_idents$sample_age %in% c(samp_age) & compiled_clust_df_idents$Clone_Index %in% "Index3")
#Index 3 pct
#    EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#0.04663212   0.60492228   0.06735751   0.28108808

#plot INs by age, but remove IN only clones
compiled_clust_df_idents = compiled_clust_df_idents[compiled_clust_df_idents$sample_age!=21,]
in_only_clones = compiled_clust_df_idents[apply(compiled_clust_df_idents[,c("EX", "GLIA", "OLIG")], 1, sum) == 0 & compiled_clust_df_idents_jitter$IN != 0, "Clone_IDs"]
in_only_clones
dim(compiled_clust_df_idents[!(compiled_clust_df_idents$Clone_IDs %in% in_only_clones),])
#[1] 2846   17

ggplot(compiled_clust_df_idents[!(compiled_clust_df_idents$Clone_IDs %in% in_only_clones),],
       aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age (IN_only clusters removed)') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')


#what is the percentage of clones from a given VZ that are EX only, IN only, etc?
samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=Clone_Index, fill=ex_in_presence)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=Clone_Index, fill=ex_in_presence)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


#is there a bias in direct vs. indirect neurogenesis from VZ/OSVZ?
dim(multicell_metadata)
table(multicell_metadata$subclust_idents)
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       710       1058       1859        978         12       1655         16        121       2949 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       778        607       1447         96       1010        287        437 

table(multicell_metadata$subclust_idents[multicell_metadata$Clone_Index %in% "IndexE"])
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       510        836        954        524          4        830          5         78       2493 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       605        274        663         47        465        131        274
table(multicell_metadata$subclust_idents[multicell_metadata$Clone_Index %in% "Index3"])
#        EX   EX_early       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX 
#       200        222        905        454          8        825         11         43        456 
#IPC_EX_DIV     IPC_IN IPC_IN_DIV       OLIG        OPC    OPC_DIV      RG_EX 
#       173        333        784         49        545        156        163 


ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(x=Clone_Index, fill=subclust_idents)) +
  geom_bar(position = "fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')


ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents)) +
  geom_bar(position = "dodge") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% "EX",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c("<=GW20", ">GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c("<=GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3$age_pseudo %in% c(">GW20") & ls_synthesis_v3_harmony_v3$broad_celltype %in% "EX" & !(is.na(ls_synthesis_v3_harmony_v3$Clone_Index)),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of EX subtypes across all EX cells, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')


ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% "IN",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of IN subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% "IN",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of IN subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% "IN",], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of IN subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20") & multicell_metadata$broad_celltype %in% c("GLIA","OLIG"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of GLIA subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20") & multicell_metadata$broad_celltype %in% c("GLIA","OLIG"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of GLIA subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20") & multicell_metadata$broad_celltype %in% c("GLIA","OLIG"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of GLIA subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20", ">GW20"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of subtypes in multicell clones, all samples") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c("<=GW20"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of subtypes in multicell clones, <=GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')
ggplot(multicell_metadata[multicell_metadata$age_pseudo %in% c(">GW20"),], aes(fill=Clone_Index, x=subclust_idents, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Prevalence of subtypes in multicell clones, >GW20") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Index')

## what is the "mean clone"?
head(compiled_clust_df_idents)
mean(compiled_clust_df_idents$EX_pct)
#[1] 0.2519331
mean(compiled_clust_df_idents$IN_pct)
#[1] 0.4202506
mean(compiled_clust_df_idents$GLIA_pct)
#[1] 0.2292735
mean(compiled_clust_df_idents$OLIG_pct)
#[1] 0.08977131

summary(compiled_clust_df_idents$EX_pct)
#.  Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#0.0000  0.0000  0.2857  0.4203  1.0000  1.0000 
summary(compiled_clust_df_idents$EX_pct[compiled_clust_df_idents$Clone_Index %in% "IndexE"])
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#0.0000  0.0000  0.6667  0.5523  1.0000  1.0000
summary(compiled_clust_df_idents$EX_pct[compiled_clust_df_idents$sample_age %in% c(16,19,20)])
#    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.0000  0.2000  0.6667  0.6040  1.0000  1.0000 
summary(compiled_clust_df_idents$EX_pct[compiled_clust_df_idents$sample_age %in% c(22,23,24)])
#.   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#0.00000 0.00000 0.00000 0.05395 0.00000 1.00000 
summary(compiled_clust_df_idents$EX_pct[compiled_clust_df_idents$sample_age %in% c(22,23,24) & compiled_clust_df_idents$Clone_Index %in% "IndexE"])
#.      Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#0.00000 0.00000 0.00000 0.07069 0.00000 1.00000
summary(compiled_clust_df_idents$EX_pct[compiled_clust_df_idents$sample_age %in% c(22,23,24) & compiled_clust_df_idents$Clone_Index %in% "Index3"])
#.      Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#0.00000 0.00000 0.00000 0.07069 0.00000 1.00000


## going back into the interneurons
ls_in
#27581 features across 12025 samples within 1 assay 
ls_in_unmerge = FindNeighbors(ls_in, dims=1:15)
ls_in_unmerge = FindClusters(ls_in_unmerge, resolution=.5)
ls_in_unmerge = RunUMAP(ls_in_unmerge, dims=c(1:15))
DimPlot(ls_in_unmerge, label=T)
DimPlot(ls_in_unmerge, group.by="orig.ident")
DimPlot(ls_in_unmerge, group.by="age_pseudo")
DimPlot(ls_in_unmerge, group.by="sex")
DimPlot(ls_in_unmerge, cells.highlight = rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$Clone_ize_corrected>=3,]))
#harmony on sex only
ls_in_unmerge = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% c("IN_CGE", "IN_MGE", "IN_local"))
ls_in_unmerge
#27581 features across 12025 samples within 1 assay
ls_in_unmerge <- ScaleData(ls_in_unmerge, features = all.genes)
ls_in_unmerge = FindVariableFeatures(ls_in_unmerge, selection.method="vst", nfeatures=2000)
ls_in_unmerge = RunPCA(ls_in_unmerge, features=VariableFeatures(object=ls_in_unmerge))
ElbowPlot(ls_in_unmerge)
ls_in_unmerge <- RunHarmony(ls_in_unmerge, c("sex"))
ls_in_unmerge <- FindNeighbors(ls_in_unmerge, reduction = "harmony", dims = 1:15)
ls_in_unmerge <- FindClusters(ls_in_unmerge, resolution = .5)
ls_in_unmerge <- RunUMAP(ls_in_unmerge, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in_unmerge, reduction = "umap",label=T, pt.size=1)
VlnPlot(ls_in_unmerge, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_in_unmerge, reduction = "umap",group.by="orig.ident", pt.size=1)
DimPlot(ls_in_unmerge, reduction = "umap",group.by="age_pseudo", pt.size=1)
FeaturePlot(ls_in_unmerge, label=T, features=c("GAD2", "NEUROD2", "NR2F1", "PAX6", "MKI67", "NPY", "NR2F2", "ADARB2", "MAF"))
FeaturePlot(ls_in_unmerge, label=T, features=c("nCount_RNA"))
DimPlot(ls_in_unmerge, label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_filt">=3,])) +ggtitle("Clone size >=3")
FeaturePlot(ls_in_unmerge, label=T, features=c("NPY", "PAX6", "MKI67", "LGALS1"))
#ls_in_unmerge cluster markers
ls_in_unmerge.markers <- FindAllMarkers(ls_in_unmerge, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_in_unmerge.markers)
ls_in_unmerge_clustermarkers_unfilt = as.data.frame(ls_in_unmerge.markers %>% group_by(cluster))
write.csv(ls_in_unmerge_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_unmerge_clustermarkers_v1_unfilt.csv',row.names=F)
ls_in_unmerge.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_in_unmerge_clustermarkers = as.data.frame(ls_in_unmerge.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_in_unmerge_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_unmerge_clustermarkers_v1.csv',row.names=F)
ls_in_unmerge_clustermarkers
FeaturePlot(ls_in_unmerge, label=T, features=c("NPY", "PAX6", "SCGN", "SP8"))
#putative OB markers (IN.2)
FeaturePlot(ls_in_unmerge, label=T, features=c("TSHZ1", "PBX3", "MEIS2", "CALB2", "CDCA7L", "SYNPR", "ETV1"))
#putative CIN markers (IN.3)
FeaturePlot(ls_in_unmerge, label=T, features=c("NR2F1", "NFIX", "PROX1", "NR2F2", "SOX6", "CXCR4", "MAF"))

FeaturePlot(ls_in_unmerge, label=T, features=c("TSHZ1", "PBX3", "NR2F1", "PROX1"))
VlnPlot(ls_in_unmerge, features=c("TSHZ1", "PBX3", "NR2F1", "PROX1"), ncol=4)
FeaturePlot(ls_in_unmerge, label=T, features=c("NR2F1", "SOX6", "CXCR4", "PROX1"))
FeaturePlot(ls_in_unmerge, label=T, features=c("TSHZ1", "PBX3", "MEIS2", "SYNPR"))


FeaturePlot(ls_in_unmerge, label=T, features=c("GRIK2"))

DimPlot(ls_in_unmerge, label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_corrected">=3,]), split.by="Clone_Index") +ggtitle("Clone size >=3")
DimPlot(ls_in_unmerge, label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_corrected">=2,]), split.by="Clone_Index") +ggtitle("Clone size >1")
DimPlot(ls_in_unmerge, group.by="Clone_Index")
DimPlot(ls_in_unmerge, group.by="Clone_Index", split.by="age_pseudo")

VlnPlot(ls_in_unmerge, features=c("Clone_size_corrected"))
DimPlot(subset(ls_in_unmerge, subset = Clone_Index %in% c("IndexE", "Index3")), label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_corrected">=2,]), split.by="Clone_Index") +ggtitle("Clone size >1")
DimPlot(subset(ls_in_unmerge, subset = Clone_Index %in% c("IndexE", "Index3")), label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_corrected">=3,]), split.by="Clone_Index") +ggtitle("Clone size >2")



ls_in = subset(ls_synthesis_v3_harmony_v3, clust_idents %in% c("IN_CGE", "IN_MGE", "IN_local"))
ls_in
#27581 features across 12025 samples within 1 assay
ls_in <- ScaleData(ls_in, features = all.genes)
ls_in = FindVariableFeatures(ls_in, selection.method="vst", nfeatures=2000)
ls_in = RunPCA(ls_in, features=VariableFeatures(object=ls_in))
ElbowPlot(ls_in)
ls_in <- RunHarmony(ls_in, c("orig.ident"))
ls_in <- FindNeighbors(ls_in, reduction = "harmony", dims = 1:15)
ls_in <- FindClusters(ls_in, resolution = .5)
ls_in <- RunUMAP(ls_in, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in, reduction = "umap",label=T, pt.size=1)
VlnPlot(ls_in, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_in, reduction = "umap",group.by="orig.ident", pt.size=1)
DimPlot(ls_in, reduction = "umap",group.by="age_pseudo", pt.size=1)
DimPlot(subset(ls_in, subset = Clone_Index %in% c("IndexE", "Index3")), label=T, cells.highlight=rownames(ls_in_unmerge@meta.data[ls_in_unmerge@meta.data$"Clone_size_corrected">=2,]), split.by="Clone_Index") +ggtitle("Clone size >1")
DimPlot(subset(ls_in, subset = Clone_Index %in% c("IndexE", "Index3")), group.by="Clone_Index", split.by="age_pseudo", pt.size=1)
FeaturePlot(ls_in, label=T, features=c("GAD2", "NEUROD2", "NR2F1", "PAX6", "MKI67", "NPY", "NR2F2", "ADARB2", "MAF"))
#FeaturePlot(ls_in, label=T, features=c("nCount_RNA"))
FeaturePlot(ls_in, label=T, features=c("NR2F1", "SOX6", "CXCR4", "PROX1"))
FeaturePlot(ls_in, label=T, features=c("TSHZ1", "PBX3", "MEIS2", "SYNPR"))


#differential expression based on STICR Index for INs
library(ggrepel)
ls_in = SetIdent(ls_in, value="Clone_Index")
levels(ls_in)
index.de.markers.ins = FindMarkers(subset(ls_in, Clone_size_corrected>=3 & seurat_clusters!=7), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ins, n=20)
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ins)))+
 geom_point()+
 theme_minimal()+
 geom_text_repel()+
 ggtitle("Differential expression between Index 3 (left) and Index E (right) -- INs, clone size >=3, clust7 excluded")+
 theme(plot.title = element_text(hjust = 0.5))

#check that diff expr genes are actually from index, not just age?
ls_in = SetIdent(ls_in, value="age_pseudo")
levels(ls_in)
age.de.markers.ins = FindMarkers(subset(ls_in, Clone_size_corrected>=3 & seurat_clusters!=7), ident.1 = "<=GW20", ident.2 = ">GW20")
head(age.de.markers.ins, n=20)
ggplot(data=age.de.markers.ins, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=age.de.markers.ins, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(age.de.markers.ins)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between >GW20 (left) and <=GW20 (right) -- INs, clone size >=3, clust7 excluded")+
  theme(plot.title = element_text(hjust = 0.5))


#by index:
head(index.de.markers.ins[rownames(index.de.markers.ins) %in% c('SCGN', 'CALB2', 'ETV1', 'SOX6', 'PAX6', 'NFIA'),])
#             p_val avg_log2FC pct.1 pct.2    p_val_adj
#SCGN  3.473573e-59 -1.2017664 0.458 0.765 9.580461e-55
#SOX6  1.011409e-56  0.9797867 0.903 0.656 2.789568e-52
#PAX6  2.822469e-41  1.1025245 0.567 0.314 7.784651e-37
#ETV1  2.775091e-23 -0.4977531 0.371 0.613 7.653977e-19
#NFIA  9.173633e-21  0.3078274 0.988 0.963 2.530180e-16
#CALB2 6.605250e-20 -0.7866167 0.213 0.394 1.821794e-15

#by age:
head(age.de.markers.ins[rownames(age.de.markers.ins) %in% c('SCGN', 'CALB2', 'ETV1', 'SOX6', 'PAX6', 'NFIA'),])
#             p_val avg_log2FC pct.1 pct.2    p_val_adj
#PAX6  5.560868e-67  1.2764956 0.660 0.314 1.533743e-62
#SCGN  1.167391e-54 -1.0546549 0.383 0.743 3.219780e-50
#SOX6  1.644839e-34  0.7522275 0.897 0.712 4.536629e-30
#NFIA  4.351274e-32  0.4462987 0.987 0.969 1.200125e-27
#CALB2 3.620490e-15 -0.3622932 0.187 0.371 9.985674e-11


#check diff expr. by index after restricting to one age group, and just look at non-CGE or MGE
ls_in = SetIdent(ls_in, value="seurat_clusters")
DimPlot(ls_in, label=T)
ls_in = SetIdent(ls_in, value="Clone_Index")
levels(ls_in)
index.de.markers.ins.young = FindMarkers(subset(ls_in, Clone_size_corrected>=3 & age_pseudo %in% "<=GW20" & seurat_clusters %in% c(1,3,4,5,8,9,10)), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ins.young, n=20)
ggplot(data=index.de.markers.ins.young, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ins.young, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ins.young)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- INs, clone size >=3, <=GW20 only, clusters(1,3,4,5,8,9,10)")+
  theme(plot.title = element_text(hjust = 0.5))

index.de.markers.ins.old = FindMarkers(subset(ls_in, Clone_size_corrected>=3 & age_pseudo %in% ">GW20" & seurat_clusters %in% c(1,3,4,5,8,9,10)), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ins.old, n=20)
ggplot(data=index.de.markers.ins.old, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ins.old, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ins.old)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- INs, clone size >=3, >GW20 only, clusters(1,3,4,5,8,9,10)")+
  theme(plot.title = element_text(hjust = 0.5))

#is there any difference in INs that are in shared clones?
head(compiled_clust_df_idents)
table(compiled_clust_df_idents$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#1497          949          362          449 
ls_in_meta = ls_in@meta.data
head(left_join(ls_in_meta, compiled_clust_df_idents[,c("Clone_IDs", "clonal_restriction", "ex_in_presence")]))
sum(!is.na(left_join(ls_in_meta, compiled_clust_df_idents[,c("Clone_IDs", "clonal_restriction", "ex_in_presence")])[,"ex_in_presence"]))
#[1] 2077
ls_in_meta = left_join(ls_in_meta, compiled_clust_df_idents[,c("Clone_IDs", "clonal_restriction", "ex_in_presence")])
table(ls_in_meta$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#48         1515          460           54 
ls_in_meta[ls_in_meta$ex_in_presence %in% "EX_only" & !is.na(ls_in_meta$Clone_IDs_filt),]
#weirdness is coming from IN_UNK cells or cells that had their assignment stripped due to low reads
DimPlot(ls_in, cells.highlight=rownames(ls_in@meta.data[ls_in$subclust_idents %in% c("IN_UNK"),]))
ls_in = AddMetaData(ls_in, ls_in_meta$ex_in_presence, col.name="ex_in_presence")
DimPlot(ls_in, group.by="ex_in_presence")
DimPlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), group.by="ex_in_presence")
ls_in = SetIdent(ls_in, value="seurat_clusters")
DimPlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), label=T, split.by="ex_in_presence")
#any DE genes between IN_only and EX_IN_shared?
ls_in = SetIdent(ls_in, value="ex_in_presence")
levels(ls_in)
clone.comp.de.markers = FindMarkers(subset(ls_in, Clone_size_corrected>=3 & seurat_clusters %in% c(1,3,4,7,8,9,10)), ident.1 = "EX_IN_shared", ident.2 = "IN_only")
head(clone.comp.de.markers, n=20)
ggplot(data=clone.comp.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=clone.comp.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(clone.comp.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between IN_only and EX_IN_shared -- INs, clone size >=3, clusters(1,3,4,7,8,9,10)")+
  theme(plot.title = element_text(hjust = 0.5))


FeaturePlot(ls_in, features=c("SCGN"), split.by="Clone_Index_filt")
FeaturePlot(ls_in, features=c("SOX6"), split.by="Clone_Index_filt")


ls_in = SetIdent(ls_in, value="Clone_Index_filt")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("SCGN", "SOX6"))

ls_in = SetIdent(ls_in, value="age_pseudo")
VlnPlot(ls_in, features=c("SCGN", "SOX6"))

VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("SCGN"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("SCGN"), split.by="Clone_Index", split.plot = T)
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("SOX6"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("ETV1"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("NR2F1"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("ANK2"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("SCG2"), split.by="Clone_Index")
VlnPlot(subset(ls_in, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("ADARB2"), split.by="Clone_Index")

FeaturePlot(ls_in, features=c("SCG2"))



## re-look at EX neurons for similar ideas (young vs. old, VZ vs. OSVZ)
ls_ex
DimPlot(ls_ex, label=T)
DimPlot(ls_ex, group.by="age_pseudo")
DimPlot(ls_ex, cells.highlight=rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>=3,])) + ggtitle("Clone size >= 3")
DimPlot(subset(ls_ex, Clone_Index_filt %in% c("Index3", "IndexE")), cells.highlight=rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>=3,]),
        split.by="Clone_Index_filt") + ggtitle("Clone size >= 3")
DimPlot(subset(ls_ex, Clone_Index_filt %in% c("Index3", "IndexE")), cells.highlight=rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>=3,]),
        split.by="age_pseudo") + ggtitle("Clone size >= 3")

ls_ex.markers <- FindAllMarkers(ls_ex, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_ex.markers)
ls_ex_clustermarkers_unfilt = as.data.frame(ls_ex.markers %>% group_by(cluster))
write.csv(ls_ex_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_ex_clustermarkers_v1_unfilt.csv',row.names=F)
ls_ex.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_ex_clustermarkers = as.data.frame(ls_ex.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_ex_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_ex_clustermarkers_v1.csv',row.names=F)


ls_ex = SetIdent(ls_ex, value="Clone_Index")
levels(ls_ex)
index.de.markers.ins = FindMarkers(subset(ls_ex, Clone_size_corrected>=3), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ins, n=20)
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ins)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- EX, clone size >=3, excluded")+
  theme(plot.title = element_text(hjust = 0.5))
#check that diff expr genes are actually from index, not just age?
ls_ex = SetIdent(ls_ex, value="age_pseudo")
levels(ls_ex)
age.de.markers.ex = FindMarkers(subset(ls_ex, Clone_size_corrected>=3), ident.1 = "<=GW20", ident.2 = ">GW20")
head(age.de.markers.ex, n=20)
ggplot(data=age.de.markers.ex, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=age.de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(age.de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between >GW20 (left) and <=GW20 (right) -- EX, clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))

ls_ex = SetIdent(ls_ex, value="seurat_clusters")
DimPlot(ls_ex, label=T)
ls_ex = SetIdent(ls_ex, value="Clone_Index")
levels(ls_ex)
index.de.markers.ex.young = FindMarkers(subset(ls_ex, Clone_size_corrected>=3 & age_pseudo %in% "<=GW20"), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ex.young, n=20)
ggplot(data=index.de.markers.ex.young, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ex.young, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ex.young)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- INs, clone size >=3, <=GW20 only")+
  theme(plot.title = element_text(hjust = 0.5))

index.de.markers.ex.old = FindMarkers(subset(ls_ex, Clone_size_corrected>=3 & age_pseudo %in% ">GW20"), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers.ex.old, n=20)
ggplot(data=index.de.markers.ex.old, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ex.old, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ex.old)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- INs, clone size >=3, >GW20 only")+
  theme(plot.title = element_text(hjust = 0.5))

FeaturePlot(ls_ex, features=c("TLE4", "SYT6", "PCDH11X", "ZFHX3"))
FeaturePlot(ls_ex, features=c("PPP1R17", "MARCH1", "CUX2", "SATB2"))
DimPlot(ls_ex, group.by="clust_idents")

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TLE4", "SYT6", "PCDH11X", "ZFHX3"), label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("PPP1R17", "MARCH1", "CUX2", "SATB2"), label=T)

DotPlot(ls_synthesis_v3_harmony_v3, features=unique(c("CRYM","TMEM178A","TRPM3","NGEF","OSBPL1A","PCDH17",
                                               "HS3ST5","HCRTR2","CDH10","TOX","NR4A2","CCN2","GABRA5",
                                               "ROBO2","NLGN1","DPP10","LRP1B","ROBO1","OPCML","PCDH9",
                                               "RALYL","TRPM3","MDGA2","NRG1","ARPP21","LRRTM4","HS3ST4",
                                               "NEGR1","RGS7","PDE1A","NFIA","MACROD2","FRMPD4","SORCS1",
                                               "MGAT4C","TLE4","KIAA1217","KHDRBS3","GRIN2B","ASTN2"))) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("ARPP21", "GRIN2B", "CDH10", "TOX"), label=T)

DotPlot(ls_synthesis_v3_harmony_v3, features=unique(c("CDH18","PPM1E","SYN3","KCNMA1","FOXP2","CNTNAP5",
                                                      "DOCK4","KCNQ5","GRM5","LINGO2","ZFPM2","LMO3",
                                                      "ADAMTSL1","C8orf34","PRKG1","RGS6","EFNA5","GRIK3",
                                                      "PDZD2","SLC35F1","KLHL1","PRICKLE2","CSMD2","GAS7",
                                                      "DOCK9","SLC8A1","ASIC2","MGST1","CDH10","PCDH11Y",
                                                      "SOX5","OSBPL10","ZFHX3","ZFPM2","EPB41L4A","SLIT3",
                                                      "ZNF385B","POU6F2","ADAMTS19","SEMA3E","FAM160A1","ST18",
                                                      "ZNF804B","SULF1","HCRTR2","PAPPA2","PLEKHH2","NFIB",
                                                      "OCA2","RYR2"))) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("KCNQ5", "CSMD2", "DOCK9", "RYR2"), label=T)

DotPlot(ls_synthesis_v3_harmony_v3, features=c("EIF1B","NEUROD2","NEUROD6","BHLHE22","CSRP2","STMN2","SATB2-AS1",
                                               "SRM","NEFM","UCHL1","SLA","ZBTB18","PRSS12","CALM1","BCL11A",
                                               "SATB2","PCSK1N","ADRA2A","NSG1","CDK5R1")) +
  ggtitle("Cluster 0 markers") +
  theme(axis.text.x=element_text(angle=45, hjust=1))

ls_ex = SetIdent(ls_ex, value="seurat_clusters")
DimPlot(ls_ex, label=T)

DimPlot(ls_ex, cells.highlight = rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>1,]), cols.highlight="#022678") + 
  ggtitle("Clone size > 1")
DimPlot(ls_ex, cells.highlight = rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>1,]), 
        cols.highlight="#022678", split.by="age_pseudo") + 
  ggtitle("Clone size > 1")

#differential expression of ls_ex to see what the differences are above and below the "split"
levels(ls_ex)
ex.subclust.de.markers = FindMarkers(subset(ls_ex), ident.1 = c(0,2,4,5), ident.2 = c(1,3,7))
head(ex.subclust.de.markers, n=20)
ggplot(data=ex.subclust.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=ex.subclust.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(ex.subclust.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between EX top half [right] (clusters 0,2,4,5) and EX bottom half [left] (cluster 1,3,7) -- EX, ")+
  theme(plot.title = element_text(hjust = 0.5))
write.csv(ex.subclust.de.markers, "/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ex_subclust_de_markers.csv")

FeaturePlot(ls_ex, features=c("CUX2", "SRM", "GABARAP", "FTL"))
FeaturePlot(ls_ex, features=c("nCount_RNA"), pt.size=0.5)
VlnPlot(ls_ex, features=c("nCount_RNA"), pt.size=0.5)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CUX2", "SRM", "GABARAP", "FTL"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("nCount_RNA"), pt.size=0.5, raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1,]),
        raster=F, cols.highlight="#022678") + 
  ggtitle("Clone size > 1")
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1,]),
        raster=F, cols.highlight="#022678", split.by="age_pseudo") + 
  ggtitle("Clone size > 1")
DimPlot(subset(ls_synthesis_v3_harmony_v3, age_pseudo %in% "<=GW20"), cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1,]),
        raster=F, cols.highlight="#022678", split.by="Clone_Index") + 
  ggtitle("Clone size > 1, <=GW20")
DimPlot(subset(ls_synthesis_v3_harmony_v3, age_pseudo %in% ">GW20"), cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1,]),
        raster=F, cols.highlight="#022678", split.by="Clone_Index") + 
  ggtitle("Clone size > 1, >GW20")


## remake core figures with corrected clone sizes 
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='IndexE'], cols.highlight=vz_col) +
  ggtitle('Index E') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='Index3'], cols.highlight=osvz_col) +
  ggtitle('Index 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, group.by = "Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=1) +
  ggtitle('STICR-labeled cells') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, split.by = "Clone_Index_filt")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected==1], cols.highlight='black') +
  ggtitle('Clone Size == 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1], cols.highlight='#2dba3b') +
  ggtitle('Clone Size > 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3], cols.highlight='#1d8027') +
  ggtitle('Clone Size >= 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=5], cols.highlight='#0f4515') +
  ggtitle('Clone Size >= 5') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2], cols.highlight='#2dba3b', split.by="Clone_Index_filt") +
  ggtitle('Large clones (>=2) by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=5], cols.highlight='#0f4515', split.by="Clone_Index_filt") +
  ggtitle('Large clones (>=5) by index') +
  theme(plot.title = element_text(hjust = 0.5))

DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2], cols.highlight='#2dba3b', split.by="age_pseudo") +
  ggtitle('Clone Size >= 2 by age') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3], cols.highlight='#1d8027', split.by="age_pseudo") +
  ggtitle('Clone Size >= 3 by age') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% ">GW20"],
        cols.highlight='#0f6040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 3 in >GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% "<=GW20"],
        cols.highlight='#0f6040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 3 in <=GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% ">GW20"],
        cols.highlight='#0f3040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 2 in >GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2 & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% "<=GW20"],
        cols.highlight='#0f3040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 2 in <=GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, 
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2 & ls_synthesis_v3_harmony_v3@meta.data$orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S")],
        cols.highlight='#0f3040', split.by="Clone_Index") +
  ggtitle('Clone Size >= 2 in == GW20 samples by index') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, group.by = "Clone_Index_filt", split.by = "age_pseudo", cols = c(osvz_col, vz_col), pt.size=1) +
  ggtitle('STICR-labeled cells') +
  theme(plot.title = element_text(hjust = 0.5))

table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18 
#49274  9354  4863  2864  1840  1332   756   480   450   260   308   264    78    70    75    80   119    36 
#19    20    21    24    25    26 
#19    40    21    24    25    26 
table(multicell_metadata$Clone_size_corrected)
#               3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   
#            4863 2864 1840 1332  756  480  450  260  308  264   78   70   75   80  119   36   
#19   20   21   24   25  26
#19   40   21   24   25  26


## misc plotting from 4/11
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))



FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR4A2", "TBR1", "ZFHX3", "TLE4"), label=T, order=F, raster=F)
FeaturePlot(ls_ex, features=c("NR4A2", "TBR1", "ZFHX3", "TLE4"), label=T, order=F, raster=F)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR4A2", "GPC4", "CCN2", "CALCA"), label=T, raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("ST3GAL1", "ETL4", "PRSS12", "GPC4"), label=T)
FeaturePlot(ls_in, features=c("PAX6", "SP8", "PROX1", "DLX2", "LHX6"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CRYM", "TBR1", "TLE4", "CUX2"), label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CRYM"), label=T, raster=F, pt.size=1)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM", "EOMES", "SATB2", "TBR1",
                                                   "DLX2", "GAD2", "NR2F2", "NXPH1",
                                                   "MKI67", "GFAP", "OLIG2", "MBP"), label=T, raster=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM", "EOMES", "SATB2", "TBR1",
                                                   "DLX2", "GAD2", "NR2F2", "NXPH1",
                                                   "MKI67", "GFAP", "OLIG2", "MBP"), label=F, raster=T)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="subclust_idents")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=T)



## add age as a metadata column
dim(seurat_meta)
colnames(seurat_meta)
table(seurat_meta$age)
colnames(ls_synthesis_v3_harmony_v3@meta.data)
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, seurat_meta$age, col.name = "sample_age")
DimPlot(ls_synthesis_v3_harmony_v3, group.by="sample_age")

ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x = sample_age, y = Clone_size_corrected)) +
  geom_bar() +
  xlab("Sample Age") +
  ylab("Clone Size Corrected") +
  ggtitle("Clone Size Corrected vs Sample Age")
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x = Clone_size_corrected, fill = as.factor(sample_age))) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  xlab("Clone Size Corrected") +
  ylab("Proportion of cells") +
  ggtitle("Proportion of cells in given clone size by age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x = Clone_size_corrected, fill = age_pseudo)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  xlab("Clone Size Corrected") +
  ylab("Proportion of cells") +
  ggtitle("Proportion of cells in given clone size by age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))  
ggplot(ls_ex@meta.data, aes(x = Clone_size_corrected, fill = age_pseudo)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  xlab("Clone Size Corrected") +
  ylab("Proportion of cells") +
  ggtitle("Proportion of cells in given clone size by age (ENs only)") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))



## look for markers of local vs. GE-derived INs
ls_in
colnames(ls_in@meta.data)
ls_in = SetIdent(ls_in, value="seurat_clusters")
DimPlot(ls_in, label=T)
DimPlot(ls_in, group.by="subclust_idents")

levels(ls_in)
local.ge.de.markers = FindMarkers(subset(ls_in, seurat_clusters %in% c(0,1,4,5,9)), ident.1 = c(1,4,5,9), ident.2 = 0)
head(local.ge.de.markers, n=20)
ggplot(data=local.ge.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=local.ge.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(local.ge.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between local INs [right] (clusters 1,4,5,9) and GE INs [left] (cluster 0) -- INs, ")+
  theme(plot.title = element_text(hjust = 0.5))
FeaturePlot(ls_in, features=c("ADARB2", "SORCS3", "NFIA", "SOX6", "LHX6", "MAF"))
FeaturePlot(ls_in, features=c("ADARB2", "LHX6", "NR2F1", "SOX6"))

## can we look at expression of specific markers (esp. deep layer?) after filtering to specific parameters
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value = "Clone_Index_filt")
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TBR1", "NR4A2"), pt.size = 0)
ls_ex = SetIdent(ls_ex, value = "Clone_Index")
VlnPlot(ls_ex, features=c("SATB2"), pt.size = 0, split.by="age_pseudo") 
VlnPlot(ls_ex, features=c("CUX2"), pt.size = 0, split.by="age_pseudo") 
VlnPlot(ls_ex, features=c("TBR1"), pt.size = 0, split.by="age_pseudo") 
VlnPlot(ls_ex, features=c("NR4A2"), pt.size = 0.1, split.by="age_pseudo") 
VlnPlot(ls_ex, features=c("CRYM"), pt.size = 0.1, split.by="age_pseudo")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("SATB2"), pt.size = 0.1, split.by="age_pseudo") + ggtitle("SATB2", subtitle = "Clone size >1")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("CUX2"), pt.size = 0.1, split.by="age_pseudo") + ggtitle("SATB2", subtitle = "Clone size >1")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("TBR1"), pt.size = 0.1, split.by="age_pseudo") + ggtitle("TBR1", subtitle = "Clone size >1")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("NR4A2"), pt.size = 0.1, split.by="age_pseudo") + ggtitle("NR4A2", subtitle = "Clone size >1")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("CRYM"), pt.size = 0.1, split.by="age_pseudo") + ggtitle("CRYM", subtitle = "Clone size >1")

ls_ex = SetIdent(ls_ex, value = "age_pseudo")
VlnPlot(ls_ex, features=c("TBR1"), pt.size = 0, split.by="Clone_Index") + scale_fill_manual(values=c(osvz_col, vz_col))
VlnPlot(ls_ex, features=c("NR4A2"), pt.size = 0.1, split.by="Clone_Index") + scale_fill_manual(values=c(osvz_col, vz_col))
VlnPlot(ls_ex, features=c("CRYM"), pt.size = 0.1, split.by="Clone_Index") + scale_fill_manual(values=c(osvz_col, vz_col))





## check clone composition and clone size as a function of age to check IPC hypothesis
#boxplot of clone sizes
dim(multicell_metadata_unique)
colnames(multicell_metadata_unique)
ggplot(multicell_metadata_unique, aes(x = age_pseudo, y = Clone_size_corrected)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x = age_pseudo, y = Clone_size_corrected)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age -- ALL cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_ex@meta.data, aes(x = age_pseudo, y = Clone_size_corrected)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age -- ALL EX cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_ex@meta.data, aes(x = age_pseudo, y = Clone_size_corrected, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age -- ALL EX cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_ex@meta.data, aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL EX cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_ex@meta.data, ls_ex@meta.data$age_pseudo %in% "<=GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge",  binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL <=GW20 EX cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_ex@meta.data, ls_ex@meta.data$age_pseudo %in% ">GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL >GW20 EX cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_synthesis_v3_harmony_v3@meta.data, ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% "<=GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL <=GW20 cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_synthesis_v3_harmony_v3@meta.data, ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% ">GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL >GW20 cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(ls_in@meta.data, aes(x = age_pseudo, y = Clone_size_corrected, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age -- ALL IN cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_in@meta.data, aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL IN cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_in@meta.data, ls_in@meta.data$age_pseudo %in% "<=GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge",  binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL <=GW20 IN cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_in@meta.data, ls_in@meta.data$age_pseudo %in% ">GW20"), aes(x = Clone_size_corrected, fill=Clone_Index)) +
  geom_histogram(position = "dodge", binwidth=1) +
  scale_fill_manual(values=c(osvz_col, vz_col)) +
  xlab("Clone Size") +
  ylab("Counts") +
  ggtitle("Clone Size by Index -- ALL >GW20 IN cells, not >=3 Clone_size") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))

table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected)
#.   1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18 
#49274  9354  4863  2864  1840  1332   756   480   450   260   308   264    78    70    75    80   119    36 
#19    20    21    24    25    26 
#19    40    21    24    25    26

table(multicell_metadata$Clone_size_corrected)
table(multicell_metadata_unique$Clone_size_corrected)
colnames(compiled_clust_df_idents)
dim(compiled_clust_df_idents)

head(left_join(multicell_metadata_unique, compiled_clust_df_idents, by="Clone_IDs"))
multicell_full_metadata_unique = left_join(multicell_metadata_unique, compiled_clust_df_idents, by="Clone_IDs")

dim(multicell_full_metadata_unique)
multicell_full_metadata_unique = multicell_full_metadata_unique[!is.na(multicell_full_metadata_unique$ex_in_presence),]
ggplot(multicell_full_metadata_unique, aes(x = age_pseudo, y = Clone_size_corrected, fill = ex_in_presence)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(multicell_full_metadata_unique, aes(x = ex_in_presence, y = Clone_size_corrected, fill = age_pseudo)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(multicell_full_metadata_unique, ex_in_presence %in% c("EX_only", "EX_IN_shared")), aes(x = ex_in_presence, y = Clone_size_corrected, fill = age_pseudo)) +
  geom_boxplot() +
  xlab("Sample age") +
  ylab("Clone Size") +
  ggtitle("Clone Size vs Age") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5))

ggplot(multicell_full_metadata_unique, aes(x=ex_in_presence, fill=age_pseudo, group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('EX/IN presence by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(subset(multicell_full_metadata_unique, ex_in_presence %in% c("EX_only", "EX_IN_shared")), aes(x=ex_in_presence, fill=age_pseudo, group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('EX/IN presence by age, clones with EX only') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(subset(multicell_full_metadata_unique, ex_in_presence %in% c("EX_only", "EX_IN_shared")), aes(x=ex_in_presence, fill=age_pseudo, group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(count))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('EX/IN presence by age, clones with EX only') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)

DimPlot(ls_synthesis_v3_harmony_v3,
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[
          ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% multicell_full_metadata_unique$Clone_IDs_filt[
            multicell_full_metadata_unique$ex_in_presence %in% c("EX_IN_shared")
          ]
          ,]),
        raster=F) +
  ggtitle("Cells in EX_IN_shared clones")
DimPlot(ls_synthesis_v3_harmony_v3,
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[
          ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% multicell_full_metadata_unique$Clone_IDs_filt[
            multicell_full_metadata_unique$ex_in_presence %in% c("EX_only")
          ]
          ,]),
        raster=F) +
  ggtitle("Cells in EX_only clones")

DimPlot(ls_synthesis_v3_harmony_v3,
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[
          ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% multicell_full_metadata_unique$Clone_IDs_filt[
            multicell_full_metadata_unique$ex_in_presence %in% c("EX_only")
            ]
          ,]),
        split.by="age_pseudo", raster=F) +
  ggtitle("Cells in EX_only clones")
DimPlot(ls_synthesis_v3_harmony_v3,
        cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data[
          ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% multicell_full_metadata_unique$Clone_IDs_filt[
            multicell_full_metadata_unique$ex_in_presence %in% c("EX_IN_shared")
          ]
          ,]),
        split.by="age_pseudo", raster=F) +
  ggtitle("Cells in EX_IN_shared clones")
DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)


FeaturePlot(ls_in, c("GAD2", "DLX2", "LHX6", "SORCS3", "NR2F2", "ADARB2", "MEIS2", "PROX1", "SOX6", "TSHZ1", "PBX3", "NR2F1"))
FeaturePlot(ls_synthesis_v3_harmony_v3, c("NPY", "MKI67"))
FeaturePlot(ls_in, c("NPY", "MKI67"))
FeaturePlot(ls_glia, c("NPY", "MKI67", "OLIG2", "EOMES"), label=T)
FeaturePlot(ls_glia, c("NPY", "DLX2", "OLIG2", "EOMES"), label=T)






## 4/26/23 work

## try to convert Seurat object to AnnData for use with CoSpar
library("SeuratDisk")
SaveH5Seurat(ls_synthesis_v3_harmony_v3, filename = '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v3.h5Seurat')
Convert('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v3.h5Seurat', dest = "h5ad")
## to do manually...
#adata.X: state count matrix, shape (n_cell, n_gene). This should not be log-normalized.
#adata.var_names: list of gene names, shape (n_genes,)
#adata.obs[time_info]: time annotation (type: str) for each cell, shape (n_cell,)
#adata.obs[state_info]: state annotation for each cell, shape (n_cell, 1). [Optional. Can be generated in preprocessing
#adata.obsm[X_clone]: clonal labels for each cell in the form of np.array or sparse matrix, shape (n_cell, n_clone).
#adata.obsm[X_pca]: PCA matrix, shape (n_cell, n_pcs). [Optional. Can be generated in preprocessing.
#adata.obsm[X_emb]: two-dimensional embedding, shape (n_cell, 2). [Optional. Can be generated in preprocessing

#adata.X: state count matrix, shape (n_cell, n_gene). This should not be log-normalized.
ls_synthesis_v3_harmony_v3@assays$RNA[1:5,1:5] #this is already log-normalized.... need to get non-normalized
dim(ls_synthesis_v3_harmony_v3@assays$RNA)
#[1]  27581 100571
ls_v3_unnorm = merge(ls1, c(ls3,
                               ls11,
                               ls21,
                               ls22,
                               ls23,
                               ls27,
                               ls28,
                               ls29,
                               ls30,
                               ls31,
                               ls32,
                               ls33,
                               ls34,
                               ls35,
                               ls36
),
project="local_STICR_synthesis_v3")
ls_v3_unnorm@assays$RNA[1:5,1:5]
dim(ls_v3_unnorm@assays$RNA)
#[1]  27581 105254
dim(ls_v3_unnorm@assays$RNA[,colnames(ls_v3_unnorm@assays$RNA) %in% colnames(ls_synthesis_v3_harmony_v3@assays$RNA)])
subset(ls_v3_unnorm, cells = rownames(ls_synthesis_v3_harmony_v3@meta.data))
ls_v3_unnorm = subset(ls_v3_unnorm, cells = rownames(ls_synthesis_v3_harmony_v3@meta.data))
adata_x = ls_v3_unnorm@assays$RNA
dim(adata_x)
#[1]  27581 100571

#adata.var_names: list of gene names, shape (n_genes,)
adata_var_names = rownames(adata_x)
length(adata_var_names)
#[1] 27581

#adata.obs[time_info]: time annotation (type: str) for each cell, shape (n_cell,)
dim(seurat_meta)
seurat_meta$age = 16
unique(seurat_meta$orig.ident)
seurat_meta$age[seurat_meta$orig.ident %in% c('GW18_0215_3S')] = 18
seurat_meta$age[seurat_meta$orig.ident %in% c('GW24_0221_1S')] = 24
seurat_meta$age[seurat_meta$orig.ident %in% c('GW21_0502_1S')] = 21
seurat_meta$age[seurat_meta$orig.ident %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 23
seurat_meta$age[seurat_meta$orig.ident %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 22
seurat_meta$age[seurat_meta$orig.ident %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 19
seurat_meta$age[seurat_meta$orig.ident %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2', 'GW20_0308_PFC_2S')] = 20
table(seurat_meta$orig.ident)
table(seurat_meta$age)
sum(seurat_meta$CBC == colnames(adata_x))
#[1] 100571
adata_time_info = seurat_meta$age

#adata.obs[state_info]: state annotation for each cell, shape (n_cell, 1). [Optional. Can be generated in preprocessing
table(seurat_meta$subclust_idents)
adata_state_info = seurat_meta$subclust_idents

#adata.obsm[X_clone]: clonal labels for each cell in the form of np.array or sparse matrix, shape (n_cell, n_clone).
library(Matrix)
adata_x_clone = Matrix(data=0, nrow=length(seurat_meta$CBC), ncol=length(unique(seurat_meta$Clone_IDs_filt)), sparse = T)
rownames(adata_x_clone) = seurat_meta$CBC
colnames(adata_x_clone) = unique(seurat_meta$Clone_IDs_filt)
adata_x_clone[1:5,1:5]
adata_x_clone[,'LS3_Clone_45']
seurat_meta$CBC[seurat_meta$Clone_IDs_filt %in% 'LS3_Clone_45']
#[1] "AAACCCAAGCCATTTG-1_2" "AGTCATGCAAGTTCCA-1_2" "ATCGGCGTCTCTCCGA-1_2" "GAAATGACATATGCGT-1_2"
#[5] "GAGTCTACAAGTGGTG-1_2" "TTTGTTGGTCGTTCAA-1_2"
temp_cellnames = seurat_meta$CBC[seurat_meta$Clone_IDs_filt %in% 'LS3_Clone_45']
adata_x_clone[temp_cellnames, 'LS3_Clone_45'] = 1
sum(adata_x_clone[,'LS3_Clone_45'])
#[1] 6
for (bc in colnames(adata_x_clone)) {
  temp_cellnames = seurat_meta$CBC[seurat_meta$Clone_IDs_filt %in% bc]
  adata_x_clone[temp_cellnames, bc] = 1
}
sum(adata_x_clone)
#[1] 100571
sum(adata_x_clone[,2:length(colnames(adata_x_clone))])
#[1] 72658

#adata.obsm[X_pca]: PCA matrix, shape (n_cell, n_pcs). [Optional. Can be generated in preprocessing.
dim(Embeddings(ls_synthesis_v3_harmony_v3, reduction='pca'))
adata_pca = Embeddings(ls_synthesis_v3_harmony_v3, reduction='pca')

#adata.obsm[X_emb]: two-dimensional embedding, shape (n_cell, 2). [Optional. Can be generated in preprocessing
dim(ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings)
head(ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings)
adata_emb = ls_synthesis_v3_harmony_v3[["umap"]]@cell.embeddings
head(adata_emb)
data.frame(adata_emb)

adata_fp = '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/cospar_data/'
## for raw counts, might have to use SeuratDisk to save the matrix
SaveH5Seurat(ls_v3_unnorm, paste(adata_fp, 'adata_x', sep='')) #this won't actually work, use the thing below!
writeMM(adata_x[,], paste(adata_fp, 'adata_x.mtx', sep=''))
writeMM(t(adata_x[,]), paste(adata_fp, 'adata_x_t.mtx', sep='')) #use this transposed one, which has cells x genes
## cell names
write.csv(colnames(adata_x), paste(adata_fp, 'adata_obs_names.csv', sep=''))
## gene names
write.csv(adata_var_names, paste(adata_fp, 'adata_var_names.csv', sep=''))
## time
write.csv(adata_time_info, paste(adata_fp, 'adata_time_info.csv', sep=''))
## state
write.csv(adata_state_info, paste(adata_fp, 'adata_state_info.csv', sep=''))
## clone
writeMM(adata_x_clone, paste(adata_fp, 'adata_x_clone.mtx', sep=''))
## clone names
write.csv(colnames(adata_x_clone), paste(adata_fp, 'adata_x_clone_names.csv', sep=''))
## pca
write.csv(adata_pca, paste(adata_fp, 'adata_pca.csv', sep=''))
## umap coords
write.csv(adata_emb, paste(adata_fp, 'adata_emb.csv', sep=''))

##alternative method for clonal data without making a matrix: make a two column csv with cell name and clone ID
adata_clone_csv = seurat_meta[,c('CBC', 'Clone_IDs_filt')]
head(adata_clone_csv)
write.csv(adata_clone_csv, paste(adata_fp, 'adata_clone_csv.csv', sep=''))

adata_x_clone[1:30,1:5]
sum(adata_x_clone[,1] ==1)

class(t(adata_x[,]))







## 5/4 new processing
barchart_color_palette = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')
barchart_color_palette = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')
clonal_comp_stacked_barchart_count_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_count[pivot_plot$Identity=="EX"]-pivot_plot$Cell_count[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_count[pivot_plot$Identity=="EX"]-pivot_plot$Cell_count[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_count_func("GW16_1220_S1", "GW16_1220_S1")
clonal_comp_stacked_barchart_count_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                          "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_count_func(c('GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                          'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'
), "All samples >GW20")
clonal_comp_stacked_barchart_count_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                          'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")
clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2"  
), "All samples <GW20")
clonal_comp_stacked_barchart_func(c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S",'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3'
), "All samples GW20-22")
clonal_comp_stacked_barchart_func(c('GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'
), "All samples >GW22")


FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DLX2", "PAX6", "MKI67", "NPY"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR2F1", "NR2F2", "MKI67", "GAD2"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DLX2", "NR2F1", "SCGN", "NR2F2"), raster=F)

ls_in.markers <- FindAllMarkers(ls_in, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_in.markers)
ls_in_clustermarkers_unfilt = as.data.frame(ls_in.markers %>% group_by(cluster))
write.csv(ls_in_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_clustermarkers_v1_unfilt.csv',row.names=F)
ls_in.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_in_clustermarkers = as.data.frame(ls_in.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_in_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_clustermarkers_v1.csv',row.names=F)
DimPlot(ls_in, label=T)
ls_in = SetIdent(ls_in, value='subclust_idents')

colnames(ls_in@meta.data)
table(ls_in@meta.data$subclust_idents)

FeaturePlot(ls_in, features=c("GLI2", "SOX6", "CDK6", "KLHL35"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GLI2", "SOX6", "CDK6", "KLHL35"))

FeaturePlot(ls_in, features=c("LINC02487", "SOX6", "CDK6", "KLHL35"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("LINC02487"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("AFDN"), raster=F)

ls_in = SetIdent(ls_in, value="seurat_clusters")
FeaturePlot(ls_in, features=c("LINC02487", "C1orf61", "RND3", "HMGN2"), label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("LINC02487", "C1orf61", "RND3", "HMGN2"), raster=F)

DotPlot(ls_in, features=unique(c(
  "C1orf61", "RND3", "SH3BGRL3", "HMGN2", "LINC02487", 
  "TUBB3", "PFN2", "CCND2", "RPL10", "ACTG1", "DLX6-AS1",
  "BTG1", "DLX5", "SCGN", "ARL4D", "IGFBP2", "ST18", 
  "C11orf96", "PHLDA1", "FABP7"))) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

DotPlot(ls_synthesis_v3_harmony_v3, features=unique(c(
  "C1orf61", "RND3", "SH3BGRL3", "HMGN2", "LINC02487", 
  "TUBB3", "PFN2", "CCND2", "RPL10", "ACTG1", "DLX6-AS1",
  "BTG1", "DLX5", "SCGN", "ARL4D", "IGFBP2", "ST18", 
  "C11orf96", "PHLDA1", "FABP7"))) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

colnames(ls_synthesis_v3_harmony_v3@meta.data)
unique(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value='subclust_idents')
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2", "DLX2", "MKI67", "SOX6", "PAX6", "NR2F1", "NR2F2", "PROX1", "LHX6"), raster=F, label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2", "DLX2", "MKI67", "SOX6", "PAX6", "NR2F1", "NR2F2", "PROX1", "LHX6"), raster=F, label=F)

FeaturePlot(ls_in, features=c("GAD2", "DLX2", "MKI67", "SOX6", "PAX6", "NR2F1", "NR2F2", "PROX1", "LHX6"), raster=F, label=F)
FeaturePlot(ls_in, features=c("SCGN", "MEIS2", "NR2F1", "PROX1"), raster=F, label=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SCGN", "MEIS2", "NR2F1", "PROX1"), raster=F, label=F)

FeaturePlot(ls_in, features=c("PBX3", "TSHZ1", "PROX1", "SOX6"), raster=F, label=F)


FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("EMX1", "PAX6", "DLX2", "GAD2"), raster=F, label=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TLE4", "HS3ST4", "FOXP2", "TRPM3"), raster=F, label=F)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value='sample_age')
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TLE4", "TRPM3"), pt.size=0)

colnames(ls_synthesis_v3_harmony_v3@meta.data)
colnames(ls_ex@meta.data)
table(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected)
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, 'single', col.name='multi_clone')
ls_synthesis_v3_harmony_v3@meta.data$multi_clone[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1] = 'multi'
ls_synthesis_v3_harmony_v3@meta.data$multi_clone[is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected)] = NA
table(ls_synthesis_v3_harmony_v3@meta.data$multi_clone)
# multi single 
# 23384  49274
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), pt.size=0, split.by='multi_clone')
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), split.by='multi_clone')
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"), pt.size=0, split.by='multi_clone')
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"), split.by='multi_clone')

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"), split.by='multi_clone')
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), split.by='multi_clone')
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value='subclust_idents')
DimPlot(ls_synthesis_v3_harmony_v3, split.by='multi_clone', label=T, raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, split.by='multi_clone', group.by='sample_age', raster=F)

FeaturePlot(subset(ls_synthesis_v3_harmony_v3, multi_clone %in% c('multi')), features=c("TLE4"), split.by='sample_age')
FeaturePlot(subset(ls_synthesis_v3_harmony_v3, multi_clone %in% c('multi')), features=c("TBR1"), split.by='sample_age')


table(ls_synthesis_v3_harmony_v3@meta.data$sample_age)
ls_synthesis_v3_harmony_v3@meta.data$sample_age = factor(ls_synthesis_v3_harmony_v3@meta.data$sample_age, levels = c(16,18,19,20,21,22,23,24))

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2", "DLX2", "OLIG2", "MKI67", "PAX6", "PROX1", "SCGN", "SOX6", "LHX6"), raster=F)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value='subclust_idents')
DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2", "DLX2", "OLIG2", "MKI67", "PAX6", "PROX1", "SCGN", "SOX6", "LHX6"), raster=F)

ls_in = SetIdent(ls_in, value='subclust_idents')
DimPlot(ls_in, label=T, raster=F)


DimPlot(ls_ex, label=T)
FeaturePlot(ls_ex, features=c("EOMES", "TBR1", "SATB2", "BHLHE22"))


#7/12 data analysis
ls_0910 = merge(ls1, c(ls21,
                       ls22,
                       ls23
),
project="ls_0910")

# normalize data
ls_0910 <- NormalizeData(ls_0910, normalization.method = "LogNormalize", scale.factor = 10000)
# scale data based on normalization
all.genes <- rownames(ls_0910)
ls_0910 <- ScaleData(ls_0910, features = all.genes)
#ls_0910 <- ScaleData(ls_0910, features = all.genes, vars.to.regress = "nCount_RNA")
# find variably expressed genes
ls_0910 <- FindVariableFeatures(ls_0910, selection.method = "vst", nfeatures = 2000)
# perform dimensionality reduction
ls_0910 <- RunPCA(ls_0910, features = VariableFeatures(object = ls_0910))
ElbowPlot(ls_0910)
# cluster cells
ls_0910 <- FindNeighbors(ls_0910, dims = 1:15)
ls_0910 <- FindClusters(ls_0910, resolution = .4)
# make graph
ls_0910 <- RunUMAP(ls_0910, dims = 1:15)

DimPlot(ls_0910, label=T)
FeaturePlot(ls_0910, features=c("EOMES", "TBR1", "SATB2", "BHLHE22"))
FeaturePlot(ls_0910, features=c("nCount_RNA"))

ls_0308 = merge(ls1, c(ls34,
                       ls35,
                       ls36
),
project="ls_0308")
# normalize data
ls_0308 <- NormalizeData(ls_0308, normalization.method = "LogNormalize", scale.factor = 10000)
# scale data based on normalization
all.genes <- rownames(ls_0308)
ls_0308 <- ScaleData(ls_0308, features = all.genes)
#ls_0308 <- ScaleData(ls_0308, features = all.genes, vars.to.regress = "nCount_RNA")
# find variably expressed genes
ls_0308 <- FindVariableFeatures(ls_0308, selection.method = "vst", nfeatures = 2000)
# perform dimensionality reduction
ls_0308 <- RunPCA(ls_0308, features = VariableFeatures(object = ls_0308))
ElbowPlot(ls_0308)
# cluster cells
ls_0308 <- FindNeighbors(ls_0308, dims = 1:15)
ls_0308 <- FindClusters(ls_0308, resolution = .4)
# make graph
ls_0308 <- RunUMAP(ls_0308, dims = 1:15)

DimPlot(ls_0308, label=T)
FeaturePlot(ls_0308, features=c("EOMES", "TBR1", "SATB2", "BHLHE22"))
FeaturePlot(ls_0308, features=c("nCount_RNA"))




DimPlot(ls_ex)
DimPlot(ls_ex, cells.highlight = rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>1,]),
        sizes.highlight=0.5, cols.highlight = "#022678") +
  ggtitle('Clone Size > 1')
DimPlot(ls_ex, cells.highlight = rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>1,]),
        sizes.highlight=0.5, cols.highlight = "#022678", split.by="Clone_Index_filt") +
  ggtitle('Clone Size > 1')
DimPlot(subset(ls_ex, Clone_Index_filt %in% c('Index3', 'IndexE')), cells.highlight = rownames(ls_ex@meta.data[ls_ex@meta.data$Clone_size_corrected>1,]),
        sizes.highlight=0.5, cols.highlight = "#022678", split.by="Clone_Index_filt") +
  ggtitle('Clone Size > 1')


##question: is there more coupling between RG and neurons, or RG-IPC-neurons?
length(multicell_clones)
#[1] 3258
dim(multicell_metadata)
#[1] 14030    27
head(multicell_metadata)
table(multicell_metadata$Clone_size_corrected)
#   3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   24   25   26 
#4863 2864 1840 1332  756  480  450  260  308  264   78   70   75   80  119   36   19   40   21   24   25   26

##make a new multicell_metadata to include clones of 2(?)
multicell_metadata2 = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=2,]
multicell_metadata2 = multicell_metadata2[!is.na(multicell_metadata2$Clone_IDs),]
dim(multicell_metadata2)
#[1] 23384    29
length(unique(multicell_metadata2$Clone_IDs))
#[1] 7935
multicell_clones2 = unique(multicell_metadata2$Clone_IDs)
head(multicell_metadata2)
table(multicell_metadata2$Clone_size_corrected)
#   2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   24   25 
#9354 4863 2864 1840 1332  756  480  450  260  308  264   78   70   75   80  119   36   19   40   21   24   25 
#  26 
#  26 

multicell_clone_cellnames = rownames(multicell_metadata)
multicell2_clone_cellnames = rownames(multicell_metadata2)
write.csv(multicell_clone_cellnames, paste(adata_fp, 'multicell3_clone_cellnames.csv', sep=''))
write.csv(multicell2_clone_cellnames, paste(adata_fp, 'multicell2_clone_cellnames.csv', sep=''))
index_info_export = data.frame(rownames(ls_synthesis_v3_harmony_v3@meta.data),ls_synthesis_v3_harmony_v3@meta.data[,'Clone_Index_filt'])
colnames(index_info_export) = c('CBC','Clone_Index')
table(index_info_export$Clone_Index)
write.csv(index_info_export, paste(adata_fp, 'clone_index.csv', sep=''))



DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1,]),
        raster=F, cols.highlight="#022678", split.by="age_pseudo") + 
  ggtitle("Clone size > 1")


## revisiting glial subclustering
colnames(ls_synthesis_v3_harmony_v3@meta.data)
table(ls_synthesis_v3_harmony_v3@meta.data$clust_idents)
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)

ls_glia_v2 = subset(ls_synthesis_v3_harmony_v3, subclust_idents %in% c("GLIA", "GLIA_DIV"))
ls_glia_v2
#27581 features across 11647 samples within 1 assay 
ls_glia_v2 <- ScaleData(ls_glia_v2, features = all.genes)
ls_glia_v2 = FindVariableFeatures(ls_glia_v2, selection.method="vst", nfeatures=2000)
ls_glia_v2 = RunPCA(ls_glia_v2, features=VariableFeatures(object=ls_glia_v2))
ElbowPlot(ls_glia_v2)
#ls_glia_v2 <- RunHarmony(ls_div, c("orig.ident"))
#ls_glia_v2 <- FindNeighbors(ls_glia_v2, reduction = "harmony", dims = 1:15)
ls_glia_v2 <- FindNeighbors(ls_glia_v2, dims = 1:15)
ls_glia_v2 <- FindClusters(ls_glia_v2, resolution = .5)
#ls_glia_v2 <- RunUMAP(ls_glia_v2, reduction = "harmony", dims=c(1:15))
ls_glia_v2 <- RunUMAP(ls_glia_v2, dims=c(1:15))
DimPlot(ls_glia_v2, reduction = "umap",label=T)
DimPlot(ls_glia_v2, reduction = "umap",group.by="orig.ident")
DimPlot(ls_glia_v2, reduction = "umap",group.by="age_pseudo")
DimPlot(ls_glia_v2, reduction = "umap",group.by="sample_age")
FeaturePlot(ls_glia_v2, features=c("VIM", "HOPX", "EOMES", "DLX2", "OLIG2", "NEUROD2", "ID3", "PPP1R17", "MKI67"))
FeaturePlot(ls_glia_v2, features=c("CRYAB"), order=T)
FeaturePlot(ls_glia_v2, features=c("HOPX"), order=T)
VlnPlot(ls_glia_v2, features=c("CRYAB"))
VlnPlot(ls_glia_v2, features=c("HOPX"))
FeaturePlot(ls_glia_v2, features=c("VIM"))
FeaturePlot(ls_glia_v2, features=c("AQP4"))
FeaturePlot(ls_glia_v2, features=c("AQP4","GFAP","VIM","HOPX"))

ls_glia_v2_harmony <- RunHarmony(ls_glia_v2, c("orig.ident"))
ls_glia_v2_harmony <- FindNeighbors(ls_glia_v2_harmony, reduction = "harmony", dims = 1:15)
ls_glia_v2_harmony <- FindClusters(ls_glia_v2_harmony, resolution = .5)
ls_glia_v2_harmony <- RunUMAP(ls_glia_v2_harmony, reduction = "harmony", dims=c(1:15))
DimPlot(ls_glia_v2_harmony, reduction = "umap",label=T)
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="orig.ident")
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="age_pseudo")
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="sample_age")
FeaturePlot(ls_glia_v2_harmony, features=c("VIM", "HOPX", "EOMES", "DLX2", "OLIG2", "NEUROD2", "ID3", "PPP1R17", "MKI67"))
FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB"), order=T, label=T)


ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="sample_age")
VlnPlot(ls_glia_v2_harmony, features="NFIA")
VlnPlot(ls_glia_v2_harmony, features="NFIX")
VlnPlot(ls_glia_v2_harmony, features="NFIB")

ls_glia_v2 = RunPCA(ls_glia_v2, features=VariableFeatures(object=ls_glia_v2), ndims.print=1:15, nfeatures.print = 10)
ls_glia_v2 = CellCycleScoring(ls_glia_v2, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
RidgePlot(ls_glia_v2, features = c("PCNA", "TOP2A", "MCM6", "MKI67"), ncol = 2)
ls_glia_v2 = RunPCA(ls_glia_v2, features=c(s.genes, g2m.genes))
DimPlot(ls_glia_v2)
ls_glia_v2 <- ScaleData(ls_glia_v2, vars.to.regress = c("S.Score", "G2M.Score"), features = rownames(ls_glia_v2))
ls_glia_v2 <- RunPCA(ls_glia_v2, features = VariableFeatures(ls_glia_v2), nfeatures.print = 10)
ls_glia_v2 <- RunPCA(ls_glia_v2, features = c(s.genes, g2m.genes))
DimPlot(ls_glia_v2)
head(ls_glia_v2@meta.data)

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
ls_glia_v2_harmony = CellCycleScoring(ls_glia_v2_harmony, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
head(ls_glia_v2_harmony[[]])
RidgePlot(ls_glia_v2_harmony, features = c("PCNA", "TOP2A", "MCM6", "MKI67"), ncol = 2)
ls_glia_v2_harmony = RunPCA(ls_glia_v2_harmony, features=c(s.genes, g2m.genes))
DimPlot(ls_glia_v2_harmony)
ls_glia_v2_harmony <- ScaleData(ls_glia_v2_harmony, vars.to.regress = c("S.Score", "G2M.Score"), features = rownames(ls_glia_v2_harmony))
ls_glia_v2_harmony <- RunPCA(ls_glia_v2_harmony, features = VariableFeatures(ls_glia_v2_harmony), nfeatures.print = 10)
DimPlot(ls_glia_v2_harmony)
ls_glia_v2_harmony <- FindNeighbors(ls_glia_v2_harmony, reduction = "harmony", dims = 1:15)
ls_glia_v2_harmony <- FindClusters(ls_glia_v2_harmony, resolution = .5)
ls_glia_v2_harmony <- RunUMAP(ls_glia_v2_harmony, reduction = "harmony", dims=c(1:15))
DimPlot(ls_glia_v2_harmony, label=T)
DimPlot(ls_glia_v2_harmony, label=T, group.by="Phase")
#clusters 1, 4, 8, 11 are in G2M or S phase

DimPlot(ls_glia_v2_harmony,group.by="sample_age")
DimPlot(ls_glia_v2_harmony,group.by="age_pseudo")
DimPlot(ls_glia_v2_harmony,group.by="Clone_Index_filt")

FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB", "SNRPA1", "AQP4", "GDPD2"))
FeaturePlot(ls_glia_v2_harmony, features=c("ATP51", "CABLES2", "MICU2", "YPEL3"))
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
VlnPlot(ls_glia_v2_harmony, features=c("CRYAB"))
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="seurat_clusters")
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("CRYAB"))
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("CRYAB"), split.by="Clone_Index_filt")
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")), features=c("HOPX"), split.by="Clone_Index_filt")
DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$Clone_size_filt > 2,]))
DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$Clone_size_filt >= 2,])) +
  ggtitle("Clone size >= 2")
DimPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")), split.by="Clone_Index_filt", cells.highlight = rownames(ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$Clone_size_filt >= 2,])) +
  ggtitle("Clone size >= 2")
head(multicell_full_metadata_unique)
colnames(multicell_full_metadata_unique)
table(multicell_full_metadata_unique$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#  1497          949          362          449 
ls_glia_v2_harmony_meta = ls_glia_v2_harmony@meta.data
left_join(ls_glia_v2_harmony_meta, multicell_full_metadata_unique[,c("Clone_IDs_filt","clonal_restriction","ex_in_presence")])
head(ls_glia_v2_harmony@meta.data)
table(ls_glia_v2_harmony@meta.data$Clone_Index_filt)
#Index3 IndexE 
#4184   4244
length(unique(ls_glia_v2_harmony@meta.data$Clone_IDs_filt))
#[1] 6376
left_join(ls_glia_v2_harmony_meta, multicell_full_metadata_unique[,c("Clone_IDs_filt","clonal_restriction","ex_in_presence"),], by="Clone_IDs_filt")
ls_glia_v2_harmony_meta = left_join(ls_glia_v2_harmony_meta, multicell_full_metadata_unique[,c("Clone_IDs_filt","clonal_restriction","ex_in_presence"),], by="Clone_IDs_filt")
length(unique(ls_glia_v2_harmony_meta$Clone_IDs_filt))
#[1] 6376
ls_glia_v2_harmony = AddMetaData(ls_glia_v2_harmony, ls_glia_v2_harmony_meta$"ex_in_presence", col.name = "ex_in_presence")
ls_glia_v2_harmony = AddMetaData(ls_glia_v2_harmony, ls_glia_v2_harmony_meta$"clonal_restriction", col.name = "clonal_restriction")
table(ls_glia_v2_harmony$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#    776          650          312         1099
table(ls_glia_v2_harmony$clonal_restriction)
#EX_only                IN_only              GLIA_only              OLIG_only 
#    373                      0                    726                      0 
#EX_IN_shared    EX_GLIA/OLIG_shared    IN_GLIA/OLIG_shared EX_IN_GLIA/OLIG_shared 
#           0                    776                    650                    188 
#quadpotent 
#      124
DimPlot(subset(ls_glia_v2_harmony, ex_in_presence %in% c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN")), group.by="ex_in_presence", pt.size=1)
DimPlot(subset(ls_glia_v2_harmony, ex_in_presence %in% c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN")), split.by="Clone_Index_filt", group.by="ex_in_presence", pt.size=1)


head(subset(ls_glia_v2_harmony, ex_in_presence %in% c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN"))@meta.data)
subset(ls_glia_v2_harmony, Clone_IDs_filt %in% c("LS3_Clone_45"))@meta.data
DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$Clone_IDs_filt %in% c("LS3_Clone_45"),]))
sum(is.na(ls_glia_v2_harmony@meta.data$Clone_IDs_filt))
#[1] 3219
sum(is.na(ls_glia_v2_harmony@meta.data$ex_in_presence))
#[1] 8810
sum(is.na(ls_glia_v2_harmony@meta.data$ex_in_presence))
sum(is.na(multicell_full_metadata_unique$ex_in_presence))
head(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data)
table(ls_glia_v2_harmony@meta.data$Clone_size_corrected)
#   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   24 
#3871 1720  935  592  396  271  125  126   70   55   85   52   23    7   14   13   27    7    8    8    6    8 
#25 
# 9 
sum(multicell_full_metadata_unique$Clone_size_corrected>2)
#[1] 3257

## is it possible to look for diff expression in neurogenic vs. non-neurogenic cells
DimPlot(ls_glia_v2_harmony, label=T)
DimPlot(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8,10)))+
  ggtitle("G1 glia with clone size >= 3")
library(ggrepel)
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="ex_in_presence")
levels(ls_glia_v2_harmony)
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8,10)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Glial-only glia (left) and EX neurogenic glia (right) -- Glia, clone size >=3, G1 cells only")+
  theme(plot.title = element_text(hjust = 0.5))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
levels(ls_glia_v2_harmony)
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8,10)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index3-labeled glia (left) and IndexE-labeled glia (right) -- Glia, clone size >=3, G1 cells only")+
  theme(plot.title = element_text(hjust = 0.5))

ggplot(ls_glia_v2_harmony@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="seurat_clusters")
ls_glia_v2_harmony.markers <- FindAllMarkers(ls_glia_v2_harmony, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_glia_v2_harmony.markers)
ls_glia_v2_harmony_clustermarkers_unfilt = as.data.frame(ls_glia_v2_harmony.markers %>% group_by(cluster))
write.csv(ls_glia_v2_harmony_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_glia_v2_harmony_clustermarkers_v1_unfilt.csv',row.names=F)
ls_glia_v2_harmony.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_glia_v2_harmony_clustermarkers = as.data.frame(ls_glia_v2_harmony.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_glia_v2_harmony_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_glia_v2_harmony_clustermarkers_v1.csv',row.names=F)
ls_glia_v2_harmony_clustermarkers

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="sample_age")
VlnPlot(subset(ls_glia_v2_harmony, sample_age %in% c(16,19,20,22,23,24)), features=c("EOMES"))
VlnPlot(subset(ls_glia_v2_harmony, sample_age %in% c(16,19,20,22,23,24)), features=c("HOPX"))
VlnPlot(subset(ls_glia_v2_harmony, sample_age %in% c(16,19,20,22,23,24)), features=c("CRYAB"))
VlnPlot(subset(ls_glia_v2_harmony, sample_age %in% c(16,19,20,22,23,24)), features=c("AQP4"))
VlnPlot(subset(ls_glia_v2_harmony, sample_age %in% c(16,19,20,22,23,24)), features=c("GJA1"))
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="seurat_clusters")
FeaturePlot(ls_glia_v2_harmony, features=c("GJA1", "ID3", "AQP4", "CRYAB"), label=T)
FeaturePlot(ls_glia_v2_harmony, features=c("CD44", "SPARCL1", "ITGB4", "ANGPTL4"), label=T)
FeaturePlot(ls_glia_v2_harmony, features=c("CDON", "GJA1"), label=T, pt.size=1)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CDON"), label=T, raster=F)
FeaturePlot(ls_glia_v2_harmony, features=c("HES1", "HES5", "HES6", "EOMES"), label=T, pt.size=1)


ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
levels(ls_glia_v2_harmony)
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,2,5,7)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index3-labeled glia (left) and IndexE-labeled glia (right) -- Glia, clone size >=3, Clusters 0,2,5,7")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index3-labeled glia (left) and IndexE-labeled glia (right) -- Glia, clone size >=3, Clusters 0")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,2,5,7) & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index3-labeled glia (left) and IndexE-labeled glia (right) -- Glia, clone size >=3, Clusters 0,2,5,7, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,5,7,8) & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index3-labeled glia (left) and IndexE-labeled glia (right) -- Glia, clone size >=3, Clusters 0,1,5,7,8, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="ex_in_presence")
levels(ls_glia_v2_harmony)
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8) & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia glia (right) -- Glia, clone size >=3, Clusters 0,1,5,7,8, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))



FeaturePlot(ls_synthesis_v3_harmony_v3, features=c(
  "VIM", "EOMES", "SATB2", "TBR1",
  "DLX2", "GAD2", "NR2F2", "NXPH1",
  "MKI67", "GFAP", "OLIG2", "MBP"
), raster=T)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="broad_celltype", raster=F, label=T)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="broad_celltype", raster=F, label=T, cols=barchart_color_palette)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents", raster=F, label=T)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="Clone_Index_filt", raster=F, cols=c(osvz_col, vz_col))
DimPlot(ls_synthesis_v3_harmony_v3, group.by="Clone_Index_filt", split.by="Clone_Index_filt", raster=F, cols=c(osvz_col, vz_col))

ls_synthesis_v3_harmony_v3@meta.data$subclust_idents = factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents,
                                                              levels=c("GLIA_DIV", "GLIA", "RG_EX", 
                                                                       "IPC_EX_DIV", "IPC_EX", "EX_EARLY", "EX",
                                                                       "IPC_IN_DIV", "IPC_IN", "IN_local",
                                                                       "OPC_DIV", "OPC", "OLIG",
                                                                       "IN_CGE", "IN_MGE", "IN_UNK"))
ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents %in% c("GLIA_DIV", "GLIA", "IPC_EX_DIV", "IPC_EX", "EX_EARLY", "EX","IPC_IN_DIV", "IPC_IN", "IN_local","OPC_DIV", "OPC", "OLIG","IN_CGE", "IN_MGE"))
       @meta.data, aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents %in% c("GLIA_DIV", "GLIA", "IPC_EX_DIV", "IPC_EX", "EX_EARLY", "EX","IPC_IN_DIV", "IPC_IN", "IN_local","OPC_DIV", "OPC", "OLIG","IN_CGE", "IN_MGE") & Clone_Index_filt %in% c("Index3", "IndexE"))
@meta.data, aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=subclust_idents, fill=sample_age)) + geom_bar(position = "fill") +
  ggtitle('Sample contribution to clusters') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS35_Clone_78",])
        , raster=F, cols.highlight=vz_col, sizes.highlight=6, label=T) + 
  ggtitle("LS35_Clone_78")
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS35_Clone_104",])
        , raster=F, cols.highlight=osvz_col, sizes.highlight=6, label=T) + 
  ggtitle("LS35_Clone_104")

ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS35_Clone_104",]

FeaturePlot(ls_in, features=c("ADARB2", "NR2F1", "LHX6", "SOX6"), label=T)
FeaturePlot(ls_in, features=c("PAX6", "NPY", "NR2F1", "SOX6"), label=T)
VlnPlot(ls_in, features=c("PAX6"))
VlnPlot(ls_in, features=c("NPY"))
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("PAX6"), group.by="subclust_idents", pt.size=0)

FeaturePlot(ls_in, features=c("NKX2-1", "ASCL1", "GSX2", "PAX6"), label=T)
FeaturePlot(ls_in, features=c("GSX2", "BCL11B", "NKX2-1", "LHX6"), label=T)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NKX2-1", "ASCL1", "GSX2", "PAX6"), label=T, raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("GSX2"), pt.size=0, group.by="subclust_idents")
VlnPlot(ls_in, features=c("GSX2"), pt.size=0.1, group.by="orig.ident")
VlnPlot(subset(ls_synthesis_v3_harmony_v3, broad_celltype %in% c("IN")), features=c("GSX2"), pt.size=0, group.by="sample_age")

DimPlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), group.by="ex_in_presence")
FeaturePlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), split.by="ex_in_presence", features=c("GSX2"))
VlnPlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), split.by="ex_in_presence", features=c("GSX2"))

FeaturePlot(ls_in, features=c("ETV1", "NKX6-2", "GSX2", "ISL1"), label=T)
FeaturePlot(ls_in, features=c("SP9", "SOX6", "NPY", "TOX3"), label=T)


subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE") & ex_in_presence %in% "EX_IN_shared")@meta.data
ls_in@meta.data$ex_in_presence[is.na(ls_in@meta.data$Clone_size_corrected)] = NA
DimPlot(subset(ls_in, ex_in_presence %in% c("IN_only", "EX_IN_shared")), group.by="ex_in_presence")
DimPlot(subset(ls_in, Clone_size_corrected>=3 & ex_in_presence %in% c("EX_IN_shared", "IN_only")), group.by="ex_in_presence")

subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE") &
         ex_in_presence %in% c("EX_IN_shared") &
         Clone_size_corrected >=3)@meta.data$Clone_IDs_filt

mge_cge_en_shared_clones = subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE") &
                                    ex_in_presence %in% c("EX_IN_shared") &
                                    Clone_size_corrected >=3)@meta.data$Clone_IDs_filt
mge_cge_en_shared_clones

clone="LS21_Clone_256"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)
clone="LS21_Clone_30"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)
clone="LS22_Clone_47"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)
clone="LS23_Clone_206"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)

DimPlot(subset(ls_in, Clone_size_corrected>=3 &
                 ex_in_presence %in% c("EX_IN_shared", "IN_only") &
                 read_counts_filt > 500), group.by="ex_in_presence")
subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE") &
         ex_in_presence %in% c("EX_IN_shared") &
         Clone_size_corrected >=3 &
         read_counts_filt > 500)@meta.data$Clone_IDs_filt
#[1] "LS21_Clone_256" "LS35_Clone_189" "LS36_Clone_140"
clone="LS21_Clone_256"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)
clone="LS35_Clone_189"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("sample_age", "seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)
clone="LS36_Clone_140"
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),c("sample_age", "seurat_clusters", "Barcode_UMI_Count_filt", "read_counts_filt", "subclust_idents")]
DimPlot(ls_synthesis_v3_harmony_v3, 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% c(clone),]), raster=F) +
  ggtitle(clone)

DimPlot(subset(ls_in, read_counts_filt > 500), 
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >3,]),
        split.by="Clone_Index_filt")
DimPlot(ls_in, group.by="subclust_idents", label=T)        


FeaturePlot(ls_in, features=c("SCGN", "SOX6", "PROX1", "PAX6"), label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SCGN", "SOX6", "PROX1", "PAX6"), raster=F)
FeaturePlot(ls_in, features=c("DLX2"), label=T)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MKI67", "DLX2", "PAX6"), raster=F)


DimPlot(ls_glia_v2_harmony, group.by="seurat_clusters", label=T)
ls_glia_v2_harmony$glial_subtype = NA
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(3)] = "S_phase_div_glia"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(4,9,6)] = "G2M_phase_div_glia"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(2)] = "ASTRO"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(1)] = "Transitioning OPCs/IPCs"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(8)] = "Transitioning IPCs"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(0)] = "RG"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(5)] = "RG (putative tRG)"
ls_glia_v2_harmony$glial_subtype[ls_glia_v2_harmony$seurat_clusters %in% c(7)] = "RG (putative oRG)"

DimPlot(ls_glia_v2_harmony, group.by="glial_subtype", label=T, pt.size=0.7)
FeaturePlot(ls_glia_v2_harmony, features=c("PTPRZ1", "HOPX", "FAM107A", "EGFR"))
FeaturePlot(ls_glia_v2_harmony, features=c("VIM", "HOPX", "CRYAB", 
                                           "MKI67", "EOMES", "OLIG2",
                                           "AQP4", "GJA1", "FAM107A"))
ls_glia_v2_harmony$glial_subtype = factor(ls_glia_v2_harmony$glial_subtype, levels = c("S_phase_div_glia", "G2M_phase_div_glia", "RG", "RG (putative tRG)", "RG (putative oRG)", "Transitioning IPCs", "Transitioning OPCs/IPCs", "ASTRO"))
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="glial_subtype")
ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & glial_subtype %in% c("S_phase_div_glia", "G2M_phase_div_glia", "ASTRO", "Transitioning OPCs/IPCs", "Transitioning IPCs", "RG", "RG (putative tRG)", "RG (putative oRG)"))@meta.data, aes(x=glial_subtype, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
DimPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")), group.by="Clone_Index_filt", pt.size=1, cols=c(osvz_col, vz_col))

FeaturePlot(ls_glia_v2_harmony, features=c("TAGLN2", "FABP7", "CDON"))


#ERBB4: neuregulin receptor, directs migration to ctx
#EPHA5: migration
#ROBO2: interneuron migration marker
#TSHZ1 and FOXP2: OB markers
#PROX1: kind of promiscuous
#HTR3A: CGE in mouse, not super specific but should be expressed
#PDZRN3: putative CGE


FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("ROBO2", "ERBB4"),raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DCX"),raster=F)

FeaturePlot(ls_in, features=c("ERBB4", "ROBO2", "ZEB2", "LHX6"))
FeaturePlot(ls_in, features=c("PDZRN3", "HTR3A"))

#CGE markers
FeaturePlot(ls_in, features=c("PID1", "EYS", "SHISA9", "LUZP2"))
FeaturePlot(ls_in, features=c("EPHA5", "NR2F1", "CNTN5", "ZEB1"))

FeaturePlot(ls_in, features=c("EPHA5", "ZEB2", "ERBB4", "ZEB1"))

FeaturePlot(ls_in, features=c("NR2F2", "SHISA9", "CNTN5", "MAML3"))


#LGE markers
FeaturePlot(ls_in, features=c("PCDH9", "DCC", "ETV1", "TOX3"))
FeaturePlot(ls_in, features=c("NRP2", "TSHZ1", "FOXP2", "GSX2"))
FeaturePlot(ls_in, features=c("SCGN"))
FeaturePlot(ls_in, features=c("BTG1", "DCLK2"))



FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VAX1"),raster=F)
FeaturePlot(ls_glia_v2_harmony, features=c("VAX1"),raster=F)
FeaturePlot(ls_glia_v2_harmony, features=c("ZFP36", "BEX4", "LGALS3", "ANXA2"),raster=F)
FeaturePlot(ls_glia_v2_harmony, features=c("CLMN", "ATP13A4", "EPAS8", "PCDH7", "SYNE1"),raster=F)
#putative mouse ventral
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("ADGR13", "SLIT2", "PTPRN2", "RBMS1", "SNTB1"),raster=F)
#putative mouse dorsal
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("RLBP1", "GM29260", "PAX6", "DCC"),raster=F)




## 9/18/23
## redoing a lot of the figures for colors and just to make them again...

#barcharts for index inclusion in different clusters
load('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/20230401_local_STICR_synthesis_v3_harmony_v4.RData')
osvz_col = "#FF6666" #OSVZ (Index 3) is a light purple
vz_col = "#2128eb" #VZ (Index E) is a dark blue
DimPlot(ls_synthesis_v3_harmony_v3, group.by="Clone_Index_filt", raster=FALSE)
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=seurat_clusters, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="subclust_idents")
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))

#plotting STICR labeled cells and clone sizes on UMAP
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='IndexE'], cols.highlight=vz_col) +
  ggtitle('Index E') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt=='Index3'], cols.highlight=osvz_col) +
  ggtitle('Index 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, group.by = "Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=1) +
  ggtitle('STICR-labeled cells') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, split.by = "Clone_Index_filt")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt==1], cols.highlight='black') +
  ggtitle('Clone Size == 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>1], cols.highlight='#2dba3b') +
  ggtitle('Clone Size > 1') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>3], cols.highlight='#1d8027') +
  ggtitle('Clone Size > 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>5], cols.highlight='#0f4515') +
  ggtitle('Clone Size > 5') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(subset(ls_synthesis_v3_harmony_v3, Clone_Index_filt %in% c('IndexE','Index3')), raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_filt>5], cols.highlight='#0f4515', split.by="Clone_Index_filt") +
  ggtitle('Large clones (>5) by index') +
  theme(plot.title = element_text(hjust = 0.5))

#stacked barcharts of clonal composition with new color palette
library(RColorBrewer)
brewer.pal(4,'Set3')
#[1] "#8DD3C7" "#FFFFB3" "#BEBADA" "#FB8072"
#barchart_color_palette = c('#ED936B', '#FFFDBB', '#629FC7', '#7DBF73')
barchart_color_palette = brewer.pal(4,'Set3')
clonal_comp_stacked_barchart_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")

#stacked barcharts with individual cells, not percentages -- ordered by clone size
barchart_color_palette = brewer.pal(4,'Set3')
clonal_comp_stacked_barchart_counts_func = function(current_sample, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/4
  length(pivot_plot[,1])/4
  clone_order = pivot_plot[order(multicell_metadata_unique$Clone_size_corrected[multicell_metadata_unique$orig.ident %in% current_sample & multicell_metadata_unique$Clone_Index_filt %in% "IndexE"], decreasing = T),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_col(position='stack', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$orig.ident %in% c(current_sample)
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/4
  length(pivot_plot[,1])/4
  clone_order = pivot_plot[order(multicell_metadata_unique$Clone_size_corrected[multicell_metadata_unique$orig.ident %in% current_sample & multicell_metadata_unique$Clone_Index_filt %in% "Index3"], decreasing = T),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_col(position='stack', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_counts_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2", 'GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S', "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples")

clonal_comp_stacked_barchart_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                    "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_counts_func(c("GW16_1220_S1", "GW16_1220_S1", "GW19_0221_S1", "GW19_0221_S2",  
                                    "GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S"), "All samples <=GW20")
clonal_comp_stacked_barchart_counts_func(c('GW21_0502_1S','GW22_1118_S1','GW22_1118_S2','GW22_1118_S3',
                                    'GW23_0910_S1','GW23_0910_S2','GW23_0910_S4','GW24_0221_1S'), "All samples >GW20")

#boxplots of clonal competency by index or age
ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clone index')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=EX_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of EX cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=IN_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=GLIA_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of glial/progenitor cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')
ggplot(compiled_clust_df_idents_jitter, aes(x=as.factor(sample_age), y=OLIG_pct, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of OPCs/oligos in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

#barcharts of clonal potency by age and index
ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index, GW', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

#barcharts of broad clonal potency (ex/in presence) by age and index
ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=factor(sample_age))) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(16,19,20)
samp_description="<=GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

samp_age=c(22,23,24)
samp_description=">GW20"
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$sample_age %in% c(samp_age),], aes(x=ex_in_presence, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle(paste('Clonal potency by index,', as.character(samp_description))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

#revisiting glia subclustering story
ls_glia
DimPlot(ls_glia_v2_harmony, label=T)
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="age_pseudo", pt.size=1)
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="seurat_clusters", pt.size=1, label=T)
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="seurat_clusters", pt.size=1, label=T)
DimPlot(ls_glia_v2_harmony, reduction = "umap",group.by="glial_subtype", pt.size=1, label=T)

ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data, aes(x=glial_subtype, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="ex_in_presence")
levels(ls_glia_v2_harmony)
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8) & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia glia (right) -- Glia, clone size >=3, Clusters 0,1,5,7,8, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))
write.csv(de.markers.glia, '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/neurogenic_gliogenic_DEGs_v1.csv')

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia glia (right) -- Glia, clone size >=3, All subclusters, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "IN_only")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between IN-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, All subclusters, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & glial_subtype %in% c('RG','RG (putative tRG)', 'RG (putative oRG)') & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "IN_only")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between IN-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & glial_subtype %in% c('RG','RG (putative tRG)', 'RG (putative oRG)') & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, glial_subtype %in% c('ASTRO') & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived Index3 (left) and VZ_derived IndexE (right) -- Glia, Astrocytes only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c('IndexE','Index3')), features=c("ANGPTL4","ITGB4"))
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c('IndexE','Index3') & glial_subtype %in% c('ASTRO','RG','RG (putative tRG)', 'RG (putative oRG)')), features=c("ANGPTL4","ITGB4"))
VlnPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c('IndexE','Index3') & glial_subtype %in% c('ASTRO','RG','RG (putative tRG)', 'RG (putative oRG)')), features=c("ITGB4","TMEM158","MGMT","CELSR1"), ncol=2)

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="age_pseudo")
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, sample_age %in% c(16, 18, 19, 20, 21, 22, 23, 24) & seurat_clusters %in% c(0,1,2,5,7,8)), ident.1 = ">GW20", ident.2 = "<=GW20")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between <=GW20 (left) and >GW20 (right) -- Glia, subclusters 0,1,2,5,7,8")+
  theme(plot.title = element_text(hjust = 0.5))

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, sample_age %in% c(20)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived Index3 (left) and VZ_derived IndexE (right) -- Glia, all subclusters, GW20 only")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, sample_age %in% c(21,22,23,24)), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived Index3 (left) and VZ_derived IndexE (right) -- Glia, all subclusters, >GW20 samples only")+
  theme(plot.title = element_text(hjust = 0.5))



ls_ex
DimPlot(ls_ex)



ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=subclust_idents, fill=age_pseudo)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% c('<=GW20'),], aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters, <=GW20') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% c('>GW20'),], aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters, >GW20') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))

ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% c('<=GW20'),], aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters, <=GW20') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% c('>GW20'),], aes(x=1, fill=Clone_Index_filt)) + geom_bar(position = "fill", width = 1) +
  xlim(0,14) +
  ggtitle('Overall index contribution to clusters, >GW20') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))


#what if we randomly shuffle VZ/OSVZ identity
ls_meta_copy = ls_synthesis_v3_harmony_v3@meta.data
ls_meta_copy$Clone_Index_shuffled=NA
ls_meta_copy$Clone_Index_shuffled[ls_meta_copy$Clone_Index_filt %in% c("IndexE","Index3")] = sample(ls_meta_copy$Clone_Index_filt[ls_meta_copy$Clone_Index_filt %in% c("IndexE","Index3")])
table(ls_meta_copy$Clone_Index_filt)
#Index3 IndexE 
#34915  37743
table(ls_meta_copy$Clone_Index_shuffled)
#Index3 IndexE 
#34915  37743
ls_meta_copy$Clone_Index_filt[20000:20010]
# [1] "Index3" "IndexE" "Index3" "Index3" NA       "Index3" "Index3" "Index3" "Index3" "Index3" "Index3"
ls_meta_copy$Clone_Index_shuffled[20000:20010]
# [1] "IndexE" "IndexE" "IndexE" "Index3" NA       "Index3" "IndexE" "Index3" "Index3" "Index3" "Index3"
sum(rownames(ls_synthesis_v3_harmony_v3@meta.data) == rownames(ls_meta_copy))
#[1] 100571
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3'),], aes(x=subclust_idents, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))
ggplot(ls_meta_copy[ls_meta_copy$Clone_Index_shuffled %in% c('IndexE','Index3'),], aes(x=subclust_idents, fill=Clone_Index_shuffled)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters -- single random shuffle') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x=element_text(angle=45, hjust=1))


DimPlot(ls_ex)
FeaturePlot(ls_ex, features="nCount_RNA", max.cutoff="q90")
FeaturePlot(ls_synthesis_v3_harmony_v3, features="nCount_RNA", max.cutoff="q90", raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="age_pseudo", raster=F)


#trying to use SCENIC?
devtools::install_github("aertslab/SCENIC@v1.1.2")
library(SCENIC)
setwd("/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/230919_SCENIC")
exprMat=ls_glia_v2_harmony@assays$RNA[0:5,0:5]
cellInfo = ls_glia_v2_harmony$glial_subtype
cellInfo=data.frame(cellInfo)
colnames(cellInfo) = "CellType"
dir.create("int")
saveRDS(cellInfo, file="int/cellInfo.Rds")
DimPlot(ls_glia_v2_harmony, group.by="glial_subtype", label=T)
colVars <- list(CellType=c("S_phase_div_glia"="forestgreen", 
                           "G2M_phase_div_glia-mural"="darkorange", 
                           "RG"="magenta4", 
                           "RG (putative tRG)"="hotpink", 
                           "RG (putative oRG)"="red3", 
                           "Transitioning IPCs"="skyblue", 
                           "Transitioning OPCs/IPCs"="darkblue",
                           "ASTRO"="turquoise",
                           "NA"="lightgrey"))
colVars$CellType <- colVars$CellType[intersect(names(colVars$CellType), cellInfo$CellType)]
saveRDS(colVars, file="int/colVars.Rds")
plot.new(); legend(0,1, fill=colVars$CellType, legend=names(colVars$CellType))

org <- "hgnc" # or hgnc, or dmel
dbDir <- "/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/230319_SCENIC/cisTarget_databases" # RcisTarget databases location
dbDir <- "/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/230919_SCENIC"
myDatasetTitle <- "230919_SCENIC_ls_glia_harmony_v2" # choose a name for your analysis
data(defaultDbNames)
dbs <- defaultDbNames[[org]]
scenicOptions <- initializeScenic(org=org, dbDir=dbDir, dbs=dbs, datasetTitle=myDatasetTitle, nCores=10) 

dbs = c("500bp"="hg19-500bp-upstream-7species.mc9nr.genes_vs_motifs.rankings.feather",
             "10kb"="hg19-tss-centered-10kb-7species.mc9nr.genes_vs_motifs.rankings.feather")

dbs

dbFiles <- c("https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-500bp-upstream-7species.mc9nr.feather",
             "https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-tss-centered-10kb-7species.mc9nr.feather")
# dir.create("cisTarget_databases"); setwd("cisTarget_databases") # if needed
for(featherURL in dbFiles)
{
  download.file(featherURL, destfile=basename(featherURL)) # saved in current dir
}


## 9/25/23
#check clonal output of tRG and oRG
DimPlot(ls_glia_v2_harmony, label=T)
colnames(ls_glia_v2_harmony@meta.data)
tRG_clone_list = ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$glial_subtype %in% 'RG (putative tRG)', 'Clone_IDs_filt']
tRG_clone_list = unique(tRG_clone_list)
oRG_clone_list = ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$glial_subtype %in% 'RG (putative oRG)', 'Clone_IDs_filt']
oRG_clone_list = unique(oRG_clone_list)

library('RColorBrewer')
barchart_color_palette = brewer.pal(4,'Set3')
clonal_comp_stacked_barchart_func_by_clones = function(clone_list, chart_title) {
  index_filter = clone_cluster_pivot_idents$Clone_Index=="IndexE" & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idxE_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"Index E"))
  index_filter = clone_cluster_pivot_idents$Clone_Index=="Index3" & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[index_filter,]
  n_idx3_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, "Index 3"))
  total_clones = n_idxE_clones+n_idx3_clones
  pct_idxE_clones = n_idxE_clones/total_clones
  pct_idx3_clones = n_idx3_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_idxE_clones, 
                       pct_idx3_clones)
  )
}
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func_by_clones(c(tRG_clone_list, oRG_clone_list), "tRG- and oRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones(tRG_clone_list, "tRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones(oRG_clone_list, "oRG-containing clones")

barchart_color_palette = brewer.pal(4,'Set3')
clonal_comp_stacked_barchart_func_by_clones_indices_combined = function(clone_list, chart_title) {
  clone_filter = clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[clone_filter,]
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(chart_title)
}
clonal_comp_stacked_barchart_func_by_clones_indices_combined(c(tRG_clone_list, oRG_clone_list), "tRG- and oRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones_indices_combined(tRG_clone_list, "tRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones_indices_combined(oRG_clone_list, "oRG-containing clones")

head(compiled_clust_df_idents)
compiled_clust_df_idents$tRG_oRG_clone = NA
compiled_clust_df_idents$tRG_oRG_clone[compiled_clust_df_idents$Clone_IDs %in% tRG_clone_list] = 'tRG'
compiled_clust_df_idents$tRG_oRG_clone[compiled_clust_df_idents$Clone_IDs %in% oRG_clone_list] = 'oRG'

#need to fix this characterization -- previously, GLIA/OLIG clones were being called EX_only :(
head(compiled_clust_df_idents)
compiled_clust_df_idents$clonal_restriction = NA
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN == 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & apply(compiled_clust_df_idents[,c("GLIA","OLIG")], 1, sum) != 0] = "EX_IN_GLIA/OLIG_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "EX_IN_shared"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "EX_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG == 0] = "IN_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG == 0] = "GLIA_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0 & compiled_clust_df_idents$GLIA == 0 & compiled_clust_df_idents$OLIG != 0] = "OLIG_only"
compiled_clust_df_idents$clonal_restriction[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0 & compiled_clust_df_idents$GLIA != 0 & compiled_clust_df_idents$OLIG != 0] = "quadpotent"
compiled_clust_df_idents$clonal_restriction = factor(compiled_clust_df_idents$clonal_restriction, levels=(
  c("EX_only", "IN_only", "GLIA_only", "OLIG_only", "GLIA/OLIG_shared", "EX_IN_shared", "EX_GLIA/OLIG_shared", "IN_GLIA/OLIG_shared", "EX_IN_GLIA/OLIG_shared", "quadpotent")
))
table(compiled_clust_df_idents$clonal_restriction)
#EX_only                IN_only              GLIA_only              OLIG_only 
#    939                    411                    226                     54 
#GLIA/OLIG_shared           EX_IN_shared    EX_GLIA/OLIG_shared    IN_GLIA/OLIG_shared 
#             169                    148                    558                    538 
#EX_IN_GLIA/OLIG_shared             quadpotent 
#                   160                     54 

compiled_clust_df_idents$ex_in_presence = NA
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN == 0] = "EX_only"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN != 0] = "IN_only"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX == 0 & compiled_clust_df_idents$IN == 0] = "no_EX_or_IN"
compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$EX != 0 & compiled_clust_df_idents$IN != 0] = "EX_IN_shared"
table(compiled_clust_df_idents$ex_in_presence)
compiled_clust_df_idents$ex_in_presence = factor(compiled_clust_df_idents$ex_in_presence, levels=(
  c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN")
))
table(compiled_clust_df_idents$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#   1497          949          362          449

osvz_col = "#FF6666" #OSVZ (Index 3) is salmon
vz_col = "#2128eb" #VZ (Index E) is a dark blue
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG'),], aes(x=clonal_restriction, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG'),], aes(x=ex_in_presence, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

young_sample_list = unique(ls_synthesis_v3_harmony_v3@meta.data$orig.ident[ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% '<=GW20'])
old_sample_list = unique(ls_synthesis_v3_harmony_v3@meta.data$orig.ident[ls_synthesis_v3_harmony_v3@meta.data$age_pseudo %in% '>GW20'])

clonal_comp_stacked_barchart_func_by_clones_age_sep = function(clone_list, chart_title) {
  age_filter = clone_cluster_pivot_idents$orig.ident %in% young_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_young_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"<=GW20"))
  age_filter = clone_cluster_pivot_idents$orig.ident %in% old_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_old_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, ">GW20"))
  total_clones = n_young_clones+n_old_clones
  pct_young_clones = n_young_clones/total_clones
  pct_old_clones = n_old_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_young_clones, 
                       pct_old_clones)
  )
}
library(ggplot2)
library(ggpubr)

clonal_comp_stacked_barchart_func_by_clones_age_sep(c(tRG_clone_list, oRG_clone_list), "tRG- and oRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones_age_sep(tRG_clone_list, "tRG-containing clones")
clonal_comp_stacked_barchart_func_by_clones_age_sep(oRG_clone_list, "oRG-containing clones")

ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG') & compiled_clust_df_idents$orig.ident %in% young_sample_list,], aes(x=clonal_restriction, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone, <=GW20 samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG') & compiled_clust_df_idents$orig.ident %in% old_sample_list,], aes(x=clonal_restriction, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone, >GW20 samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG') & compiled_clust_df_idents$orig.ident %in% young_sample_list,], aes(x=ex_in_presence, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone, <=GW20 samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggplot(compiled_clust_df_idents[compiled_clust_df_idents$tRG_oRG_clone %in% c('tRG', 'oRG') & compiled_clust_df_idents$orig.ident %in% old_sample_list,], aes(x=ex_in_presence, fill=tRG_oRG_clone, group=tRG_oRG_clone)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by tRG/oRG membership in clone, >GW20 samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

colnames(ls_glia_v2_harmony@meta.data)
DimPlot(subset(ls_glia_v2_harmony, ex_in_presence %in% c('EX_only', 'IN_only', 'EX_IN_shared', 'no_EX_or_IN')), group.by='ex_in_presence', pt.size=2)
ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & ex_in_presence %in% c('EX_only', 'IN_only', 'EX_IN_shared', 'no_EX_or_IN'))@meta.data, aes(x=glial_subtype, fill=ex_in_presence)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data, aes(x=glial_subtype, fill=ex_in_presence)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))

## trying some pseudotime with monocle
## first going to try to go in and map the new glial subcluster identitites
DimPlot(ls_synthesis_v3_harmony_v3, raster=F)
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
table(ls_div$subclust_idents)
#  GLIA_DIV IPC_EX_DIV IPC_IN_DIV    OPC_DIV 
#      2303       2060       2334        543
table(ls_in$subclust_idents)
#    IN_CGE   IN_local     IN_MGE     IN_UNK IPC_IN_DIV 
#      2860       5445       2621        828        271
table(ls_glia_v2_harmony$glial_subtype)
#S_phase_div_glia      G2M_phase_div_glia                      RG       RG (putative tRG) 
#            1359                    2374                    2412                    1068 
#RG (putative oRG)      Transitioning IPCs Transitioning OPCs/IPCs                   ASTRO 
#              802                     725                    1439                    1368
table(ls_ex$subclust_idents)
#      EX EX_early   IPC_EX 
#   49657     7235     8636 
## compile all subclust_idents and add to the main object
all_subclust_idents = data.frame(subclust_idents = ls_glia_v2_harmony$glial_subtype)
all_subclust_idents = rbind(all_subclust_idents, data.frame(subclust_idents = ls_in$subclust_idents))
all_subclust_idents = rbind(all_subclust_idents, data.frame(subclust_idents = ls_ex$subclust_idents))
all_subclust_idents = rbind(all_subclust_idents, data.frame(subclust_idents = ls_div$subclust_idents))
all_subclust_idents$CBC = rownames(all_subclust_idents)
colnames(all_subclust_idents) = c("subclust_idents_v2", "CBC")
head(all_subclust_idents)
dim(all_subclust_idents)
seurat_meta = ls_synthesis_v3_harmony_v3@meta.data
dim(seurat_meta)
length(unique(all_subclust_idents)$CBC)
table(all_subclust_idents$subclust_idents_v2)
table(seurat_meta$subclust_idents)
colnames(seurat_meta)
seurat_meta$CBC = rownames(seurat_meta)
tail(left_join(seurat_meta, all_subclust_idents))
seurat_meta = left_join(seurat_meta, all_subclust_idents)
sum(is.na(seurat_meta$subclust_idents_v2))
seurat_meta$subclust_idents_v2[seurat_meta$subclust_idents %in% 'OPC'] = 'OPC'
seurat_meta$subclust_idents_v2[seurat_meta$subclust_idents %in% 'OLIG'] = 'OLIG'
seurat_meta$subclust_idents_v2[seurat_meta$subclust_idents %in% 'RG_EX'] = 'RG_UNK'
table(seurat_meta$subclust_idents_v2)

rownames(seurat_meta) = seurat_meta$CBC
length(rownames(seurat_meta))
sum(rownames(seurat_meta) == rownames(ls_synthesis_v3_harmony_v3@meta.data))
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, seurat_meta$subclust_idents_v2, col.name = 'subclust_idents_v2')
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)
#ASTRO                      EX                EX_early      G2M_phase_div_glia 
#1368                   49657                    7235                    2374 
#IN_CGE                IN_local                  IN_MGE                  IN_UNK 
#2860                    5445                    2621                     828 
#IPC_EX              IPC_EX_DIV              IPC_IN_DIV                    OLIG 
#8636                    2060                    2605                     755 
#OPC                 OPC_DIV                      RG       RG (putative oRG) 
#2249                     543                    2412                     802 
#RG (putative tRG)        S_phase_div_glia      Transitioning IPCs Transitioning OPCs/IPCs 
#1068                    1359                     725                    1439 
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2))
#[1] 3530
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents == 'IPC_IN'] = 'IPC_IN'
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2))
#[1] 2338
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)])
#GLIA_DIV       GLIA      RG_EX IPC_EX_DIV     IPC_EX   EX_EARLY         EX IPC_IN_DIV     IPC_IN 
#       1         99          0          0        709          0          0          0          0 
#IN_local    OPC_DIV        OPC       OLIG     IN_CGE     IN_MGE     IN_UNK 
#       0          0          0          0          0          0          0
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents == 'IPC_EX'] = 'IPC_EX'
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2))
#[1] 1629
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)]=rep("unk", sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)))
sum(is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2))
#[1] 0

DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_v2", raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents", raster=F)
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)] = 'unk'
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_v2", raster=F)


library(tidyverse)
library(monocle3)
library(SeuratWrappers)
library(ggridges)
cds <- as.cell_data_set(ls_synthesis_v3_harmony_v3)
fData(cds)
#DataFrame with 27581 rows and 0 columns
rownames(fData(cds))[1:10]
fData(cds)$gene_short_name <- rownames(fData(cds))
head(fData(cds))
head(counts(cds))
#6 x 100571 sparse Matrix of class "dgCMatrix"
recreate.partitions <- c(rep(1, length(cds@colData@rownames)))
names(recreate.partitions) <- cds@colData@rownames
recreate.partitions <- as.factor(recreate.partitions)
recreate.partitions
cds@clusters@listData[["UMAP"]][["partitions"]] <- recreate.partitions
##Assign cluster information
colnames(ls_synthesis_v3_harmony_v3@meta.data)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="subclust_idents_v2")
list.cluster <- ls_synthesis_v3_harmony_v3@active.ident
list.cluster
cds@clusters@listData[["UMAP"]][["clusters"]] <- list.cluster
##Assign UMAP coordinates
cds@int_colData@listData[["reducedDims"]]@listData[["UMAP"]] <- ls_synthesis_v3_harmony_v3@reductions$umap@cell.embeddings
cluster.before.traj <-plot_cells(cds, color_cells_by = "cluster", label_groups_by_cluster = F, 
                                 group_label_size = 5) + theme(legend.position = "right")
cluster.before.traj
##learn trajectory
cds <- learn_graph(cds, use_partition = F)
plot_cells(cds, color_cells_by = "cluster", label_groups_by_cluster = F,
           label_branch_points = T, label_roots = T, label_leaves = F,
           group_label_size = 5)
##order cells in pseudotime
cds <- order_cells(cds, reduction_method = "UMAP", root_cells = colnames(cds[, clusters(cds) == "RG"]))
plot_cells(cds, color_cells_by = "pseudotime", label_groups_by_cluster = T,
           label_branch_points = T, label_roots = F, label_leaves = F)
##look at clusters by pseudotime
cds$monocle3_pseudotime <- pseudotime(cds)
data.pseudo <- as.data.frame(colData(cds))
ggplot(data.pseudo, aes(monocle3_pseudotime, subclust_idents, fill = subclust_idents)) + geom_boxplot()
ggplot(data.pseudo, aes(monocle3_pseudotime, reorder(subclust_idents, monocle3_pseudotime), fill = subclust_idents)) + geom_boxplot()
#find genes that change with pseudotime
deg <- graph_test(cds, neighbor_graph = "principal_graph")
deg %>% arrange(q_value) %>% filter(status == "OK") %>% head()

#differential expression based on STICR Index for ENs?
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
levels(ls_synthesis_v3_harmony_v3)
index.de.markers = FindMarkers(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("EX")), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers, n=20)
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- EX, no clone size filter")+
  theme(plot.title = element_text(hjust = 0.5))

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
levels(ls_synthesis_v3_harmony_v3)
index.de.markers = FindMarkers(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("EX","EX_early")), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers, n=20)
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- EX and EX_early, no clone size filter")+
  theme(plot.title = element_text(hjust = 0.5))

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
levels(ls_synthesis_v3_harmony_v3)
index.de.markers = FindMarkers(subset(ls_synthesis_v3_harmony_v3, Clone_size_filt >1 & subclust_idents_v2 %in% c("EX","EX_early")), ident.1 = "IndexE", ident.2 = "Index3")
head(index.de.markers, n=20)
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Index 3 (left) and Index E (right) -- EX and EX_early, Clone size > 1")+
  theme(plot.title = element_text(hjust = 0.5))

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="age_pseudo")
levels(ls_synthesis_v3_harmony_v3)
index.de.markers = FindMarkers(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("EX","EX_early")), ident.1 = ">GW20", ident.2 = "<=GW20")
head(index.de.markers, n=20)
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between <=GW20 (left) and >GW20 (right) -- EX and EX_early, no clone size filter")+
  theme(plot.title = element_text(hjust = 0.5))
write.csv(index.de.markers, "/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ex_subclust_age_de_markers_230925.csv")

FeaturePlot(ls_glia_v2_harmony, features=c("CUL3", "ITGA2", "GPX3", "CREB5"))
FeaturePlot(ls_glia_v2_harmony, features=c("LGALS3", "ANXA2", "ITGA2", "ITGB5"))

FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB", "EOMES", "PPP1R17"))


VlnPlot(ls_glia_v2_harmony, features=c("GPX3"))
VlnPlot(ls_glia_v2_harmony, features=c("CRYAB"))
VlnPlot(ls_glia_v2_harmony, features=c("ITGA2"))
VlnPlot(ls_glia_v2_harmony, features=c("HOPX"))
VlnPlot(ls_glia_v2_harmony, features=c("EGFR"))
VlnPlot(ls_glia_v2_harmony, features=c("FAM107A"))
VlnPlot(ls_glia_v2_harmony, features=c("CXCL12"))
DimPlot(ls_glia_v2_harmony, group.by="seurat_clusters", label=T)

ls_ex = SetIdent(ls_ex, value = "Clone_Index_filt")
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("TLE4"))
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("TBR1"))
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("HS3ST4"))
VlnPlot(subset(ls_ex, Clone_size_corrected>1), features=c("CRYM"))


ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value = "Clone_Index_filt")
VlnPlot(subset(ls_synthesis_v3_harmony_v3, Clone_size_corrected>1 & subclust_idents_v2 %in% c('EX')), features=c("TLE4"))
VlnPlot(subset(ls_synthesis_v3_harmony_v3, Clone_size_corrected>1 & subclust_idents_v2 %in% c('EX', 'EX_early')), features=c("TLE4"))
VlnPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c('EX')), features=c("TLE4"))
VlnPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c('EX')), features=c("TBR1"))
VlnPlot(subset(ls_synthesis_v3_harmony_v3, Clone_size_corrected>1 & subclust_idents_v2 %in% c('EX')), features=c("TBR1"))
VlnPlot(subset(ls_synthesis_v3_harmony_v3, Clone_size_corrected>1 & subclust_idents_v2 %in% c('EX')), features=c("CRYM"))

FeaturePlot(ls_synthesis_v3_harmony_v3, features="S100A13", raster=F)
FeaturePlot(ls_glia_v2_harmony, features="S100A13", raster=F)

##bookmark to come back to for later

## 10/3/23
DimPlot(ls_ex)
DimPlot(ls_ex, group.by="orig.ident")
DimPlot(ls_ex, group.by="age_pseudo")
FeaturePlot(ls_ex, features=c("NFIA", "NFIX", "NFIB", "FEZF2"))
head(ls_glia_v2_harmony@meta.data)

setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231003_figs')
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="subclust_idents")
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 == 'EX_early'] = 'EX_EARLY'
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)])
DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)
#install.packages('svglite')
library(svglite)
image = DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)
ggsave(file="test.svg", plot=image, width=10, height=8)
ggsave(file="full_umap_subclust_labels.png", plot=image, width=12, height=8)

head(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% compiled_clust_df_idents$Clone_IDs[compiled_clust_df_idents$clonal_restriction %in% c('quadpotent')],])
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% 'LS36_Clone_1'], raster=F, sizes.highlight = 3) +
  ggtitle('LS36_Clone_1')
image = DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs %in% 'LS36_Clone_1'], raster=F, sizes.highlight = 3) +
  ggtitle('LS36_Clone_1')
ggsave(file="example_clone.png", plot=image, width=12, height=8)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GFAP", "EOMES", "NEUROD6", "OLIG2", "MKI67", "GAD2"), ncol=3, raster=F)
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GFAP", "EOMES", "NEUROD6", "OLIG2", "MKI67", "GAD2"), ncol=3, raster=F)
ggsave(file="full_umap_marker_genes.png", plot=image, width=12, height=8)


length(clone_list)
all_clone_list = unique(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3, 'Clone_IDs_filt'])
length(all_clone_list)
#[1] 3259
library(ggplot2)
library(ggpubr)
clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "All multicellular clones")
image = clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "All multicellular clones")
ggsave(file="all_clones_pct_barchart.svg", plot=image, width=12, height=8)

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction, fill=Clone_Index, group=Clone_Index)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


ggplot(compiled_clust_df_idents, aes(x=sample_age, fill=ex_in_presence)) +
  geom_bar(position="fill", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3,]), split.by="Clone_Index_filt", raster=F)


colnames(compiled_clust_df_idents)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("FOXP2"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("FOXP2"), raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"), raster=F)
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("SATB2"), raster=F)

DimPlot(ls_synthesis_v3_harmony_v3, raster=F)

DimPlot(ls_in, cells.highlight = rownames(ls_in@meta.data)[ls_in@meta.data$Clone_size_filt>=3])
head(ls_in@meta.data[ls_in@meta.data$Clone_size_filt>=3,])
head(ls_in@meta.data[ls_in@meta.data$Clone_size_filt>=3 & ls_in@meta.data$clust_idents %in% 'IN_unk',])
subset(ls_in@meta.data, Clone_size_filt>=3 & subclust_idents %in% 'IN_UNK')
table(ls_in@meta.data$subclust_idents)

DotPlot(ls_in, features=c("CXCR4", "ERBB4", "NRG1", "NRP2", "EPHA3", "EPHA5", "EPHA6", "SLIT1", "SLIT2", "ARX", "ROBO1", "DLX1", "DLX2", "LHX6", "MAF", "ZEB2")) + 
  theme(axis.text.x=element_text(angle=45, hjust=1))
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("NRG1"), raster=F)


setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231003_figs')
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1], cols.highlight='#2dba3b') +
  ggtitle('Cells in multicellular clones') +
  theme(plot.title = element_text(hjust = 0.5))
image = DimPlot(ls_synthesis_v3_harmony_v3, raster=F, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>1], cols.highlight='#2dba3b') +
  ggtitle('Cells in multicellular clones') +
  theme(plot.title = element_text(hjust = 0.5))
ggsave(file="clone_size_greater_than_1_umap.svg", plot=image, width=12, height=8)
ggsave(file="clone_size_greater_than_1_umap.png", plot=image, width=12, height=8)

#going to delete IN_UNK from the object........
#this might result in some unintended consequences, but they are just doublets
#the original is preserved in 20231005_local_STICR_all_objects
FeaturePlot(ls_in, features="nCount_RNA", max.cutoff="q90")
FeaturePlot(ls_in, features="NEUROD2", max.cutoff="q90")
ls_in = subset(ls_in, subclust_idents %in% c("IN_local", "IN_MGE", "IN_CGE", "IPC_IN_DIV"))
library(harmony)
ls_in_new = RunPCA(ls_in, features=VariableFeatures(object=ls_in))
ls_in_new <- RunHarmony(ls_in_new, c("orig.ident"))
ls_in_new <- FindNeighbors(ls_in_new, reduction = "harmony", dims = 1:15)
ls_in_new <- FindClusters(ls_in_new, resolution = .5)
ls_in_new <- RunUMAP(ls_in_new, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in_new, reduction = "umap",label=T)
#inverting the UMAP_2 embedding to make it look more like the original
ls_in_new[["umap"]]@cell.embeddings[,2] = -ls_in_new[["umap"]]@cell.embeddings[,2]

#look at INs in multicellular clones
DimPlot(ls_in_new, cells.highlight = rownames(ls_in_new@meta.data)[ls_in_new@meta.data$Clone_size_corrected > 1], cols.highlight='#2dba3b') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(plot.title = element_text(hjust=0.5))
image = DimPlot(ls_in_new, cells.highlight = rownames(ls_in_new@meta.data)[ls_in_new@meta.data$Clone_size_corrected > 1], cols.highlight='#2dba3b') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_sub_clone_size_greater_than_1_umap.svg", plot=image, width=12, height=8)
ggsave(file="in_sub_clone_size_greater_than_1_umap.png", plot=image, width=12, height=8)
DimPlot(ls_in_new, group.by="subclust_idents", label=T)
image = DimPlot(ls_in_new, group.by="subclust_idents", label=T) +
  ggtitle('IN subclustering') +
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_sub_umap.png", plot=image, width=12, height=8)
ggsave(file="in_sub_umap.svg", plot=image, width=12, height=8)
colnames(ls_in_new@meta.data)
ls_in_old_meta = ls_in@meta.data
rm(ls_in)
ls_in = ls_in_new
rm(ls_in_new)
DimPlot(ls_in)
colnames(ls_in@meta.data)
ls_in = SetIdent(ls_in, value="subclust_idents")
DimPlot(ls_in)
FeaturePlot(ls_in, features=c("PAX6", "NR2F2", "LHX6", "SOX6"))

#look at INs in EX/IN shared clones
DimPlot(ls_in, cells.highlight = rownames(ls_in@meta.data)[ls_in@meta.data$ex_in_presence %in% "EX_IN_shared"]) + 
  ggtitle('IN Cells in EX/IN shared clones') +
  theme(plot.title = element_text(hjust=0.5))
ls_in@meta.data[ls_in@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in@meta.data$subclust_idents %in% "IN_MGE",]
ls_in@meta.data$ex_in_presence[is.na(ls_in@meta.data$Clone_size_corrected)]=NA
table(ls_in@meta.data$Barcode_UMI_Count_filt)
DimPlot(ls_in, cells.highlight = rownames(ls_in@meta.data)[ls_in@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in@meta.data$Barcode_UMI_Count_filt>2]) + 
  ggtitle('IN Cells in EX/IN shared clones (Barcode UMI Count >2)') +
  theme(plot.title = element_text(hjust=0.5))
image = DimPlot(ls_in, cells.highlight = rownames(ls_in@meta.data)[ls_in@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in@meta.data$Barcode_UMI_Count_filt>2]) + 
  ggtitle('IN Cells in EX/IN shared clones (Barcode UMI Count >2)') +
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_cells_in_ex_in_shared_clones.svg", plot=image, width=12, height=8)

#some IN marker genes to show that "local" identity is strange
FeaturePlot(ls_in, features=c("GAD2", "PAX6", "NR2F2", "LHX6", "SOX6", "ERBB4"), ncol=3)
image=FeaturePlot(ls_in, features=c("GAD2", "PAX6", "NR2F2", "LHX6", "SOX6", "ERBB4"), ncol=3)
ggsave(file="in_cells_marker_gene_umap_v1.png", plot=image, width=12, height=8)

DotPlot(subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE", "IN_local")), 
        features=c("GAD1", "GAD2", "DLX1", "DLX2", "DLX5", #general INs
                   "NR2F2", "NR2F1", "ADARB2", #CGE
                   "ERBB4", "ZEB2", #general GE-derived
                   "NKX2-1", "LHX6", "MAF", "MAFB", "SOX6", #MGE
                   "PAX6","NPY", #local
                   )) + 
  theme(axis.text.x=element_text(angle=45, hjust=1))

FeaturePlot(ls_in, features=c("ETV1", "CALB2", "PBX3", "TSHZ1"))
FeaturePlot(ls_in, features=c("SP8", "CALB2", "SCGN", "ANXA5"))


VlnPlot(subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE", "IN_local")), 
        features=c("GAD2", "ADARB2", "NR2F1", "LHX6", "SOX6"), pt.size=0, ncol=5) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
image=VlnPlot(subset(ls_in, subclust_idents %in% c("IN_CGE", "IN_MGE", "IN_local")), 
              features=c("GAD2", "ADARB2", "NR2F1", "LHX6", "SOX6"), pt.size=0, ncol=5) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
ggsave(file="in_cells_marker_gene_vlnplot_v1.svg", plot=image, width=12, height=5)


#plots to show the prevalence of IN neurogenesis at older ages?
head(compiled_clust_df_idents)
compiled_clust_df_idents$age_pseudo=NA
compiled_clust_df_idents$age_pseudo[compiled_clust_df_idents$sample_age<=20] = "<=GW20"
compiled_clust_df_idents$age_pseudo[compiled_clust_df_idents$sample_age>20] = ">GW20"
table(compiled_clust_df_idents$age_pseudo)
table(compiled_clust_df_idents$ex_in_presence)
head(compiled_clust_df_idents)

ggplot(compiled_clust_df_idents, aes(x=sample_age, fill=ex_in_presence)) +
  geom_bar(position="fill", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=age_pseudo, fill=ex_in_presence)) +
  geom_bar(position="fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(compiled_clust_df_idents, aes(x=age_pseudo, fill=clonal_restriction)) +
  geom_bar(position="fill") +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')


ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

#GW16
table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==16])/sum(compiled_clust_df_idents$sample_age==16)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#0.73983740   0.06504065   0.09485095   0.10027100
ex_in_freq_df = data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==16])/sum(compiled_clust_df_idents$sample_age==16))
colnames(ex_in_freq_df)=c("ex_in_presence", "GW16")
ex_in_freq_df = cbind(ex_in_freq_df, "GW19"=data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==19])/sum(compiled_clust_df_idents$sample_age==19))[,2])
ex_in_freq_df = cbind(ex_in_freq_df, "GW20"=data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==20])/sum(compiled_clust_df_idents$sample_age==20))[,2])
ex_in_freq_df = cbind(ex_in_freq_df, "GW22"=data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==22])/sum(compiled_clust_df_idents$sample_age==22))[,2])
ex_in_freq_df = cbind(ex_in_freq_df, "GW23"=data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==23])/sum(compiled_clust_df_idents$sample_age==23))[,2])
ex_in_freq_df = cbind(ex_in_freq_df, "GW24"=data.frame(table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$sample_age==24])/sum(compiled_clust_df_idents$sample_age==24))[,2])
ex_in_freq_df
#  ex_in_presence       GW16       GW19       GW20       GW22       GW23       GW24
#1        EX_only 0.73983740 0.67045455 0.64602804 0.03324100 0.08498024 0.01809955
#2        IN_only 0.06504065 0.10227273 0.13785047 0.77008310 0.53557312 0.59276018
#3   EX_IN_shared 0.09485095 0.09090909 0.13960280 0.08310249 0.05928854 0.09049774
#4    no_EX_or_IN 0.10027100 0.13636364 0.07651869 0.11357341 0.32015810 0.29864253

ex_in_freq_df_pivot = data.frame("ex_in_presence"=rep(ex_in_freq_df$ex_in_presence, 6),
                                 "ex_in_pct" = c(ex_in_freq_df$GW16, ex_in_freq_df$GW19, ex_in_freq_df$GW20,
                                                 ex_in_freq_df$GW22, ex_in_freq_df$GW23, ex_in_freq_df$GW24),
                                 "sample_age"=c(rep("GW16",4), rep("GW19",4), rep("GW20",4),
                                                rep("GW20",4), rep("GW22",4), rep("GW24",4)),
                                 "age_pseudo"=c(rep("<=GW20", 12), rep(">GW20",12)),
                                 "x_coords"=c(rep(c(0.75,1.75,2.75,3.75),3),
                                              rep(c(1.25,2.25,3.25,4.25),3))
)
ex_in_freq_df_pivot
ggplot(ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  geom_point()

ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  geom_point(data=ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
image=ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  geom_point(data=ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency by index, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')
ggsave(file="ex_in_presence_by_age_barchart_with_points.svg", plot=image, width=12, height=8)


#plotting ls_glia stuff for figures
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight=rownames(subset(ls_synthesis_v3_harmony_v3@meta.data, subclust_idents %in% c("GLIA", "GLIA_DIV"))), raster=F)
DimPlot(ls_glia_v2_harmony)
VlnPlot(ls_glia_v2_harmony, features=c("OLIG2"))
VlnPlot(ls_glia_v2_harmony, features=c("CXCR4"))

FeaturePlot(ls_glia_v2_harmony, features=c("MKI67", "VIM", "OLIG2", "EOMES"))
FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"), split.by="age_pseudo")
FeaturePlot(ls_glia_v2_harmony, features=c("nCount_RNA"), max.cutoff="q90")
FeaturePlot(ls_glia_v2_harmony, features=c("HOPX", "CRYAB", "AQP4", "GFAP"))
DimPlot(ls_glia_v2_harmony, group.by="seurat_clusters")

FeaturePlot(ls_glia_v2_harmony, features=c("THY1","PDGFRA", "ASCL1", "EGFR"))
FeaturePlot(ls_glia_v2_harmony, features=c("CXCR4","PDGFRA", "ASCL1", "EGFR"))

FeaturePlot(ls_glia_v2_harmony, features=c("THY1","CD24", "EGFR", "CXCR4"))

table(ls_glia_v2_harmony@meta.data$glial_subtype)
ls_glia_v2_harmony@meta.data$glial_subtype[ls_glia_v2_harmony@meta.data$glial_subtype %in% c("Transitioning OPCs/IPCs")] = "Transitioning OPCs"
ls_glia_v2_harmony@meta.data$glial_subtype[ls_glia_v2_harmony@meta.data$glial_subtype %in% c("G2M_phase_div_glia")] = "G2M dividing"
ls_glia_v2_harmony@meta.data$glial_subtype[ls_glia_v2_harmony@meta.data$glial_subtype %in% c("S_phase_div_glia")] = "S dividing"
ls_glia_v2_harmony@meta.data$glial_subtype[ls_glia_v2_harmony@meta.data$glial_subtype %in% c("ASTRO")] = "Astrocytes"
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="glial_subtype")
DimPlot(ls_glia_v2_harmony)
VlnPlot(ls_glia_v2_harmony, features=c("HOPX"))
VlnPlot(ls_glia_v2_harmony, features=c("CRYAB"))
VlnPlot(ls_glia_v2_harmony, features=c("ANXA1"))


DimPlot(ls_glia_v2_harmony, label=T)
image=DimPlot(ls_glia_v2_harmony, label=T)
ggsave(filename='ls_glia_umap_v1.png', image, width=12, height=8)
FeaturePlot(ls_glia_v2_harmony, features=c("VIM", "GFAP", "MKI67", "OLIG2", "EOMES", "PAX6"), ncol=3)
image=FeaturePlot(ls_glia_v2_harmony, features=c("VIM", "GFAP", "MKI67", "OLIG2", "EOMES", "PAX6"), ncol=3)
ggsave(filename='ls_glia_umap_marker_genes_v1.png', image, width=12, height=8)

unique(ls_glia_v2_harmony@meta.data$glial_subtype)

ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & glial_subtype %in% c("G2M dividing","Transitioning OPCs","Astrocytes","S dividing","Transitioning IPCs","RG (putative tRG)","RG","RG (putative oRG)"))@meta.data, 
       aes(x=glial_subtype, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('Index contribution to glial subclusters (multicellular clones only)') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
image=ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & glial_subtype %in% c("G2M dividing","Transitioning OPCs","Astrocytes","S dividing","Transitioning IPCs","RG (putative tRG)","RG","RG (putative oRG)"))@meta.data, 
             aes(x=glial_subtype, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('Index contribution to glial subclusters (multicellular clones only)') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
ggsave(filename='glia_sub_index_contribution_barchart.svg', width=12, height=8)


ls_glia_v2_harmony_meta_copy = ls_glia_v2_harmony@meta.data
ls_glia_v2_harmony_meta_copy$Clone_Index_shuffled=NA
ls_glia_v2_harmony_meta_copy$Clone_Index_shuffled[ls_glia_v2_harmony_meta_copy$Clone_Index_filt %in% c("IndexE","Index3")] = sample(ls_glia_v2_harmony_meta_copy$Clone_Index_filt[ls_glia_v2_harmony_meta_copy$Clone_Index_filt %in% c("IndexE","Index3")])

ggplot(subset(ls_glia_v2_harmony_meta_copy, Clone_Index_filt %in% c("Index3", "IndexE") & glial_subtype %in% c("G2M dividing","Transitioning OPCs","Astrocytes","S dividing","Transitioning IPCs","RG (putative tRG)","RG","RG (putative oRG)")), 
       aes(x=glial_subtype, fill=Clone_Index_shuffled)) +
  geom_bar(position = "fill") +
  ggtitle('Index contribution to glial subclusters (multicellular clones only)') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
image=ggplot(subset(ls_glia_v2_harmony_meta_copy, Clone_Index_filt %in% c("Index3", "IndexE") & glial_subtype %in% c("G2M dividing","Transitioning OPCs","Astrocytes","S dividing","Transitioning IPCs","RG (putative tRG)","RG","RG (putative oRG)")), 
             aes(x=glial_subtype, fill=Clone_Index_shuffled)) +
  geom_bar(position = "fill") +
  ggtitle('Index contribution to glial subclusters (multicellular clones only)') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1))
ggsave(filename='glia_sub_index_contribution_shuffled_barchart.svg', width=12, height=8)

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, sample_age %in% c(19,20,22,23,24) & glial_subtype %in% c("RG", "RG (putative tRG)", "RG (putative oRG)")), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived Index3 (left) and VZ_derived IndexE (right) -- RG subclusters only, no GW16")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, sample_age %in% c(16, 19,20,22,23,24) & glial_subtype %in% c("RG", "RG (putative tRG)", "RG (putative oRG)")), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived Index3 (left) and VZ_derived IndexE (right) -- RG subclusters only")+
  theme(plot.title = element_text(hjust = 0.5))

VlnPlot(subset(ls_glia_v2_harmony, Clone_index_filt %in% c("IndexE", "Index3") & glial_subtype %in% c("RG", "RG (putative tRG)", "RG (putative oRG)")), features=c("CRYAB"), pt.size=0)


library(dplyr)
ls_meta_copy = ls_synthesis_v3_harmony_v3@meta.data
colnames(compiled_clust_df_idents)
colnames(ls_meta_copy)
head(compiled_clust_df_idents[,c("Clone_IDs", "ex_in_presence")])
compiled_clust_df_idents$Clone_IDs_filt = compiled_clust_df_idents$Clone_IDs
tail(left_join(ls_meta_copy, compiled_clust_df_idents[,c("Clone_IDs_filt", "ex_in_presence")]))
ls_meta_copy = left_join(ls_meta_copy, compiled_clust_df_idents[,c("Clone_IDs_filt", "ex_in_presence")])
ls_meta_copy[ls_meta_copy$Clone_IDs_filt %in% "LS36_Clone_69", ]
table(ls_meta_copy$ex_in_presence)
#     EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#        5887         4198         2229         1713 
sum(is.na(ls_meta_copy$ex_in_presence))
#[1] 86544
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, ls_meta_copy$ex_in_presence, col.name = "ex_in_presence")
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% c("EX_only")], raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% c("IN_only")], raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% c("EX_IN_shared")], raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% c("no_EX_or_IN")], raster=F)

table(ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% 'EX_only'])
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents[ls_synthesis_v3_harmony_v3@meta.data$ex_in_presence %in% 'EX_only'])

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("IN_UNK")], raster=F)
sum(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("IN_UNK"))
#[1] 828



DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$])

FeaturePlot(ls_in, features=c("DLX2", "PAX6", "SCGN", "SOX6"))
DoHeatmap(ls_in, features=c("GAD2", "DLX2", "PAX6", "SCGN", "CALB2", "PROX1", "NPY", "SOX6", "ERBB4", "NR2F2", "ADARB2", "LHX6"))
DoHeatmap(ls_in, features=c("GAD2", "DLX2", "PAX6", "SCGN", "CALB2", "PROX1", "NPY", "SOX6", "ERBB4", "NR2F2", "ADARB2", "LHX6")) + 
  scale_fill_gradientn(colors = c("blue", "white", "red"))

ggplot(subset(compiled_clust_df_idents, age_pseudo==">GW20"), aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  #geom_point(data=ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle('Clonal potency for >GW20 samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

#can we make a Venn diagram for clonal identity? going to combine glia and olig for this
head(compiled_clust_df_idents)
table(compiled_clust_df_idents$ex_in_presence)
table(subset(compiled_clust_df_idents, age_pseudo==">GW20")$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#59          680           80          269
table(subset(compiled_clust_df_idents, age_pseudo==">GW20")$clonal_restriction)
#EX_only                IN_only              GLIA_only              OLIG_only 
#19                    341                    112                     37 
#GLIA/OLIG_shared           EX_IN_shared    EX_GLIA/OLIG_shared    IN_GLIA/OLIG_shared 
#120                     42                     40                    339 
#EX_IN_GLIA/OLIG_shared             quadpotent 
#32                      6 

EX_count = 19+42+40+32+6 #139
IN_count = 341+42+339+6 #728
GLIA_count = 112+37+120+40+339+32+6 #686

139-42-32-6
59+32
139+339
787+112+37+120

#install.packages("ggVennDiagram")
library(ggVennDiagram)
venn_list = list('EX'=1:139, 'IN'=59:787, 'GLIA'=c(19:59,59:91,91:97,139:478,787:1056))
ggVennDiagram(venn_list, color = 1, lwd = 0.7) + 
  scale_fill_gradient(low = "#F4FAFE", high = "#F4FAFE") +
  theme(legend.position = "none")

colnames(ls_synthesis_v3_harmony_v3@meta.data)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="age_pseudo")
VlnPlot(ls_synthesis_v3_harmony_v3, features=c("TBR1", "SATB2"), pt.size=0)
VlnPlot(subset(ls_synthesis_v3_harmony_v3, multi_clone=='multi'), features=c("TBR1", "SATB2", "TLE4"))
#FeaturePlot(subset(ls_synthesis_v3_harmony_v3, multi_clone=='multi'), features=c("TBR1"), split.by="age_pseudo", pt.size=0.5, order=T, cols=c("gray","red"))
FeaturePlot(subset(ls_synthesis_v3_harmony_v3, multi_clone=='multi'), features=c("TBR1"), split.by="age_pseudo", pt.size=0.5, order=T)

## 11/6/23
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="subclust_idents_v2")
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=T)
DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=T)
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("IN_UNK"), invert=T), raster=F, label=T)

#trying to generate IN subset including IN IPCs this time
DimPlot(ls_in)
ls_in_v2 = subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("IN_CGE", "IN_MGE", "IN_local", "IPC_IN"))
ls_in_v2
#27581 features across 12118 samples within 1 assay
ls_in_v2 <- ScaleData(ls_in_v2, features = all.genes)
ls_in_v2 = FindVariableFeatures(ls_in_v2, selection.method="vst", nfeatures=2000)
ls_in_v2 = RunPCA(ls_in_v2, features=VariableFeatures(object=ls_in_v2))
ElbowPlot(ls_in_v2)
ls_in_v2 <- RunHarmony(ls_in_v2, c("orig.ident"))
ls_in_v2 <- FindNeighbors(ls_in_v2, reduction = "harmony", dims = 1:15)
ls_in_v2 <- FindClusters(ls_in_v2, resolution = .5)
ls_in_v2 <- RunUMAP(ls_in_v2, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in_v2, reduction = "umap",label=T, pt.size=1)
VlnPlot(ls_in_v2, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_in_v2, reduction = "umap",group.by="orig.ident", pt.size=1)
DimPlot(ls_in_v2, reduction = "umap",group.by="age_pseudo", pt.size=1)
DimPlot(subset(ls_in_v2, subset = Clone_Index %in% c("IndexE", "Index3")), label=T, cells.highlight=rownames(ls_in_v2@meta.data[ls_in_v2@meta.data$"Clone_size_corrected">=2,]), split.by="Clone_Index") +ggtitle("Clone size >1")
DimPlot(subset(ls_in_v2, subset = Clone_Index %in% c("IndexE", "Index3")), group.by="Clone_Index", split.by="age_pseudo", pt.size=1)
FeaturePlot(ls_in_v2, label=T, features=c("GAD2", "NEUROD2", "NR2F1", "PAX6", "MKI67", "NPY", "NR2F2", "ADARB2", "MAF"))
ls_in_v2[["umap"]]@cell.embeddings[,1] = -ls_in_v2[["umap"]]@cell.embeddings[,1]
ls_in_v2[["umap"]]@cell.embeddings[,2] = -ls_in_v2[["umap"]]@cell.embeddings[,2]
FeaturePlot(ls_in_v2, label=T, features=c("GAD2", "NEUROD2", "NR2F1", "PAX6", "MKI67", "NPY", "NR2F2", "ADARB2", "MAF"))
DimPlot(ls_in_v2, group.by="subclust_idents_v2", label=T)
FeaturePlot(ls_in_v2, features=c("GAD2", "NR2F2", "LHX6", "ERBB4", "PAX6", "SOX6"), ncol=3)

setwd("/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231106_figs")
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in_v2@meta.data$Barcode_UMI_Count_filt>2]) + 
  ggtitle('IN Cells in EX/IN shared clones (Barcode UMI Count >2)') +
  theme(plot.title = element_text(hjust=0.5))
image = DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in_v2@meta.data$Barcode_UMI_Count_filt>2]) + 
  ggtitle('IN Cells in EX/IN shared clones (Barcode UMI Count >2)') +
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_cells_in_v2_ex_in_shared_clones.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_size_corrected > 1], cols.highlight='#2dba3b') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(plot.title = element_text(hjust=0.5))
image = DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_size_corrected > 1], cols.highlight='#2dba3b') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_sub_clone_size_greater_than_1_umap.svg", plot=image, width=12, height=8)

#FeaturePlot(ls_in_v2, label=T, features=c("nCount_RNA"))
FeaturePlot(ls_in_v2, label=T, features=c("NR2F1", "SOX6", "CXCR4", "PROX1"))
FeaturePlot(ls_in_v2, label=T, features=c("TSHZ1", "PBX3", "MEIS2", "SYNPR"))


#working on some stuff for SFN poster
table(compiled_clust_df_idents$ex_in_presence)
compiled_clust_df_idents$in_binary = NA
compiled_clust_df_idents$in_binary[compiled_clust_df_idents$ex_in_presence %in% c('IN_only', 'EX_IN_shared')] = "INs_present"
compiled_clust_df_idents$in_binary[compiled_clust_df_idents$ex_in_presence %in% c('EX_only', 'no_EX_or_IN')] = "no_INs"
ggplot(compiled_clust_df_idents, aes(x=in_binary, fill=Clone_Index)) +
  geom_bar() +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of clones containing IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

table(compiled_clust_df_idents$in_binary[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#INs_present      no_INs 
#172         334
table(compiled_clust_df_idents$in_binary[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#INs_present      no_INs 
#379        1284 
table(compiled_clust_df_idents$in_binary[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#INs_present      no_INs 
#519         253 
table(compiled_clust_df_idents$in_binary[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#INs_present      no_INs 
#241          75

in_binary_df = data.frame(c(172, 334, 379, 1284, 519, 253, 241, 75))
colnames(in_binary_df) = c("counts")
in_binary_df$age_pseudo = c('<=GW20', '<=GW20', '<=GW20', '<=GW20', '>GW20', '>GW20', '>GW20', '>GW20')
in_binary_df$Clone_Index = c('Index3', 'Index3', 'IndexE', 'IndexE', 'Index3', 'Index3', 'IndexE', 'IndexE')
in_binary_df$in_presence = c('INs_present', 'no_INs', 'INs_present', 'no_INs', 'INs_present', 'no_INs', 'INs_present', 'no_INs')
in_binary_df
ggplot(in_binary_df, aes(x=age_pseudo, y=counts, fill=in_presence, group_by=age_pseudo)) +
  geom_point() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

ggplot(data=in_binary_df, aes(x=age_pseudo, y=counts, fill=in_presence, group_by=Clone_Index)) +
  geom_bar(stat="identity", position=position_dodge())
ggplot(data=in_binary_df, aes(x=age_pseudo, y=counts, group=interaction(in_presence,Clone_Index))) +
  geom_bar(stat="identity", position=position_dodge())

ggplot(data=subset(in_binary_df, age_pseudo == "<=GW20"), aes(x=Clone_Index, y=counts, fill=in_presence)) +
  geom_bar(stat="identity", position=position_dodge())
ggplot(data=subset(in_binary_df, age_pseudo == ">GW20"), aes(x=Clone_Index, y=counts, fill=in_presence)) +
  geom_bar(stat="identity", position=position_dodge())

in_binary_df$pct = c(172/(172+334), 334/(172+334), 
                     379/(379+1284), 1284/(379+1284),
                     519/(519+253), 253/(519+253),
                     241/(241+75), 75/(241+75))
ggplot(data=subset(in_binary_df, in_presence == "INs_present"), aes(x=age_pseudo, y=pct, fill=Clone_Index)) +
  geom_bar(stat="identity", position=position_dodge()) +
  ggtitle("Percentage of clones with at least one IN by age and index") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample age') +
  ylab('Percent of clones')



head(compiled_clust_df_idents)
ggplot(compiled_clust_df_idents, aes(x=in_binary, fill=factor(age_pseudo), group_by=Clone_Index)) +
  geom_bar() +
  ggtitle('Clonal potency by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Clonal potency')

compiled_clust_df_idents_jitter$age_pseudo=NA
table(compiled_clust_df_idents_jitter$sample_age)
compiled_clust_df_idents_jitter$age_pseudo[compiled_clust_df_idents_jitter$sample_age %in% c(16,19,20)] = "<=GW20"
compiled_clust_df_idents_jitter$age_pseudo[compiled_clust_df_idents_jitter$sample_age %in% c(22,23,24)] = ">GW20"
ggplot(compiled_clust_df_idents_jitter, aes(x=age_pseudo, fill=Clone_Index)) +
  geom_boxplot(outlier.shape=NA) +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of IN cells in clones by age') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample Age')

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MKI67", "DLX2", "PAX6"), raster=F, ncol=3)
DotPlot(ls_synthesis_v3_harmony_v3, features=c("MKI67", "DLX2", "PAX6", "OLIG2", "EOMES", "VIM"))
ls_synthesis_v3_harmony_v3@active.ident = factor(ls_synthesis_v3_harmony_v3@active.ident, levels=c(
  "ASTRO",
  "RG",
  "RG (putative oRG)",
  "RG (putative tRG)",
  "S_phase_div_glia",
  "G2M_phase_div_glia",
  "Transitioning IPCs",
  "IPC_EX",
  "IPC_EX_DIV",
  "IPC_IN_DIV",
  "IPC_IN",
  "IN_local",
  "IN_MGE",
  "IN_CGE",
  "EX",
  "EX_early",
  "OPC_DIV",
  "Transitioning OPCs/IPCs",
  "OPC",
  "OLIG"
))
DotPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("unk", "IN_UNK", "Transitioning OPCs/IPCs"), invert=T), features=c("MKI67", "DLX2", "PAX6", "OLIG2", "EOMES", "VIM"))
DotPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("unk", "IN_UNK", "Transitioning OPCs/IPCs"), invert=T), features=c("MKI67", "DLX2", "PAX6"))


colnames(ls_in_v2@meta.data)
head(ls_in_v2@meta.data)
ls_in_v2@active.ident = SetIdent(ls_in_v2, value="subclust_idents_v2")
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c("IPC_IN", "IN_local", "IN_CGE", "IN_MGE"))
VlnPlot(ls_in_v2, features=c("GAD2", "ADARB2", "LHX6", "ERBB4", "NR2F1", "SOX6", "SCGN", "PAX6"), ncol=4, pt.size=0)

DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=T)
image = DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=T)
ggsave(file="ls_synthesis_v3_full_umap.svg", plot=image, width=10, height=8)


theme_set(theme_bw(base_size = 18))
ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, '#00bbc2'))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=20),
        axis.title = element_text(size=18),
        axis.text = element_text(size=14),
        legend.text = element_text(size=14),
        legend.title = element_text(size=16)
        ) +
  xlab('Cell type')+
  ylab('Percentage of cell type in clones')
image = ggplot(clone_cluster_pivot_idents, aes(x=Identity, y=Cell_percentage, fill=Clone_Index)) +
  geom_boxplot() +
  scale_fill_manual(values=c(osvz_col, '#00bbc2'))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Clone composition by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=20),
        axis.title = element_text(size=18),
        axis.text = element_text(size=14),
        legend.text = element_text(size=14),
        legend.title = element_text(size=16)
  ) +
  xlab('Cell type')+
  ylab('Percentage of cell type in clones')
ggsave(file="celltypes_in_clone_by_index.svg", plot=image, width=10, height=8)

real_subclusts = c("ASTRO","RG","RG (putative oRG)","RG (putative tRG)","S_phase_div_glia","G2M_phase_div_glia",
                   "Transitioning IPCs","IPC_EX","IPC_EX_DIV","IPC_IN_DIV","IPC_IN","IN_local","IN_MGE","IN_CGE",
                   "EX","EX_early","OPC_DIV","Transitioning OPCs/IPCs","OPC","OLIG")
ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% real_subclusts,], aes(x=subclust_idents_v2, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, '#00bbc2', 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1),
        axis.title.x = element_blank()) +
  ylab('Percentage')
image = ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_Index_filt %in% c('IndexE','Index3') & ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% real_subclusts,], aes(x=subclust_idents_v2, fill=Clone_Index_filt)) + geom_bar(position = "fill") +
  ggtitle('Index contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, '#00bbc2', 'grey'))+
  theme(plot.title = element_text(hjust = 0.5), axis.text.x = element_text(angle = 45, hjust=1),
        axis.title.x = element_blank()) +
  ylab('Percentage')
ggsave(file="index_contribution_to_clusters.svg", plot=image, width=10, height=8)

DimPlot(ls_in_v2, label=T) + 
  ggtitle('Interneuron subclusters')+
  theme(plot.title = element_text(hjust=0.5))
image = DimPlot(ls_in_v2, label=T) + 
  ggtitle('Interneuron subclusters')+
  theme(plot.title = element_text(hjust=0.5))
ggsave(file="in_v2_subclustering_umap.svg", plot=image, width=10, height=8)

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & glial_subtype %in% c('RG','RG (putative tRG)', 'RG (putative oRG)') & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia$labels = rownames(de.markers.glia)
custom_labels = de.markers.glia |> filter(labels %in% c("AC131571.1", "PDGFRA", "GLI3"))
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_label_repel(aes(label=labels), data=custom_labels)+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))


tf_list = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231106_figs/tf_list.csv')
head(tf_list)
sum(de.markers.glia$labels %in% tf_list)
tf_int = intersect(de.markers.glia$labels, tf_list$TF)
custom_labels = de.markers.glia |> filter(labels %in% tf_int)
ggplot(data=de.markers.glia, aes(x =avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_label_repel(aes(label=labels), data=custom_labels)+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))
image = ggplot(data=de.markers.glia, aes(x =avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_label_repel(aes(label=labels), data=custom_labels)+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- Glia, clone size >=3, RG subclusters only, GW16 excluded")+
  theme(plot.title = element_text(hjust = 0.5))
ggsave(file="DE_genes_radial_glia_tf_labels_only.svg", plot=image, width=12, height=8)


library('RColorBrewer')
barchart_color_palette = brewer.pal(4,'Set3')
barchart_color_palette = c("#FFFFB3","#BEBADA","#FB8072","#8DD3C7")
clonal_comp_stacked_barchart_func_by_clones_age_sep = function(clone_list, chart_title) {
  age_filter = clone_cluster_pivot_idents$orig.ident %in% young_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_young_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"<=GW20"))
  age_filter = clone_cluster_pivot_idents$orig.ident %in% old_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_old_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, ">GW20"))
  total_clones = n_young_clones+n_old_clones
  pct_young_clones = n_young_clones/total_clones
  pct_old_clones = n_old_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_young_clones, 
                       pct_old_clones)
  )
}
library(ggplot2)
library(ggpubr)
all_clone_list = unique(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3, 'Clone_IDs_filt'])
clonal_comp_stacked_barchart_func_by_clones_age_sep(all_clone_list, "All multicellular clones")
image = clonal_comp_stacked_barchart_func_by_clones_age_sep(all_clone_list, "All multicellular clones")
ggsave(file="clonal_comp_barchart_age_sep.png", plot=image, width=12, height=8)
ggsave(file="clonal_comp_barchart_age_sep.svg", plot=image, width=12, height=8)

clonal_comp_stacked_barchart_func_by_clones_indices_combined = function(clone_list, chart_title) {
  clone_filter = clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[clone_filter,]
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(chart_title)
}

clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "All multicellular clones")
image = clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "All multicellular clones")
ggsave(file="clonal_comp_barchart_all_clones_no_sep.png", plot=image, width=12, height=8)
ggsave(file="clonal_comp_barchart_all_clones_no_sep.svg", plot=image, width=12, height=8)

ggplot(compiled_clust_df_idents, aes(x=clonal_restriction)) +
  geom_bar(aes(y = (..count..)/sum(..count..))) +
  scale_fill_manual()+
  ggtitle('Clonal potency') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=20),
        axis.title = element_text(size=24),
        axis.text = element_text(size=24, angle = 45, hjust=1),
        axis.title.x = element_blank())+
  ylab('Percentage of all clones') 
image = ggplot(compiled_clust_df_idents, aes(x=clonal_restriction)) +
  geom_bar(aes(y = (..count..)/sum(..count..))) +
  scale_fill_manual()+
  ggtitle('Clonal potency') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=20),
        axis.title = element_text(size=24),
        axis.text = element_text(size=24, angle = 45, hjust=1),
        axis.title.x = element_blank())+
  ylab('Percentage of all clones') 
ggsave(file="clonal_potency_no_sep.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=F)
image = DimPlot(ls_synthesis_v3_harmony_v3, raster=F, label=F)
ggsave(file="full_umap_no_labels.png", plot=image, width=12, height=8)

VlnPlot(ls_in_v2, features=c("GAD2", "ADARB2", "LHX6", "ERBB4", "NR2F1", "SOX6", "SCGN", "PAX6"), ncol=4, pt.size=0)
image = VlnPlot(ls_in_v2, features=c("GAD2", "ADARB2", "LHX6", "ERBB4", "NR2F1", "SOX6", "SCGN", "PAX6"), ncol=4, pt.size=0)
ggsave(file="in_v2_vlnplot_markers.svg", plot=image, width=12, height=8)



BiocManager::install("eulerr")
library(eulerr)
head(compiled_clust_df_idents)
head(compiled_clust_df_idents$IN)>0
euler_mat = cbind(
  EX=compiled_clust_df_idents$EX>0,
  IN=compiled_clust_df_idents$IN>0,
  GLIA=compiled_clust_df_idents$GLIA+compiled_clust_df_idents$OLIG>0
)
euler(euler_mat)

plot(euler(euler_mat),
     fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
image = plot(euler(euler_mat),
             fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
ggsave(file='venn_diagram_mrs.svg', plot=image, width=12, height=8)
ggsave(file='venn_diagram_mrs.png', plot=image, width=12, height=8)


plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
image = plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
ggsave(file='venn_diagram_young_samples_mrs.svg', plot=image, width=12, height=8)
ggsave(file='venn_diagram_young_samples_mrs.png', plot=image, width=12, height=8)

plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
image = plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
ggsave(file='venn_diagram_old_samples_mrs.svg', plot=image, width=12, height=8)
ggsave(file='venn_diagram_old_samples_mrs.png', plot=image, width=12, height=8)

save.image("/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/20231005_local_STICR_main_objects.RData")



## 11/28/23 ##

ggplot(ls_glia_v2_harmony@meta.data, aes(x=glial_subtype, fill=age_pseudo)) + geom_bar(position = "fill") +
  ggtitle('Age contribution to clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
ggplot(subset(ls_glia_v2_harmony, ex_in_presence %in% c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN"))@meta.data, aes(x=glial_subtype, fill=ex_in_presence)) + geom_bar(position = "fill") +
  ggtitle('Clonal potency by subtype') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))

## Figure 1 panels ##
#adding a new subclust metadata column for the figure
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231129_figs')
colnames(ls_synthesis_v3_harmony_v3@meta.data)
table(ls_synthesis_v3_harmony_v3$subclust_idents_v2)
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, ls_synthesis_v3_harmony_v3$subclust_idents_v2, "subclust_idents_paper")
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("RG (putative oRG)", "RG (putative tRG)")] = "RG"
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_paper", raster=F)
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("IPC_EX_DIV", "IPC_IN_DIV", "OPC_DIV", "S_phase_div_glia", "G2M_phase_div_glia")] = "Dividing"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("ASTRO")] = "Astrocytes"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("OLIG")] = "Oligodendrocytes"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("Transitioning IPCs", "Transitioning OPCs/IPCs")] = "RG"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("RG")] = "Radial glia"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("RG")] = "Radial glia"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("IPC_EX")] = "Excitatory IPCs"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("IPC_IN")] = "Inhibitory IPCs"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("EX")] = "Excitatory neurons"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("Excitatory neurons")] = "EX"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2 %in% 
                                                             c("EX_early")] = "EX"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("Excitatory IPCs")] = "EX IPCs"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("Inhibitory IPCs")] = "IN IPCs"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("Inhibitory IPCs")] = "IN IPCs"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 
                                                             c("OPC")] = "OPCs"

DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), group.by="subclust_idents_paper", raster=F)

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value = "subclust_idents_paper")
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper)
#      Astrocytes         Dividing               EX          EX IPCs          IN IPCs           IN_CGE         IN_local 
#1368             8941            56892             9345             1192             2860             5445 
#IN_MGE           IN_UNK Oligodendrocytes             OPCs      Radial glia              unk 
#2621              828              755             2249             6446             1629 
ls_synthesis_v3_harmony_v3@active.ident = factor(ls_synthesis_v3_harmony_v3@active.ident, levels=c(
  "Astrocytes",
  "Radial glia",
  "Dividing",
  "OPCs",
  "Oligodendrocytes",
  "EX IPCs",
  "EX",
  "IN IPCs",
  "IN_local",
  "IN_CGE",
  "IN_MGE",
  "IN_UNK",
  "unk"
))
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F)
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F)
ggsave(file="ls_synthesis_v3_full_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_full_umap.png", plot=image, width=10, height=8)

#EX highlight
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("EX")],
        cols.highlight = "#f2cb30") + 
  ggtitle("EX") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("EX")],
                cols.highlight = "#f2cb30") + 
  ggtitle("EX") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
ggsave(file="ls_synthesis_v3_umap_ex_highlight.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_ex_highlight.png", plot=image, width=10, height=8)

#oligo highlight
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("OPCs", "Oligodendrocytes")],
        cols.highlight = "#52ba86") + 
  ggtitle("Oligos") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("OPCs", "Oligodendrocytes")],
                cols.highlight = "#52ba86") + 
  ggtitle("Oligos") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
ggsave(file="ls_synthesis_v3_umap_oligo_highlight.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_oligo_highlight.png", plot=image, width=10, height=8)

#IN highlight
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("IN IPCs", "IN_local", "IN_CGE", "IN_MGE")],
        cols.highlight = "#fc7253") + 
  ggtitle("IN") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("IN IPCs", "IN_local", "IN_CGE", "IN_MGE")],
                cols.highlight = "#fc7253") + 
  ggtitle("IN") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
ggsave(file="ls_synthesis_v3_umap_in_highlight.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_in_highlight.png", plot=image, width=10, height=8)

#glia highlight
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("Astrocytes", "Radial glia")],
        cols.highlight = "#9056d6") + 
  ggtitle("Glia") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("Astrocytes", "Radial glia")],
                cols.highlight = "#9056d6") + 
  ggtitle("Glia") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
ggsave(file="ls_synthesis_v3_umap_glia_highlight.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_glia_highlight.png", plot=image, width=10, height=8)

#dividing highlight
DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("Dividing")],
        cols.highlight = "#e091e3") + 
  ggtitle("Dividing") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("IN_UNK", "unk"), invert=T), raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("Dividing")],
                cols.highlight = "#e091e3") + 
  ggtitle("Dividing") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=50))
ggsave(file="ls_synthesis_v3_umap_dividing_highlight.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_dividing_highlight.png", plot=image, width=10, height=8)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM", "GFAP",
                                                   "NEUROD2", "NEUROD6",
                                                   "DLX2", "GAD2",
                                                   "OLIG2","MBP"),
            ncol=4, raster=F, cols=c("gray", "darkred"))


table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper)
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, NA, "lineage_trajectories_paper")
ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c('Astrocytes', 'Radial glia')] = 'RG/Astro'
ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c('EX', 'EX IPCs')] = 'EX'
ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c('IN IPCs', 'IN_CGE', 'IN_local', 'IN_MGE')] = 'IN'
ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c('Oligodendrocytes', 'OPCs')] = 'Oligo'
ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c('Dividing')] = 'Dividing'
table(ls_synthesis_v3_harmony_v3@meta.data$lineage_trajectories_paper)
#Dividing       EX       IN    Oligo RG/Astro 
#8941    66237    12118     3004     7814
div_col = "#e091e3"
ex_col = "#f2cb30"
in_col = '#fc7253'
olig_col = '#52ba86'
glia_col = '#9056d6'

DimPlot(subset(ls_synthesis_v3_harmony_v3, lineage_trajectories_paper %in% c(NA), invert=T), 
        group.by = "lineage_trajectories_paper", raster=F, cols = c('#e3b1d9', '#ebda96', '#f7a38f', '#8bad9b', '#b6a3cc')) +
  ggtitle('Lineage trajectories') +
  theme(legend.position = 'none', axis.text = element_blank(), axis.ticks = element_blank(), plot.title = element_text(size=30, hjust=0.5))
image = DimPlot(subset(ls_synthesis_v3_harmony_v3, lineage_trajectories_paper %in% c(NA), invert=T), 
                group.by = "lineage_trajectories_paper", raster=F, cols = c('#e3b1d9', '#ebda96', '#f7a38f', '#8bad9b', '#b6a3cc')) +
  ggtitle('Lineage trajectories') +
  theme(legend.position = 'none', axis.text = element_blank(), axis.ticks = element_blank(), plot.title = element_text(size=30, hjust=0.5))
ggsave('ls_synthesis_umap_lineage_trajectories.png', image, width=12, height=8)
ggsave('ls_synthesis_umap_lineage_trajectories.svg', image, width=12, height=8)

DimPlot(subset(ls_synthesis_v3_harmony_v3, lineage_trajectories_paper %in% c(NA), invert=T), 
                group.by = "broad_celltype", raster=F, cols = c(ex_col, glia_col, in_col, olig_col)) +
  ggtitle('Broad celltypes') +
  theme(legend.position = 'none', axis.text = element_blank(), axis.ticks = element_blank(), plot.title = element_text(size=30, hjust=0.5))

#plot some individual marker genes
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "#275c42")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "#275c42")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_olig2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_olig2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MBP"),raster=F, cols=c("#e6e8e7", "#275c42")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MBP"),raster=F, cols=c("#e6e8e7", "#275c42")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_mbp.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_mbp.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NEUROD2"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NEUROD2"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_neurod2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_neurod2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SATB2"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SATB2"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_satb2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_satb2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_vim.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_vim.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GFAP"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GFAP"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_gfap.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_gfap.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("HES1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("HES1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_hes1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_hes1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("AQP4"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("AQP4"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_aqp4.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_aqp4.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DLX2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DLX2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_dlx2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_dlx2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GAD2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_gad2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_gad2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("LHX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("LHX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_lhx6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_lhx6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR2F2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR2F2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_nr2f2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_nr2f2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MKI67"),raster=F, cols=c("#e6e8e7", "#b028b5")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("MKI67"),raster=F, cols=c("#e6e8e7", "#b028b5")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_mki67.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_mki67.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TOP2A"),raster=F, cols=c("#e6e8e7", "#b028b5")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TOP2A"),raster=F, cols=c("#e6e8e7", "#b028b5")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_top2a.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_top2a.png", plot=image, width=4, height=3.2)



head(multicell_full_metadata_unique)
table(multicell_full_metadata_unique$clonal_restriction)
multicell_full_metadata_unique[multicell_full_metadata_unique$clonal_restriction %in% "quadpotent",]
DimPlot(ls_synthesis_v3_harmony_v3, raster=F,
        cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS34_Clone_3"],
        sizes.highlight = 3) + 
  ggtitle("Representative clone") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=30))
image = DimPlot(ls_synthesis_v3_harmony_v3, raster=F,
                cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS34_Clone_3"],
                sizes.highlight = 3) + 
  ggtitle("Representative clone") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = 'none', plot.title = element_text(hjust = 0.5, size=30))
ggsave(file="ls_synthesis_v3_umap_rep_clone.svg", plot=image, width=10, height=8)
ggsave(file="ls_synthesis_v3_umap_rep_clone.png", plot=image, width=10, height=8)

ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS34_Clone_3",]
barchart_color_palette = c("#FFFFB3","#9056d6","#fc7253","#8DD3C7")
#barchart_color_palette = c("#FFFFB3","#BEBADA","#FB8072","#8DD3C7")

ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS34_Clone_3",], aes(fill = broad_celltype, x=Clone_IDs_filt)) +
  geom_bar(position = 'fill', width=0.1) +
  scale_fill_manual(values = barchart_color_palette, name = "Broad celltype") +
  ylab("Percentage of cell type in clone") +
  xlab("Clone") +
  ggtitle("Clonal composition -- representative clone") + 
  theme_bw() +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5, size=25), 
        legend.text = element_text(size=15), legend.title = element_text(size=20), axis.title = element_text(size=20), axis.text.y = element_text(size=15))
image = ggplot(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS34_Clone_3",], aes(fill = broad_celltype, x=Clone_IDs_filt)) +
  geom_bar(position = 'fill', width=0.1) +
  scale_fill_manual(values = barchart_color_palette, name = "Broad celltype") +
  ylab("Percentage of cell type in clone") +
  xlab("Clone") +
  ggtitle("Clonal composition -- representative clone") + 
  theme_bw() +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5, size=25), 
        legend.text = element_text(size=15), legend.title = element_text(size=20), axis.title = element_text(size=20), axis.text.y = element_text(size=15))
ggsave(file="rep_clone_barchart.svg", plot=image, width=10, height=8)

all_clone_list = unique(ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3, 'Clone_IDs_filt'])
clonal_comp_stacked_barchart_func_by_clones_indices_combined = function(clone_list, chart_title) {
  clone_filter = clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[clone_filter,]
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette, name = "Broad celltype") +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5, size=25), 
          legend.text = element_text(size=15), legend.title = element_text(size=20), axis.title = element_text(size=20), axis.text.y = element_text(size=15))+
    ylab("Percentage of cell type in clone") +
    xlab("Clone") +
    ggtitle(chart_title)
}
clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "Clonal composition -- all multicellular clones")
image = clonal_comp_stacked_barchart_func_by_clones_indices_combined(all_clone_list, "Clonal composition -- all multicellular clones")
ggsave(file="all_clones_barchart.svg", plot=image, width=14, height=8)


## 12/6/23
## moving on to figure 2 and 3 high quality charts(?)
## also need to try to do the upset plot for Fig 1
library(Seurat)
library(dplyr)
library(ggplot2)
DimPlot(ls_in_v2)
ls_in_v2 = AddMetaData(ls_in_v2, NA, "subclust_idents_paper")
colnames(ls_in_v2@meta.data)
table(ls_in_v2@meta.data$subclust_idents_v2)
ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$subclust_idents_v2 %in% "IN_CGE"] = "IN_CGE"
ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$subclust_idents_v2 %in% "IN_MGE"] = "IN_MGE"
ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$subclust_idents_v2 %in% "IN_local"] = "IN_local"
ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$subclust_idents_v2 %in% "IPC_IN"] = "IN IPCs"
ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$seurat_clusters %in% c(8)] = "IN_OB"
ls_in_v2 = SetIdent(ls_in_v2, value="subclust_idents_paper")
DimPlot(ls_in_v2)
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231206_figs')
image = DimPlot(ls_in_v2, raster=F)
ggsave(file="ls_in_v2_full_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_in_v2_full_umap.png", plot=image, width=10, height=8)

FeaturePlot(ls_in_v2, features=c("PBX3", "ETV1", "MEIS2", "ERBB4"))
FeaturePlot(ls_in_v2, features=c("PBX3", "TSHZ1", "KLHL35", "CXCR4"))
ggplot(ls_in_v2@meta.data, aes(x=seurat_clusters, fill=orig.ident)) + geom_bar(position = "fill") +
  ggtitle('Sample contribution to clusters') +
  #scale_fill_manual(values=c(osvz_col, vz_col, 'grey'))+
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_in_v2, group.by="seurat_clusters", label=T)
FeaturePlot(ls_in_v2, features=c("PAX6"), order=T)
FeaturePlot(ls_in_v2, features=c("SOX6"), order=T)
FeaturePlot(ls_in_v2, features=c("SOX6"))

FeaturePlot(ls_in_v2, features=c("GAD2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("GAD2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_gad2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_gad2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("NPY"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("NPY"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_npy.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_npy.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("NR2F2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("NR2F2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_nr2f2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_nr2f2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("LHX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("LHX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_lhx6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_lhx6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("SOX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("SOX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_sox6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_sox6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("PAX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("PAX6"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_pax6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_pax6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("ERBB4"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("ERBB4"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_erbb4.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_erbb4.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("PBX3"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("PBX3"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_pbx3.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_pbx3.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_in_v2, features=c("SCGN"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("SCGN"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_scgn.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_scgn.png", plot=image, width=4, height=3.2)

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_size_corrected > 1], cols.highlight='#c44e33') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_size_corrected > 1], cols.highlight='#c44e33') + 
  ggtitle('IN Cells in multicellular clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="in_sub_clone_size_greater_than_1_umap.svg", plot=image, width=10, height=8)
ggsave(file="in_sub_clone_size_greater_than_1_umap.png", plot=image, width=10, height=8)

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in_v2@meta.data$Barcode_UMI_Count_filt>2], cols.highlight = "#f27527") + 
  ggtitle('IN Cells in EX/IN shared clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$ex_in_presence %in% "EX_IN_shared" & ls_in_v2@meta.data$Barcode_UMI_Count_filt>2], cols.highlight = "#f27527") + 
  ggtitle('IN Cells in EX/IN shared clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="in_cells_in_ex_in_shared_clones.svg", plot=image, width=10, height=8)
ggsave(file="in_cells_in_ex_in_shared_clones.png", plot=image, width=10, height=8)

barchart_color_palette = c("#FFFFB3","#9056d6","#fc7253","#8DD3C7")
clonal_comp_stacked_barchart_func_by_clones_age_sep = function(clone_list, chart_title) {
  age_filter = clone_cluster_pivot_idents$orig.ident %in% young_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_young_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot1 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title,"<=GW20"))
  age_filter = clone_cluster_pivot_idents$orig.ident %in% old_sample_list & clone_cluster_pivot_idents$Clone_IDs %in% clone_list
  pivot_plot = clone_cluster_pivot_idents[age_filter,]
  n_old_clones = length(pivot_plot[,1])/3
  length(pivot_plot[,1])/3
  clone_order = pivot_plot[order(pivot_plot$Cell_percentage[pivot_plot$Identity=="EX"]-pivot_plot$Cell_percentage[pivot_plot$Identity=="GLIA"]),1]
  plot2 = ggplot(pivot_plot, aes(fill=Identity, x=Clone_IDs, y=Cell_count)) +
    scale_x_discrete(limits = clone_order) +
    geom_bar(position='fill', stat='identity', width=1) +
    scale_fill_manual(values = barchart_color_palette) +
    theme_bw() +
    theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(), plot.title = element_text(hjust = 0.5))+
    ggtitle(paste(chart_title, ">GW20"))
  total_clones = n_young_clones+n_old_clones
  pct_young_clones = n_young_clones/total_clones
  pct_old_clones = n_old_clones/total_clones
  ggarrange(plot1, plot2, ncol=2, nrow=1, common.legend=TRUE, legend="right", 
            widths = c(pct_young_clones, 
                       pct_old_clones)
  )
}
clonal_comp_stacked_barchart_func_by_clones_age_sep(all_clone_list, "Clonal composition -- all multicellular clones")
image = clonal_comp_stacked_barchart_func_by_clones_age_sep(all_clone_list, "Clonal composition -- all multicellular clones")
ggsave(file="all_clones_age_sep_barchart.svg", plot=image, width=14, height=8)

DotPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_v2 %in% c("unk", "IN_UNK", "Transitioning OPCs/IPCs"), invert=T), features=c("MKI67", "DLX2", "PAX6", "SCGN", "SOX6"))
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c(
  "IN_MGE",
  "IN_OB",
  "IN_CGE",
  "IN_local",
  "IN IPCs"))
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c(
  "IN IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"))
DotPlot(ls_in_v2, features=c("GAD2", "MKI67", "DLX2", "PAX6", "SCGN", "SOX6", "ERBB4", "NR2F2", "LHX6", "PBX3")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
        )
image = DotPlot(ls_in_v2, features=c("GAD2", "MKI67", "DLX2", "PAX6", "SCGN", "SOX6", "ERBB4", "NR2F2", "LHX6", "PBX3")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
ggsave(file="in_marker_dotplot.svg", plot=image, width=14, height=8)

FeaturePlot(ls_in_v2, features=c("ERBB4"))

ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  geom_point(data=ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  scale_fill_manual(values=c("#cccccc","#6b6b6b"))+
  ggtitle('Clonal potency by age, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), 
        axis.text = element_text(size=20), 
        axis.title = element_text(size=25),
        legend.title = element_text(size=25),
        legend.text = element_text(size=20)
        ) +
  xlab('Clonal potency') +
  ylab('Percentage of clones from sample') +
  labs(fill="Sample age")
image = ggplot(compiled_clust_df_idents, aes(x=ex_in_presence, fill=as.factor(age_pseudo), group=age_pseudo)) +
  geom_bar(position="dodge", aes(y=after_stat(prop))) +
  geom_point(data=ex_in_freq_df_pivot, aes(x=x_coords, y=ex_in_pct, fill=age_pseudo))+
  scale_fill_manual(values=c("#cccccc","#6b6b6b"))+
  ggtitle('Clonal potency by age, all samples') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), 
        axis.text = element_text(size=20), 
        axis.title = element_text(size=25),
        legend.title = element_text(size=25),
        legend.text = element_text(size=20)
  ) +
  xlab('Clonal potency') +
  ylab('Percentage of clones from sample') +
  labs(fill="Sample age")
ggsave(file="clonal_potency_by_age_with_sample_dots.svg", plot=image, width=14, height=8)


#adding column to compiled_clust_df_idents to make an upset plot
dim(compiled_clust_df_idents)
head(compiled_clust_df_idents)
compiled_clust_df_idents$upset = NA
table(compiled_clust_df_idents$clonal_restriction)
#insane dplyr usage to get column names in compiled_clust_df_idents that don't have a 0 to get upset plot syntax right
paste(colnames(compiled_clust_df_idents[1,2:5] %>% select(where(~ any(. != 0)))), collapse="&")
#for loop to get all rows in the table:
upset_notations = c()
for(r in 1:nrow(compiled_clust_df_idents)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_idents[r,2:5] %>% select(where(~ any(. != 0)))), collapse="&"))
}
upset_notations
compiled_clust_df_idents$upset = upset_notations
head(compiled_clust_df_idents)
#install.packages("UpSetR")
library(UpSetR)

table(compiled_clust_df_idents$upset)
upset(fromExpression(table(compiled_clust_df_idents$upset)),
      order.by = "freq",
      decreasing = T)
svg('all_clones_upset_plot.svg', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_idents$upset)),
              order.by = "freq",
              decreasing = T)
dev.off()
pdf('all_clones_upset_plot.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_idents$upset)),
      order.by = "freq",
      decreasing = T)
dev.off()

upset(fromExpression(table(compiled_clust_df_idents$upset[compiled_clust_df_idents$age_pseudo %in% "<=GW20"])),
      order.by = "freq",
      decreasing = T,
      empty.intersections = "on")
upset(fromExpression(table(compiled_clust_df_idents$upset[compiled_clust_df_idents$age_pseudo %in% ">GW20"])),
      order.by = "freq",
      decreasing = T,
      empty.intersections = "on")


manual_quant_scgn_df = data.frame(c(17,17,17,19.5,19.5,19.5,20,20,20,24.5,24.5,24.5,25,25,25), 
                                  row.names = paste(c("17","17","17","19.5","19.5","19.5","20","20","20","24.5","24.5","24.5","25","25","25"), c("_1", "_2", "_3"), sep="")
                                  )
colnames(manual_quant_scgn_df) = "age"
manual_quant_scgn_df$triple_counts = c(0,0,0,0,0,3,2,0,2,10,7,6,7,1,3)
manual_quant_scgn_df$dlx2_counts = c(65,40,68,169,154,208,316,283,315,219,235,306,129,59,162)
manual_quant_scgn_df$pct_triple = c(0,0,0,0,0,0.014423077,0.006329114,0,0.006349206,0.0456621,0.029787234,0.019607843,0.054263566,0.016949153,0.018518519)
manual_quant_scgn_df$age_pseudo = c("<=GW20", "<=GW20", "<=GW20", "<=GW20", "<=GW20", "<=GW20", "<=GW20", "<=GW20", "<=GW20", ">GW20", ">GW20", ">GW20", ">GW20", ">GW20", ">GW20")
ggplot(manual_quant_scgn_df, aes(x=age_pseudo, y=pct_triple)) +
  geom_bar()

paste(c("17","17","17","19.5","19.5","19.5","20","20","20","24.5","24.5","24.5","25","25","25"), c("_1", "_2", "_3"), sep="")


## Figure 3?
table(ls_glia_v2_harmony@meta.data$glial_subtype)
DimPlot(ls_glia_v2_harmony, label=T, group.by="Phase")
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="glial_subtype")
RidgePlot(ls_glia_v2_harmony, features = c("PCNA", "TOP2A", "MCM6", "MKI67"), ncol = 2)

FeaturePlot(ls_glia_v2_harmony, features=c("TOP2A", "OLIG2", "EOMES", "DLX2"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("VIM","EOMES","AQP4"), raster=F)
DimPlot(ls_glia_v2_harmony)
VlnPlot(ls_glia_v2_harmony, features=c("EOMES"))
VlnPlot(ls_glia_v2_harmony, features=c("OLIG2"))
FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"))
FeaturePlot(ls_glia_v2_harmony, features=c("HOPX"))
FeaturePlot(ls_glia_v2_harmony, features=c("MT3"))
FeaturePlot(ls_glia_v2_harmony, features=c("GJA1"))
FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB"))

FeaturePlot(ls_glia_v2_harmony, features=c("TOP2A"))
FeaturePlot(ls_glia_v2_harmony, features=c("EOMES"))
colnames(ls_glia_v2_harmony@meta.data)

#really trying to isolate the EOMES+ from OLIG2+ in the middle clusters
ls_glia_v2_harmony <- FindClusters(ls_glia_v2_harmony, resolution = .6)
ls_glia_v2_harmony <- RunUMAP(ls_glia_v2_harmony, reduction = "harmony", dims=c(1:15))
DimPlot(ls_glia_v2_harmony, reduction = "umap",label=T)
FeaturePlot(ls_glia_v2_harmony, features=c("EOMES"))
FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"))
VlnPlot(ls_glia_v2_harmony, features=c("EOMES", "OLIG2"))
FeaturePlot(ls_glia_v2_harmony, features=c("nCount_RNA"), max.cutoff = "q90")

ls_glia_v2_harmony <- FindClusters(ls_glia_v2_harmony, resolution = .8)
ls_glia_v2_harmony <- RunUMAP(ls_glia_v2_harmony, reduction = "harmony", dims=c(1:15))
DimPlot(ls_glia_v2_harmony, reduction = "umap",label=T)
FeaturePlot(ls_glia_v2_harmony, features=c("EOMES"))
FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"))
DimPlot(ls_glia_v2_harmony, group.by="glial_subtype",label=T)
table(ls_glia_v2_harmony@meta.data$seurat_clusters)

table(ls_glia_v2_harmony$glial_subtype)
#Astrocytes       G2M dividing                 RG  RG (putative oRG)  RG (putative tRG)         S dividing 
#1368               2374               2412                802               1068               1359 
#Transitioning IPCs Transitioning OPCs 
#725               1439
sum(ls_glia_v2_harmony$glial_subtype %in% c('Transitioning IPCs', 'Transitioning OPCs'))
#[1] 2164
ls_glia_sub_sub = subset(ls_glia_v2_harmony, glial_subtype %in% c('Transitioning IPCs', 'Transitioning OPCs'))
ls_glia_sub_sub <- ScaleData(ls_glia_sub_sub, features = all.genes)
ls_glia_sub_sub = FindVariableFeatures(ls_glia_sub_sub, selection.method="vst", nfeatures=2000)
ls_glia_sub_sub = RunPCA(ls_glia_sub_sub, features=VariableFeatures(object=ls_glia_sub_sub))
ElbowPlot(ls_glia_sub_sub)
ls_glia_sub_sub <- RunHarmony(ls_glia_sub_sub, c("orig.ident"))
ls_glia_sub_sub <- FindNeighbors(ls_glia_sub_sub, reduction = "harmony", dims = 1:15)
ls_glia_sub_sub <- FindNeighbors(ls_glia_sub_sub, dims = 1:15)
ls_glia_sub_sub <- FindClusters(ls_glia_sub_sub, resolution = .5)
#ls_glia_sub_sub <- RunUMAP(ls_glia_sub_sub, reduction = "harmony", dims=c(1:15))
ls_glia_sub_sub <- RunUMAP(ls_glia_sub_sub, dims=c(1:15))
DimPlot(ls_glia_sub_sub, reduction = "umap",label=T)
FeaturePlot(ls_glia_sub_sub, features=c("EOMES", "OLIG2"), label=T)
colnames(ls_glia_sub_sub@meta.data)
DimPlot(ls_glia_sub_sub, group.by="glial_subtype")
DimPlot(ls_glia_sub_sub, group.by="orig.ident")
table(ls_glia_sub_sub@meta.data$glial_subtype)
#Transitioning IPCs Transitioning OPCs 
#725               1439
ls_glia_sub_sub = AddMetaData(ls_glia_sub_sub, NA, "glial_subtype_v2")
ls_glia_sub_sub@meta.data$glial_subtype[ls_glia_sub_sub@meta.data$seurat_clusters %in% c(0,4,5)] = "Transitioning IPCs"
ls_glia_sub_sub@meta.data$glial_subtype[ls_glia_sub_sub@meta.data$seurat_clusters %in% c(1,2,3,6)] = "Transitioning OPCs"
table(ls_glia_sub_sub@meta.data$glial_subtype)
#Transitioning IPCs Transitioning OPCs 
#941               1223

ls_glia_v2_harmony = AddMetaData(ls_glia_v2_harmony, ls_glia_v2_harmony$glial_subtype, "glial_subtype_v2")
table(ls_glia_v2_harmony@meta.data$glial_subtype_v2)
#Astrocytes       G2M dividing                 RG  RG (putative oRG)  RG (putative tRG)         S dividing 
#1368               2374               2412                802               1068               1359 
#Transitioning IPCs Transitioning OPCs 
#725               1439 
ls_glia_v2_harmony@meta.data$glial_subtype_v2[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_sub@meta.data)[ls_glia_sub_sub@meta.data$glial_subtype %in% "Transitioning IPCs"]] = "Transitioning IPCs"
ls_glia_v2_harmony@meta.data$glial_subtype_v2[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_sub@meta.data)[ls_glia_sub_sub@meta.data$glial_subtype %in% "Transitioning OPCs"]] = "Transitioning OPCs"
table(ls_glia_v2_harmony@meta.data$glial_subtype_v2)
#Astrocytes       G2M dividing                 RG  RG (putative oRG)  RG (putative tRG)         S dividing 
#1368               2374               2412                802               1068               1359 
#Transitioning IPCs Transitioning OPCs 
#941               1223
DimPlot(ls_glia_v2_harmony, group.by="glial_subtype_v2")


DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data)[ls_glia_v2_harmony@meta.data$Clone_size_corrected > 1], cols.highlight='#522985') + 
  ggtitle('Glial cells in multicellular clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))

FeaturePlot(ls_glia_v2_harmony, features=c("GFAP"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("GFAP"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_gfap.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_gfap.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("VIM"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("VIM"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_VIM.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_VIM.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_OLIG2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_OLIG2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("PPP1R17"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("PPP1R17"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_PPP1R17.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_PPP1R17.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("EOMES"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("EOMES"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_EOMES.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_EOMES.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_CRYAB.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_CRYAB.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("HOPX"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("HOPX"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_HOPX.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_HOPX.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("TOP2A"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("TOP2A"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_TOP2A.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_TOP2A.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("PCNA"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("PCNA"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_PCNA.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_PCNA.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("MT3"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("MT3"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_MT3.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_MT3.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("GJA1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("GJA1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_GJA1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_GJA1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("AQP4"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("AQP4"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_AQP4.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_AQP4.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("NEUROD2"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("NEUROD2"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_NEUROD2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_NEUROD2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("NEUROD6"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("NEUROD6"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_NEUROD6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_NEUROD6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("HES1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("HES1"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_HES1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_HES1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("HES5"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("HES5"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_HES5.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_HES5.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_glia_v2_harmony, features=c("HES6"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_glia_v2_harmony, features=c("HES6"),raster=F, cols=c("#e6e8e7", "#522985")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_glia_umap_HES6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_glia_HES6.png", plot=image, width=4, height=3.2)


ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="glial_subtype_v2")
DimPlot(ls_glia_v2_harmony, label=F, raster=F)
FeaturePlot(ls_glia_v2_harmony, features=c("EOMES", "OLIG2"))
ls_glia_v2_harmony = AddMetaData(ls_glia_v2_harmony, ls_glia_v2_harmony$glial_subtype_v2, "subclust_idents_paper")
ls_glia_v2_harmony@meta.data$subclust_idents_paper = ls_glia_v2_harmony@meta.data$glial_subtype_v2
ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$glial_subtype %in% c('G2M dividing')] =
  "Dividing (G2M phase)"
ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$glial_subtype %in% c('S dividing')] =
  "Dividing (S phase)"
ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$glial_subtype %in% c('RG (putative tRG)')] =
  "tRG"
ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$glial_subtype %in% c('RG (putative oRG)')] =
  "oRG"
ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$seurat_clusters %in% c(13)] =
  NA
#ls_glia_v2_harmony@meta.data$subclust_idents_paper[ls_glia_v2_harmony@meta.data$glial_subtype %in% c('RG')] =
#  "Astrocytes" ##hard decision here, but based on lack of multicellular clones, thinking is now that these "RG" are actually immature astrocytes
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="subclust_idents_paper")
DimPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% NA, invert=T), raster=F) +
  theme(legend.text = element_text(size=16))
image = DimPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% NA, invert=T), raster=F) +
  theme(legend.text = element_text(size=16))
ggsave(file="ls_glia_full_umap.svg", plot=image, width=12, height=8)
ggsave(file="ls_glia_full_umap.png", plot=image, width=12, height=8)


osvz_col = "#ff6666"
vz_col = "#00bcc2"
table(ls_glia_v2_harmony@meta.data$subclust_idents_paper)
#Astrocytes Dividing (G2M phase)   Dividing (S phase)                  oRG                   RG   Transitioning IPCs 
#1368                 2374                 1359                  769                 2412                  970 
#Transitioning OPCs                  tRG 
#1194                 1068 
table(ls_glia_v2_harmony@meta.data$Clone_Index_filt)
#Index3 IndexE 
#4184   4244 
dim(ls_glia_v2_harmony@meta.data)
#[1] 11647    40

ls_glia_v2_harmony@active.ident = factor(ls_glia_v2_harmony@active.ident, levels=c(
  "Dividing (S phase)",
  "Dividing (G2M phase)",
  "RG", 
  "oRG",
  "tRG",
  "Astrocytes",
  "Transitioning OPCs",
  "Transitioning IPCs"))
ls_glia_v2_harmony$subclust_idents_paper = factor(ls_glia_v2_harmony$subclust_idents_paper, levels = c(
  "Dividing (S phase)",
  "Dividing (G2M phase)",
  "RG", 
  "oRG",
  "tRG",
  "Astrocytes",
  "Transitioning OPCs",
  "Transitioning IPCs"
))
ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG"))@meta.data, 
       aes(x=subclust_idents_paper, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('GZ origin') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
image = ggplot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG"))@meta.data, 
               aes(x=subclust_idents_paper, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('GZ origin') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggsave(file="ls_glia_gz_contribution_to_clusters.svg", plot=image, width=14, height=8)
ggsave(file="ls_glia_gz_contribution_to_clusters.png", plot=image, width=14, height=8)

table(ls_glia_v2_harmony@meta.data$subclust_idents_paper)
dim(subset(ls_glia_v2_harmony@meta.data, Clone_Index_filt %in% c("Index3", "IndexE") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG")))

ls_glia_v2_harmony = SetIdent()
dim(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE")))
table(ls_glia_v2_harmony$Clone_Index_filt)
DimPlot(ls_glia_v2_harmony)
table(ls_glia_v2_harmony$ex_in_presence)

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="subclust_idents_paper")
DimPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","Transitioning IPCs","RG","Transitioning OPCs","tRG")),
               group.by="Clone_Index_filt", cols=c(osvz_col, vz_col, "#e6e8e7"), pt.size=1) +
  ggtitle('GZ origin') +
  theme(plot.title = element_text(size=30, hjust=0.5), legend.text=element_text(size=20))
image = DimPlot(subset(ls_glia_v2_harmony, Clone_Index_filt %in% c("Index3", "IndexE") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG")),
                group.by="Clone_Index_filt", cols=c(osvz_col, vz_col, "#e6e8e7"), pt.size=1) +
  ggtitle('GZ origin') +
  theme(plot.title = element_text(size=30, hjust=0.5), legend.text=element_text(size=20))
ggsave(file="ls_glia_umap_gz_origin.svg", plot=image, width=14, height=8)
ggsave(file="ls_glia_umap_gz_origin.png", plot=image, width=14, height=8)

table(ls_glia_v2_harmony@meta.data$Clone_Index)
table(ls_glia_v2_harmony@meta.data$Clone_Index_filt)
ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="Clone_Index_filt")
VlnPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("RG","tRG","oRG") & Clone_Index_filt %in% c("Index3","IndexE")), features=c("CRYAB", "HOPX", "MT3"), pt.size=0)

VlnPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("RG","tRG","oRG") & Clone_Index_filt %in% c("Index3","IndexE")), features=c("ANGPTL1"), pt.size=0)
VlnPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("RG","tRG","oRG") & Clone_Index_filt %in% c("Index3","IndexE")), features=c("ITGA2"), pt.size=0)

FeaturePlot(ls_glia_v2_harmony, features=c("EGFR", "PDGFRA","F3","HOPX"))
DimPlot(ls_glia_v2_harmony, group.by="subclust_idents_paper", label=T)


FeaturePlot(ls_glia_v2_harmony, features=c("ANGPTL1", "GLI3", "ITGA2", "FBXO32"))
FeaturePlot(ls_glia_v2_harmony, features=c("CRYAB", "ANGPTL1", "HOPX", "MT3"))
FeaturePlot(ls_glia_v2_harmony, features=c("CCL2", "S100A6", "ID3", "NAMPT"))

VlnPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("RG","tRG","oRG") & Clone_Index_filt %in% c("Index3","IndexE")), features=c("CRYAB", "HOPX", "MT3"), pt.size=0)
image = VlnPlot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("RG","tRG","oRG") & Clone_Index_filt %in% c("Index3","IndexE")), features=c("CRYAB", "HOPX", "MT3"), pt.size=0)
ggsave(file="ls_glia_rg_markers_vlnplot_gz_origin.svg", plot=image, width=14, height=8)
ggsave(file="ls_glia_rg_markers_vlnplot_gz_origin.png", plot=image, width=14, height=8)

tRG_clone_list = ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$subclust_idents_paper %in% 'tRG', 'Clone_IDs_filt']
tRG_clone_list = unique(tRG_clone_list)
oRG_clone_list = ls_glia_v2_harmony@meta.data[ls_glia_v2_harmony@meta.data$subclust_idents_paper %in% 'oRG', 'Clone_IDs_filt']
oRG_clone_list = unique(oRG_clone_list)

pdf('trg_clones_upset_plot.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_idents$upset[compiled_clust_df_idents$Clone_IDs_filt %in% tRG_clone_list])),
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('org_clones_upset_plot.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_idents$upset[compiled_clust_df_idents$Clone_IDs_filt %in% oRG_clone_list])),
      order.by = "freq",
      decreasing = T,
      main.bar.color = osvz_col)
dev.off()

table(compiled_clust_df_idents$upset[compiled_clust_df_idents$tRG_oRG_clone %in% c("tRG")])
table(compiled_clust_df_idents$upset[compiled_clust_df_idents$tRG_oRG_clone %in% c("oRG")])

compiled_clust_df_idents$upset = factor(compiled_clust_df_idents$upset, levels=c(
  "GLIA",
  "EX&GLIA",
  "IN&GLIA",
  "GLIA&OLIG",
  "IN&GLIA&OLIG",
  "EX&IN&GLIA",
  "EX&GLIA&OLIG",
  "EX&IN&GLIA&OLIG"
))
ggplot(subset(compiled_clust_df_idents, tRG_oRG_clone %in% c("oRG", "tRG")), aes(x=upset, fill=tRG_oRG_clone)) +
  geom_bar(position="dodge")
image = ggplot(subset(compiled_clust_df_idents, tRG_oRG_clone %in% c("oRG", "tRG")), aes(x=upset, fill=tRG_oRG_clone)) +
  geom_bar(position="dodge")
ggsave(file="trg_org_barcharts_for_comb_upset.svg", plot=image, width=14, height=8)

ls_glia_v2_harmony = SetIdent(ls_glia_v2_harmony, value="ex_in_presence")
levels(ls_glia_v2_harmony)
#de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & seurat_clusters %in% c(0,1,2,5,7,8) & sample_age %in% c(18, 19, 20, 21, 22, 23, 24)), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & subclust_idents_paper %in% c("tRG", "oRG", "RG")), ident.1 = "EX_only", ident.2 = "no_EX_or_IN")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))
image = ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))
ggsave('volcano_ls_glia_rg_clusts_gliogenic_vs_exgenic.svg', image, width=12, height=8)

de.markers.glia$labels = rownames(de.markers.glia)
tf_int = intersect(de.markers.glia$labels, tf_list$TF)
custom_labels = de.markers.glia |> filter(labels %in% tf_int)
ggplot(data=de.markers.glia, aes(x =avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_label_repel(aes(label=labels), data=custom_labels)+
  ggtitle("Differential expression between Gliogenic-only glia (left) and EX-neurogenic glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))

de.markers.glia = FindMarkers(subset(ls_glia_v2_harmony, Clone_size_corrected>=3 & subclust_idents_paper %in% c("tRG", "oRG", "RG")), ident.1 = "EX_only", ident.2 = "IN_only")
head(de.markers.glia, n=20)
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between IN-neurogenic glia (left) and EX-neurogenic glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))
image = ggplot(data=de.markers.glia, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between IN-neurogenic glia (left) and EX-neurogenic glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))
ggsave('volcano_ls_glia_rg_clusts_inhgenic_vs_exgenic.svg', image, width=12, height=8)


de.markers.glia$labels = rownames(de.markers.glia)
tf_int = intersect(de.markers.glia$labels, tf_list$TF)
custom_labels = de.markers.glia |> filter(labels %in% tf_int)
ggplot(data=de.markers.glia, aes(x =avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.glia)))+
  geom_point()+
  theme_minimal()+
  geom_label_repel(aes(label=labels), data=custom_labels)+
  ggtitle("Differential expression between IN-neurogenic glia (left) and EX-neurogenic glia (right) -- tRG, oRG, and RG clone size >=3")+
  theme(plot.title = element_text(hjust = 0.5))


ggplot(subset(ls_glia_v2_harmony, ex_in_presence %in% c("EX_only", "IN_only", "EX_IN_shared", "no_EX_or_IN") & subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG"))@meta.data, 
       aes(x=subclust_idents_paper, fill=ex_in_presence)) +
  geom_bar(position = "fill") +
  ggtitle('Cluster composition by GZ origin') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")

ggplot(subset(ls_glia_v2_harmony, subclust_idents_paper %in% c("Astrocytes","Dividing (G2M phase)","Dividing (S phase)","oRG","RG","Transitioning IPCs","RG","Transitioning OPCs","tRG"))@meta.data, 
       aes(x=subclust_idents_paper, fill=orig.ident)) +
  geom_bar(position = "fill") +
  ggtitle('Sample origin') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="Sample", y="Proportion of cells")

DotPlot(ls_glia_v2_harmony, features=c("VIM", "GFAP", "TOP2A", "PCNA", "CRYAB", "HOPX", "GJA1", "AQP4")) +
  ggtitle('Astrocyte marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  ) ##can't make a dotplot with a harmony for some reason???


clonal_comp_stacked_barchart_func_by_clones_indices_combined(tRG_clone_list, "tRG-containing clones")
image = clonal_comp_stacked_barchart_func_by_clones_indices_combined(tRG_clone_list, "tRG-containing clones")
ggsave('clonal_comp_barchart_tRG_clones.svg', image, width=10, height=8)

clonal_comp_stacked_barchart_func_by_clones_indices_combined(oRG_clone_list, "oRG-containing clones")
image = clonal_comp_stacked_barchart_func_by_clones_indices_combined(oRG_clone_list, "oRG-containing clones")
ggsave('clonal_comp_barchart_oRG_clones.svg', image, width=10, height=8)

length(tRG_clone_list)
length(oRG_clone_list)

DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% tRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in tRG-containing clones") +
  theme(plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% tRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in tRG-containing clones") +
  theme(plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_tRG_clones.svg', image, width=12, height=8)
ggsave('ls_full_umap_tRG_clones.png', image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#f382bc", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% oRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in oRG-containing clones") +
  theme(plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#f382bc", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% oRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in oRG-containing clones") +
  theme(plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_oRG_clones.svg', image, width=12, height=8)
ggsave('ls_full_umap_oRG_clones.png', image, width=12, height=8)

length(tRG_clone_list_v2)



## Figure 4?
#making a binary graph for EX presence
table(compiled_clust_df_idents$ex_in_presence)
compiled_clust_df_idents$ex_binary = NA
compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$ex_in_presence %in% c('EX_only', 'EX_IN_shared')] = "EX_present"
compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$ex_in_presence %in% c('IN_only', 'no_EX_or_IN')] = "no_EX"
ggplot(compiled_clust_df_idents, aes(x=ex_binary, fill=Clone_Index)) +
  geom_bar(position = "dodge") +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  #geom_jitter(position=position_jitter(0.05)) +
  ggtitle('Percentage of clones containing EX cells in clones by index') +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('EX presence')

table(compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#EX_present      no_EX 
#316        190
table(compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#EX_present      no_EX 
#1404        259
table(compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#EX_present      no_EX 
#88        684
table(compiled_clust_df_idents$ex_binary[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#EX_present      no_EX 
#51        265
ex_binary_df = data.frame(c(316, 190, 1404, 259, 88, 684, 51, 265))
colnames(ex_binary_df) = c("counts")
ex_binary_df$age_pseudo = c('<=GW20', '<=GW20', '<=GW20', '<=GW20', '>GW20', '>GW20', '>GW20', '>GW20')
ex_binary_df$Clone_Index = c('Index3', 'Index3', 'IndexE', 'IndexE', 'Index3', 'Index3', 'IndexE', 'IndexE')
ex_binary_df$ex_presence = c('EX_present', 'no_EX', 'EX_present', 'no_EX', 'EX_present', 'no_EX', 'EX_present', 'no_EX')

ex_binary_df$pct = c(316/(316+190), 190/(316+190), 
                     1404/(1404+259), 259/(1404+259),
                     88/(88+684), 684/(88+684),
                     51/(51+265), 265/(51+265))
ggplot(data=subset(ex_binary_df, ex_presence == "EX_present"), aes(x=age_pseudo, y=pct, fill=Clone_Index)) +
  geom_bar(stat="identity", position=position_dodge()) +
  ggtitle("Percentage of clones with at least one EX cell") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), 
        axis.text = element_text(size=20), 
        axis.title = element_text(size=25),
        legend.title = element_text(size=25),
        legend.text = element_text(size=20)
  ) +
  xlab('Sample age') +
  ylab('Percent of clones') +
  labs('GZ labeled')
image = ggplot(data=subset(ex_binary_df, ex_presence == "EX_present"), aes(x=age_pseudo, y=pct, fill=Clone_Index)) +
  geom_bar(stat="identity", position=position_dodge()) +
  ggtitle("Percentage of clones with at least one EX cell") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), 
        axis.text = element_text(size=20), 
        axis.title = element_text(size=25),
        legend.title = element_text(size=25),
        legend.text = element_text(size=20)
  ) +
  xlab('Sample age') +
  ylab('Percent of clones') +
  labs('GZ labeled')
ggsave(file="pct_clones_ex_binary.svg", plot=image, width=10, height=8)

(51/(51+265))/(88/(88+684))
(1404/(1404+259))/(316/(316+190))


table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#263          119           53           71
table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$age_pseudo == '<=GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#1175          150          229          109
table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'Index3'] )
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#36          467           52          217
table(compiled_clust_df_idents$ex_in_presence[compiled_clust_df_idents$age_pseudo == '>GW20' & compiled_clust_df_idents$Clone_Index == 'IndexE'] )
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#23          213           28           52
ex_only_binary_df = data.frame(c(263, 119+53+71, 1175, 150+229+109, 36, 467+52+217, 23, 213+28+52))
colnames(ex_only_binary_df) = c("counts")
ex_only_binary_df$age_pseudo = c('<=GW20', '<=GW20', '<=GW20', '<=GW20', '>GW20', '>GW20', '>GW20', '>GW20')
ex_only_binary_df$Clone_Index = c('Index3', 'Index3', 'IndexE', 'IndexE', 'Index3', 'Index3', 'IndexE', 'IndexE')
ex_only_binary_df$ex_presence = c('EX_present', 'no_EX', 'EX_present', 'no_EX', 'EX_present', 'no_EX', 'EX_present', 'no_EX')

ex_only_binary_df$pct = c(263/(316+190), (119+53+71)/(316+190), 
                          1175/(1404+259), (150+229+109)/(1404+259),
                          36/(88+684), (467+52+217)/(88+684),
                          23/(51+265), (213+28+52)/(51+265))
ggplot(data=subset(ex_only_binary_df, ex_presence == "EX_present"), aes(x=age_pseudo, y=pct, fill=Clone_Index)) +
  geom_bar(stat="identity", position=position_dodge()) +
  ggtitle("Percentage of clones with only EX by age and index") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab('Sample age') +
  ylab('Percent of clones')


library(eulerr)
svg('venn_diagram_young_clones.svg', width=12, height=8)
plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
dev.off()
svg('venn_diagram_old_clones.svg', width=12, height=8)
plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$EX>0,
  IN=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$IN>0,
  GLIA=subset(compiled_clust_df_idents, age_pseudo %in% '>GW20')$GLIA+subset(compiled_clust_df_idents, age_pseudo %in% '<=GW20')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
dev.off()

svg('venn_diagram_vz_clones.svg', width=12, height=8)
plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, Clone_Index %in% 'IndexE')$EX>0,
  IN=subset(compiled_clust_df_idents, Clone_Index %in% 'IndexE')$IN>0,
  GLIA=subset(compiled_clust_df_idents, Clone_Index %in% 'IndexE')$GLIA+subset(compiled_clust_df_idents, Clone_Index %in% 'IndexE')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
dev.off()
svg('venn_diagram_osvz_clones.svg', width=12, height=8)
plot(euler(cbind(
  EX=subset(compiled_clust_df_idents, Clone_Index %in% 'Index3')$EX>0,
  IN=subset(compiled_clust_df_idents, Clone_Index %in% 'Index3')$IN>0,
  GLIA=subset(compiled_clust_df_idents, Clone_Index %in% 'Index3')$GLIA+subset(compiled_clust_df_idents, Clone_Index %in% 'Index3')$OLIG>0)),
  fill = c("#FFFFB3", "#FB8072", "#BEBADA"))
dev.off()

ggplot(subset(ls_synthesis_v3_harmony_v3, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data, 
       aes(x=subclust_idents_paper, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('GZ origin') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
image = ggplot(subset(ls_synthesis_v3_harmony_v3, Clone_Index_filt %in% c("Index3", "IndexE"))@meta.data, 
               aes(x=subclust_idents_paper, fill=Clone_Index_filt)) +
  geom_bar(position = "fill") +
  ggtitle('GZ origin') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggsave(file="cluster_composition_by_gz_origin_whole_umap.svg", plot=image, width=10, height=8)

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
de.markers.ex = FindMarkers(subset(ls_synthesis_v3_harmony_v3, Clone_Index_filt %in% c("IndexE", "Index3") & subclust_idents_paper %in% c("EX", "EX IPCs")), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.ex, n=20)
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived EX (left) and VZ-derived EX (right)")+
  theme(plot.title = element_text(hjust = 0.5))
image = ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived EX (left) and VZ-derived EX (right)")+
  theme(plot.title = element_text(hjust = 0.5))
ggsave('volcano_ex_ipc_by_gz.svg', image, width=10, height=8)

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
de.markers.ex = FindMarkers(subset(ls_synthesis_v3_harmony_v3, Clone_Index_filt %in% c("IndexE", "Index3") & subclust_idents_paper %in% c("EX")), ident.1 = "IndexE", ident.2 = "Index3")
head(de.markers.ex, n=20)
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived EX (left) and VZ-derived EX (right)")+
  theme(plot.title = element_text(hjust = 0.5))
image = ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between OSVZ-derived EX (left) and VZ-derived EX (right)")+
  theme(plot.title = element_text(hjust = 0.5))
ggsave('volcano_ex_only_by_gz.svg', image, width=10, height=8)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_tbr1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_tbr1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("EOMES"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("EOMES"),raster=F, cols=c("#e6e8e7", "#b8af12")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_eomes.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_eomes.png", plot=image, width=4, height=3.2)


FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SLC17A7"),raster=F, cols=c("#e6e8e7", "#b8af12"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SLC17A7"),raster=F, cols=c("#e6e8e7", "#b8af12"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_synthesis_v3_umap_slc17a7.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v3_umap_slc17a7.png", plot=image, width=4, height=3.2)

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected > 1 &ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("EX IPCs", "EX")], cols.highlight='#b8af12', raster=F) + 
  ggtitle('EX Cells in multicellular clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected > 1 &ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% c("EX IPCs", "EX")], cols.highlight='#b8af12', raster=F) + 
  ggtitle('EX Cells in multicellular clones') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="ex_sub_clone_size_greater_than_1_umap.svg", plot=image, width=10, height=8)
ggsave(file="ex_sub_clone_size_greater_than_1_umap.png", plot=image, width=10, height=8)

## 12/10/23

## doing some random stuff for a bit
ls_synthesis_v3_harmony_v3@meta.data[10000:10005,]
table(ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count_filt)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count_filt <2], raster=F, sizes.highlight =0.1)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Barcode_UMI_Count_filt <2 & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected>=3], raster=F, sizes.highlight =0.1)

DimPlot(ls_synthesis_v3_harmony_v3, raster=F)
help(PCASigGenes)

TopFeatures(object = ls_synthesis_v3_harmony_v3[["pca"]], dim = 1, nfeatures = 20, balanced = TRUE)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("DBI", "SOX2", "RBFOX1", "CUX2"), raster=T)
DimPlot(ls_synthesis_v3_harmony_v3, group.by="seurat_clusters", raster=F, label=T)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="seurat_clusters")
de.markers.ex = FindMarkers(ls_synthesis_v3_harmony_v3, ident.1 = 0, ident.2 = 3)
head(de.markers.ex, n=20)
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=de.markers.ex, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(de.markers.ex)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between EX cluster 0 and cluster 3")+
  theme(plot.title = element_text(hjust = 0.5))
write.csv(de.markers.ex, '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/diff_expr_genes_ex_cluster0_vs_cluster3.csv')
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SOX11"), raster=F)

ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX', 'EX IPCs', 'IN IPCs', 'IN_CGE', 'IN_local', 'IN_MGE', 'Oligodendrocytes', 'OPCs', 'Radial glia'))@meta.data,
       aes(x=subclust_idents_paper, fill=orig.ident)) +
  geom_bar(position = "fill") +
  ggtitle('Sample origin') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="Sample", y="Proportion of cells")
image = ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX', 'EX IPCs', 'IN IPCs', 'IN_CGE', 'IN_local', 'IN_MGE', 'Oligodendrocytes', 'OPCs', 'Radial glia'))@meta.data,
               aes(x=subclust_idents_paper, fill=orig.ident)) +
  geom_bar(position = "fill") +
  ggtitle('Sample origin') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="Sample", y="Proportion of cells")
ggsave(file="cluster_composition_by_sample_origin_whole_umap.svg", plot=image, width=10, height=8)
gc()

ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX', 'EX IPCs', 'IN IPCs', 'IN_CGE', 'IN_local', 'IN_MGE', 'Oligodendrocytes', 'OPCs', 'Radial glia'))@meta.data,
       aes(x=orig.ident, fill=subclust_idents_paper)) +
  geom_bar(position = "fill") +
  ggtitle('Cell type composition of samples') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="Subtype", y="Proportion of cells")
gc()
image = ggplot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX', 'EX IPCs', 'IN IPCs', 'IN_CGE', 'IN_local', 'IN_MGE', 'Oligodendrocytes', 'OPCs', 'Radial glia'))@meta.data,
               aes(x=orig.ident, fill=subclust_idents_paper)) +
  geom_bar(position = "fill") +
  ggtitle('Cell type composition of samples') +
  #scale_fill_manual(values=c(osvz_col, vz_col))+
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="Subtype", y="Proportion of cells")
ggsave(file="sample_origin_by_cluster_composition_whole_umap.svg", plot=image, width=10, height=8)
gc()

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$orig.ident %in% c("GW20_0308_PFC_2S")], raster=F, sizes.highlight =0.1)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2")], raster=F, sizes.highlight =0.1)

ls_ex = SetIdent(ls_ex, value="seurat_clusters")
DimPlot(ls_ex, label=T)
FeaturePlot(ls_ex, features="nCount_RNA", max.cutoff="q90")
FeaturePlot(ls_synthesis_v3_harmony_v3, features="nCount_RNA", max.cutoff="q95", raster=F)

DimPlot(ls_ex, cells.highlight = rownames(ls_ex@meta.data)[ls_ex@meta.data$Clone_size_corrected > 2], cols.highlight='#2dba3b') + 
  ggtitle('EX Cells in multicellular clones') +
  theme(plot.title = element_text(hjust=0.5))

VlnPlot(ls_ex, features = c("nCount_RNA"))

intersect(ls_synthesis_v3_harmony_v3@meta.data$Clone_barcodes_filt[ls_synthesis_v3_harmony_v3@meta.data$orig.ident %in% c('GW20_0308_PFC_2S')], 
          ls_synthesis_v3_harmony_v3@meta.data$Clone_barcodes_filt![ls_synthesis_v3_harmony_v3@meta.data$orig.ident %in% c('GW20_0308_PFC_2S')])

length(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW20_0308_PFC_2S'))$Clone_barcodes_filt))

length(intersect(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW20_0308_PFC_2S'))$Clone_barcodes_filt),
          unique(subset(ls_synthesis_v3_harmony_v3@meta.data, !(orig.ident %in% c('GW20_0308_PFC_2S')))$Clone_barcodes_filt)))
#[1] 10

length(intersect(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW20_0308_V1_S1'))$Clone_barcodes_filt),
          unique(subset(ls_synthesis_v3_harmony_v3@meta.data, !(orig.ident %in% c('GW20_0308_V1_S1')))$Clone_barcodes_filt)))
#[1] 13

length(intersect(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW24_0221_1S'))$Clone_barcodes_filt),
          unique(subset(ls_synthesis_v3_harmony_v3@meta.data, !(orig.ident %in% c('GW24_0221_1S')))$Clone_barcodes_filt)))
#[1] 1

length(intersect(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW18_0215_3S'))$Clone_barcodes_filt),
                 unique(subset(ls_synthesis_v3_harmony_v3@meta.data, !(orig.ident %in% c('GW18_0215_3S')))$Clone_barcodes_filt)))
#[1] 1

head(unique(subset(ls_synthesis_v3_harmony_v3@meta.data, orig.ident %in% c('GW20_0308_V1_S1'))$Clone_barcodes_filt))

ls_meta = ls_synthesis_v3_harmony_v3@meta.data
ls_meta$Clone_barcodes_filt_comp = ls_meta$Clone_barcodes_filt
ls_meta$Clone_barcodes_filt_comp[ls_meta$orig.ident %in% "GW24_0221_1S"] = paste(ls_meta$Clone_barcodes_filt_comp[ls_meta$orig.ident %in% "GW24_0221_1S"], 't', sep="")
head(unique(subset(ls_meta, orig.ident %in% c('GW24_0221_1S'))$Clone_barcodes_filt_comp))

unique(ls_meta$orig.ident)
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW18_0215_3S'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW18_0215_3S')))$Clone_barcodes_filt)))
#[1] 1
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW24_0221_1S'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW24_0221_1S')))$Clone_barcodes_filt)))
#[1] 1
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW21_0502_1S'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW21_0502_1S')))$Clone_barcodes_filt)))
#[1] 1
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW23_0910_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW23_0910_S1')))$Clone_barcodes_filt)))
#[1] 7
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW23_0910_S2'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW23_0910_S2')))$Clone_barcodes_filt)))
#[1] 5
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW23_0910_S4'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW23_0910_S4')))$Clone_barcodes_filt)))
#[1] 3
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW22_1118_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW22_1118_S1')))$Clone_barcodes_filt)))
#[1] 1
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW22_1118_S2'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW22_1118_S2')))$Clone_barcodes_filt)))
#[1] 4
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW22_1118_S3'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW22_1118_S3')))$Clone_barcodes_filt)))
#[1] 2
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW16_1220_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW16_1220_S1')))$Clone_barcodes_filt)))
#[1] 10
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW16_1220_S2'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW16_1220_S2')))$Clone_barcodes_filt)))
#[1] 10
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW19_0221_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW19_0221_S1')))$Clone_barcodes_filt)))
#[1] 4
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW19_0221_S2'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW19_0221_S2')))$Clone_barcodes_filt)))
#[1] 4
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW20_0308_V1_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW20_0308_V1_S1')))$Clone_barcodes_filt)))
#[1] 13
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW20_0308_V1_S2'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW20_0308_V1_S2')))$Clone_barcodes_filt)))
#[1] 12
length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW20_0308_PFC_2S'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in% c('GW20_0308_PFC_2S')))$Clone_barcodes_filt)))

length(intersect(unique(subset(ls_meta, orig.ident %in% c('GW20_0308_PFC_2S', 'GW20_0308_V1_S2', 'GW20_0308_V1_S1'))$Clone_barcodes_filt),
                 unique(subset(ls_meta, !(orig.ident %in%  c('GW20_0308_PFC_2S', 'GW20_0308_V1_S2', 'GW20_0308_V1_S1')))$Clone_barcodes_filt)))
intersect(unique(subset(ls_meta, orig.ident %in% c('GW20_0308_PFC_2S', 'GW20_0308_V1_S2', 'GW20_0308_V1_S1'))$Clone_barcodes_filt),
          unique(subset(ls_meta, !(orig.ident %in%  c('GW20_0308_PFC_2S', 'GW20_0308_V1_S2', 'GW20_0308_V1_S1')))$Clone_barcodes_filt))

ls_meta[ls_meta$Clone_barcodes_filt %in% c("IndexE_Bit1_F_277-Bit2_F_218-Bit3_F_274t",
"IndexE_Bit1_F_113-Bit2_F_435-Bit3_F_494t", "Index3_Bit1_F_281-Bit2_F_482-Bit3_F_033t",
"IndexE_Bit1_F_134-Bit2_F_088-Bit3_F_443t", "IndexE_Bit1_F_282-Bit2_F_262-Bit3_F_229t",
"Index3_Bit1_F_047-Bit2_F_272-Bit3_F_184t", "Index3_Bit1_F_019-Bit2_F_488-Bit3_F_145t",
"IndexE_Bit1_F_129-Bit2_F_037-Bit3_F_411t", "Index3_Bit1_F_252-Bit2_F_097-Bit3_F_204t",
"IndexE_Bit1_F_390-Bit2_F_265-Bit3_F_349t"),]


intersect(unique(subset(ls_meta, orig.ident %in% c('GW16_1220_S2'))$Clone_barcodes_filt),
          unique(subset(ls_meta, !(orig.ident %in% c('GW16_1220_S2')))$Clone_barcodes_filt))
ls_meta[ls_meta$Clone_barcodes_filt %in% c("IndexE_Bit1_F_214-Bit2_F_291-Bit3_F_491t", "IndexE_Bit1_F_432-Bit2_F_057-Bit3_F_389t", "IndexE_Bit1_F_195-Bit2_F_328-Bit3_F_457t",
                                           "IndexE_Bit1_F_390-Bit2_F_265-Bit3_F_349t", "IndexE_Bit1_F_047-Bit2_F_187-Bit3_F_278t",
                                           "IndexE_Bit1_F_221-Bit2_F_486-Bit3_F_257t", "Index3_Bit1_F_241-Bit2_F_113-Bit3_F_128t",
                                           "IndexE_Bit1_F_468-Bit2_F_492-Bit3_F_443t", "IndexE_Bit1_F_430-Bit2_F_046-Bit3_F_123t"),c('orig.ident', 'Clone_barcodes_filt', 'Clone_size_filt')]


write.csv(ls_meta, '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_v3_metadata.csv')
write.csv(compiled_clust_df_idents, '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/compiled_clust_df_idents.csv')


## can I regress out nCount_RNA from subclustering?
ls_ex_v2 = subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c("EX IPCs", "EX"))
ls_ex_v2
#27581 features across 66237 samples within 1 assay 
ls_ex_v2 <- ScaleData(ls_ex_v2, features = all.genes, vars.to.regress = "nCount_RNA")
ls_ex_v2 = FindVariableFeatures(ls_ex_v2, selection.method="vst", nfeatures=2000)
ls_ex_v2 = RunPCA(ls_ex_v2, features=VariableFeatures(object=ls_ex_v2))
ElbowPlot(ls_ex_v2)
ls_ex_v2 <- RunHarmony(ls_ex_v2, "orig.ident")
ls_ex_v2 <- FindNeighbors(ls_ex_v2, reduction = "harmony", dims = 1:15)
ls_ex_v2 <- FindClusters(ls_ex_v2, resolution = .5)
ls_ex_v2 <- RunUMAP(ls_ex_v2, reduction = "harmony", dims=c(1:15))
DimPlot(ls_ex_v2, reduction = "umap",label=T)
VlnPlot(ls_ex_v2, features = c("nCount_RNA", "percent.mt"), ncol = 2)
DimPlot(ls_ex_v2, reduction = "umap",group.by="orig.ident", pt.size=0.5)
DimPlot(ls_ex_v2, reduction = "umap",group.by="age_pseudo", pt.size=0.5)
FeaturePlot(ls_ex_v2, label=T, features=c("EOMES", "NEUROD2", "MARCH1", "TBR1", "BCL11B", "SATB2", "CUX2", "CUX1", "TLE4"))
FeaturePlot(ls_ex_v2, label=T, features=c("TLE4", "NR4A2", "TBR1", "OTX1", "BCL11B", "RORB", "SATB2", "POU3F2", "CUX2"))
FeaturePlot(ls_ex_v2, label=T, features=c("VIM", "HOPX", "NEUROD1", "NEUROD2"))
DimPlot(ls_ex_v2, reduction = "umap", group.by="Clone_Index_filt", pt.size=0.5)
DimPlot(ls_ex_v2, reduction = "umap", split.by="Clone_Index_filt", pt.size=0.5)



DimPlot(ls_glia_v2_harmony, raster=F, 
        cells.highlight=rownames(ls_glia_v2_harmony@meta.data)[ls_glia_v2_harmony@meta.data$Clone_size_corrected>=3],
        cols.highlight='#0f6040') +
  ggtitle('Clone Size >= 3') +
  theme(plot.title = element_text(hjust = 0.5))
DimPlot(ls_glia_v2_harmony, group.by="subclust_idents_paper")
DimPlot(ls_glia_v2_harmony, group.by="age_pseudo")


ggplot(ls_glia_v2_harmony, aes(x=subclust_idents_paper, fill=)
        cells.highlight=rownames(ls_glia_v2_harmony@meta.data)[ls_glia_v2_harmony@meta.data$Clone_size_corrected>=3],
        cols.highlight='#0f6040') +
  ggtitle('Clone Size >= 3') +
  theme(plot.title = element_text(hjust = 0.5))


DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 'IN_UNK'], raster=F)
in_unk_cells = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 'IN_UNK']
DimPlot(ls_in_v2, cells.highlight=in_unk_cells, raster=F)

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight=rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 'unk'], raster=F)
unk_cells = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper %in% 'unk']
DimPlot(ls_glia_v2_harmony, cells.highlight=unk_cells, raster=F)

DimPlot(ls_glia, cells.highlight=unk_cells, raster=F)
FeaturePlot(ls_glia, features=c("nCount_RNA"))
FeaturePlot(ls_glia, features=c("NEUROD2", "EOMES", "SATB2"))

FeaturePlot(ls_glia_v2_harmony, features=c("OLIG2", "EOMES", "VIM", "HOPX"))
##going to try to get better resolution on the dividing glia
table(ls_glia_v2_harmony$subclust_idents_paper)
ls_glia_sub_div_s = subset(ls_glia_v2_harmony, subclust_idents_paper %in% c('Dividing (S phase)'))
ls_glia_sub_div_s <- ScaleData(ls_glia_sub_div_s, features = all.genes)
ls_glia_sub_div_s = FindVariableFeatures(ls_glia_sub_div_s, selection.method="vst", nfeatures=2000)
ls_glia_sub_div_s = RunPCA(ls_glia_sub_div_s, features=VariableFeatures(object=ls_glia_sub_div_s))
ElbowPlot(ls_glia_sub_div_s)
ls_glia_sub_div_s <- RunHarmony(ls_glia_sub_div_s, c("orig.ident"))
ls_glia_sub_div_s <- FindNeighbors(ls_glia_sub_div_s, reduction = "harmony", dims = 1:15)
ls_glia_sub_div_s <- FindNeighbors(ls_glia_sub_div_s, dims = 1:15)
ls_glia_sub_div_s <- FindClusters(ls_glia_sub_div_s, resolution = 0.8)
#ls_glia_sub_div_s <- RunUMAP(ls_glia_sub_div_s, reduction = "harmony", dims=c(1:15))
ls_glia_sub_div_s <- RunUMAP(ls_glia_sub_div_s, dims=c(1:15))
DimPlot(ls_glia_sub_div_s, reduction = "umap",label=T)
FeaturePlot(ls_glia_sub_div_s, features=c("EOMES", "OLIG2", "VIM", "HOPX"), label=T)
DimPlot(ls_glia_sub_div_s, group.by="orig.ident") #very interesting separation by orig.ident even after harmony...
ls_glia_sub_div_s = AddMetaData(ls_glia_sub_div_s, NA, "glia_lineage_trajectory")
ls_glia_sub_div_s@meta.data$glia_lineage_trajectory[ls_glia_sub_div_s@meta.data$seurat_clusters %in% c(7)] = "EX"
ls_glia_sub_div_s@meta.data$glia_lineage_trajectory[ls_glia_sub_div_s@meta.data$seurat_clusters %in% c(1,6)] = "OLIG"
ls_glia_sub_div_s@meta.data$glia_lineage_trajectory[ls_glia_sub_div_s@meta.data$seurat_clusters %in% c(0,2,3,4,5)] = "PROG"
DimPlot(ls_glia_sub_div_s, group.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_s, features=c("EOMES"), label=T, split.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_s, features=c("OLIG2"), label=T, split.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_s, features=c("VIM"), label=T, split.by = "glia_lineage_trajectory")

ls_glia_sub_div_g2m = subset(ls_glia_v2_harmony, subclust_idents_paper %in% c('Dividing (G2M phase)'))
ls_glia_sub_div_g2m <- ScaleData(ls_glia_sub_div_g2m, features = all.genes)
ls_glia_sub_div_g2m = FindVariableFeatures(ls_glia_sub_div_g2m, selection.method="vst", nfeatures=2000)
ls_glia_sub_div_g2m = RunPCA(ls_glia_sub_div_g2m, features=VariableFeatures(object=ls_glia_sub_div_g2m))
ElbowPlot(ls_glia_sub_div_g2m)
ls_glia_sub_div_g2m <- RunHarmony(ls_glia_sub_div_g2m, c("orig.ident"))
ls_glia_sub_div_g2m <- FindNeighbors(ls_glia_sub_div_g2m, reduction = "harmony", dims = 1:15)
ls_glia_sub_div_g2m <- FindNeighbors(ls_glia_sub_div_g2m, dims = 1:15)
ls_glia_sub_div_g2m <- FindClusters(ls_glia_sub_div_g2m, resolution = 0.8)
#ls_glia_sub_div_g2m <- RunUMAP(ls_glia_sub_div_g2m, reduction = "harmony", dims=c(1:15))
ls_glia_sub_div_g2m <- RunUMAP(ls_glia_sub_div_g2m, dims=c(1:15))
DimPlot(ls_glia_sub_div_g2m, reduction = "umap",label=T)
FeaturePlot(ls_glia_sub_div_g2m, features=c("EOMES", "OLIG2", "VIM", "HOPX"), label=T)
DimPlot(ls_glia_sub_div_g2m, group.by="orig.ident") #very interesting separation by orig.ident even after harmony...
ls_glia_sub_div_g2m = AddMetaData(ls_glia_sub_div_g2m, "PROG", "glia_lineage_trajectory")
ls_glia_sub_div_g2m@meta.data$glia_lineage_trajectory[ls_glia_sub_div_g2m@meta.data$seurat_clusters %in% c(1)] = "EX"
ls_glia_sub_div_g2m@meta.data$glia_lineage_trajectory[ls_glia_sub_div_g2m@meta.data$seurat_clusters %in% c(2,9)] = "OLIG"
DimPlot(ls_glia_sub_div_g2m, group.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_g2m, features=c("EOMES"), label=T, split.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_g2m, features=c("OLIG2"), label=T, split.by = "glia_lineage_trajectory")
FeaturePlot(ls_glia_sub_div_g2m, features=c("VIM"), label=T, split.by = "glia_lineage_trajectory")


ls_glia_v2_harmony = AddMetaData(ls_glia_v2_harmony, NA, "glia_lineage_trajectory")
table(ls_glia_v2_harmony$subclust_idents_paper)
ls_glia_v2_harmony$glia_lineage_trajectory[ls_glia_v2_harmony$subclust_idents_paper %in% c("tRG", "oRG")] = "PROG"
ls_glia_v2_harmony$glia_lineage_trajectory[ls_glia_v2_harmony$subclust_idents_paper %in% c("Astrocytes", "RG")] = "ASTRO"
ls_glia_v2_harmony$glia_lineage_trajectory[ls_glia_v2_harmony$subclust_idents_paper %in% c("Transitioning OPCs")] = "OLIG"
ls_glia_v2_harmony$glia_lineage_trajectory[ls_glia_v2_harmony$subclust_idents_paper %in% c("Transitioning IPCs")] = "EX"
length(rownames(ls_glia_v2_harmony@meta.data)[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_div_g2m@meta.data)] == rownames(ls_glia_sub_div_g2m@meta.data))
sum(rownames(ls_glia_v2_harmony@meta.data)[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_div_g2m@meta.data)] == rownames(ls_glia_sub_div_g2m@meta.data))
sum(is.na(ls_glia_v2_harmony@meta.data$glia_lineage_trajectory)[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_div_g2m@meta.data)])
ls_glia_v2_harmony@meta.data$glia_lineage_trajectory[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_div_g2m@meta.data)] = ls_glia_sub_div_g2m@meta.data$glia_lineage_trajectory
ls_glia_v2_harmony@meta.data$glia_lineage_trajectory[rownames(ls_glia_v2_harmony@meta.data) %in% rownames(ls_glia_sub_div_s@meta.data)] = ls_glia_sub_div_s@meta.data$glia_lineage_trajectory
table(ls_glia_v2_harmony@meta.data$glia_lineage_trajectory)
#ASTRO    EX  OLIG  PROG 
# 3780  1303  1995  4436
DimPlot(ls_glia_v2_harmony, group.by="glia_lineage_trajectory")
write.csv(ls_glia_v2_harmony@meta.data, '/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_glia_v2_harmony_metadata.csv')



FeaturePlot(ls_glia_v2_harmony, features=c("FOXP1"), pt.size=1)
FeaturePlot(ls_glia_v2_harmony, features=c("HES1"), pt.size=1)
FeaturePlot(ls_glia_v2_harmony, features=c("HES6"), pt.size=1)
FeaturePlot(ls_glia_v2_harmony, features=c("GJA1"), pt.size=1)


FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SOX5"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TENM2"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("BCL11B"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("FOXP1"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("FOXP2"), raster=F)

gc()

colnames(ls_glia_sub_div_s@meta.data)
DimPlot(ls_glia_sub_div_s, group.by="glial_subtype")
table(ls_glia_sub_div_s@meta.data$glial_subtype)
#Transitioning IPCs Transitioning OPCs 
#725               1439
ls_glia_sub_div_s = AddMetaData(ls_glia_sub_div_s, NA, "glial_subtype_v2")
ls_glia_sub_div_s@meta.data$glial_subtype[ls_glia_sub_div_s@meta.data$seurat_clusters %in% c(0,4,5)] = "Transitioning IPCs"
ls_glia_sub_div_s@meta.data$glial_subtype[ls_glia_sub_div_s@meta.data$seurat_clusters %in% c(1,3,6)] = "Transitioning OPCs"
table(ls_glia_sub_div_s@meta.data$glial_subtype)
#Transitioning IPCs Transitioning OPCs 
#970               1194

#have to go fix a tiny mistake in ls_div that's causing some dividing IN IPCs to be labeled as dividing EX IPCs
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS3_Clone_0"], raster=F)
DimPlot(ls_div, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS3_Clone_0"], raster=F)
FeaturePlot(ls_div, features=c("EOMES", "DLX2", "VIM"))
head(ls_div@meta.data)
DimPlot(ls_div, group.by="seurat_clusters")
DimPlot(ls_div, group.by="subclust_idents")
table(ls_div@meta.data$subclust_idents)

ls_div <- FindClusters(ls_div, resolution = .8)
ls_div <- RunUMAP(ls_div, reduction = "harmony", dims=c(1:15))
DimPlot(ls_div, reduction = "umap",label=T)
DimPlot(ls_div, reduction = "umap",group.by="orig.ident")
FeaturePlot(ls_div, features=c("EOMES", "DLX2", "VIM", "OLIG2"), label=T)
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(1,8,7,9,4,6)] = "GLIA_DIV"
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(2,3,5,12,16)] = "IPC_IN_DIV"
ls_div$subclust_idents[ls_div$seurat_clusters %in% c(0,10,11,13,14)] = "EX_IPC_DIV"
DimPlot(ls_div, reduction = "umap",group.by="RNA_snn_res.0.5")
ls_div$subclust_idents[ls_div$RNA_snn_res.0.5 %in% c(7)] = "OPC_DIV"
DimPlot(ls_div, reduction = "umap",group.by="subclust_idents")



table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
#GLIA_DIV       GLIA      RG_EX IPC_EX_DIV     IPC_EX   EX_EARLY         EX IPC_IN_DIV     IPC_IN   IN_local    OPC_DIV 
#2303       9344          0       2060       9345          0      49657       2605       1192       5445        543 
#OPC       OLIG     IN_CGE     IN_MGE     IN_UNK 
#2249        755       2860       2621        828

length(rownames(ls_synthesis_v3_harmony_v3@meta.data)[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_div@meta.data)] == rownames(ls_div@meta.data))
sum(rownames(ls_synthesis_v3_harmony_v3@meta.data)[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_div@meta.data)] == rownames(ls_div@meta.data))
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_div@meta.data)] = ls_div@meta.data$subclust_idents
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents)
#.   EX EX_IPC_DIV       GLIA   GLIA_DIV     IN_CGE   IN_local     IN_MGE     IN_UNK     IPC_EX     IPC_IN IPC_IN_DIV 
# 49657       1829       9344       2443       2860       5445       2621        828       9345       1192       2696 
#OLIG        OPC    OPC_DIV 
#755       2249        543

table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)
# ASTRO                      EX                EX_early      G2M_phase_div_glia                  IN_CGE 
#1368                   49657                    7235                    2374                    2860 
#IN_local                  IN_MGE                  IN_UNK                  IPC_EX              IPC_EX_DIV 
#5445                    2621                     828                    9345                    2060 
#IPC_IN              IPC_IN_DIV                    OLIG                     OPC                 OPC_DIV 
#1192                    2605                     755                    2249                     543 
#RG       RG (putative oRG)       RG (putative tRG)        S_phase_div_glia      Transitioning IPCs 
#2412                     802                    1068                    1359                     725 
#Transitioning OPCs/IPCs                     unk 
#1439                    1629
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% 'EX_IPC_DIV'] = 'IPC_EX_DIV'
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% 'IPC_IN_DIV'] = 'IPC_IN_DIV'
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_v2)
# ASTRO                      EX                EX_early      G2M_phase_div_glia                  IN_CGE 
#1368                   49657                    7235                    2372                    2860 
#IN_local                  IN_MGE                  IN_UNK                  IPC_EX              IPC_EX_DIV 
#5445                    2621                     828                    9345                    1833 
#IPC_IN              IPC_IN_DIV                    OLIG                     OPC                 OPC_DIV 
#1192                    2834                     755                    2249                     543 
#RG       RG (putative oRG)       RG (putative tRG)        S_phase_div_glia      Transitioning IPCs 
#2412                     802                    1068                    1359                     725 
#Transitioning OPCs/IPCs                     unk 
#1439                    1629 

ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("EX", "EX_early", "IPC_EX", "EX_IPC_DIV", "RG_EX")
] = "EX"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("IN_CGE", "IN_local", "IN_MGE", "IPC_IN", "IPC_IN_DIV")
] = "IN"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("GLIA", "GLIA_DIV")
] = "GLIA"
ls_synthesis_v3_harmony_v3@meta.data$broad_celltype[
  ls_synthesis_v3_harmony_v3@meta.data$subclust_idents %in% c("OPC", "OPC_DIV", "OLIG")
] = "OLIG"
table(ls_synthesis_v3_harmony_v3$broad_celltype)
# EX  GLIA    IN  OLIG 
#68066 13233 14814  3547


DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% "LS36_Clone_451"], raster=F)
DimPlot(ls_div, cells.highlight = rownames(ls_div@meta.data)[ls_div@meta.data$Clone_IDs_filt %in% "LS36_Clone_451"], raster=F)
DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data)[ls_glia_v2_harmony@meta.data$Clone_IDs_filt %in% "LS36_Clone_451"], raster=F)
DimPlot(ls_glia_sub_div_g2m, cells.highlight = rownames(ls_glia_sub_div_g2m@meta.data)[ls_glia_sub_div_g2m@meta.data$Clone_IDs_filt %in% "LS36_Clone_451"], raster=F)
FeaturePlot(ls_glia_sub_div_g2m, features=c("VIM", "EOMES", "OLIG2", "DLX2"))
VlnPlot(subset(ls_glia_sub_div_g2m, Clone_IDs_filt %in% "LS36_Clone_451"), features = c("VIM", "EOMES", "OLIG2", "DLX2"))

DimPlot(ls_in_v2, cells.highlight = rownames(ls_glia_sub_div_g2m@meta.data)[ls_glia_sub_div_g2m@meta.data$Clone_IDs_filt %in% "LS36_Clone_451"], raster=F)

DimPlot(ls_div, cells.highlight = "TTCCTCTGTGAACTAA-1_16", raster=F)
DimPlot(ls_glia_v2_harmony, cells.highlight = "TTCCTCTGTGAACTAA-1_16", raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = "TTCCTCTGTGAACTAA-1_16", raster=F)

ls_synthesis_v3_harmony_v3@meta.data[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% "TTCCTCTGTGAACTAA-1_16",]


ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, NA, "subclust_idents_final")
DimPlot(ls_synthesis_v3_harmony_v3, group.by="seurat_clusters", raster=F, label=T)
ls_synthesis_v3_harmony_v3$subclust_idents_final[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters %in% c(0,1,2,3,11)] = "EX"
ls_synthesis_v3_harmony_v3$subclust_idents_final[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters %in% c(7)] = "EX IPCs"
ls_synthesis_v3_harmony_v3$subclust_idents_final[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters %in% c(12)] = "Oligodendrocytes"
ls_synthesis_v3_harmony_v3$subclust_idents_final[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters %in% c(4)] = "OPCs"
ls_synthesis_v3_harmony_v3$subclust_idents_final[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_div@meta.data)] = ls_div@meta.data$subclust_idents
ls_synthesis_v3_harmony_v3$subclust_idents_final[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_in_v2@meta.data)] = ls_in_v2@meta.data$subclust_idents_paper
ls_synthesis_v3_harmony_v3$subclust_idents_final[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_glia_v2_harmony@meta.data)] = ls_glia_v2_harmony@meta.data$subclust_idents_paper
DimPlot(ls_glia_v2_harmony, group.by="subclust_idents_paper")
DimPlot(ls_glia_v2_harmony, group.by="glia_lineage_trajectory")
colnames(ls_glia_v2_harmony@meta.data)
colnames(ls_glia_sub_div_g2m@meta.data)

table(ls_synthesis_v3_harmony_v3$subclust_idents_final)
sum(is.na(ls_synthesis_v3_harmony_v3$subclust_idents_final))
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_final", raster=F)

DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% unk_cells_to_fix], raster=F)

DimPlot(ls_div, cells.highlight = rownames(ls_div@meta.data)[rownames(ls_div@meta.data) %in% unk_cells_to_fix], raster=F)
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[rownames(ls_in_v2@meta.data) %in% unk_cells_to_fix], raster=F)
DimPlot(ls_glia_v2_harmony, cells.highlight = rownames(ls_glia_v2_harmony@meta.data)[rownames(ls_glia_v2_harmony@meta.data) %in% unk_cells_to_fix], raster=F)
DimPlot(ls_in, cells.highlight = rownames(ls_in@meta.data)[rownames(ls_in@meta.data) %in% unk_cells_to_fix], raster=F)
DimPlot(ls_glia, cells.highlight = rownames(ls_glia@meta.data)[rownames(ls_glia@meta.data) %in% unk_cells_to_fix], raster=F)
DimPlot(ls_glia)
FeaturePlot(ls_glia, features=c("OLIG2", "HOPX"))


DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$seurat_clusters %in% c(8)], raster=F)

##going to try to make some final "definitive" cell type calls because this is still affecting everything downstream of it
##going to make some overlapping subclustered objects so that ambiguous cells can be plotted in both and determined

#ls_prog -- glia, EX IPCS, IN IPCS, dividing, OPCs, oligos
#from ls_prog, subset down to ls_prog_div based on cell cycle scores

#ls_in_v3 -- IN CGE, IN MGE, IN local, IN IPCs, dividing, OPCs

DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_paper", raster=F)
ls_prog = subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX IPCs', 'IN IPCs', 'OPCs', 'Radial glia', 'unk', 'Oligodendrocytes'))
ls_prog
#27581 features across 31925 samples within 1 assay 
ls_prog <- ScaleData(ls_prog, features = all.genes)
ls_prog = FindVariableFeatures(ls_prog, selection.method="vst", nfeatures=2000)
ls_prog = RunPCA(ls_prog, features=VariableFeatures(object=ls_prog))
ElbowPlot(ls_prog)
ls_prog <- RunHarmony(ls_prog, c("orig.ident"))
ls_prog <- FindNeighbors(ls_prog, reduction = "harmony", dims = 1:15)
ls_prog <- FindClusters(ls_prog, resolution = .5)
ls_prog <- RunUMAP(ls_prog, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog, reduction = "umap",label=T)
DimPlot(ls_prog, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog, features=c("nCount_RNA"), max.cutoff="q90")
#cluster 5 is likely all doublets
FeaturePlot(ls_prog, features=c("MKI67", "TOP2A", "CENPF", "PCNA"), label=T)
FeaturePlot(ls_prog, features=c("VIM", "OLIG2", "EOMES", "DLX2"), label=T)
FeaturePlot(ls_prog, features=c("MBP", "OLIG2", "PLP1", "DLX2"), label=T)
FeaturePlot(ls_prog, features=c("VIM", "GJA1", "HOPX", "CRYAB"), label=T)

DimPlot(ls_prog, group.by="subclust_idents_paper")
ls_prog = AddMetaData(ls_prog, NA, 'subclust_idents_definitive')
ls_prog@meta.data$subclust_idents_definitive[ls_prog@meta.data$seurat_clusters %in% c(1,3,8)] = 'EX IPCs'
ls_prog@meta.data$subclust_idents_definitive[ls_prog@meta.data$seurat_clusters %in% c(12)] = 'Oligodendrocytes'

VlnPlot(ls_prog, features=c("MKI67", "PCNA"))
ls_prog_div = subset(ls_prog, seurat_clusters %in% c(2,4,6,7,9,10))
ls_prog_div <- ScaleData(ls_prog_div, features = all.genes)
ls_prog_div = FindVariableFeatures(ls_prog_div, selection.method="vst", nfeatures=2000)
ls_prog_div = RunPCA(ls_prog_div, features=VariableFeatures(object=ls_prog_div))
ElbowPlot(ls_prog_div)
ls_prog_div <- RunHarmony(ls_prog_div, c("orig.ident"))
ls_prog_div <- FindNeighbors(ls_prog_div, reduction = "harmony", dims = 1:15)
ls_prog_div <- FindClusters(ls_prog_div, resolution = .6)
ls_prog_div <- RunUMAP(ls_prog_div, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_div, reduction = "umap",label=T)
DimPlot(ls_prog_div, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog_div, features=c("nCount_RNA"), max.cutoff="q90")
FeaturePlot(ls_prog_div, features=c("MKI67", "TOP2A", "CENPF", "PCNA"), label=T)
FeaturePlot(ls_prog_div, features=c("VIM", "OLIG2", "EOMES", "DLX2"), label=T)
FeaturePlot(ls_prog_div, features=c("MBP", "OLIG2", "PLP1", "DLX2"), label=T)
FeaturePlot(ls_prog_div, features=c("VIM", "GJA1", "HOPX", "CRYAB"), label=T)


ls_prog_div@meta.data$subclust_idents_definitive[ls_prog_div@meta.data$seurat_clusters %in% c(2,9)] = 'EX IPCs'
ls_prog_div@meta.data$subclust_idents_definitive[ls_prog_div@meta.data$seurat_clusters %in% c(1,3,6,12,4)] = 'IN IPCs'
ls_prog_div@meta.data$subclust_idents_definitive[ls_prog_div@meta.data$seurat_clusters %in% c(11)] = 'OPCs'
ls_prog_div@meta.data$subclust_idents_definitive[ls_prog_div@meta.data$seurat_clusters %in% c(8,5,10)] = 'RG'
#need to split off 4 for further subclustering
#need to split off 0 for further subclustering

ls_prog_div_sub = subset(ls_prog_div, seurat_clusters %in% c(0,4,7))
ls_prog_div_sub
ls_prog_div_sub <- ScaleData(ls_prog_div_sub, features = all.genes)
ls_prog_div_sub = FindVariableFeatures(ls_prog_div_sub, selection.method="vst", nfeatures=2000)
ls_prog_div_sub = RunPCA(ls_prog_div_sub, features=VariableFeatures(object=ls_prog_div_sub))
ElbowPlot(ls_prog_div_sub)
ls_prog_div_sub <- RunHarmony(ls_prog_div_sub, c("orig.ident"))
ls_prog_div_sub <- FindNeighbors(ls_prog_div_sub, reduction = "harmony", dims = 1:15)
ls_prog_div_sub <- FindClusters(ls_prog_div_sub, resolution = .5)
ls_prog_div_sub <- RunUMAP(ls_prog_div_sub, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_div_sub, reduction = "umap",label=T)
DimPlot(ls_prog_div_sub, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog_div_sub, features=c("nCount_RNA"), max.cutoff="q90")
#cluster 5 is likely all doublets
FeaturePlot(ls_prog_div_sub, features=c("MKI67", "TOP2A", "CENPF", "PCNA"), label=T)
FeaturePlot(ls_prog_div_sub, features=c("VIM", "OLIG2", "EOMES", "DLX2"), label=T)
FeaturePlot(ls_prog_div_sub, features=c("MBP", "OLIG2", "PLP1", "DLX2"), label=T)
FeaturePlot(ls_prog_div_sub, features=c("VIM", "OLIG2", "HOPX", "PLP1"), label=T)
VlnPlot(ls_prog_div_sub, features=c("VIM", "OLIG2", "EOMES", "DLX2"))
ls_prog_div_sub@meta.data$subclust_idents_definitive[ls_prog_div_sub@meta.data$seurat_clusters %in% c(1)] = 'IN IPCs'
ls_prog_div_sub@meta.data$subclust_idents_definitive[ls_prog_div_sub@meta.data$seurat_clusters %in% c(2,5,8)] = 'OPCs'
ls_prog_div_sub@meta.data$subclust_idents_definitive[ls_prog_div_sub@meta.data$seurat_clusters %in% c(0,3,4,6,7,9,10)] = 'RG'
sum(is.na(ls_prog_div_sub@meta.data$subclust_idents_definitive))

##start to roll the identities back up the chain
ls_prog_div@meta.data$subclust_idents_definitive[rownames(ls_prog_div@meta.data) %in% rownames(ls_prog_div_sub@meta.data)] = ls_prog_div_sub@meta.data$subclust_idents_definitive
DimPlot(ls_prog_div, group.by="subclust_idents_definitive")
ls_prog_div = SetIdent(ls_prog_div, value="subclust_idents_definitive")
VlnPlot(ls_prog_div, features=c("VIM", "OLIG2", "EOMES", "DLX2"))

ls_prog@meta.data$subclust_idents_definitive[rownames(ls_prog@meta.data) %in% rownames(ls_prog_div@meta.data)] = ls_prog_div@meta.data$subclust_idents_definitive
DimPlot(ls_prog, group.by="subclust_idents_definitive")

#split off 0 and 13 for glial subclustering
ls_prog_glia = subset(ls_prog, seurat_clusters %in% c(0,13))
ls_prog_glia <- ScaleData(ls_prog_glia, features = all.genes)
ls_prog_glia = FindVariableFeatures(ls_prog_glia, selection.method="vst", nfeatures=2000)
ls_prog_glia = RunPCA(ls_prog_glia, features=VariableFeatures(object=ls_prog_glia))
ElbowPlot(ls_prog_glia)
ls_prog_glia <- RunHarmony(ls_prog_glia, c("orig.ident"))
ls_prog_glia <- FindNeighbors(ls_prog_glia, reduction = "harmony", dims = 1:15)
ls_prog_glia <- FindClusters(ls_prog_glia, resolution = .5)
ls_prog_glia <- RunUMAP(ls_prog_glia, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_glia, reduction = "umap",label=T)
DimPlot(ls_prog_glia, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog_glia, features=c("nCount_RNA"), max.cutoff="q90", label=T)
DimPlot(ls_prog_glia, reduction = "umap", group.by = "Clone_Index_filt", pt.size=1)
FeaturePlot(ls_prog_glia, features=c("VIM", "GFAP", "OLIG2", "EOMES"), label=T)
FeaturePlot(ls_prog_glia, features=c("VIM", "HOPX", "CRYAB", "GJA1"), label=T)
ls_prog_glia.markers <- FindAllMarkers(ls_prog_glia, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_prog_glia.markers)
ls_prog_glia_clustermarkers_unfilt = as.data.frame(ls_prog_glia.markers %>% group_by(cluster))
write.csv(ls_prog_glia_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_prog_glia_clustermarkers_v1_unfilt.csv',row.names=F)
ls_prog_glia.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_prog_glia_clustermarkers = as.data.frame(ls_prog_glia.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_prog_glia_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_prog_glia_clustermarkers_v1.csv',row.names=F)
ls_prog_glia_clustermarkers
FeaturePlot(ls_prog_glia, features=c("HES1", "HES5", "HES6", "NFIA"), label=T)
FeaturePlot(ls_prog_glia, features=c("OLIG2", "PDGFRA", "HOPX", "GAD1"), label=T)
FeaturePlot(ls_prog_glia, features=c("C3", "AIF1", "CCL2", "S100A11"), label=T)
FeaturePlot(ls_prog_glia, features=c("MKI67", "PCNA", "BRIP1", "LHX6"), label=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("BRIP1"), raster=F)


#subclust calls
#0: misc. radial glia, likely oRG biased 
#1: misc. radial glia, likely oRG biased and similar to 0
#2: "transitioning OPCs"... PDGFRA, OLIG1, OLIG2 are top5 cluster markers but it's also high in HOPX and VIM
#3: astrocyte (SPON1, GJA1, APOE, AQP4)
#4: tRG (ANXA1, ANXA2, CRYAB)
#5: lower quality cells with ribosomal genes, etc.; just general RG?
#6: astrocytes?
#7: general RG
#8: S phase
#9: G2M phase
#10: IPCs

##have to split off 8 and and 9.....
ls_prog_glia_s_phase = subset(ls_prog_glia, seurat_clusters %in% c(8))
ls_prog_glia_s_phase <- ScaleData(ls_prog_glia_s_phase, features = all.genes)
ls_prog_glia_s_phase = FindVariableFeatures(ls_prog_glia_s_phase, selection.method="vst", nfeatures=2000)
ls_prog_glia_s_phase = RunPCA(ls_prog_glia_s_phase, features=VariableFeatures(object=ls_prog_glia_s_phase))
ElbowPlot(ls_prog_glia_s_phase)
ls_prog_glia_s_phase <- RunHarmony(ls_prog_glia_s_phase, c("orig.ident"))
ls_prog_glia_s_phase <- FindNeighbors(ls_prog_glia_s_phase, reduction = "harmony", dims = 1:15)
ls_prog_glia_s_phase <- FindClusters(ls_prog_glia_s_phase, resolution = .5)
ls_prog_glia_s_phase <- RunUMAP(ls_prog_glia_s_phase, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_glia_s_phase, reduction = "umap",label=T)
DimPlot(ls_prog_glia_s_phase, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog_glia_s_phase, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_glia_s_phase, features=c("VIM", "GAD1", "OLIG2", "HOPX"), label=T)
FeaturePlot(ls_prog_glia_s_phase, features=c("MKI67", "PCNA", "BRIP1", "HOPX"), label=T)
ls_prog_glia_s_phase.markers <- FindAllMarkers(ls_prog_glia_s_phase, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
ls_prog_glia_s_phase.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_prog_glia_s_phase_clustermarkers = as.data.frame(ls_prog_glia_s_phase.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
ls_prog_glia_s_phase_clustermarkers
#can't find anything definitive to separate these, separation seems to be primarily on the basis of S phase or G2M phase

ls_prog_glia_g2m_phase = subset(ls_prog_glia, seurat_clusters %in% c(9))
ls_prog_glia_g2m_phase <- ScaleData(ls_prog_glia_g2m_phase, features = all.genes)
ls_prog_glia_g2m_phase = FindVariableFeatures(ls_prog_glia_g2m_phase, selection.method="vst", nfeatures=2000)
ls_prog_glia_g2m_phase = RunPCA(ls_prog_glia_g2m_phase, features=VariableFeatures(object=ls_prog_glia_g2m_phase))
ElbowPlot(ls_prog_glia_g2m_phase)
ls_prog_glia_g2m_phase <- RunHarmony(ls_prog_glia_g2m_phase, c("orig.ident"))
ls_prog_glia_g2m_phase <- FindNeighbors(ls_prog_glia_g2m_phase, reduction = "harmony", dims = 1:15)
ls_prog_glia_g2m_phase <- FindClusters(ls_prog_glia_g2m_phase, resolution = .5)
ls_prog_glia_g2m_phase <- RunUMAP(ls_prog_glia_g2m_phase, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_glia_g2m_phase, reduction = "umap",label=T)
DimPlot(ls_prog_glia_g2m_phase, reduction = "umap", group.by = "orig.ident")
FeaturePlot(ls_prog_glia_g2m_phase, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_glia_g2m_phase, features=c("VIM", "HES5", "HES1", "HES6"), label=T)
FeaturePlot(ls_prog_glia_g2m_phase, features=c("MKI67", "PCNA", "BRIP1", "HOPX"), label=T)
DimPlot(ls_prog_glia_g2m_phase, reduction = "umap", group.by = "age_pseudo")
ls_prog_glia_g2m_phase.markers <- FindAllMarkers(ls_prog_glia_g2m_phase, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
ls_prog_glia_g2m_phase.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_prog_glia_g2m_phase_clustermarkers = as.data.frame(ls_prog_glia_g2m_phase.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
ls_prog_glia_g2m_phase_clustermarkers
#can't find anything definitive to separate these out from general radial glia, seems like there may be a bias for ex_neurogenic (AUTS2, TOX, ROBO2, HES6) but not strong enough to claim right now

ls_prog_glia_clust2 = subset(ls_prog_glia, seurat_clusters %in% c(2))
ls_prog_glia_clust2 <- ScaleData(ls_prog_glia_clust2, features = all.genes)
ls_prog_glia_clust2 = FindVariableFeatures(ls_prog_glia_clust2, selection.method="vst", nfeatures=2000)
ls_prog_glia_clust2 = RunPCA(ls_prog_glia_clust2, features=VariableFeatures(object=ls_prog_glia_clust2))
ElbowPlot(ls_prog_glia_clust2)
ls_prog_glia_clust2 <- RunHarmony(ls_prog_glia_clust2, c("orig.ident"))
ls_prog_glia_clust2 <- FindNeighbors(ls_prog_glia_clust2, reduction = "harmony", dims = 1:15)
ls_prog_glia_clust2 <- FindClusters(ls_prog_glia_clust2, resolution = .5)
ls_prog_glia_clust2 <- RunUMAP(ls_prog_glia_clust2, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_glia_clust2, reduction = "umap",label=T)
DimPlot(ls_prog_glia_clust2, reduction = "umap", group.by = "orig.ident")
DimPlot(ls_prog_glia_clust2, reduction = "umap", group.by = "Clone_Index_filt")
DimPlot(ls_prog_glia_clust2, reduction = "umap", group.by = "subclust_idents_final")
FeaturePlot(ls_prog_glia_clust2, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_glia_clust2, features=c("MKI67", "PCNA", "BRIP1", "HOPX"), label=T)
FeaturePlot(ls_prog_glia_clust2, features=c("VIM", "HOPX", "MT3", "PDGFRA"), label=T)
FeaturePlot(ls_prog_glia_clust2, features=c("VIM", "OLIG2", "CLU", "PDGFRA"), label=T)
#for now, going to call 0, 2, 3, 6 as oRG based on HOPX, CLU, MT3, and relatively low PDGFRA and OLIG2 compared to 1 and 3. 4 is dividing, but seems more OPC-like
ls_prog_glia_clust2@meta.data$subclust_idents_definitive[ls_prog_glia_clust2@meta.data$seurat_clusters %in% c(0,2,5,6)] = "oRG"
ls_prog_glia_clust2@meta.data$subclust_idents_definitive[ls_prog_glia_clust2@meta.data$seurat_clusters %in% c(1,3,4)] = "OPCs"


ls_prog_glia@meta.data$subclust_idents_definitive[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust2@meta.data)] = ls_prog_glia_clust2@meta.data$subclust_idents_definitive
ls_prog_glia@meta.data$subclust_idents_definitive[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust6@meta.data)] = ls_prog_glia_clust6@meta.data$subclust_idents_definitive

DimPlot(ls_prog_glia, group.by = "subclust_idents_final", pt.size=1)
DimPlot(ls_prog_glia, group.by = "subclust_idents_final", pt.size=1, split.by='seurat_clusters')
ggplot(ls_prog_glia@meta.data, aes(x=seurat_clusters, fill=subclust_idents_final)) +
  geom_bar()
ls_prog_glia@meta.data$subclust_idents_definitive[ls_prog_glia@meta.data$seurat_clusters %in% c(0, 1, 5, 7, 8, 9)] = "RG"
ls_prog_glia@meta.data$subclust_idents_definitive[ls_prog_glia@meta.data$seurat_clusters %in% c(4)] = "tRG"
ls_prog_glia@meta.data$subclust_idents_definitive[ls_prog_glia@meta.data$seurat_clusters %in% c(10)] = "EX IPCs"
ls_prog_glia@meta.data$subclust_idents_definitive[ls_prog_glia@meta.data$seurat_clusters %in% c(3)] = "Astrocytes"
DimPlot(ls_prog_glia, label=T)
DimPlot(ls_prog_glia, group.by = "subclust_idents_definitive", pt.size=1)
table(ls_prog_glia$subclust_idents_definitive)
#Astrocytes    EX IPCs         OPCs        oRG         RG        tRG 
#.     786         52          422        487       4140        714 
table(ls_prog_glia$subclust_idents_final)
#Astrocytes Dividing (G2M phase)   Dividing (S phase)              EX IPCs           EX_IPC_DIV             GLIA_DIV 
#     1367                   85                   69                    3                    4                    1 
#IN IPCs                 OPCs                  oRG                   RG   Transitioning IPCs   Transitioning OPCs 
#     5                  204                  732                 2389                  321                  468 
#tRG 
#1055

ggplot(ls_prog_glia@meta.data, aes(x=subclust_idents_definitive, fill=Clone_Index_filt)) +
  geom_bar()

ls_prog@meta.data$subclust_idents_definitive[rownames(ls_prog@meta.data) %in% rownames(ls_prog_glia@meta.data)] = ls_prog_glia@meta.data$subclust_idents_definitive

DimPlot(ls_prog, group.by="subclust_idents_definitive", pt.size=1)
DimPlot(ls_prog, label=T)

ls_prog_subclust_11 = subset(ls_prog, seurat_clusters %in% c(11))
ls_prog_subclust_11 <- ScaleData(ls_prog_subclust_11, features = all.genes)
ls_prog_subclust_11 = FindVariableFeatures(ls_prog_subclust_11, selection.method="vst", nfeatures=2000)
ls_prog_subclust_11 = RunPCA(ls_prog_subclust_11, features=VariableFeatures(object=ls_prog_subclust_11))
ElbowPlot(ls_prog_subclust_11)
ls_prog_subclust_11 <- RunHarmony(ls_prog_subclust_11, c("orig.ident"))
ls_prog_subclust_11 <- FindNeighbors(ls_prog_subclust_11, reduction = "harmony", dims = 1:15)
ls_prog_subclust_11 <- FindClusters(ls_prog_subclust_11, resolution = .5)
ls_prog_subclust_11 <- RunUMAP(ls_prog_subclust_11, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_subclust_11, reduction = "umap",label=T)
DimPlot(ls_prog_subclust_11, reduction = "umap", group.by = "orig.ident")
DimPlot(ls_prog_subclust_11, reduction = "umap", group.by = "Clone_Index_filt")
DimPlot(ls_prog_subclust_11, reduction = "umap", group.by = "subclust_idents_final")
FeaturePlot(ls_prog_subclust_11, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_subclust_11, features=c("OLIG2", "PDGFRA", "GAD1", "NPY"), label=T)
#seem like pretty much all OPCs.... maybe GE origin, but just going to go with OPCs for now?
ls_prog@meta.data$subclust_idents_definitive[ls_prog@meta.data$seurat_clusters %in% c(11)] = "OPCs"
DimPlot(ls_prog, group.by="subclust_idents_definitive", pt.size=1)
table(ls_prog@meta.data$subclust_idents_definitive)
#Astrocytes          EX IPCs          IN IPCs  Oligodendrocytes             OPCs              oRG 
#     786            11048             4456                 772             2859              487 
#RG              tRG 
#8698              714 
table(ls_prog@meta.data$subclust_idents_final)
#Astrocytes Dividing (G2M phase)   Dividing (S phase)                   EX              EX IPCs           EX_IPC_DIV 
#1368                 2374                 1359                 1931                 6705                 1827 
#GLIA_DIV              IN IPCs           IPC_IN_DIV     Oligodendrocytes              OPC_DIV                 OPCs 
#142                 1192                 2425                  915                  543                 3473 
#oRG                   RG   Transitioning IPCs   Transitioning OPCs                  tRG 
#769                 2412                  941                 1223                 1068


#ls_in_v3 -- IN CGE, IN MGE, IN local, IN IPCs, dividing, OPCs
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_paper", raster=F)
ls_in_v3 = subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('IN_CGE', 'IN_MGE', 'IN_local', 'IN IPCs', 'IN_UNK', 'OPCs'))
ls_in_v3
#27581 features across 15195 samples within 1 assay 
ls_in_v3 <- ScaleData(ls_in_v3, features = all.genes)
ls_in_v3 = FindVariableFeatures(ls_in_v3, selection.method="vst", nfeatures=2000)
ls_in_v3 = RunPCA(ls_in_v3, features=VariableFeatures(object=ls_in_v3))
ElbowPlot(ls_in_v3)
ls_in_v3 <- RunHarmony(ls_in_v3, c("orig.ident"))
ls_in_v3 <- FindNeighbors(ls_in_v3, reduction = "harmony", dims = 1:15)
ls_in_v3 <- FindClusters(ls_in_v3, resolution = .5)
ls_in_v3 <- RunUMAP(ls_in_v3, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in_v3, reduction = "umap",label=T)
DimPlot(ls_in_v3, reduction = "umap", group.by = "orig.ident")
DimPlot(ls_in_v3, reduction = "umap", group.by = "Clone_Index_filt")
DimPlot(ls_in_v3, reduction = "umap", group.by = "subclust_idents_final")
FeaturePlot(ls_in_v3, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_in_v3, features=c("OLIG2", "PDGFRA", "GAD1", "DLX2"), label=T)
FeaturePlot(ls_in_v3, features=c("MKI67", "NPY", "PAX6", "DLX2"), label=T)
FeaturePlot(ls_in_v3, features=c("NR2F2", "ADARB2", "LHX6", "MAF"), label=T)
FeaturePlot(ls_in_v3, features=c("NR2F2", "PBX3", "TSHZ1", "NR2F1"), label=T)
FeaturePlot(ls_in_v3, features=c("PAX6", "SCGN", "SOX6", "DLX2"), label=T)


ls_in_v3 = AddMetaData(ls_in_v3, ls_in_v3@meta.data$subclust_idents_final, "subclust_idents_definitive")
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(0)] = "IN_MGE"
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(2)] = "IN_CGE"
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(8)] = "IN_OB"
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(4)] = "IN IPCs"
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(7)] = NA
ls_in_v3@meta.data$subclust_idents_definitive[ls_in_v3@meta.data$seurat_clusters %in% c(1,5,11)] = "IN_local"


ls_in_v3_subclust_2_6_10 = subset(ls_in_v3, seurat_clusters %in% c(2,6,10))
ls_in_v3_subclust_2_6_10 <- ScaleData(ls_in_v3_subclust_2_6_10, features = all.genes)
ls_in_v3_subclust_2_6_10 = FindVariableFeatures(ls_in_v3_subclust_2_6_10, selection.method="vst", nfeatures=2000)
ls_in_v3_subclust_2_6_10 = RunPCA(ls_in_v3_subclust_2_6_10, features=VariableFeatures(object=ls_in_v3_subclust_2_6_10))
ElbowPlot(ls_in_v3_subclust_2_6_10)
ls_in_v3_subclust_2_6_10 <- RunHarmony(ls_in_v3_subclust_2_6_10, c("orig.ident"))
ls_in_v3_subclust_2_6_10 <- FindNeighbors(ls_in_v3_subclust_2_6_10, reduction = "harmony", dims = 1:15)
ls_in_v3_subclust_2_6_10 <- FindClusters(ls_in_v3_subclust_2_6_10, resolution = .5)
ls_in_v3_subclust_2_6_10 <- RunUMAP(ls_in_v3_subclust_2_6_10, reduction = "harmony", dims=c(1:15))
DimPlot(ls_in_v3_subclust_2_6_10, reduction = "umap",label=T)
FeaturePlot(ls_in_v3_subclust_2_6_10, features=c("NR2F2", "ADARB2", "DLX2", "PAX6"), label=T)
FeaturePlot(ls_in_v3_subclust_2_6_10, features=c("SCGN", "SOX6", "ERBB4", "GAD2"), label=T)
DimPlot(ls_in_v3_subclust_2_6_10, reduction = "umap", group.by = "subclust_idents_final")
DimPlot(ls_in_v3_subclust_2_6_10, cells.highlight = rownames(ls_in_v3_subclust_2_6_10@meta.data)[ls_in_v3_subclust_2_6_10@meta.data$Clone_size_corrected > 2])
ls_in_v3_subclust_2_6_10@meta.data$subclust_idents_definitive[ls_in_v3_subclust_2_6_10@meta.data$seurat_clusters %in% c(2,3,4)] = "IN_local"
ls_in_v3_subclust_2_6_10@meta.data$subclust_idents_definitive[ls_in_v3_subclust_2_6_10@meta.data$seurat_clusters %in% c(0,1,5,6)] = "IN_CGE"

ls_in_v3@meta.data$subclust_idents_definitive[rownames(ls_in_v3@meta.data) %in% rownames(ls_in_v3_subclust_2_6_10@meta.data)] = ls_in_v3_subclust_2_6_10@meta.data$subclust_idents_definitive
DimPlot(ls_in_v3, reduction = "umap", group.by = "subclust_idents_definitive")

DimPlot(ls_synthesis_v3_harmony_v3, label=T, raster=F)
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, NA, "subclust_idents_definitive")
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_final %in% c('EX')] = "EX"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_final %in% c('Oligodendrocytes')] = "Oligodendrocytes"
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_in_v3@meta.data)] = ls_in_v3@meta.data$subclust_idents_definitive
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_prog@meta.data)] = ls_prog@meta.data$subclust_idents_definitive
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_definitive", label=T, raster=F)






FeaturePlot(ls_in_v2, features=c("PAX6", "LAMP5", "VIP", "SCGN"))
FeaturePlot(ls_in_v2, features=c("PAX6", "LAMP5", "NR2F2", "ADARB2"))
FeaturePlot(ls_in_v2, features=c("SCGN", "CALB1", "CALB2", "NOS1"))
FeaturePlot(ls_in_v2, features=c("SCGN", "LHX6", "CALB2", "NOS1"))
FeaturePlot(ls_in_v2, features=c("SCGN", "LHX6", "LAMP5", "NPY"))
FeaturePlot(ls_in_v2, features=c("NKX2-1", "LHX6", "LAMP5", "NPY"))
FeaturePlot(ls_in_v2, features=c("RELN", "NR2F2", "LAMP5", "NPY"))
FeaturePlot(ls_in_v2, features=c("SCGN", "CALB1", "CALB2", "SP8"))
FeaturePlot(ls_in_v2, features=c("SCGN", "SP8", "CALB2", "NPY"))
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("HS3ST4", "NR4A2", "CRYM", "TRPM3"))

FeaturePlot(ls_in_v2, features=c("SCGN", "CALB1", "CALB2", "SP8"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("HS3ST4", "NR4A2", "CRYM", "TRPM3"), raster=F)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NRG1", "NR4A2", "CRYM", "TRPM3"), raster=F, order=T)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CRYM"), raster=F, order=T, split.by="Clone_Index_filt")

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SORCS1"), raster=F, order=T, split.by="Clone_Index_filt")

FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CRYM"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR4A2"), raster=F, order=T, split.by="Clone_Index_filt")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("NR4A2"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("FOXP2"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("HS3ST4"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("ZFHX3"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TLE4"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("TBR1"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("PAPPA2"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("GRIK3"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CUX2"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CUX2"), raster=F, order=F, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("CUX2"), raster=F, order=T, split.by="age_pseudo")
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("SATB2"), raster=F, order=T, split.by="age_pseudo")
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="age_pseudo")

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="age_pseudo")
VlnPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_definitive %in% "EX"), features=c("CRYM", "NR4A2", "TLE4", "CUX2", "SATB2", "NRG1"), pt.size=0)
ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="Clone_Index_filt")
VlnPlot(subset(ls_synthesis_v3_harmony_v3, subclust_idents_definitive %in% "EX"), features=c("CRYM", "NR4A2", "TLE4", "CUX2", "SATB2", "NRG1"), pt.size=0)

DimPlot(ls_synthesis_v3_harmony_v3, group.by="seurat_clusters", raster=F, label=T)
VlnPlot(subset(ls_synthesis_v3_harmony_v3, seurat_clusters %in% c(0)), features=c("CRYM", "NR4A2", "TLE4", "CUX2", "SATB2", "NRG1"), pt.size=0.1)

ls_synthesis_v3_harmony_v3 = SetIdent(ls_synthesis_v3_harmony_v3, value="orig.ident")
ls_in_v2 = SetIdent(ls_in_v2, value="orig.ident")
DimPlot(ls_in_v2, split.by="age_pseudo")
DimPlot(ls_in_v2, split.by="sample_age")
ls_in_v2 = SetIdent(ls_in_v2, value="subclust_idents_paper")
DimPlot(subset(ls_in_v2, orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S")), split.by="orig.ident")

ls_in_v2 = SetIdent(ls_in_v2, value="orig.ident")
VlnPlot(subset(ls_in_v2, orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2", "GW20_0308_PFC_2S")), features=c("SCGN", "PAX6", "DLX2", "CALB2", "LAMP5", "LHX6"))



DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive)], raster=F)
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_final)], raster=F)

DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_definitive", raster=F)
DimPlot(ls_prog_glia)
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive")
DimPlot(ls_prog_glia, group.by="subclust_idents_final")

DimPlot(ls_prog)
table(ls_prog@meta.data$seurat_clusters)
FeaturePlot(ls_prog, features=c("VIM", "EOMES", "SATB2", "NEUROD2"))
FeaturePlot(ls_prog, features="nCount_RNA", max.cutoff = "q90")
DimPlot(ls_prog, group.by="subclust_idents_final", label=T)
DimPlot(ls_prog, group.by="subclust_idents_definitive", label=T)
DimPlot(ls_prog, group.by="orig.ident")
DimPlot(ls_prog, cells.highlight = rownames(ls_prog@meta.data)[ls_prog@meta.data$Clone_size_corrected >2], sizes.highlight = 0.1)

table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_final[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive)])
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_paper[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive)])
table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[is.na(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_final)])
DimPlot(ls_in_v3, group.by="subclust_idents_definitive")


## 12/21/23
## figure generation
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/231221_figs')
table(ls_synthesis_v3_harmony_v3$subclust_idents_definitive)
#Astrocytes               EX          EX IPCs          IN IPCs           IN_CGE         IN_local           IN_MGE 
#       786            56582            11048             4547             2080             5341             2610 
#IN_OB               Oligodendrocytes             OPCs              oRG               RG              tRG 
#  798                          772             2859              487             8698              714 
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_definitive", raster=F)
## going to try to use these identities, but first going to change some of the nomenclature 
## EX --> ENs
## IN IPCs --> IN_IPCs
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% 'EX'] = 'ENs'
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% 'EX IPCs'] = 'EX_IPCs'
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% 'IN IPCs'] = 'IN_IPCs'
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_definitive", raster=F)

ls_synthesis_v3_harmony_v4 = subset(ls_synthesis_v3_harmony_v3, subclust_idents_definitive %in% NA, invert=T)
ls_synthesis_v3_harmony_v4
DimPlot(ls_synthesis_v3_harmony_v4, group.by="subclust_idents_definitive", raster=F)
ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value="subclust_idents_definitive")
ls_synthesis_v3_harmony_v4@active.ident = factor(ls_synthesis_v3_harmony_v4@active.ident, levels=c(
  "RG",
  "oRG",
  "tRG",
  "Astrocytes",
  "OPCs",
  "Oligodendrocytes",
  "EX_IPCs",
  "ENs",
  "IN_IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"
))
DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
image = DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
ggsave(file="full_umap_subclust_idents.png", plot=image, width=12, height=8)
ggsave(file="full_umap_subclust_idents.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5, group.by="orig.ident", raster=F)
ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, ls_synthesis_v3_harmony_v4@meta.data$orig.ident, col.name = "orig.ident.merge")
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW18_0215_3S')] = 'GW18'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW24_0221_1S')] = 'GW24'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW21_0502_1S')] = 'GW21'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 'GW23'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 'GW22'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW16_1220_S1', 'GW16_1220_S2')] = 'GW16'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 'GW19'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2')] = 'GW20_V1'
ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% c('GW20_0308_PFC_2S')] = 'GW20_PFC'
DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5, group.by="orig.ident.merge", raster=F)
DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.1, group.by="orig.ident.merge", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20), plot.title = element_blank())
image = DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5, group.by="orig.ident.merge", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20), plot.title = element_blank())
ggsave(file="full_umap_sample_idents.png", plot=image, width=12, height=8)
ggsave(file="full_umap_sample_idents.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5, group.by="age_pseudo", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20), plot.title = element_blank())
image = DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5, group.by="age_pseudo", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20), plot.title = element_blank())
ggsave(file="full_umap_pseudoage_idents.png", plot=image, width=12, height=8)
ggsave(file="full_umap_pseudoage_idents.svg", plot=image, width=12, height=8)

#note: code to generate upset plots can be found here: 231220_local_STICR_clone_reprocessing.R

## Figure 2
ls_in_v2 = SetIdent(ls_in_v2, value="subclust_idents_paper")
DimPlot(ls_in_v2, pt.size=0.5) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30))
image = DimPlot(ls_in_v2, pt.size=1) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30))
ggsave(file="in_umap_subclust_idents.png", plot=image, width=12, height=8)
ggsave(file="in_umap_subclust_idents.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, pt.size=0.5, group.by="orig.ident", raster=F)
ls_in_v2 = AddMetaData(ls_in_v2, ls_in_v2@meta.data$orig.ident, col.name = "orig.ident.merge")
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW18_0215_3S')] = 'GW18'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW24_0221_1S')] = 'GW24'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW21_0502_1S')] = 'GW21'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 'GW23'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 'GW22'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW16_1220_S1', 'GW16_1220_S2')] = 'GW16'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 'GW19'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2')] = 'GW20_V1'
ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$orig.ident.merge %in% c('GW20_0308_PFC_2S')] = 'GW20_PFC'
DimPlot(ls_in_v2, pt.size=0.5, group.by="orig.ident.merge", raster=F)
DimPlot(ls_in_v2, pt.size=2, group.by="orig.ident.merge", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=40), plot.title = element_blank())
image = DimPlot(ls_in_v2, pt.size=2, group.by="orig.ident.merge", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=40), plot.title = element_blank())
ggsave(file="in_umap_sample_idents.png", plot=image, width=12, height=8)
ggsave(file="in_umap_sample_idents.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, pt.size=3, group.by="age_pseudo", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=40), plot.title = element_blank())
image = DimPlot(ls_in_v2, pt.size=3, group.by="age_pseudo", raster=F, shuffle=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=40), plot.title = element_blank())
ggsave(file="in_umap_pseudoage_idents.png", plot=image, width=12, height=8)
ggsave(file="in_umap_pseudoage_idents.svg", plot=image, width=12, height=8)


FeaturePlot(ls_in_v2, features=c("ADARB2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_in_v2, features=c("ADARB2"),raster=F, cols=c("#e6e8e7", "#c44e33")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_in_v2_umap_adarb2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_in_v2_umap_adarb2.png", plot=image, width=4, height=3.2)

#note: code to generate UMAPs with multicell clones highlighted can be found here: 231220_local_STICR_clone_reprocessing.R
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c(
  "IN IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"))
DotPlot(ls_in_v2, features=c("GAD2", "DLX1", "DLX2", "DLX5", "FOXG1", "ERBB4",
                             "NR2F2", "ADARB2", "PROX1", "CALB2", "VIP", "RELN", 
                             "LHX6", "MAF", "SST",
                             "MEIS2", "SP8",
                             "PBX3", "TSHZ1",
                             "SOX6", "NR2F1", "SCGN", "PAX6", "NPY", "MKI67")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=18, angle=30, hjust=1, vjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
image = DotPlot(ls_in_v2, features=c("GAD2", "DLX1", "DLX2", "DLX5", "FOXG1", "ERBB4",
                                     "NR2F2", "ADARB2", "PROX1", "CALB2", "VIP", "RELN", 
                                     "LHX6", "MAF", "SST",
                                     "MEIS2", "SP8",
                                     "PBX3", "TSHZ1",
                                     "SOX6", "NR2F1", "SCGN", "PAX6", "NPY", "MKI67")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=18, angle=30, hjust=1, vjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
ggsave(file="in_marker_dotplot.svg", plot=image, width=14, height=8)

ls_in_v2 = SetIdent(ls_in_v2, value = "subclust_idents_paper")
DotPlot(ls_in_v2, features=c("GAD2", "DLX1", "DLX2", "DLX5", "FOXG1", "ERBB4",
                             "NR2F2", "ADARB2", "PROX1", "CALB2", "VIP", "RELN", 
                             "LHX6", "MAF", "SST",
                             "MEIS2", "SP8",
                             "PBX3", "TSHZ1",
                             "SOX6", "NR2F1", "SCGN", "PAX6", "ST18", "NPY", "MKI67")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=18, angle=30, hjust=1, vjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
image = DotPlot(ls_in_v2, features=c("GAD2", "DLX1", "DLX2", "DLX5", "FOXG1", "ERBB4",
                                     "NR2F2", "ADARB2", "PROX1", "CALB2", "VIP", "RELN", 
                                     "LHX6", "MAF", "SST",
                                     "MEIS2", "SP8",
                                     "PBX3", "TSHZ1",
                                     "SOX6", "NR2F1", "SCGN", "PAX6", "ST18", "NPY", "MKI67")) +
  ggtitle('Interneuron marker gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=18, angle=30, hjust=1, vjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
ggsave(file="in_marker_dotplot_with_st18.svg", plot=image, width=14, height=8)

## Figure 3?

# install.packages("remotes")
remotes::install_github("davidsjoberg/ggsankey")
library(ggsankey)

head(filt_df_v3)
test_sankey = filt_df_v3 %>% make_long(Clone_Index_filt, subclust_idents_definitive)
dim(test_sankey)
head(test_sankey)

ggplot(test_sankey, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
image = ggplot(test_sankey, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
ggsave('sankey_clone_composition_by_gz_v1.svg', plot = image, width=12, height=8)

head(filt_df_v3)
sankey_multicell_clones = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2,] %>% make_long(Clone_Index_filt, subclust_idents_definitive)
dim(sankey_multicell_clones)

ggplot(sankey_multicell_clones, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
image = ggplot(sankey_multicell_clones, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
ggsave('sankey_clone_composition_by_gz_2cell_clones_v2.svg', plot = image, width=12, height=8)


sankey_multicell_clones_young = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$age_pseudo %in% c('<=GW20'),] %>% make_long(Clone_Index_filt, subclust_idents_definitive)

ggplot(sankey_multicell_clones_young, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
image = ggplot(sankey_multicell_clones_young, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
ggsave('sankey_clone_composition_by_gz_young_samples_2cell_clones.svg', plot = image, width=12, height=8)


sankey_multicell_clones_old = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$age_pseudo %in% c('>GW20'),] %>% make_long(Clone_Index_filt, subclust_idents_definitive)
ggplot(sankey_multicell_clones_old, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
image = ggplot(sankey_multicell_clones_old, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
ggsave('sankey_clone_composition_by_gz_old_samples_2cell_clones.svg', plot = image, width=12, height=8)


sankey_multicell_clones_by_age = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2,] %>% make_long(age_pseudo, subclust_idents_definitive)
ggplot(sankey_multicell_clones_by_age, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
image = ggplot(sankey_multicell_clones_by_age, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')
ggsave('sankey_clone_composition_by_age_2cell_clones.svg', plot = image, width=12, height=8)

DimPlot(ls_prog_glia)
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive", label=T)
FeaturePlot(ls_prog_glia, features=c("VIM", "HES1", "CRYAB", "HOPX"))
DimPlot(ls_glia_v2_harmony)


tRG_clone_list = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% 'tRG', 'Clone_IDs_filt']
tRG_clone_list = unique(tRG_clone_list)
length(tRG_clone_list)
#[1] 455
oRG_clone_list = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% 'oRG', 'Clone_IDs_filt']
oRG_clone_list = unique(oRG_clone_list)
length(oRG_clone_list)
#[1] 366
DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% tRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in tRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% tRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in tRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
       plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_tRG_clones_v2.svg', image, width=12, height=8)
ggsave('ls_full_umap_tRG_clones_v2.png', image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% oRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in oRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v3, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% oRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3], raster=F) +
  ggtitle("Cells in oRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_oRG_clones_v2.svg', image, width=12, height=8)
ggsave('ls_full_umap_oRG_clones_v2.png', image, width=12, height=8)


filt_df_v3$trg_org_containing = NA
filt_df_v3$trg_org_containing[filt_df_v3$Clone_IDs_filt %in% tRG_clone_list] = "tRG"
filt_df_v3$trg_org_containing[filt_df_v3$Clone_IDs_filt %in% oRG_clone_list] = "oRG"

sankey_multicell_clones_by_rg = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$trg_org_containing %in% c('tRG', 'oRG'),] %>% make_long(trg_org_containing, subclust_idents_definitive)
ggplot(sankey_multicell_clones_by_rg, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')


vz_rg_clone_list = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% c('RG', 'tRG', 'oRG') & ls_synthesis_v3_harmony_v3@meta.data$Clone_Index %in% 'IndexE', 'Clone_IDs_filt']
vz_rg_clone_list = unique(vz_rg_clone_list)
length(vz_rg_clone_list)
#[1] 2991
osvz_rg_clone_list = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% c('RG', 'tRG', 'oRG') & ls_synthesis_v3_harmony_v3@meta.data$Clone_Index %in% 'Index3', 'Clone_IDs_filt']
osvz_rg_clone_list = unique(osvz_rg_clone_list)
length(osvz_rg_clone_list)
#[1] 4496
filt_df_v3$vz_osvz_prog_containing = NA
filt_df_v3$vz_osvz_prog_containing[filt_df_v3$Clone_IDs_filt %in% vz_rg_clone_list] = "VZ_RG"
filt_df_v3$vz_osvz_prog_containing[filt_df_v3$Clone_IDs_filt %in% osvz_rg_clone_list] = "OSVZ_RG"

sankey_multicell_clones_by_rg = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$vz_osvz_prog_containing %in% c('VZ_RG', 'OSVZ_RG'),] %>% make_long(vz_osvz_prog_containing, subclust_idents_definitive)
ggplot(sankey_multicell_clones_by_rg, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  ggtitle('All ages, 2-cell clones') +
  theme(plot.title=element_text(hjust=0.5))

sankey_multicell_clones_by_rg_young = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$vz_osvz_prog_containing %in% c('VZ_RG', 'OSVZ_RG') & filt_df_v3$age_pseudo %in% '<=GW20',] %>% make_long(vz_osvz_prog_containing, subclust_idents_definitive)
ggplot(sankey_multicell_clones_by_rg_young, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  ggtitle('<=GW20 samples, 2-cell clones') +
  theme(plot.title=element_text(hjust=0.5))

sankey_multicell_clones_by_rg_old = filt_df_v3[filt_df_v3$Clone_size_corr_3umi >= 2 & filt_df_v3$vz_osvz_prog_containing %in% c('VZ_RG', 'OSVZ_RG') & filt_df_v3$age_pseudo %in% '>GW20',] %>% make_long(vz_osvz_prog_containing, subclust_idents_definitive)
ggplot(sankey_multicell_clones_by_rg_old, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  ggtitle('>GW20 samples, 2-cell clones') +
  theme(plot.title=element_text(hjust=0.5))



library(circlize)
#try to make a chord diagram for tRG and oRG?
##approach: restrict to cells in tRG or oRG clones, then get counts of all other cell types
tRG_clone_list = tRG_clone_list[!is.na(tRG_clone_list)]
trg_clone_data = ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% tRG_clone_list & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=3,]
dim(trg_clone_data)
table(trg_clone_data$subclust_idents_definitive)
#Astrocytes              ENs          EX_IPCs          IN_IPCs         IN_local           IN_MGE Oligodendrocytes 
#        7               10               34               22               16                1                2 
#OPCs              oRG               RG              tRG 
#29                6               93                95
subclust_names = rownames(table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive))
length(subclust_names)
chord_matrix = matrix(rep(0, length(subclust_names)*length(subclust_names)), ncol=length(subclust_names))
dim(chord_matrix)
rownames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
colnames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")

clust_id = "RG"
clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
  "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
))
clust_interactions_temp

clust_id = "IN_OB"
clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
  "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
))
clust_interactions_temp

for (clust_id in subclust_names) {
  clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
library(circlize)
chordDiagram(chord_matrix, transparency=0.5)


#what if we got rid of the self-clones?
chord_matrix_cross_only = chord_matrix
for (nc in c(1:13)) {
  chord_matrix_cross_only[nc,nc] = 0
}
chord_matrix_cross_only
chordDiagram(chord_matrix_cross_only, transparency=0.5)


clust_id = "RG"
clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
  "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
))
clust_interactions_temp
clust_id = "oRG"
clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
  "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
))
clust_interactions_temp

## dual interactions are coming because there are slightly different numbers -- ex. if a clone has 2x RG and 1x oRG, then it will count as 1 from RG to oRG but as 2 from oRG to RG
## not necessarily a problem, but not great for readability

## will look cleaner if we do individual plots from only a signle origin, ex. tRG
subclust_names = rownames(table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive))
length(subclust_names)
chord_matrix = matrix(rep(0, length(subclust_names)*length(subclust_names)), ncol=length(subclust_names))
dim(chord_matrix)
rownames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
colnames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")

for (clust_id in c("tRG")) {
  clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chord_matrix_cross_only = chord_matrix
for (nc in c(1:13)) {
  chord_matrix_cross_only[nc,nc] = 0
}
chord_matrix_cross_only
chordDiagram(chord_matrix_cross_only, transparency=0.5)
## still sucks....

for (clust_id in c("oRG", "tRG")) {
  clone_list_temp = ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive %in% clust_id & ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected >=2]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chord_matrix_cross_only = chord_matrix
for (nc in c(1:13)) {
  chord_matrix_cross_only[nc,nc] = 0
}
chord_matrix_cross_only
chordDiagram(chord_matrix_cross_only, transparency=0.5)









FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("Clone_size_corrected"), raster=F, max.cutoff = "q50")
ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected, col.name = "Clone_size_corrected_zeroes")
ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected_zeroes[is.na(ls_synthesis_v3_harmony_v3@meta.data$Clone_size_corrected_zeroes)] = 0
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("Clone_size_corrected_zeroes"), raster=F, min.cutoff=0, max.cutoff = "q85", pt.size=0.2)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("Clone_size_corrected_zeroes"), raster=F, min.cutoff=0, max.cutoff = "q95", pt.size=0.2)
FeaturePlot(ls_synthesis_v3_harmony_v3, features=c("Clone_size_corrected_zeroes"), raster=F, min.cutoff=0, max.cutoff = "q95", pt.size=0.2)
DimPlot(ls_synthesis_v3_harmony_v3, group.by=c("Clone_size_corrected"), raster=F, pt.size=0.2)
DimPlot(ls_synthesis_v3_harmony_v3, group.by=c("Clone_size_corrected"), raster=F, pt.size=0.2, cols=DiscretePalette(length(unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_size)),"glasbey"))
DimPlot(ls_synthesis_v3_harmony_v3, group.by=c("Clone_size_corrected"), raster=F, pt.size=0.2, cols=sample(DiscretePalette(length(unique(ls_synthesis_v3_harmony_v3@meta.data$Clone_size)),"glasbey")))





## work on figure 3 starting on 1/7/24
FeaturePlot(ls_in_v2, features=c("EMX2"), pt.size=1, order=T)
FeaturePlot(ls_glia_v2_harmony, features=c("EMX2"), pt.size=1, order=T)
FeaturePlot(ls_prog, features=c("EMX2"), pt.size=1, order=T)
DimPlot(ls_prog, group.by="subclust_idents_definitive", label=T)
FeaturePlot(ls_prog_glia, features=c("EMX2"), pt.size=1, order=T)
DimPlot(ls_prog, label=T)
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive", label=T)
FeaturePlot(ls_prog_glia, features=c("OLIG2", "GJA1", "MT3", "PDGFRA"))
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_size_corrected>2])
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_size_corrected>1])
DimPlot(ls_prog_glia)
FeaturePlot(ls_prog_glia, features=c("EMX1", "EMX2", "PAX6", "GLI3"))
FeaturePlot(ls_prog_glia, features=c("GSX2", "NKX2-1", "OTX2", "GLI3"))
FeaturePlot(ls_prog_glia, features=c("EMX2", "GLI3", "DMRT5", "DMRT3"))
ls_prog_glia = SetIdent(ls_prog_glia, value="subclust_idents_definitive")
VlnPlot(ls_prog_glia, features=c("EMX2", "GSX2"))
ls_prog_glia = SetIdent(ls_prog_glia, value="orig.ident")
VlnPlot(ls_prog_glia, features=c("GSX2", "PDGFRA"))

FeaturePlot(ls_prog_glia, features=c("VIM", "HES1", "GFAP", "SOX2"))
DimPlot(ls_prog_glia, group.by="subclust_idents_final", pt.size=1)
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$subclust_idents_final %in% "oRG"], pt.size=1)
FeaturePlot(ls_glia_v2_harmony, features=c("EMX2", "GSX2", "HOPX", "PDGFRA"))

FeaturePlot(ls_in_v2, features=c("EMX2", "GSX2", "NPY", "PAX6"))




ls_prog_olig = subset(ls_prog, subclust_idents_definitive %in% c("OPCs", "Oligodendrocytes"))
ls_prog_olig <- ScaleData(ls_prog_olig, features = all.genes)
ls_prog_olig = FindVariableFeatures(ls_prog_olig, selection.method="vst", nfeatures=2000)
ls_prog_olig = RunPCA(ls_prog_olig, features=VariableFeatures(object=ls_prog_olig))
ElbowPlot(ls_prog_olig)
ls_prog_olig <- RunHarmony(ls_prog_olig, c("orig.ident"))
ls_prog_olig <- FindNeighbors(ls_prog_olig, reduction = "harmony", dims = 1:15)
ls_prog_olig <- FindClusters(ls_prog_olig, resolution = .5)
ls_prog_olig <- RunUMAP(ls_prog_olig, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_olig, reduction = "umap",label=T)
DimPlot(ls_prog_olig, reduction = "umap", group.by = "orig.ident")
DimPlot(ls_prog_olig, reduction = "umap", group.by = "Clone_Index_filt")
DimPlot(ls_prog_olig, reduction = "umap", group.by = "subclust_idents_definitive")
DimPlot(ls_prog_olig, reduction = "umap", group.by = "age_pseudo")
FeaturePlot(ls_prog_olig, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_olig, features=c("MKI67", "PCNA", "BRIP1", "HOPX"), label=T)
FeaturePlot(ls_prog_olig, features=c("HOPX", "AQP4", "MT3", "VIM"), label=T)
FeaturePlot(ls_prog_olig, features=c("OLIG2", "OLIG1", "PDGFRA", "MBP"), label=T)
FeaturePlot(ls_prog_olig, features=c("EMX2", "OTX2", "NKX2-1", "GSX2"), label=T)
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$Clone_size_corrected>1])
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$Clone_size_corrected>2])
colnames(ls_prog_olig@meta.data)
table(ls_prog_olig@meta.data$ex_in_presence)
#EX_only      IN_only EX_IN_shared  no_EX_or_IN 
#114          523          158          525
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$ex_in_presence %in% c("EX_only")])
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$ex_in_presence %in% c("IN_only")])
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$ex_in_presence %in% c("EX_IN_shared")])
DimPlot(ls_prog_olig, cells.highlight=rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$ex_in_presence %in% c("no_EX_or_IN")])

library(ggrepel)
ls_prog_olig = SetIdent(ls_prog_olig, value="ex_in_presence")
levels(ls_prog_olig)
exin.de.markers.olig = FindMarkers(ls_prog_olig, ident.1 = "IN_only", ident.2 = "EX_only")
head(index.de.markers.ins, n=20)
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=index.de.markers.ins, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(index.de.markers.ins)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between EX_only (left) and IN_only (right) -- Oligo lineage")+
  theme(plot.title = element_text(hjust = 0.5))
VlnPlot(ls_prog_olig, features=c("GSX2"))
VlnPlot(ls_prog_olig, features=c("EMX2"))

FeaturePlot(ls_prog_olig, features=c("EMX2"), split.by="ex_in_presence", pt.size=1)
FeaturePlot(ls_prog_olig, features=c("GSX2"), split.by="ex_in_presence", pt.size=1)
FeaturePlot(ls_prog_olig, features=c("EMX2", "GSX2", "OLIG2", "MBP"))
FeaturePlot(ls_prog_olig, features=c("MKI67", "PCNA", "PDGFRA", "OLIG2"))




colnames(ls_synthesis_v3_harmony_v4@meta.data)
DimPlot(ls_synthesis_v3_harmony_v4, group.by="Clone_Index_filt", cols=c(osvz_col, vz_col), raster=F)
ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, NA, "Clone_Index_filt_multicell")
ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt_multicell[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)] = ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)]
table(ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt_multicell)
#Index3 IndexE 
#7109  11562
DimPlot(ls_synthesis_v3_harmony_v4, group.by="Clone_Index_filt_multicell", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, order=T, na.value='lightgray')
DimPlot(ls_synthesis_v3_harmony_v4, group.by="Clone_Index_filt_multicell", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray')

## going to add a column of all NAs, and get a plot of everything as NA
## then going to make a plot of subsetted Index E and Index 3 so that I can shuffle the order, and overlay that on top of the NA plot

ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, NA, "dummy_NA")
ls_synthesis_v3_harmony_v4@meta.data$dummy_NA = factor(ls_synthesis_v3_harmony_v4@meta.data$dummy_NA, levels = c("IndexE", "Index3", NA))
DimPlot(ls_synthesis_v3_harmony_v4, group.by="dummy_NA", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray')


DimPlot(ls_synthesis_v3_harmony_v4, group.by="dummy_NA", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray') + 
  ggtitle('GZ origin') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(ls_synthesis_v3_harmony_v4, group.by="dummy_NA", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray') + 
  ggtitle('GZ origin') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="umap_gz_origin_na_background.png", plot=image, width=10, height=8)
ggsave(file="umap_gz_origin_na_background.svg", plot=image, width=10, height=8)

DimPlot(subset(ls_synthesis_v3_harmony_v4, Clone_Index_filt_multicell %in% c("IndexE", "Index3")), group.by="Clone_Index_filt_multicell", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray') + 
  ggtitle('GZ origin') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(subset(ls_synthesis_v3_harmony_v4, Clone_Index_filt_multicell %in% c("IndexE", "Index3")), group.by="Clone_Index_filt_multicell", cols=c(osvz_col, vz_col), raster=F, pt.size = 1, shuffle=T, na.value='lightgray') + 
  ggtitle('GZ origin') +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="umap_gz_origin_multicell.png", plot=image, width=10, height=8)
ggsave(file="umap_gz_origin_multicell.svg", plot=image, width=10, height=8)




## 1/10/24 figure making
# did this on the other doc, copying here for posterity:
#ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive, col.name = "subclust_idents_upset_final")
#ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final == 'tRG'] = 'RG'
#ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final == 'oRG'] = 'RG'
#table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive)
#table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final)

DimPlot(ls_synthesis_v3_harmony_v4, group.by="subclust_idents_upset_final")
ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value="subclust_idents_upset_final")
ls_synthesis_v3_harmony_v4@active.ident = factor(ls_synthesis_v3_harmony_v4@active.ident, levels=c(
  "RG",
  "Astrocytes",
  "OPCs",
  "Oligodendrocytes",
  "EX_IPCs",
  "ENs",
  "IN_IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"
))
DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
image = DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
ggsave(file="full_umap_subclust_idents.png", plot=image, width=12, height=8)
ggsave(file="full_umap_subclust_idents.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20), legend.position="bottom")


## 1/11/24
## finalizing RG calls
#ls_prog -- glia, EX IPCS, IN IPCS, dividing, OPCs, oligos
#ls_prog = subset(ls_synthesis_v3_harmony_v3, subclust_idents_paper %in% c('Astrocytes', 'Dividing', 'EX IPCs', 'IN IPCs', 'OPCs', 'Radial glia', 'unk', 'Oligodendrocytes'))
ls_prog
#27581 features across 31925 samples within 1 assay 
DimPlot(ls_prog)
FeaturePlot(ls_prog, features=c("MKI67", "PCNA", "PDGFRA", "EOMES"), label=T)
FeaturePlot(ls_prog, features=c("OLIG2", "OLIG1", "PDGFRA", "GSX2"), label=T)

FeaturePlot(ls_prog, features=c("VIM", "HES1", "GFAP", "HOPX"), label=T)
VlnPlot(ls_prog, features=c("VIM", "HES1"))
#potential RG clusters: 0, 2, 5, 7, 8, 9, 10, 13, 14 (VIM)
#potential RG clusters: 0, 2, 5, 7, 8, 9, 10, 13, 14 (HES1)
#PDGFRA and OLIG2 have substantial staining in 2 (as does GSX2)
#4,6,7,9,10 are all dividing, 5 is unknown (likely EN doublets), and 14 is a small unknown

#ls_prog_div = subset(ls_prog, seurat_clusters %in% c(2,4,6,7,9,10))
#ls_prog_glia = subset(ls_prog, seurat_clusters %in% c(0,13))

DimPlot(ls_prog_glia, label=T)
FeaturePlot(ls_prog_glia, features=c("MKI67", "PCNA", "PDGFRA", "EOMES"))
FeaturePlot(ls_prog_glia, features=c("OLIG2", "OLIG1", "PDGFRA", "GSX2"), label=T)
VlnPlot(ls_prog_glia, features=c("OLIG2", "PDGFRA"))
DimPlot(ls_prog_glia_clust2)
FeaturePlot(ls_prog_glia_clust2, features=c("PDGFRA", "OLIG2", "GSX2", "GFAP"))
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive", label=T)


DimPlot(ls_prog_glia, group.by="subclust_idents_final")
DimPlot(ls_prog_glia_clust2, group.by="subclust_idents_final")
DimPlot(ls_prog, cells.highlight = rownames(ls_prog@meta.data)[rownames(ls_prog@meta.data) %in% rownames(ls_prog_glia_clust2@meta.data)])
#really should probably call everything in ls_prog_glia cluster2 as OPCs

FeaturePlot(ls_prog_glia, features=c("nCount_RNA"), label=T)
VlnPlot(ls_prog_glia, features=c("nCount_RNA"))
VlnPlot(ls_prog_glia, features=c("percent.mt"))
FeaturePlot(ls_prog_glia, features=c("NES"))
FeaturePlot(ls_prog_glia, features=c("HES1"))
FeaturePlot(ls_prog_glia, features=c("PDGFRB"))
FeaturePlot(ls_prog_glia, features=c("HES1", "LIFR", "MOXD1", "INPP1", "HOPX", "CRYAB"), ncol=3, label=T)
FeaturePlot(ls_prog_glia, features=c("HOPX", "MT3", "GFAP", "HES1", "LIFR", "GPX3"), ncol=3, label=T)
FeaturePlot(ls_prog_glia, features=c("HOPX", "INPP1", "PPM1K", "PTH2R", "FSTL5", "GPX3"), ncol=3, label=T)

FeaturePlot(ls_prog_glia, features=c("CRYAB"), label=T)
DimPlot(ls_prog_glia, group.by="age_pseudo")
DimPlot(ls_prog_glia, group.by="orig.ident")
DimPlot(ls_prog_glia, group.by="Clone_Index_filt")
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(filt_df_v3_multicell)])
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(filt_df_v3_multicell)[filt_df_v3_multicell$Clone_Index %in% "IndexE"]])
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(filt_df_v3_multicell)[filt_df_v3_multicell$Clone_Index %in% "Index3"]])
DimPlot(ls_prog_glia, cells.highlight=rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt])

FeaturePlot(ls_prog_glia, features=c("GSX2"), label=T)
#0 (or at least a subset of 0???) should definitely be called as oRG based on HOPX, GPX3, (some) MT3
#INPP1 is up in 0, looks pretty good in TN data
#cluster 5 is just a low-quality cluster, low RNA counts and marker genes are all RPL and RPS genes
#4 is definitely tRG based on CRYAB

FeaturePlot(ls_prog_glia, features=c("EMX2", "PAX6", "EMX1", "GLI3", "GSX2", "OTX2"), label=T, ncol=3, pt.size=0.5)
FeaturePlot(ls_prog_glia, features=c("HES1", "VIM", "GFAP", "NES", "HOPX", "CRYAB"), ncol=3, label=T, pt.size=0.5)
FeaturePlot(ls_prog_glia, features=c("GJA1", "FGF1", "AQP4", "SPARCL1", "MT3", "APOE"), ncol=3, label=T, pt.size=0.5)
FeaturePlot(ls_prog_glia, features=c("OLIG2", "OLIG1", "PDGFRA", "MBP", "BCAS1", "LIMS2"), ncol=3, label=T, pt.size=0.5)
FeaturePlot(ls_prog_glia, features=c("HOPX", "INPP1", "PPM1K", "PTH2R", "FSTL5", "GPX3"), ncol=3, label=T, pt.size=0.5)

FeaturePlot(ls_prog_glia, features=c("HES1", "VIM", "PAX6", "SPARCL1", "APOE", "PDGFRA", "INPP1", "GPX3", "CRYAB"), ncol=3, pt.size=0.5)

FeaturePlot(ls_prog_glia, features=c("HES1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("HES1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_HES1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_HES1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("VIM"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("VIM"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_VIM.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_VIM.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("PAX6"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("PAX6"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_PAX6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_PAX6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("SPARCL1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("SPARCL1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_SPARCL1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_SPARCL1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("APOE"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("APOE"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_APOE.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_APOE.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("GJA1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("GJA1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_GJA1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_GJA1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("INPP1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("INPP1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_INPP1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_INPP1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("GPX3"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("GPX3"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_GPX3.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_GPX3.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("CRYAB"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("CRYAB"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_CRYAB.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_CRYAB.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("PDGFRA"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("PDGFRA"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_PDGFRA.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_PDGFRA.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("OLIG2"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_OLIG2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_OLIG2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("MKI67"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("MKI67"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_MKI67.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_MKI67.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("HOPX"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("HOPX"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_HOPX.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_HOPX.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("ANXA1"),raster=F, cols=c("#e6e8e7", "blue"), min.cutoff="q50") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("ANXA1"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_ANXA1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_ANXA1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("ANXA2"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = FeaturePlot(ls_prog_glia, features=c("ANXA2"),raster=F, cols=c("#e6e8e7", "blue")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="ls_prog_glia_umap_ANXA2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_ANXA2.png", plot=image, width=4, height=3.2)

DimPlot(ls_prog_glia, cells.highlight=c(rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(filt_df_v3_multicell)]),raster=F, cols.highlight="#572f89") +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
image = DimPlot(ls_prog_glia, cells.highlight=c(rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(filt_df_v3_multicell)]),raster=F, cols.highlight="#572f89") +
  theme(axis.line =element_blank(), legend.position="none", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30))
ggsave(file="ls_prog_glia_multicell_clones.svg", plot=image, width=10, height=8)
ggsave(file="ls_prog_glia_multicell_clones.png", plot=image, width=10, height=8)

ls_prog_glia = AddMetaData(ls_prog_glia, ls_prog_glia@meta.data$subclust_idents_definitive_final, col.name="subclust_names_figure")
ls_prog_glia@meta.data$subclust_names_figure[ls_prog_glia$seurat_clusters %in% c(7)] = "bRG"
ls_prog_glia@meta.data$subclust_names_figure[ls_prog_glia$seurat_clusters %in% c(9)] = "Dividing"
ls_prog_glia@meta.data$subclust_names_figure[ls_prog_glia@meta.data$subclust_names_figure %in% c("aRG")] = "tRG"
ls_prog_glia@meta.data$subclust_names_figure[ls_prog_glia@meta.data$subclust_names_figure %in% c("OPCs")] = "Early OPCs"
ls_prog_glia@meta.data$subclust_names_figure[ls_prog_glia@meta.data$subclust_names_figure %in% c("OPCs")] = "Early OPCs"

DimPlot(ls_prog_glia, group.by="subclust_names_figure") +
  ggtitle("Radial glia") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_prog_glia, group.by="subclust_names_figure") +
  ggtitle("Radial glia") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave(file="ls_prog_glia_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_prog_glia_umap.png", plot=image, width=10, height=8)

ls_prog_glia = AddMetaData(ls_prog_glia, ls_prog_glia@meta.data$orig.ident, col.name = "orig.ident.merge")
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW18_0215_3S')] = 'GW18'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW24_0221_1S')] = 'GW24'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW21_0502_1S')] = 'GW21'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 'GW23'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 'GW22'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW16_1220_S1', 'GW16_1220_S2')] = 'GW16'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 'GW19'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2')] = 'GW20_V1'
ls_prog_glia@meta.data$orig.ident.merge[ls_prog_glia@meta.data$orig.ident.merge %in% c('GW20_0308_PFC_2S')] = 'GW20_PFC'

DimPlot(ls_prog_glia, group.by="orig.ident.merge") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_prog_glia, group.by="orig.ident.merge") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave(file="ls_prog_glia_by_sample_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_prog_glia_by_sample_umap.png", plot=image, width=10, height=8)

DimPlot(ls_prog_glia, group.by="age_pseudo") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_prog_glia, group.by="age_pseudo") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave(file="ls_prog_glia_by_age_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_prog_glia_by_age_umap.png", plot=image, width=10, height=8)

DimPlot(ls_prog_glia, group.by="Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=0.5, na.value = "lightgray") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_prog_glia, group.by="Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=0.5, na.value = "lightgray") +
  ggtitle("") +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave(file="ls_prog_glia_by_gz_umap.svg", plot=image, width=10, height=8)
ggsave(file="ls_prog_glia_by_gz_umap.png", plot=image, width=10, height=8)




FeaturePlot(ls_prog_glia, features=c("THY1", "CD24"), label=T, pt.size=2)


FeaturePlot(ls_prog, features=c("GJA1", "FGF1", "AQP4", "SPARCL1", "MT3", "APOE"), ncol=3, pt.size=0.5)

FeaturePlot(ls_glia_v2_harmony, features=c("GJA1", "FGF1", "AQP4", "SPARCL1", "MT3", "APOE"), ncol=3, pt.size=0.5)


ls_prog_glia = SetIdent(ls_prog_glia, value="orig.ident")
VlnPlot(ls_prog_glia, features=c("GSX2"))

FeaturePlot(ls_prog_glia, features=c("CSPG5"))
FeaturePlot(ls_prog_glia, features=c("FABP7"))


FeaturePlot(ls_prog, features=c("VIM", "GFAP", "OLIG2", "PDGFRA"))
FeaturePlot(ls_prog, features=c("HES1", "CRYAB", "HOPX", "PDGFRA"))


DimPlot(ls_prog_glia_clust6, label=T)
FeaturePlot(ls_prog_glia_clust6, features=c("AQP4", "CRYAB", "ITGB4", "SPARCL1", "C3", "IL32"), ncol=3)
FeaturePlot(ls_prog_glia_clust6, features=c("AQP4", "CRYAB", "ZEB1", "SPARCL1", "HOPX", "VIM"), ncol=3)
DimPlot(ls_prog_glia,cells.highlight = rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust6@meta.data)[ls_prog_glia_clust6@meta.data$seurat_clusters %in% c(0)]])

# going to set ls_prog_glia_clust6 to all astrocytes (based on AQP4 and SPARCL1 and following DA's definition of VZ-derived astroyctes)

ls_prog_glia_clust238 = subset(ls_prog_glia, seurat_clusters %in% c(2, 3, 8))
ls_prog_glia_clust238 <- ScaleData(ls_prog_glia_clust238, features = all.genes)
ls_prog_glia_clust238 = FindVariableFeatures(ls_prog_glia_clust238, selection.method="vst", nfeatures=2000)
ls_prog_glia_clust238 = RunPCA(ls_prog_glia_clust238, features=VariableFeatures(object=ls_prog_glia_clust238))
ElbowPlot(ls_prog_glia_clust238)
ls_prog_glia_clust238 <- RunHarmony(ls_prog_glia_clust238, c("orig.ident"))
ls_prog_glia_clust238 <- FindNeighbors(ls_prog_glia_clust238, reduction = "harmony", dims = 1:15)
ls_prog_glia_clust238 <- FindClusters(ls_prog_glia_clust238, resolution = .5)
ls_prog_glia_clust238 <- RunUMAP(ls_prog_glia_clust238, reduction = "harmony", dims=c(1:15))
DimPlot(ls_prog_glia_clust238, reduction = "umap",label=T)
DimPlot(ls_prog_glia_clust238, reduction = "umap", group.by = "orig.ident")
DimPlot(ls_prog_glia_clust238, reduction = "umap", group.by = "Clone_Index_filt")
DimPlot(ls_prog_glia_clust238, reduction = "umap", group.by = "subclust_idents_final")
FeaturePlot(ls_prog_glia_clust238, features=c("nCount_RNA"), max.cutoff="q90", label=T)
FeaturePlot(ls_prog_glia_clust238, features=c("MKI67", "PCNA", "BRIP1", "HOPX"), label=T)
FeaturePlot(ls_prog_glia_clust238, features=c("VIM", "GSX2", "OLIG2", "PDGFRA"), label=T)
FeaturePlot(ls_prog_glia_clust238, features=c("GJA1", "SPARCL1", "ANGPTL4", "GPX3"), label=T)
#0 and 1 are astrocytes based on ANGPTL4, SPARCL1
#5 is OPCs based on OLIG2 and GSX2

DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust238@meta.data)[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(5)]])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust238@meta.data)[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(3,5)]])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust238@meta.data)[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(3,5)]])

ls_prog_glia_clust238 = AddMetaData(ls_prog_glia_clust238, NA, col.name = "subclust_idents_definitive_final")
ls_prog_glia_clust238@meta.data$subclust_idents_definitive_final[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(0)] = "Astrocytes"
ls_prog_glia_clust238@meta.data$subclust_idents_definitive_final[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(1,2,3,5)] = "OPCs"
ls_prog_glia_clust238@meta.data$subclust_idents_definitive_final[ls_prog_glia_clust238@meta.data$seurat_clusters %in% c(4,6,7)] = "oRG"

DimPlot(ls_prog_glia_clust238, group.by="subclust_idents_definitive_final")


## going to try to add a new metadata column to ls_prog_glia and eventually to everything....... but only going to change ls_prog_glia, will keep everything else the same
ls_prog_glia = AddMetaData(ls_prog_glia, ls_prog_glia@meta.data$subclust_idents_definitive, col.name = "subclust_idents_definitive_final")
ls_prog_glia@meta.data$subclust_idents_definitive_final[rownames(ls_prog_glia@meta.data) %in% rownames(ls_prog_glia_clust238@meta.data)] = ls_prog_glia_clust238@meta.data$subclust_idents_definitive_final
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(6)] = "Astrocytes"
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive_final", pt.size=1)
#after discussing with Tom, going to use the following definitions:
#0, 1, and 7 are bRG
#4 aRG
#3 and 6 are astrocytes
#2 is OPCs 
#8 is mostly OPCs and some aRG
#10 is IPCs
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$subclust_idents_definitive_final %in% c("oRG")] = "bRG"
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1,7, 5, 9)] = "bRG"
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(2)] = "OPCs"
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(3,6)] = "Astrocytes"
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(4)] = "aRG"
ls_prog_glia@meta.data$subclust_idents_definitive_final[ls_prog_glia@meta.data$seurat_clusters %in% c(10)] = "EX IPCs"
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive_final", pt.size=1)

ggplot(ls_prog_glia@meta.data, aes(x=seurat_clusters, fill=orig.ident)) +
  geom_bar(position="fill")
ggplot(ls_prog_glia@meta.data, aes(x=seurat_clusters, fill=age_pseudo)) +
  geom_bar(position="fill")
ggplot(subset(ls_prog_glia@meta.data, Clone_Index_filt %in% c('Index3', 'IndexE')), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_y_continuous(breaks=c(0,0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875,1))
ggplot(subset(ls_prog_glia@meta.data, Clone_Index_filt %in% c('Index3', 'IndexE') & age_pseudo %in% '<=GW20'), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill")+
  scale_y_continuous(breaks=c(0,0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875,1))
ggplot(subset(ls_prog_glia@meta.data, Clone_Index_filt %in% c('Index3', 'IndexE') & age_pseudo %in% '>GW20'), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_y_continuous(breaks=c(0,0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875,1))
table(subset(ls_prog_glia@meta.data, Clone_Index_filt %in% c('Index3', 'IndexE') & age_pseudo %in% '<=GW20')$Clone_Index_filt)
#Index3 IndexE 
#911   1667
#.35   #.65
ggplot(ls_prog_glia@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_y_continuous(breaks=c(0,0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875,1))

rg.de.markers = FindMarkers(ls_prog_glia, ident.1 = 0, ident.2 = 1)
head(rg.de.markers, n=20)
ggplot(data=rg.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=rg.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(rg.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Cluster 1 (left) and Cluster 0 (right) -- Radial glia")+
  theme(plot.title = element_text(hjust = 0.5))


## what about Denise's VZ/OSVZ astrocyte markers?
ls_prog_glia = SetIdent(ls_prog_glia, value="seurat_clusters")
FeaturePlot(ls_prog_glia, features=c("ITGB4", "ANGPTL4"))
VlnPlot(ls_prog_glia, features=c("ITGB4", "ANGPTL4"))
FeaturePlot(ls_prog_glia, features=c("FLCN", "NTRK2", "MASP1", "C2orf72", "TIMP3", "ISLR2"), ncol=3)

astro.de.markers = FindMarkers(ls_prog_glia, ident.1 = 3, ident.2 = 6)
head(astro.de.markers, n=20)
ggplot(data=astro.de.markers, aes(x=avg_log2FC, y=p_val_adj))+geom_point()
ggplot(data=astro.de.markers, aes(x=avg_log2FC, y=-log10(p_val_adj), label=rownames(astro.de.markers)))+
  geom_point()+
  theme_minimal()+
  geom_text_repel()+
  ggtitle("Differential expression between Cluster 6 (left) and Cluster 3 (right) -- Astroycytes")+
  theme(plot.title = element_text(hjust = 0.5))

VlnPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "ANGPTL4"))
DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "ANGPTL4"))
DotPlot(ls_prog_glia, features=c("ITGB4", "ANGPTL4"))
DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "ANGPTL4",  "TIMP3"))
VlnPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "ANGPTL4",  "TIMP3"))
VlnPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "ANGPTL4",  "TIMP3"))
VlnPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ANGPTL2", "DPYD", "TIMP1", "TIMP3"))

ls_prog_glia = SetIdent(ls_prog_glia, value = "seurat_clusters")
DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "NES", "EMP3", "MRC2", "ORMDL2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1", "MPP6"))
ls_prog_glia = SetIdent(ls_prog_glia, value = "Clone_Index_filt")
DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "NES", "EMP3", "MRC2", "ORMDL2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1", "MPP6"))
VlnPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "NES", "EMP3", "MRC2", "ORMDL2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1", "MPP6"))
DoHeatmap(subset(ls_prog_glia, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "NES", "EMP3", "MRC2", "ORMDL2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1", "MPP6")) +
  scale_fill_gradientn(colors = c("white", "blue"))
DoHeatmap(subset(ls_prog_glia, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "EMP3", "MRC2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1")) +
  scale_fill_gradientn(colors = c("white", "blue"))

ls_prog_glia = SetIdent(ls_prog_glia, value = "seurat_clusters")
DoHeatmap(subset(ls_prog_glia, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "EMP3", "MRC2", "ANGPTL4",  "TIMP3", "FLCN", "NTRK2", "NPEPL1", "MASP1")) +
  scale_fill_gradientn(colors = c("white", "blue"))

FeaturePlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "S100A11", "ANGPTL4", "NTRK2"))

table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(3,6)])
table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(3)])
table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(6)])

ggplot(subset(ls_prog_glia@meta.data, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to astrocyte clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Astrocyte subcluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
image = ggplot(subset(ls_prog_glia@meta.data, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to astrocyte clusters') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Astrocyte subcluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggsave("ls_prog_glia_astrocyte_subclusters_index_contribution_barchart.svg", width=10, height=8)


DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "EMP3", "MRC2", "ANGPTL4",  "TIMP3", "NTRK2", "NPEPL1", "MASP1")) +
  ggtitle('Astrocyte gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
image = DotPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), features=c("ITGB4", "TIMP1", "S100A11", "CRYAB", "EMP3", "MRC2", "ANGPTL4",  "TIMP3", "NTRK2", "NPEPL1", "MASP1")) +
  ggtitle('Astrocyte gene expression')+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1)
  )
ggsave("ls_prog_glia_astrocyte_subclusters_dotplot.svg", width=10, height=8)


## have to guess at aRG and bRG for the dividing cluster now.....
ls_prog_div = SetIdent(ls_prog_div, value="seurat_clusters")
DimPlot(ls_prog_div, group.by="subclust_idents_definitive")
DimPlot(ls_prog_div, group.by="Clone_Index_filt", pt.size=0.5)
DimPlot(ls_prog_div, group.by="age_pseudo", pt.size=0.5)
DimPlot(subset(ls_prog_div, subclust_idents_definitive %in% "RG"), group.by="age_pseudo", pt.size=0.5)
FeaturePlot(ls_prog_div, features=c("EMX2", "PAX6", "EMX1", "GLI3", "GSX2", "OTX2"), ncol=3, label=T)
FeaturePlot(ls_prog_div, features=c("OLIG2", "OLIG1", "PDGFRA", "DLX2", "GSX2", "NPY"), ncol=3, label=T)
FeaturePlot(ls_prog_div, features=c("HES1", "VIM", "GFAP", "NES", "HOPX", "CRYAB"), ncol=3, label=T)
FeaturePlot(ls_prog_div, features=c("HOPX", "INPP1", "PPM1K", "PTH2R", "FSTL5", "GPX3"), ncol=3, label=T)
FeaturePlot(subset(ls_prog_div, subclust_idents_definitive %in% "RG"), features=c("EMX2", "PAX6", "EMX1", "GLI3", "GSX2", "OTX2"), ncol=3, label=T)
#can't possibly do this. going to leave as general RG and will only do aRG and bRG for Fig 3 on the non-dividing cluster

## have to roll the identities all the way up the foodchain now
ls_prog = AddMetaData(ls_prog, ls_prog@meta.data$subclust_idents_definitive, col.name = "subclust_idents_definitive_final")
ls_prog@meta.data$subclust_idents_definitive_final[rownames(ls_prog@meta.data) %in% rownames(ls_prog_glia@meta.data)] = ls_prog_glia@meta.data$subclust_idents_definitive_final
DimPlot(ls_prog, group.by="subclust_idents_definitive_final", pt.size=1)

ls_synthesis_v3_harmony_v3 = AddMetaData(ls_synthesis_v3_harmony_v3, ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive, col.name = "subclust_idents_definitive_final")
ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive_final[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(ls_prog@meta.data)] = ls_prog@meta.data$subclust_idents_definitive_final
DimPlot(ls_synthesis_v3_harmony_v3, group.by="subclust_idents_definitive_final", pt.size=1, raster=F)

ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive, col.name = "subclust_idents_definitive_final")
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive_final[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(ls_prog@meta.data)] = ls_prog@meta.data$subclust_idents_definitive_final[rownames(ls_prog@meta.data) %in% rownames(ls_synthesis_v3_harmony_v4@meta.data)]
DimPlot(ls_synthesis_v3_harmony_v4, group.by="subclust_idents_definitive_final", pt.size=1, raster=F)
## IMPORTANT NOTE
## subclust_idents_definitive_final has updated calls for Astrocytes, OPCs, and RG. RG are split into aRG and bRG here.
## the aRG and bRG calls are only relevant to Fig 3, and otherwise should be grouped as RG for the most part
colnames(ls_synthesis_v3_harmony_v4@meta.data)
ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive_final, col.name = "subclust_idents_upset_definitive_final")
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final %in% c('bRG', 'aRG')] = "RG"
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final %in% c('EX IPCs')] = "EX_IPCs"
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final %in% c('IN IPCs')] = "IN_IPCs"
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final)
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final)
DimPlot(ls_synthesis_v3_harmony_v4, group.by="subclust_idents_definitive_final", pt.size=1, raster=F)



## 1/14: save.image("/media/chang/HDD-8/mgkeefe/231210_local_sticr_synthesis.RData")



## 1/12 stuff

FeaturePlot(subset(ls_synthesis_v3_harmony_v4, Clone_In), features=c("TBR1"))
##save.image("/media/chang/HDD-8/mgkeefe/231210_local_sticr_synthesis.RData")

## 1/15
library(ggplot2)
ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=subclust_idents_upset_definitive_final, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position="fill") +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  labs(fill="Sample age") +
  theme_bw()
ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=1, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position = "fill", width = 1) +
  xlim(0,11) +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  theme_bw()


ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=age_pseudo, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  labs(fill="Cell type") +
  theme_bw()

ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=age_pseudo, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% "<=GW20"), aes(x=age_pseudo, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  ggtitle("<=GW20") +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% ">GW20"), aes(x=age_pseudo, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  theme_bw()


ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  #coord_polar("y", start=0) +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% ">GW20" & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% ">GW20" & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  #coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  theme_bw()

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell) & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "IndexE"])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell) & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "Index3"])


table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final[ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% "<=GW20"])/sum(ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% "<=GW20")
#Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#0.008511841      0.669090395      0.148140587      0.007710911      0.026085181      0.020133174      0.009140021 
#IN_OB Oligodendrocytes             OPCs               RG 
#0.002465607      0.002277153      0.017102205      0.089342924 
0.008511841+0.017102205+0.002277153
#[1] 0.0278912 -- astro+OPC+oligo
0.007710911+0.026085181+0.020133174+0.009140021+0.002465607
#0.06553489
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final[ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20"])/sum(ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20")
#Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#0.02175329       0.41366757       0.04779803       0.04702853       0.08541494       0.12013141       0.06002131 
#IN_OB Oligodendrocytes             OPCs               RG 
#0.01897123       0.01855688       0.07493785       0.09171895
0.02175329+0.01855688+0.07493785
#1] 0.115248 -- astro+OPC+oligo
0.04702853+0.08541494+0.12013141+0.06002131+0.01897123
#[1] 0.3315674



ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% "<=GW20" & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  coord_polar("y", start=0) +
  ggtitle("<=GW20") +
  labs(fill="Cell type") +
  theme_bw()
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% "<=GW20" & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=Clone_Index_filt, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  #coord_polar("y", start=0) +
  ggtitle("<=GW20") +
  labs(fill="Cell type") +
  theme_bw()

ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "age_pseudo")
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("SATB2", "TBR1"), pt.size=0.1)
VlnPlot(subset(ls_synthesis_v3_harmony_v4, subclust_idents_definitive_final %in% "ENs"), features=c("SATB2", "TBR1"), pt.size=0.1)
VlnPlot(subset(ls_synthesis_v3_harmony_v4, subclust_idents_definitive_final %in% "ENs"), features=c("SATB2", "TBR1"), pt.size=0)
VlnPlot(subset(ls_synthesis_v3_harmony_v4, subclust_idents_definitive_final %in% "ENs"), features=c("SATB2", "TBR1"), pt.size=0)

VlnPlot(ls_synthesis_v3_harmony_v4, features=c("NR4A1", "CRYM"), pt.size=0.1)

ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "Clone_Index_filt")
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("SATB2", "TBR1"), pt.size=0.1)
VlnPlot(subset(ls_synthesis_v3_harmony_v4, age_pseudo %in% ">GW20" & Clone_Index_filt %in% c("IndexE", "Index3") & subclust_idents_definitive_final %in% "ENs"), features=c("SATB2", "TBR1"), pt.size=0.1)



DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)])


colnames(ls_synthesis_v3_harmony_v4@meta.data)
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=subclust_idents_upset_definitive_final, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=1, fill=Clone_Index_filt)) +
  geom_bar(position="fill", width=1) +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  xlim(c(0,11)) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & Clone_size_corrected>=2), aes(x=subclust_idents_upset_definitive_final, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & Clone_size_corrected>=2), aes(x=1, fill=Clone_Index_filt)) +
  geom_bar(position="fill", width=1) +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  xlim(c(0,11)) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & age_pseudo %in% "<=GW20"), aes(x=subclust_idents_upset_definitive_final, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & age_pseudo %in% "<=GW20"), aes(x=1, fill=Clone_Index_filt)) +
  geom_bar(position="fill", width=1) +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  xlim(c(0,11)) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & age_pseudo %in% ">GW20"), aes(x=subclust_idents_upset_definitive_final, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Clone_Index_filt %in% c("IndexE", "Index3") & age_pseudo %in% ">GW20"), aes(x=1, fill=Clone_Index_filt)) +
  geom_bar(position="fill", width=1) +
  ggtitle('GZ contribution to cell types') +
  scale_fill_manual(values=c(osvz_col, vz_col))+
  ylab("Proportion of cells in cluster") +
  xlab("Cluster") +
  xlim(c(0,11)) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))


## 1/17 random numbers for the text
table(ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge)
#GW16     GW18     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#12790      952     9692    10664    29578      395     8604    18780     6009 
table(ls_in_v2@meta.data$orig.ident.merge)
#GW16     GW18     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#671       62      213     1009     1033       88     2662     4107     2273
table(ls_in_v2@meta.data$orig.ident.merge)/table(ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge)
#GW16       GW18       GW19   GW20_PFC    GW20_V1       GW21       GW22       GW23       GW24 
#0.05246286 0.06512605 0.02197689 0.09461740 0.03492461 0.22278481 0.30939098 0.21869010 0.37826593
(0.05246286+0.06512605+0.02197689+0.09461740+0.03492461)/5
#[1] 0.05382156 -- mean INs in each sample, <=GW20
(0.22278481+0.30939098+0.21869010+0.37826593)/4
#[1] 0.282283 -- mean INs in each sample, >GW20
sd(c(0.05246286,0.06512605,0.02197689,0.09461740,0.03492461))/sqrt(length(c(0.05246286,0.06512605,0.02197689,0.09461740,0.03492461)))
#[1] 0.01257998 -- sem INs in each sample, <=GW20
sd(c(0.22278481,0.30939098,0.21869010,0.37826593))/sqrt(length(c(0.22278481,0.30939098,0.21869010,0.37826593)))
#[1] 0.03822266 -- sem INs in each sample, >GW20
DimPlot(ls_in_v2, group.by="seurat_clusters")

table(ls_in_v2@meta.data$subclust_idents_paper[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell)])
#IN IPCs   IN_CGE IN_local   IN_MGE    IN_OB 
#   715       10     2043       24       10
table(ls_in_v2@meta.data$subclust_idents_paper[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell)])/sum(rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell))
#IN IPCs      IN_CGE    IN_local      IN_MGE       IN_OB 
#0.255174875 0.003568879 0.729122056 0.008565310 0.003568879 
25.5+72.9
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell)])
mge_multicell_clones = unique(ls_in_v2@meta.data$Clone_IDs_filt[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell) & ls_in_v2@meta.data$subclust_idents_paper %in% "IN_MGE"])
compiled_clust_df_v18[compiled_clust_df_v18$Clone_IDs %in% mge_multicell_clones,]
filt_df_v3[filt_df_v3$Clone_IDs %in% mge_multicell_clones,]
table(ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$Clone_IDs_filt %in% mge_multicell_clones])
table(ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$Clone_IDs_filt %in% mge_multicell_clones & ls_in_v2@meta.data$subclust_idents_paper %in% "IN_MGE"])
#GW16 GW20_PFC  GW20_V1     GW23     GW24 
#  3        2        1       12        6
cge_multicell_clones = unique(ls_in_v2@meta.data$Clone_IDs_filt[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell) & ls_in_v2@meta.data$subclust_idents_paper %in% "IN_CGE"])

length(dorsal_in_shared_clones_filt)
#[1] 757
757/2043
#[1] 0.3705335
length(ex_in_shared_clones_filt)
315/2043
#[1] 315

table(ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$Clone_IDs_filt %in% mge_multicell_clones & ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_IPCs", "IN_local")])
#GW16 GW23 
#3   21
table(ls_in_v2@meta.data$orig.ident.merge[ls_in_v2@meta.data$Clone_IDs_filt %in% cge_multicell_clones & ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_IPCs", "IN_local")])
#GW22 GW23 
#3    3

30/2043
#[1] 0.01468429

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% c(mge_multicell_clones, cge_multicell_clones)])
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% c(dorsal_in_shared_clones_filt)])


FeaturePlot(ls_prog_glia, features=c("CD44"), pt.size=1)
FeaturePlot(ls_prog_glia, features=c("FOXG1"), pt.size=1)

table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)])/(271+116)
#  Index3    IndexE 
#0.2997416 0.7002584


## 1/18 supp figure work
## clones shared across different samples
colnames(ls_synthesis_v3_harmony_v4@meta.data)
table(ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge)
# GW16     GW18     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#12790      952     9692    10664    29578      395     8604    18780     6009
head(ls_synthesis_v3_harmony_v4@meta.data$Clone_barcodes[ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge %in% "GW24"])
ls_synthesis_v3_harmony_v4_meta = ls_synthesis_v3_harmony_v4@meta.data
ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare = ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt
sum(is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare))
table(ls_synthesis_v3_harmony_v4_meta$Barcode_UMI_Count_filt)
ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$Barcode_UMI_Count_filt<=2] = NA
head(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24"])
head(paste(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)], "t", sep=""))
length(paste(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)], "t", sep=""))

ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)] = paste(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)], "t", sep="")
gw16_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW16" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw18_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW18" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw19_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW19" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw20_pfc_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW20_PFC" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw20_v1_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW20_V1" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw21_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW21" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw22_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW22" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw23_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW23" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])
gw24_bcs = unique(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare[ls_synthesis_v3_harmony_v4_meta$orig.ident.merge %in% "GW24" & !is.na(ls_synthesis_v3_harmony_v4_meta$Clone_barcodes_filt_compare)])

bc_collision_df = data.frame("Clone_barcodes_filt_compare" = c(gw16_bcs, gw19_bcs, gw20_pfc_bcs, gw20_v1_bcs, gw21_bcs, gw22_bcs, gw23_bcs, gw24_bcs),
                             "orig.ident.merge" = c(rep("GW16", length(gw16_bcs)),
                                                    rep("GW19", length(gw19_bcs)),
                                                    rep("GW20_PFC", length(gw20_pfc_bcs)),
                                                    rep("GW20_V1", length(gw20_v1_bcs)),
                                                    rep("GW21", length(gw21_bcs)),
                                                    rep("GW22", length(gw22_bcs)),
                                                    rep("GW23", length(gw23_bcs)),
                                                    rep("GW24", length(gw24_bcs))))

length(c(gw16_bcs,gw19_bcs,gw20_pfc_bcs,gw20_v1_bcs,gw21_bcs,gw22_bcs,gw23_bcs,gw24_bcs))
#[1] 46243
length(unique(c(gw16_bcs,gw19_bcs,gw20_pfc_bcs,gw20_v1_bcs,gw21_bcs,gw22_bcs,gw23_bcs,gw24_bcs)))
#[1] 46226
46243-46226
#[1] 17
(17/46243)*100
#[1] 0.03676232

length(c(gw16_bcs,gw19_bcs))
#[1] 10285
length(unique(c(gw16_bcs,gw19_bcs)))
#[1] 10285

length(c(gw16_bcs,gw23_bcs))
#[1] 17156
length(unique(c(gw16_bcs,gw23_bcs)))
#[1] 17154

length(c(gw16_bcs,gw20_v1_bcs))
#[1] 22127
length(unique(c(gw16_bcs,gw20_v1_bcs)))
#[1] 22127

length(bc_collision_df$Clone_barcodes_filt_compare[bc_collision_df$orig.ident.merge %in% c("GW16", "GW23")])
#[1] 17156
length(unique(bc_collision_df$Clone_barcodes_filt_compare[bc_collision_df$orig.ident.merge %in% c("GW16", "GW23")]))
#[1] 17154

length(bc_collision_df$Clone_barcodes_filt_compare[bc_collision_df$orig.ident.merge %in% c("GW16", "GW20_V1")])
#[1] 22127
length(unique(bc_collision_df$Clone_barcodes_filt_compare[bc_collision_df$orig.ident.merge %in% c("GW16", "GW20_V1")]))
#[1] 22124


## going to try to make a matrix of BC collisions

sample_idents = c("GW16","GW19", "GW20_PFC", "GW20_V1", "GW21", "GW22", "GW23", "GW24")
length(sample_idents)
bc_matrix_counts = matrix(rep(0, length(sample_idents)*length(sample_idents)), ncol=length(sample_idents))
dim(bc_matrix_counts)
rownames(bc_matrix_counts) = sample_idents
colnames(bc_matrix_counts) = sample_idents

for (sample_id in sample_idents) {
  temp_collision_count = c()
  for (sample_id2 in sample_idents){
    bc_collision_df_filt = bc_collision_df[bc_collision_df$orig.ident.merge %in% c(sample_id, sample_id2),]
    n_collisions = length(bc_collision_df_filt$Clone_barcodes_filt_compare) - length(unique(bc_collision_df_filt$Clone_barcodes_filt_compare))
    temp_collision_count = c(temp_collision_count, n_collisions)
  }
  bc_matrix_counts[rownames(bc_matrix_counts) %in% sample_id,] = as.numeric(temp_collision_count)
}
bc_matrix_counts

bc_matrix_pct = matrix(rep(0, length(sample_idents)*length(sample_idents)), ncol=length(sample_idents))
dim(bc_matrix_pct)
rownames(bc_matrix_pct) = sample_idents
colnames(bc_matrix_pct) = sample_idents

for (sample_id in sample_idents) {
  temp_collision_count = c()
  for (sample_id2 in sample_idents){
    bc_collision_df_filt = bc_collision_df[bc_collision_df$orig.ident.merge %in% c(sample_id, sample_id2),]
    n_collisions = length(bc_collision_df_filt$Clone_barcodes_filt_compare) - length(unique(bc_collision_df_filt$Clone_barcodes_filt_compare))
    pct_collisions = (n_collisions/length(bc_collision_df_filt$Clone_barcodes_filt_compare))*100
    temp_collision_count = c(temp_collision_count, pct_collisions)
  }
  bc_matrix_pct[rownames(bc_matrix_pct) %in% sample_id,] = as.numeric(temp_collision_count)
}
bc_matrix_pct

heatmap(bc_matrix_counts, Rowv = NA, Colv = NA)

## 1/18
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240118_figs')


bc_collision_df_euler = cbind(
  GW16 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw16_bcs,
  GW19 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw19_bcs,
  GW20_PFC = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw20_pfc_bcs,
  GW20_V1 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw20_v1_bcs,
  GW21 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw21_bcs,
  GW22 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw22_bcs,
  GW23 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw23_bcs,
  GW24 = unique(bc_collision_df$Clone_barcodes_filt_compare) %in% gw24_bcs
)
euler(bc_collision_df_euler)
plot(euler(bc_collision_df_euler))
image = plot(euler(bc_collision_df_euler))
ggsave(file='barcode_collision_venn.svg', plot=image, width=12, height=8)



ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "subclust_idents_upset_definitive_final")
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final)
ls_synthesis_v3_harmony_v4@active.ident = factor(ls_synthesis_v3_harmony_v4@active.ident, levels=c(
  "RG",
  "Astrocytes",
  "OPCs",
  "Oligodendrocytes",
  "EX_IPCs",
  "ENs",
  "IN_IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"
))
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("percent.mt"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("percent.mt"), pt.size=0)+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_mt_pct.svg', plot=image, width=12, height=8)

VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nCount_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nCount_RNA"), pt.size=0) + 
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_ncount_rna.svg', plot=image, width=12, height=8)

VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nFeature_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nFeature_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_nfeature_rna.svg', plot=image, width=12, height=8)

ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "orig.ident.merge")
ls_synthesis_v3_harmony_v4@active.ident = factor(ls_synthesis_v3_harmony_v4@active.ident, levels=c(
  "GW16",
  "GW18",
  "GW19",
  "GW20_PFC",
  "GW20_V1",
  "GW21",
  "GW22",
  "GW23",
  "GW24"
))
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("percent.mt"), pt.size=0) + 
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
                                                                                 legend.title = element_text(size=25), legend.text = element_text(size=25), 
                                                                                 plot.title = element_text(size=30, hjust=0.5),
                                                                                 axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("percent.mt"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_mt_pct_by_sample.svg', plot=image, width=12, height=8)

VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nCount_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nCount_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_ncount_rna_by_sample.svg', plot=image, width=12, height=8)

VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nFeature_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("nFeature_RNA"), pt.size=0) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=25, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave(file='qc_violin_nfeature_rna_by_sample.svg', plot=image, width=12, height=8)

dim(ls_synthesis_v3_harmony_v3_clustermarkers)
ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "subclust_idents_upset_definitive_final")
ls_synthesis_v3_harmony_v4@active.ident = factor(ls_synthesis_v3_harmony_v4@active.ident, levels=c(
  "RG",
  "Astrocytes",
  "OPCs",
  "Oligodendrocytes",
  "EX_IPCs",
  "ENs",
  "IN_IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"
))

ls_synthesis_v3_harmony_v4.markers <- FindAllMarkers(ls_synthesis_v3_harmony_v4, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_synthesis_v3_harmony_v4.markers)
ls_synthesis_v3_harmony_v4_clustermarkers_unfilt = as.data.frame(ls_synthesis_v3_harmony_v4.markers %>% group_by(cluster))
write.csv(ls_synthesis_v3_harmony_v4_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v4_clustermarkers_v1_unfilt.csv',row.names=F)
ls_synthesis_v3_harmony_v4.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_synthesis_v3_harmony_v4_clustermarkers = as.data.frame(ls_synthesis_v3_harmony_v4.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_synthesis_v3_harmony_v4_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v4_clustermarkers_v1.csv',row.names=F)

ls_synthesis_v3_harmony_v4_top10_clustermarkers = as.data.frame(ls_synthesis_v3_harmony_v4.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC))

setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240118_figs')
random_cell_list = c()
for(sc in unique(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final)){
  cells_in_sc = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final %in% sc]
  random_cell_list = c(random_cell_list, sample(cells_in_sc, size = min(length(cells_in_sc),1000), replace = F))
}
DoHeatmap(ls_synthesis_v3_harmony_v4, features=ls_synthesis_v3_harmony_v4_top10_clustermarkers$gene, cells = random_cell_list)
image = DoHeatmap(ls_synthesis_v3_harmony_v4, features=ls_synthesis_v3_harmony_v4_top10_clustermarkers$gene, cells = random_cell_list)
ggsave('all_clusters_heatmap_top10_genes_1000cells.svg', image, width=8, height=12)
ggsave('all_clusters_heatmap_top10_genes_1000cells.png', image, width=8, height=12)

DotPlot(ls_synthesis_v3_harmony_v4, features=unique(ls_synthesis_v3_harmony_v4_top10_clustermarkers$gene)) +
  coord_flip() + 
  theme(axis.text.x = element_text(angle=30, hjust=1))
image = DotPlot(ls_synthesis_v3_harmony_v4, features=unique(ls_synthesis_v3_harmony_v4_top10_clustermarkers$gene)) +
  coord_flip() +
  theme(axis.text.x = element_text(angle=30, hjust=1))
ggsave('all_clusters_dotplot_top10_genes_1000cells.svg', image, width=8, height=12)
ggsave('all_clusters_dotplot_top10_genes_1000cells.png', image, width=8, height=12)


ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Barcode_UMI_Count_filt >= 3), aes(x=Barcode_UMI_Count_filt)) +
  geom_histogram(bins=200) +
  xlim(c(0,200))

setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240118_figs')
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Barcode_UMI_Count_filt >= 3), aes(x=log10(Barcode_UMI_Count_filt))) +
  geom_histogram(bins=24) +
  ggtitle('Barcode UMI Count, all cells with >=3 UMIs') +
  theme_bw() +
  xlab("log10(Barcode UMI Count)") +
  ylab("Number of cells") +
  xlim(c(log10(2.4), 3.1))+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, hjust=1), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Barcode_UMI_Count_filt >= 3), aes(x=log10(Barcode_UMI_Count_filt))) +
  geom_histogram(bins=24) +
  ggtitle('Barcode UMI Count, all cells with >=3 UMIs') +
  theme_bw() +
  xlab("log10(Barcode UMI Count)") +
  ylab("Number of cells") +
  xlim(c(log10(2.4), 3.1))+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, hjust=1), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave('histogram_barcode_umis_all_cells.svg', image, width=10, height=8)

ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Barcode_UMI_Count_filt >= 3 & rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)), aes(x=log10(Barcode_UMI_Count_filt))) +
  geom_histogram(bins=25) +
  ggtitle('Barcode UMI Count, cells in multicellular clones with >=3 UMIs') +
  theme_bw() +
  xlab("log10(Barcode UMI Count)") +
  ylab("Number of cells") +
  xlim(c(log10(2.4), 3.1))+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, hjust=1), axis.title = element_text(size=25), 
   plot.title = element_text(size=30, hjust=0.5),
   axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, Barcode_UMI_Count_filt >= 3 & rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)), aes(x=log10(Barcode_UMI_Count_filt))) +
  geom_histogram(bins=25) +
  ggtitle('Barcode UMI Count, cells in multicellular clones with >=3 UMIs') +
  theme_bw() +
  xlab("log10(Barcode UMI Count)") +
  ylab("Number of cells") +
    xlim(c(log10(2.4), 3.1))+
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, hjust=1), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave('histogram_barcode_umis_multicellular_clones.svg', image, width=10, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3)], sizes.highlight = 0.1) +
  scale_color_manual(labels = c("No barcode", "Barcode detected"), values=c("lightgray","limegreen")) +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3)], sizes.highlight = 0.1) +
  scale_color_manual(labels = c("No barcode", "Barcode detected"), values=c("lightgray","limegreen")) +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave('full_umap_cells_with_barcodes.png', image, width=10, height=8)
ggsave('full_umap_cells_with_barcodes.svg', image, width=10, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)], sizes.highlight = 0.5) +
  scale_color_manual(labels = c("Clone size <= 1", "Clone size >= 2"), values=c("lightgray","darkgreen")) +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)], sizes.highlight = 0.5) +
  scale_color_manual(labels = c("Clone size <= 1", "Clone size >= 2"), values=c("lightgray","darkgreen")) +
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20))
ggsave('full_umap_multicellular_clones.png', image, width=10, height=8)
ggsave('full_umap_multicellular_clones.svg', image, width=10, height=8)

colnames(ls_synthesis_v3_harmony_v4@meta.data)


colnames(ls_in_v2@meta.data)
ls_in_v2 = SetIdent(ls_in_v2, value = "subclust_idents_paper")
DotPlot(ls_in_v2, features=c("NR2F1", "NFIX", "NR2F2", "SOX6", "KLHL35", "CXCR4",
                             "TSHZ1", "HOMER3", "ETV1", "CALB2", "PBX3"))

ls_in_v2 = SetIdent(ls_in_v2, value = "seurat_clusters")
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c(5,3,2,4,9,7,0,8,1,6))
DimPlot(ls_in_v2, label=T)
DotPlot(ls_in_v2, features=c("NR2F1", "NFIX", "NR2F2", "SOX6", "KLHL35", "CXCR4",
                             "TSHZ1", "HOMER3", "ETV1", "CALB2", "PBX3"))
DotPlot(ls_in_v2, features=c("SCGN", "PAX6", "NR2F1", "MEIS2", "SP8", "GSX2", "NPY", "TAGLN3", "BTG1"))
DotPlot(subset(ls_in_v2, seurat_clusters %in% c(2,3,4,5,7,9)), features=c("NR2F1", "NFIX", "NR2F2", "SOX6", "KLHL35", "CXCR4",
                             "TSHZ1", "HOMER3", "ETV1", "CALB2", "PBX3"))
DoHeatmap(subset(ls_in_v2, seurat_clusters %in% c(2,3,4,5,7,9)), features=c("NR2F1", "NFIX", "NR2F2", "SOX6", "KLHL35", "CXCR4",
                                                                          "TSHZ1", "HOMER3", "ETV1", "CALB2", "PBX3"))

DotPlot(subset(ls_in_v2, subclust_idents_paper %in% c("IN_local", "IN IPCs") & seurat_clusters %in% c(2,3,4,5,7,9)), features=c("COL1A2", "LGALS1", "NPY",
                                                                                                                                "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6",
                                                                                                                                "MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1", "PAX6", "SCGN", "PROX1", "SP8", "CALB2"))
DotPlot(subset(ls_in_v2, subclust_idents_paper %in% c("IN_local", "IN IPCs") & seurat_clusters %in% c(2,3,4,5,7,9)), features=c("COL1A2", "LGALS1", "NPY",
                                                                          "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6",
                                                                          "MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1"))
image = DotPlot(subset(ls_in_v2, subclust_idents_paper %in% c("IN_local", "IN IPCs") & seurat_clusters %in% c(2,3,4,5,7,9)), features=c("COL1A2", "LGALS1", "NPY",
                                                                                                                                        "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6",
                                                                                                                                        "MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1"))
#OB (IN.2) markers: c("MEIS2", "PBX3", "RUNX1T1", "ETV1", "TSHZ1")
#CGE-like(IN.3) markers: c("NRXN1", "MEG3", "KLHL35", "MYT1L", "NR2F1", "NFIX", "NR2F2", "SOX6")

DoHeatmap(subset(ls_in_v2, subclust_idents_paper %in% c("IN_local", "IN IPCs") & seurat_clusters %in% c(2,3,4,5,7,9)), features=c("TAGLN3", "BTG1", "NSG1", "NR2F1", "NFIX", "NR2F2", "SOX6", "KLHL35", "CXCR4",
                                                                            "TSHZ1", "HOMER3", "ETV1", "CALB2", "PBX3"))
DoHeatmap(subset(ls_in_v2, subclust_idents_paper %in% c("IN_local", "IN IPCs") & seurat_clusters %in% c(7,9)), features=c("PAX6", "SOX6", "NFIX", "SCGN", "TAGLN3", "BTG1", "NSG1", "NR2F2", "KLHL35")
                                                                                                                                )


table(ls_in_v2@meta.data$subclust_idents_paper)

gc()


## 1/22/24 figures

## turning scgn counts into graphs

scgn_raw_counts = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/local_STICR_scgn_raw_counts.csv', header = T)
scgn_raw_counts
colnames(scgn_raw_counts)[1] = "image"

scgn_avg_counts = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/local_STICR_scgn_average_counts.csv', header = T)
scgn_avg_counts
colnames(scgn_avg_counts)[1] = "image"


ggplot() +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=DLX2_SCGN_of_DLX2, size=2), color="#ffc561", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange", method = "lm") +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2_SCGN), color="#1a6d78") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=triple_of_DLX2, size=2), color="cyan", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2), color="#1a6d78", method = "lm") +
  ylab('Ratio of cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+ and DLX2+/SCGN+/PAX6+ cells') +
  labs(colour="DLX2+SCGN+PAX6+ of DLX2+") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=100), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5))

ggplot() +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=DLX2_SCGN_of_DLX2, size=2), color="#ffc561", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange", method = "lm") +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2_SCGN), color="#1a6d78") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=triple_of_DLX2, size=2), color="cyan", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2), color="#1a6d78", method = "lm") +
  ylab('Ratio of cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+ and DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot() +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=DLX2_SCGN_of_DLX2, size=2), color="#ffc561", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange", method = "lm") +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2_SCGN), color="#1a6d78") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=triple_of_DLX2, size=2), color="cyan", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2), color="#1a6d78", method = "lm") +
  ylab('Ratio of cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+ and DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=100), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5))
ggsave('scgn_counts_emergence_of_scgn_and_pax6.svg', image, width=10, height=8)

ggplot() +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange") +
  #geom_jitter(data = scgn_raw_counts, aes(x=age, y=DLX2_SCGN_of_DLX2, size=2), color="#ffc561", width=0.1) +
  #geom_smooth(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange", method = "lm") +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2_SCGN), color="#1a6d78") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=triple_of_DLX2, size=2), color="cyan", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2), color="#1a6d78", method = "lm") +
  ylab('Ratio of DLX2+ cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot() +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange") +
  #geom_jitter(data = scgn_raw_counts, aes(x=age, y=DLX2_SCGN_of_DLX2, size=2), color="#ffc561", width=0.1) +
  #geom_smooth(data = scgn_avg_counts, aes(x=age, y=DLX2_SCGN_of_DLX2), color="darkorange", method = "lm") +
  #geom_point(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2_SCGN), color="#1a6d78") +
  geom_jitter(data = scgn_raw_counts, aes(x=age, y=triple_of_DLX2, size=2), color="cyan", width=0.1) +
  geom_smooth(data = scgn_avg_counts, aes(x=age, y=triple_of_DLX2), color="#1a6d78", method = "lm") +
  ylab('Ratio of DLX2+ cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=20), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5))
ggsave('scgn_counts_emergence_of_dlx2_scgn_pax6_only_lineplot.svg', image, width=10, height=8)



scgn_raw_counts$age_pseudo = NA
scgn_raw_counts$age_pseudo[scgn_raw_counts$age <= 20] = "<=GW20"
scgn_raw_counts$age_pseudo[scgn_raw_counts$age > 20] = ">GW20"
scgn_avg_counts$age_pseudo = NA
scgn_avg_counts$age_pseudo[scgn_avg_counts$age <= 20] = "<=GW20"
scgn_avg_counts$age_pseudo[scgn_avg_counts$age > 20] = ">GW20"

  
ggplot(data = scgn_avg_counts, aes(x=age_pseudo, y=triple_of_DLX2_SCGN, fill=age_pseudo, color=age_pseudo)) +
  geom_bar(stat="summary", alpha=0.6) +
  geom_jitter(width=0.1, height=0, stroke=1) +
  scale_fill_manual(values=c("#c9c9c9", "#999999")) +
  scale_color_manual(values=c("#c9c9c9", "#999999")) +
  theme_bw() +
  ggtitle("Emergence of DLX2+/SCGN+/PAX6+ cells") +
  labs(color = "Sample ages", fill ="Sample ages") +
  xlab("Sample ages (GW)") +
  ylab("Ratio of DLX2+SCGN+ cells") +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot(data = scgn_avg_counts, aes(x=age_pseudo, y=triple_of_DLX2_SCGN, fill=age_pseudo, color=age_pseudo)) +
  geom_bar(stat="summary", alpha=0.6) +
  geom_jitter(width=0.1, height=0, stroke=1) +
  scale_fill_manual(values=c("#c9c9c9", "#999999")) +
  scale_color_manual(values=c("#c9c9c9", "#999999")) +
  theme_bw() +
  ggtitle("Emergence of DLX2+/SCGN+/PAX6+ cells") +
  labs(color = "Sample ages", fill ="Sample ages") +
  xlab("Sample ages (GW)") +
  ylab("Ratio of DLX2+SCGN+ cells") +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5))
ggsave('scgn_counts_emergence_of_scgn_pax6_of_dlx2_scgn.svg', image, width=10, height=8)

scgn_raw_counts

young_scgn_pct = scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age <= 20]
mean(young_scgn_pct)
#[1] 0.003011266
sd(young_scgn_pct)/sqrt(length(young_scgn_pct))
#[1] 0.001694705
old_scgn_pct = scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age > 20]
mean(old_scgn_pct)
#[1] 0.05874983
sd(old_scgn_pct)/sqrt(length(old_scgn_pct))
#[1] 0.01142364

## stats for scgn???
scgn_raw_counts

?t.test

t.test(scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age <=20], scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age >20])
scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age <=20]
scgn_raw_counts$triple_of_DLX2[scgn_raw_counts$age >20]
scgn_raw_counts


t.test(scgn_raw_counts$triple_of_DLX2_SCGN[scgn_raw_counts$age <=20], scgn_raw_counts$triple_of_DLX2_SCGN[scgn_raw_counts$age >20])
#data:  scgn_raw_counts$triple_of_DLX2_SCGN[scgn_raw_counts$age <= 20] and scgn_raw_counts$triple_of_DLX2_SCGN[scgn_raw_counts$age > 20]
#t = -4.6294, df = 13.135, p-value = 0.0004594
#alternative hypothesis: true difference in means is not equal to 0
#95 percent confidence interval:
#  -0.20731808 -0.07548275
#sample estimates:
#  mean of x  mean of y 
#0.01659544 0.15799585 

library("ggpubr")
ggscatter(scgn_raw_counts, x = "age", y = "triple_of_DLX2", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Sample age", ylab = "Ratio of triple positives to DLX2+") +
  theme_bw()
ggscatter(scgn_raw_counts, x = "age", y = "triple_of_DLX2", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson") +
  ylab('Ratio of DLX2+ cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=20), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5, ))
image = ggscatter(scgn_raw_counts, x = "age", y = "triple_of_DLX2", 
                  add = "reg.line", conf.int = TRUE, 
                  cor.coef = TRUE, cor.method = "pearson") +
  ylab('Ratio of DLX2+ cells') +
  xlab('Sample age') +
  ggtitle('Emergence of DLX2+/SCGN+/PAX6+ cells') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=20), legend.text = element_text(size=20), #note: doing this with legend title to make space to put real legend title in
        plot.title = element_text(size=30, hjust=0.5, ))
ggsave('scgn_counts_emergence_of_dlx2_scgn_pax6_only_lineplot_with_corr.svg', image, width=10, height=8)



## 1/23/24 -- breaking out RDS object for UCSC
colnames(ls_synthesis_v3_harmony_v4@meta.data)
## metadata to preserve:
  # orig.ident ---- rename to LS__ identifiers?
  # cell ID
  # orig.ident.merge
  # nCount_RNA
  # nFeature_RNA
  # percent.mt
  # sex
  # sample_age
  # age_pseudo
  # Clone_IDs_filt ----- apply filt_df_v3 filtering
  # Clone_Index_filt ----- apply filt_df_v3 filtering, and change IndexE --> VZ and Index3 --> OSVZ
  # Clone_size_corrected  ----- apply filt_df_v3 filtering and copy clone sizes from there
  # Barcode_UMI_Count_filt ----- apply filt_df_v3 filtering
  # multi_clone ----- apply filt_df_v3 filtering
  # subclust_idents_upset_definitive_final ----- rename to "subcluster" or "celltype" or something

cortex_midgestation = ls_synthesis_v3_harmony_v4
DimPlot(cortex_midgestation)
cortex_midgestation@assays$RNA@scale.data = matrix(0)
DimPlot(cortex_midgestation)
FeaturePlot(cortex_midgestation, features=c("SATB2", "VIM", "DLX5", "PDGFRA"))
dim(cortex_midgestation@meta.data)
length(rownames(cortex_midgestation@meta.data)[rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3)])
#[1] 58165
length(rownames(cortex_midgestation@meta.data)[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))])
#[1] 39299
cortex_midgestation@meta.data$Clone_IDs_filt[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))] = NA
cortex_midgestation@meta.data$Clone_Index_filt[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))] = NA
cortex_midgestation@meta.data$Clone_size_corrected[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))] = NA
cortex_midgestation@meta.data$Barcode_UMI_Count_filt[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))] = NA
cortex_midgestation@meta.data$multi_clone[!(rownames(cortex_midgestation@meta.data) %in% rownames(filt_df_v3))] = NA

table(cortex_midgestation@meta.data$orig.ident)
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW16_1220_S1"] = "GW16_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW16_1220_S2"] = "GW16_L2"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW18_0215_3S"] = "GW18_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW19_0221_S1"] = "GW19_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW19_0221_S2"] = "GW19_L2"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW20_0308_PFC_2S"] = "GW20_PFC_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW20_0308_V1_S1"] = "GW20_V1_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW20_0308_V1_S2"] = "GW20_V1_L2"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW21_0502_1S"] = "GW21_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW22_1118_S1"] = "GW22_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW22_1118_S2"] = "GW22_L2"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW22_1118_S3"] = "GW22_L3"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW23_0910_S1"] = "GW23_L1"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW23_0910_S2"] = "GW23_L2"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW23_0910_S4"] = "GW23_L3"
cortex_midgestation@meta.data$orig.ident[cortex_midgestation@meta.data$orig.ident %in% "GW24_0221_1S"] = "GW24_L1"
table(cortex_midgestation@meta.data$orig.ident)

cortex_midgestation_meta = cortex_midgestation@meta.data
cortex_midgestation_meta$Clone_IDs_actual = NA
cortex_midgestation_meta$Clone_Sizes_actual = NA
sum(rownames(cortex_midgestation_meta)[rownames(cortex_midgestation_meta) %in% rownames(filt_df_v3)] == rownames(filt_df_v3)[rownames(filt_df_v3) %in% rownames(cortex_midgestation_meta)])
#[1] 58165
cortex_midgestation_meta$Clone_Sizes_actual[rownames(cortex_midgestation_meta) %in% rownames(filt_df_v3)] = filt_df_v3$Clone_size_corr_3umi[rownames(filt_df_v3) %in% rownames(cortex_midgestation_meta)]
cortex_midgestation_meta$Clone_IDs_actual[rownames(cortex_midgestation_meta) %in% rownames(filt_df_v3)] = filt_df_v3$Clone_IDs[rownames(filt_df_v3) %in% rownames(cortex_midgestation_meta)]

head(cortex_midgestation_meta[,c("Clone_IDs_actual", "Clone_Sizes_actual")][rownames(cortex_midgestation_meta) %in% rownames(filt_df_v3),])
head(filt_df_v3[,c("Clone_IDs", "Clone_size_corr_3umi")][rownames(filt_df_v3) %in% rownames(cortex_midgestation_meta),])

cortex_midgestation@meta.data$Clone_size_filt = cortex_midgestation_meta$Clone_Sizes_actual
cortex_midgestation@meta.data$Clone_IDs_filt = cortex_midgestation_meta$Clone_IDs_actual
cortex_midgestation@meta.data$multi_clone = NA
cortex_midgestation@meta.data$multi_clone[cortex_midgestation@meta.data$Clone_size_filt >=2] = "multi_cell_clone"
cortex_midgestation@meta.data$multi_clone[cortex_midgestation@meta.data$Clone_size_filt ==1] = "single_cell_clone"
table(cortex_midgestation@meta.data$multi_clone)

cortex_midgestation@meta.data$Clone_Index_filt[cortex_midgestation@meta.data$Clone_Index_filt %in% "IndexE"] = "VZ"
cortex_midgestation@meta.data$Clone_Index_filt[cortex_midgestation@meta.data$Clone_Index_filt %in% "Index3"] = "OSVZ"
table(filt_df_v3$Clone_Index_filt)

colnames(cortex_midgestation@meta.data)
cortex_midgestation@meta.data = cortex_midgestation@meta.data[,c("orig.ident", "orig.ident.merge", "nCount_RNA", "nFeature_RNA", "percent.mt", "seurat_clusters", "sample_age", "age_pseudo", "Clone_IDs_filt", "Clone_size_filt", "Clone_Index_filt", "multi_clone", "broad_celltype", "subclust_idents_upset_definitive_final")]
colnames(cortex_midgestation@meta.data)
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "orig.ident.merge"] = "sample_ID"
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "age_pseudo"] = "age_binned"
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "Clone_IDs_filt"] = "Clone_IDs"
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "Clone_Index_filt"] = "GZ_origin"
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "Clone_size_filt"] = "Clone_size"
colnames(cortex_midgestation@meta.data)[colnames(cortex_midgestation@meta.data) %in% "subclust_idents_upset_definitive_final"] = "subcluster_identity"

colnames(cortex_midgestation@meta.data)

saveRDS(cortex_midgestation, file="/media/chang/HDD-8/mgkeefe/cortex_midgestation_v1.RDS")
#save.image("/media/chang/HDD-8/mgkeefe/240123_local_sticr_synthesis.RData")

## testing the RDS:
readRDS("/media/chang/HDD-8/mgkeefe/cortex_midgestation_v1.RDS")
cortex_midgestation = readRDS("/media/chang/HDD-8/mgkeefe/cortex_midgestation_v1.RDS")

## 1/24


## making final venn diagrams for supplement
setwd("/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240123_figs")
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final = factor(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final, levels=c(
  "RG",
  "Astrocytes",
  "OPCs",
  "Oligodendrocytes",
  "EX_IPCs",
  "ENs",
  "IN_IPCs",
  "IN_local",
  "IN_CGE",
  "IN_OB",
  "IN_MGE"
))
## code here works to make a nice plot, but need to change it to get labels
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% "<=GW20"), aes(x=age_pseudo, fill=subclust_idents_upset_definitive_final, color="white")) +
  geom_bar(position="fill") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  ggtitle("<=GW20") +
  labs(fill="Cell type") +
  ggtitle("Cluster composition pre-midgestation") +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))

pie_chart_age = "<=GW20"
pie_chart_df = data.frame((table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final))/sum(table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final)))
pie_chart_df$Freq = round((pie_chart_df$Freq*100), 1)
ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", pie_chart_age)) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", pie_chart_age)) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
ggsave('pie_chart_<=GW20.svg', image, width=10, height=8)

pie_chart_age = ">GW20"
pie_chart_df = data.frame((table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final))/sum(table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final)))
pie_chart_df$Freq = round((pie_chart_df$Freq*100), 1)
ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", pie_chart_age)) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", pie_chart_age)) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
ggsave('pie_chart_>GW20.svg', image, width=10, height=8)

pie_chart_age = c("<=GW20", ">GW20")
pie_chart_df = data.frame((table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final))/sum(table(subset(ls_synthesis_v3_harmony_v4@meta.data, age_pseudo %in% pie_chart_age)$subclust_idents_upset_definitive_final)))
pie_chart_df$Freq = round((pie_chart_df$Freq*100), 1)
ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", "all ages")) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
image = ggplot(pie_chart_df, aes(x=0, y=Freq, fill=Var1, color="white", label=paste(Freq, "%", sep=""))) +
  geom_bar(stat="identity", position="fill") +
  geom_text_repel(size=6, position = position_fill(vjust=0.5), color="black") +
  scale_color_manual(values = "white") +
  coord_polar("y", start=0) +
  ggtitle(">GW20") +
  labs(fill="Cell type") +
  ggtitle(paste("Cluster composition", "all ages")) +
  theme_bw() + 
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.ticks = element_blank(), axis.line = element_blank(), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.title.x = element_blank(), axis.title.y = element_blank(),
        plot.title = element_text(size=30, hjust=0.5))
ggsave('pie_chart_all_ages.svg', image, width=10, height=8)

ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=subclust_idents_upset_definitive_final, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position="fill") +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  labs(fill="Sample age") +
  ylab('Ratio of cluster') +
  xlab('Cell type') +
  ggtitle('Cluster composition by sample age') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=subclust_idents_upset_definitive_final, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position="fill") +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  labs(fill="Sample age") +
  ylab('Ratio of cluster') +
  xlab('Cell type') +
  ggtitle('Cluster composition by sample age') +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave('stacked_barchart_cluster_compositon_by_age_all_samples.svg', image, width=10, height=8)

ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=1, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position = "fill", width = 1) +
  xlim(0,11) +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  theme_bw()
image = ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=1, fill=factor(age_pseudo, levels=c(">GW20", "<=GW20")))) +
  geom_bar(position = "fill", width = 1) +
  xlim(0,11) +
  scale_fill_manual(values=c("#9c9391", "#d9d3d3")) +
  theme_bw()
ggsave('stacked_barchart_cluster_compositon_by_age_single_bar.svg', image, width=10, height=8)

ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=orig.ident.merge, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  labs(fill="Cell type") +
  ylab("Proportion of cluster") +
  xlab("Sample") +
  ggtitle("Cluster composition by sample") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(ls_synthesis_v3_harmony_v4@meta.data, aes(x=orig.ident.merge, fill=subclust_idents_upset_definitive_final)) +
  geom_bar(position="fill") +
  labs(fill="Cell type") +
  ylab("Proportion of cluster") +
  xlab("Sample") +
  ggtitle("Cluster composition by sample") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=15, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave('stacked_barchart_cluster_compositon_by_sample_all_samples.svg', image, width=10, height=8)


## remaking small single-gene umaps for the whole dataset
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_HES1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_HES1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_EOMES.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_EOMES.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MKI67.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_MKI67.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SPARCL1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_SPARCL1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SLC17A7.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_SLC17A7.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_GAD2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_GAD2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_OLIG2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_OLIG2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MBP.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_MBP.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_LHX6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_LHX6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SPARCL1_order.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_synthesis_v4_umap_SPARCL1_order.png", plot=image, width=4, height=3.2)

DimPlot(ls_synthesis_v3_harmony_v4, pt.size=0.5) +
  scale_fill_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))

## revisiting interneuron markers for supp

colnames(ls_in_v2@meta.data)
ls_in_v2 = SetIdent(ls_in_v2, value = "subclust_idents_paper")
ls_in_v2.markers <- FindAllMarkers(ls_in_v2, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_in_v2.markers)
ls_in_v2_clustermarkers_unfilt = as.data.frame(ls_in_v2.markers %>% group_by(cluster))
write.csv(ls_in_v2_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_v2_clustermarkers_v1_unfilt.csv',row.names=F)
ls_in_v2.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_in_v2_clustermarkers = as.data.frame(ls_in_v2.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_in_v2_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_v2_clustermarkers_v1.csv',row.names=F)

ls_in_v2_clustermarkers_top10_clustermarkers = as.data.frame(ls_in_v2.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC))

random_cell_list_ins = c()
for(sc in unique(ls_in_v2@meta.data$subclust_idents_paper)){
  cells_in_sc = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% sc]
  random_cell_list_ins = c(random_cell_list_ins, sample(cells_in_sc, size = min(length(cells_in_sc),1000), replace = F))
}
DoHeatmap(ls_in_v2, features=ls_in_v2_clustermarkers_top10_clustermarkers$gene)
image = DoHeatmap(ls_in_v2, features=ls_in_v2_clustermarkers_top10_clustermarkers$gene)
ggsave('ls_in_v2_clusters_heatmap_top10_genes_all_cells.svg', image, width=8, height=12)
ggsave('ls_in_v2_clusters_heatmap_top10_genes_all_cells.png', image, width=8, height=12)

DoHeatmap(ls_in_v2, features=ls_in_v2_clustermarkers_top10_clustermarkers$gene, cells = random_cell_list_ins)
image = DoHeatmap(ls_in_v2, features=ls_in_v2_clustermarkers_top10_clustermarkers$gene, cells = random_cell_list_ins)
ggsave('ls_in_v2_clusters_heatmap_top10_genes_1000cells.svg', image, width=8, height=12)
ggsave('ls_in_v2_clusters_heatmap_top10_genes_1000cells.png', image, width=8, height=12)

FeaturePlot(ls_in_v2, features=c("PAX6"), order=T)
FeaturePlot(ls_in_v2, features=c("SCGN"), order=T)

# what about looking at seurat clusters instead of assigned clusters?
ls_in_v2 = SetIdent(ls_in_v2, value="seurat_clusters")
ls_in_v2.seurat_cluster.markers <- FindAllMarkers(ls_in_v2, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(ls_in_v2.seurat_cluster.markers)
ls_in_v2_seurat_clustermarkers_unfilt = as.data.frame(ls_in_v2.seurat_cluster.markers %>% group_by(cluster))
write.csv(ls_in_v2_seurat_clustermarkers_unfilt,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_v2_clustermarkers_seurat_clusters_v1_unfilt.csv',row.names=F)
ls_in_v2.seurat_cluster.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC)
ls_in_v2_seurat_clustermarkers = as.data.frame(ls_in_v2.seurat_cluster.markers %>% group_by(cluster) %>% top_n(n = 20, wt = avg_log2FC))
write.csv(ls_in_v2_seurat_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_in_v2_clustermarkers_seurat_clusters_v1.csv',row.names=F)

ls_in_v2_seurat_clustermarkers_top10_clustermarkers = as.data.frame(ls_in_v2.seurat_cluster.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC))
DoHeatmap(ls_in_v2, features=ls_in_v2_seurat_clustermarkers_top10_clustermarkers$gene)
image = DoHeatmap(ls_in_v2, features=ls_in_v2_seurat_clustermarkers_top10_clustermarkers$gene)
ggsave("heatmap_ls_in_v2_by_seurat_cluster.png", width=8, height=12)
ggsave("heatmap_ls_in_v2_by_seurat_cluster.svg", width=8, height=12)

DimPlot(ls_in_v2) +
  #scale_colour_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
image = DimPlot(ls_in_v2) +
  #scale_colour_brewer(palette="Set1") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=20))
ggsave("ls_in_v2_umap_by_seurat_cluster.png", width=10, height=8)
ggsave("ls_in_v2_umap_by_seurat_cluster.svg", width=10, height=8)


dorsal_in_shared_clones_filt
ggplot(subset(ls_in_v2@meta.data, Clone_IDs_filt %in% dorsal_in_shared_clones_filt), aes(x=seurat_clusters)) +
  geom_bar() +
  scale_x_discrete(drop=F) +
  ylab("Number of cells") +
  xlab("Subcluster") +
  ggtitle("Cells with shared dorsal barcodes") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(ls_in_v2@meta.data, Clone_IDs_filt %in% dorsal_in_shared_clones_filt), aes(x=seurat_clusters)) +
  geom_bar() +
  scale_x_discrete(drop=F) +
  ylab("Number of cells") +
  xlab("Subcluster") +
  ggtitle("Cells with shared dorsal barcodes") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("ls_in_v2_barchart_cells_with_dorsal_barcodes.svg", image, width=10, height=8)

ggplot(subset(ls_in_v2@meta.data, Clone_IDs_filt %in% filt_df_v3_multicell$Clone_IDs), aes(x=seurat_clusters)) +
  geom_bar()+
  scale_x_discrete(drop=F) +
  ylab("Number of cells") +
  xlab("Subcluster") +
  ggtitle("Cells in multicellular clones") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(ls_in_v2@meta.data, Clone_IDs_filt %in% filt_df_v3_multicell$Clone_IDs), aes(x=seurat_clusters)) +
  geom_bar()+
  scale_x_discrete(drop=F) +
  ylab("Number of cells") +
  xlab("Subcluster") +
  ggtitle("Cells in multicellular clones") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("ls_in_v2_barchart_cells_in_multicellular_clones.svg", image, width=10, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5, family="Helvetica"))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5, family="Helvetica"))
ggsave(file="ls_in_v2_umap_dorsal_in_shared_clones_single_sample_clones_removed.png", plot=image, width=10, height=8)
ggsave(file="ls_in_v2_umap_dorsal_in_shared_clones_single_sample_clones_removed.svg", plot=image, width=10, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones_filt], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5, family="Helvetica"))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones_filt], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5, family="Helvetica"))
ggsave(file="ls_in_v2_umap_multicellular_clones_single_sample_clones_removed.png", plot=image, width=10, height=8)
ggsave(file="ls_in_v2_umap_multicellular_clones_single_sample_clones_removed.svg", plot=image, width=10, height=8)

ls_in_v2 = SetIdent(ls_in_v2, value="seurat_clusters")
ls_in_v2@active.ident = factor(ls_in_v2@active.ident, levels=c(5,3,2,4,9,7,0,8,1,6))
head(ls_in_v2@active.ident)
table(ls_in_v2@active.ident)
length(ls_in_v2@active.ident[ls_in_v2@active.ident == 8])
ls_in_v2@active.ident[ls_in_v2@active.ident == 8] = 10
length(c(5,3,2,4,9,7,0,8,1,6))
DotPlot(ls_in_v2, features=c("PBX3", "TSHZ1"))
DotPlot(ls_in_v2, features=c("NR2F2", "LHX6", "COL1A2", "LGALS1", "NPY",
                             "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6",
                             "PROX1","MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1"))

DotPlot(subset(ls_in_v2, seurat_clusters %in% c(2,3,4,5,7,9)), features=c("COL1A2", "LGALS1", "NPY",
                                                                          "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6", "PROX1",
                                                                          "MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1")) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = DotPlot(subset(ls_in_v2, seurat_clusters %in% c(2,3,4,5,7,9)), features=c("COL1A2", "LGALS1", "NPY",
                                                                                  "MEG3", "KLHL35", "NR2F1", "NFIX", "SOX6", "PROX1",
                                                                                  "MEIS2", "RUNX1T1", "ETV1", "PBX3", "TSHZ1")) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("ls_in_v2_dotplot_local_clusters_only.svg", image, width=10, height=8)

## trying to make a Sankey for barcode recovery
library(ggsankey)
sankey_barcode_recovery_metadata = cortex_midgestation@meta.data
sankey_barcode_recovery_metadata$cell = "Cell"
sankey_barcode_recovery_metadata$barcode_recovered = NA
sankey_barcode_recovery_metadata$barcode_recovered[sankey_barcode_recovery_metadata$multi_clone %in% c("multi_cell_clone", "single_cell_clone")] = "Barcode recovered"
sankey_barcode_recovery_metadata$barcode_recovered[!(sankey_barcode_recovery_metadata$multi_clone %in% c("multi_cell_clone", "single_cell_clone"))] = "Barcode lost"
colnames(sankey_barcode_recovery_metadata)
sankey_barcode_recovery = sankey_barcode_recovery_metadata %>% make_long(cell, barcode_recovered, age_binned, multi_clone)
ggplot(sankey_barcode_recovery, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.title = element_blank(), 
        legend.position = "none", plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(sankey_barcode_recovery, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.title = element_blank(), 
        legend.position = "none", plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("sankey_barcode_recovery_by_cells.svg", image, width=10, height=8)

table(sankey_barcode_recovery_metadata$barcode_recovered)
#     Barcode lost Barcode recovered 
#     39299             58165
table(sankey_barcode_recovery_metadata$age_binned[sankey_barcode_recovery_metadata$barcode_recovered %in% "Barcode recovered"])
#<=GW20  >GW20 
#37794  20371
table(sankey_barcode_recovery_metadata$multi_clone[sankey_barcode_recovery_metadata$age_binned %in% "<=GW20" & sankey_barcode_recovery_metadata$barcode_recovered %in% "Barcode recovered"])
# multi_cell_clone single_cell_clone 
# 12475             25319
table(sankey_barcode_recovery_metadata$multi_clone[sankey_barcode_recovery_metadata$age_binned %in% ">GW20" & sankey_barcode_recovery_metadata$barcode_recovered %in% "Barcode recovered"])
# multi_cell_clone single_cell_clone 
# 6196             14175

sankey_barcode_recovery_metadata = cortex_midgestation@meta.data
head(sankey_barcode_recovery_metadata)
sankey_barcode_recovery_metadata = sankey_barcode_recovery_metadata[!(is.na(sankey_barcode_recovery_metadata$Clone_IDs)),]
dim(sankey_barcode_recovery_metadata)
#[1] 58165    14
dim(sankey_barcode_recovery_metadata[!duplicated(sankey_barcode_recovery_metadata$Clone_IDs),])
#[1] 45905    14
sankey_barcode_recovery_metadata = sankey_barcode_recovery_metadata[!duplicated(sankey_barcode_recovery_metadata$Clone_IDs),]
table(sankey_barcode_recovery_metadata$multi_clone)
# multi_cell_clone single_cell_clone 
# 6411             39494
sankey_barcode_recovery_metadata$clones = "Clone recovered"
sankey_barcode_recovery = sankey_barcode_recovery_metadata %>% make_long(clones, age_binned, multi_clone)
ggplot(sankey_barcode_recovery, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.title = element_blank(), 
        legend.position = "none", plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(sankey_barcode_recovery, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes') +
  theme(axis.text.y = element_blank(), axis.text.x = element_blank(), axis.title = element_blank(), 
        legend.position = "none", plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("sankey_barcode_recovery_by_clones.svg", image, width=10, height=8)

table(sankey_barcode_recovery_metadata$age_binned)
#<=GW20  >GW20 
#29552  16353
table(sankey_barcode_recovery_metadata$multi_clone[sankey_barcode_recovery_metadata$age_binned %in% ">GW20"])
# multi_cell_clone single_cell_clone 
# 2178             14175
table(sankey_barcode_recovery_metadata$multi_clone[sankey_barcode_recovery_metadata$age_binned %in% "<=GW20"])
# multi_cell_clone single_cell_clone 
# 4233             25319

dim(sankey_barcode_recovery_metadata)

## finalizing supp for RG subclustering
head(ls_prog_glia.markers)
head(ls_prog_glia_clustermarkers)
ls_prog_glia_top10_clustermarkers = as.data.frame(ls_prog_glia.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC))
DoHeatmap(ls_prog_glia, features=intersect(ls_prog_glia_top10_clustermarkers$gene, rownames(ls_prog_glia@assays$RNA@scale.data)))


FeaturePlot(ls_prog_glia, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_MKI67.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_MKI67.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("PCNA")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("PCNA")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_PCNA.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_PCNA.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("CENPE")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("CENPE")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_CENPE.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_CENPE.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_HES1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_HES1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("GFAP")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("GFAP")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_GFAP.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_GFAP.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("EMX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("EMX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_EMX2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_EMX2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("LHX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("LHX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_LHX6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_LHX6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("NXPH1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("NXPH1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_NXPH1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_NXPH1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("VIM")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("VIM")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_VIM.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_VIM.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("NES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("NES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_NES.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_NES.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("INPP1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("INPP1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_INPP1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_INPP1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("CRYAB")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("CRYAB")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_CRYAB.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_CRYAB.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("APOE")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("APOE")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_APOE.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_APOE.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("AQP4")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("AQP4")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_AQP4.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_AQP4.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("FOXG1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("FOXG1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_FOXG1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_FOXG1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("PAX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("PAX6")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_PAX6.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_PAX6.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_OLIG2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_OLIG2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("CD44")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("CD44")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_CD44.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_CD44.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_EOMES.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_EOMES.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("OTX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("OTX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_OTX2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_OTX2.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("GJA1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_prog_glia, features=c("GJA1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_glia_umap_GJA1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_glia_umap_GJA1.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_prog_glia, features=c("LIFR")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_EOMES.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_EOMES.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_OLIG2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_OLIG2.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("VIM")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("VIM")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_VIM.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_VIM.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("DLX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("DLX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_DLX2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_DLX2.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_HES1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_HES1.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_MKI67.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_MKI67.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("FOXG1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("FOXG1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_FOXG1.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_FOXG1.png", plot=image, width=4, height=3.2)

FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("EMX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(subset(ls_prog, seurat_clusters %in% c(5), invert=T), features=c("EMX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_prog_umap_EMX2.svg", plot=image, width=4, height=3.2)
ggsave(file="ls_prog_umap_EMX2.png", plot=image, width=4, height=3.2)



DotPlot(ls_prog_glia, features=c("PAX6", "FOXG1", "EMX2", 
                                 "OTX2", "LHX6",
                                 "VIM", "HES1", "AQP4", "GFAP", "GLI3", "NES", "HOPX",
                                 "CRYAB", "ANXA1",
                                 "INPP1", "PPM1K",
                                 "APOE", "CD44", "SPARCL1", "GJA1",
                                 "MKI67", "CENPE",
                                 "OLIG2", "PDGFRA", "EOMES")) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = DotPlot(ls_prog_glia, features=c("PAX6", "FOXG1", "EMX2", 
                                         "OTX2", "LHX6",
                                         "VIM", "HES1", "AQP4", "GFAP", "GLI3", "NES", "HOPX",
                                         "CRYAB", "ANXA1",
                                         "INPP1", "PPM1K",
                                         "APOE", "CD44", "SPARCL1", "GJA1",
                                         "MKI67", "CENPE",
                                         "OLIG2", "PDGFRA", "EOMES")) +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20, angle=30, hjust=1), axis.title = element_text(size=25), 
        legend.title = element_text(size=25), legend.text = element_text(size=20), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("ls_prog_glia_dotplot_by_subcluster.svg", image, width=12, height=8)

DimPlot(ls_prog, group.by="seurat_clusters", label=T)
FeaturePlot(ls_prog, features=c("nCount_RNA"), label=T, max.cutoff="q80")
FeaturePlot(ls_prog_glia, features=c("nCount_RNA"), label=T, max.cutoff="q80")


ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "seurat_clusters")
DimPlot(ls_synthesis_v3_harmony_v4, label=T)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("DLX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("DLX2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_DLX2.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_MKI67.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_GAD2.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_OLIG2.png", image, width=4, height=3.2)
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(ls_in_v2@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(ls_in_v2@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
ggsave("ls_full_umap_ls_in_v2_highlighted.png", image, width=4, height=3.2)



FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_EOMES.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_HES1.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_MKI67.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NPY")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NPY")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_NPY.png", image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_SLC17A7.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_SLC17A7_orderT.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SATB2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SATB2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_SATB2.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("TBR1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("TBR1")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_TBR1.png", image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NEUROD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NEUROD2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_NEUROD2.png", image, width=4, height=3.2)


image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NPY")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave("ls_full_umap_NPY.png", image, width=4, height=3.2)


DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(ls_prog@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(ls_prog@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
ggsave("ls_full_umap_ls_prog_highlighted.png", image, width=4, height=3.2)

DimPlot(ls_prog, cells.highlight = rownames(ls_prog@meta.data)[rownames(ls_prog@meta.data) %in% rownames(ls_prog_glia@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
image = DimPlot(ls_prog, cells.highlight = rownames(ls_prog@meta.data)[rownames(ls_prog@meta.data) %in% rownames(ls_prog_glia@meta.data)]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
ggsave("ls_prog_umap_ls_prog_glia_highlighted.png", image, width=4, height=3.2)

DimPlot(ls_prog_glia, pt.size=1.5) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30))
image = DimPlot(ls_prog_glia, pt.size=1.5) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30))
ggsave("ls_prog_glia_umap_seurat_clusters.png", image, width=10, height=8)
ggsave("ls_prog_glia_umap_seurat_clusters.svg", image, width=10, height=8)


colnames(ls_prog_glia@meta.data)

DimPlot(ls_prog_glia, pt.size=1.5, group.by="subclust_names_figure") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
image = DimPlot(ls_prog_glia, pt.size=1.5, group.by="subclust_names_figure") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
ggsave("ls_prog_glia_umap_grouped_subclusts.png", image, width=10, height=8)
ggsave("ls_prog_glia_umap_grouped_subclusts.svg", image, width=10, height=8)

DimPlot(ls_prog_glia, pt.size=1, group.by="orig.ident.merge") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
image = DimPlot(ls_prog_glia, pt.size=1, group.by="orig.ident.merge") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
ggsave("ls_prog_glia_umap_grouped_sample.png", image, width=10, height=8)
ggsave("ls_prog_glia_umap_grouped_sample.svg", image, width=10, height=8)

DimPlot(ls_prog_glia, pt.size=1, group.by="age_pseudo") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
image = DimPlot(ls_prog_glia, pt.size=1, group.by="age_pseudo") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.text = element_text(size=30), plot.title = element_blank())
ggsave("ls_prog_glia_umap_grouped_age_pseudo.png", image, width=10, height=8)
ggsave("ls_prog_glia_umap_grouped_age_pseudo.svg", image, width=10, height=8)

ggplot(subset(ls_prog_glia@meta.data, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to astrocyte clusters') +
  scale_fill_manual(values=c("#05c4c7", "#ff6666"))+
  ylab("Proportion of cells in cluster") +
  xlab("Astrocyte subcluster") +
  theme_bw() +
  labs(fill = "GZ") +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=15), legend.title = element_text(size=20),
        axis.text = element_text(size=15), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20))
image = ggplot(subset(ls_prog_glia@meta.data, seurat_clusters %in% c(3,6) & Clone_Index_filt %in% c("IndexE", "Index3")), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  ggtitle('GZ contribution to astrocyte clusters') +
  scale_fill_manual(values=c("#05c4c7", "#ff6666"))+
  ylab("Proportion of cells in cluster") +
  xlab("Astrocyte subcluster") +
  theme_bw() +
  labs(fill = "GZ") +
  theme(plot.title = element_text(hjust = 0.5, size=25), legend.text = element_text(size=20), legend.title = element_text(size=20),
        axis.text = element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=25))
ggsave("ls_prog_glia_astrocyte_subclusters_index_contribution_barchart.svg", width=6, height=8)



## 
dim(ls_in_v2@meta.data)
#[1] 12118    37
dim(ls_in_v2@meta.data[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local",])
#[1] 5393   37
head(ls_in_v2@assays$RNA@counts)
length(ls_in_v2@assays$RNA@counts["NR2F1",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]])
sum(ls_in_v2@assays$RNA@counts["NR2F1",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 3533
3533/5393
sum(ls_in_v2@assays$RNA@counts["PROX1",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 3533
2472/5393
sum(ls_in_v2@assays$RNA@counts["SCGN",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 3480
3480/5393
sum(ls_in_v2@assays$RNA@counts["CALB2",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 1576
1576/5393
sum(ls_in_v2@assays$RNA@counts["ADARB2",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 1220
1220/5393
sum(ls_in_v2@assays$RNA@counts["RELN",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 23
23/5393
sum(ls_in_v2@assays$RNA@counts["LHX6",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 50
50/5393
sum(ls_in_v2@assays$RNA@counts["MAF",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 284
284/5393
sum(ls_in_v2@assays$RNA@counts["SST",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 106
106/5393
sum(ls_in_v2@assays$RNA@counts["SOX6",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 3713
3713/5393
sum(ls_in_v2@assays$RNA@counts["PBX3",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 1103
1103/5393
sum(ls_in_v2@assays$RNA@counts["TSHZ1",colnames(ls_in_v2)[colnames(ls_in_v2) %in% rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% "IN_local"]]] > 0)
#[1] 743
743/5393


table(cortex_midgestation@meta.data$subcluster_identity)
#      Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#           1277            56582            11048             2080             4547             5341             2610 
#IN_OB Oligodendrocytes             OPCs               RG 
#  798              772             3621             8788

table(cortex_midgestation@meta.data$subcluster_identity[cortex_midgestation@meta.data$multi_clone %in% 'multi_cell_clone'])
#Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#167             2848             5234                7             2886             1980               25 
#IN_OB Oligodendrocytes             OPCs               RG 
#14              173             1900             3437
table(cortex_midgestation@meta.data$subcluster_identity[cortex_midgestation@meta.data$multi_clone %in% 'multi_cell_clone'])/table(cortex_midgestation@meta.data$subcluster_identity)
#      Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#     0.130775255      0.050334028      0.473750905      0.003365385      0.634704201      0.370717094      0.009578544 
#       IN_OB Oligodendrocytes             OPCs               RG 
#0.017543860      0.224093264      0.524716929      0.391101502

sum(compiled_clust_df_v18$RG >=1 & compiled_clust_df_v18$IN_OB >=1)
rg_containing_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$RG >=1]
table(ls_in_v2@meta.data$seurat_clusters[ls_in_v2@meta.data$Clone_IDs_filt %in% rg_containing_clones])
#  0   1   2   3   4   5   6   7   8   9 
#  1   1  80 177  23 301   2   2   0  30 

sum(compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local + compiled_clust_df_v18$IN_CGE + compiled_clust_df_v18$IN_MGE + compiled_clust_df_v18$IN_OB >=1)
#[1] 2079
in_containing_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local + compiled_clust_df_v18$IN_CGE + compiled_clust_df_v18$IN_MGE + compiled_clust_df_v18$IN_OB >=1]

sum(compiled_clust_df_v18$IN_IPCs >=1)
#[1] 1535
sum(compiled_clust_df_v18$IN_local >=1)
#[1] 1120
sum(compiled_clust_df_v18$IN_CGE >=1)
#[1] 6
sum(compiled_clust_df_v18$IN_MGE >=1)
#[1] 18
sum(compiled_clust_df_v18$IN_OB >=1)
#[1] 12

DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones])
table(cortex_midgestation@meta.data$subcluster_identity[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones])
#Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#       16              149              499                7             2886             1980               25 
#IN_OB Oligodendrocytes             OPCs               RG 
#   14               32              862              962
sum(table(cortex_midgestation@meta.data$subcluster_identity[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones]))
#[1] 7432
7 + 2886 + 1980 + 25 + 14
#[1] 4912
table(cortex_midgestation@meta.data$subcluster_identity[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones])/4912
#IN_CGE          IN_IPCs         IN_local           IN_MGE      IN_OB
#0.001425081      0.587540717      0.403094463      0.005089577 0.002850163


## 1/27 figure work
## representative clones for Fig 1a

#multicellular clone:
compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$EX_IPCs >=1 & compiled_clust_df_v18$RG >=1 & compiled_clust_df_v18$IN_IPCs >= 1]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS36_Clone_28"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS36_Clone_28"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
ggsave('ls_umap_example_multipotent_clone.png', image, width=4, height=3.2)
#two-cell EN clone:
compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$ENs == 2]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS35_Clone_2285"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS35_Clone_2285"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
ggsave('ls_umap_example_two_en_clone.png', image, width=4, height=3.2)
#one-cell EN clone:
ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_definitive_final %in% "ENs" & ls_synthesis_v3_harmony_v4@meta.data$Clone_size == 1]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS3_Clone_3446"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% "LS3_Clone_3446"], sizes.highlight = 4, cols.highlight = "darkgreen") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_blank())
ggsave('ls_umap_example_one_en_clone.png', image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_OLIG2_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MBP_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image =FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_HES1_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MKI67_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SLC17A7_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_EOMES_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_GAD2_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SPARCL1_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q80') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_LHX6_large_dots_q80_ordered.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_OLIG2_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MBP_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image =FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_HES1_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MKI67_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SLC17A7_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_EOMES_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_GAD2_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SPARCL1_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q50') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_LHX6_large_dots_q50_ordered.png", plot=image, width=4, height=3.2)

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_OLIG2_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("BCAS1"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MBP"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MBP_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image =FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_HES1_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("MKI67"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_MKI67_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SLC17A7"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SLC17A7_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("EOMES"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_EOMES_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("GAD2"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_GAD2_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_SPARCL1_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("LHX6"), order=T, pt.size=1, min.cutoff='q70') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_LHX6_large_dots_q70_ordered.png", plot=image, width=4, height=3.2)


image = FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2")) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"))
ggsave(file="ls_synthesis_v4_umap_OLIG2_large_dots.png", plot=image, width=4, height=3.2)





ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "subclust_idents_upset_definitive_final")
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), pt.size=0) +
  theme(axis.text = element_text(size=20, family="Helvetica"), axis.title = element_text(size=30, family="Helvetica"),
        plot.title = element_text(size=30, family="Helvetica"), legend.text = element_text(size=20, family="Helvetica"))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("OLIG2"), pt.size=0) +
  theme(axis.text = element_text(size=20, family="Helvetica"), axis.title = element_text(size=30, family="Helvetica"),
        plot.title = element_text(size=30, family="Helvetica"), legend.text = element_text(size=20, family="Helvetica"))
ggsave(file="ls_synthesis_violin_plot_OLIG2", plot=image, width=4, height=3.2)
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), pt.size=0) +
  theme(axis.text = element_text(size=20, family="Helvetica"), axis.title = element_text(size=30, family="Helvetica"),
        plot.title = element_text(size=30, family="Helvetica"), legend.text = element_text(size=20, family="Helvetica"))
image = VlnPlot(ls_synthesis_v3_harmony_v4, features=c("HES1"), pt.size=0) +
  theme(axis.text = element_text(size=20, family="Helvetica"), axis.title = element_text(size=30, family="Helvetica"),
        plot.title = element_text(size=30, family="Helvetica"), legend.text = element_text(size=20, family="Helvetica"))
ggsave(file="ls_synthesis_violin_plot_HES1", plot=image, width=4, height=3.2)



ls_synthesis_v3_harmony_v4 = SetIdent(ls_synthesis_v3_harmony_v4, value = "subclust_idents_upset_definitive_final")
VlnPlot(ls_synthesis_v3_harmony_v4, features=c("SPARCL1"), pt.size=0)


DimPlot(ls_synthesis_v3_harmony_v4, group.by="subclust_idents_upset_definitive_final")

sum(compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age>20] >=1)





ls_synthesis_v3_harmony_v3.markers


packageVersion("Seurat")
#[1] 4.3.0.9002
packageVersion("harmony")
#[1] 0.1.1

table(cortex_midgestation@meta.data$sample_ID)
#    GW16     GW18     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#  12790      952     9692    10664    29578      395     8604    18780     6009
table(cortex_midgestation@meta.data$sample_ID[!is.na(cortex_midgestation@meta.data$Clone_IDs)])
# GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#8972     3078     6202    19542      228     5039    11449     3655 
table(cortex_midgestation@meta.data$sample_ID[cortex_midgestation@meta.data$GZ_origin %in% "VZ"])
#    GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#6742     2197     2539    12938       86     2982     2553      358 
table(cortex_midgestation@meta.data$sample_ID[cortex_midgestation@meta.data$GZ_origin %in% "VZ" & cortex_midgestation@meta.data$multi_clone %in% "multi_cell_clone"])
#    GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#.  2212      407     1134     5958       11     1282      416      142
table(cortex_midgestation@meta.data$sample_ID[cortex_midgestation@meta.data$GZ_origin %in% "OSVZ"])
#    GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#    2230      881     3663     6604      142     2057     8896     3297 
table(cortex_midgestation@meta.data$sample_ID[cortex_midgestation@meta.data$GZ_origin %in% "OSVZ" & cortex_midgestation@meta.data$multi_clone %in% "multi_cell_clone"])
#    GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
# 230      197      910     1427       16      817     2414     1098 



FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("TBR1", "SATB2"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("SATB2", "TBR1"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(ls_ex, features=c("SATB2", "TBR1"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(ls_ex, features=c("SATB2", "TLE4"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(ls_ex, features=c("TBR1", "TLE4"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("TBR1", "TLE4"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)

DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% clones_vz_old])

FeaturePlot(subset(cortex_midgestation, Clone_IDs %in% clones_vz_old), features=c("TBR1", "TLE4"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(subset(cortex_midgestation, Clone_IDs %in% clones_vz_old), features=c("TBR1", "SATB2"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(subset(cortex_midgestation, Clone_IDs %in% clones_osvz_old), features=c("TBR1", "SATB2"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(subset(cortex_midgestation, GZ_origin %in% "VZ"), features=c("TBR1", "SATB2"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)
FeaturePlot(subset(cortex_midgestation, GZ_origin %in% "OSVZ"), features=c("TBR1", "SATB2"), blend = T, blend.threshold = 0.1, order=T, pt.size=2)


VlnPlot(cortex_midgestation, features=c("TBR1", "SATB2"), split.by="GZ_origin", pt.size=0)


table(ls_in_v2@meta.data$subclust_idents_paper)
# IN IPCs   IN_CGE IN_local   IN_MGE    IN_OB 
#1192     2369     5393     2621      543
table(ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$age_pseudo %in% "<=GW20"])
# IN IPCs   IN_CGE IN_local   IN_MGE    IN_OB 
# 440      532     1310      585      121
table(ls_in_v2@meta.data$subclust_idents_paper[ls_in_v2@meta.data$age_pseudo %in% ">GW20"])
# IN IPCs   IN_CGE IN_local   IN_MGE    IN_OB 
#     752     1837     4083     2036      422 
table(ls_in_v2@meta.data$age_pseudo[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones_filt])
#<=GW20 >GW20 
#299      38 
299/(440+1310)
#[1] 0.1708571
38/(752+4083)
#[1] 0.007859359

table(ls_in_v2@meta.data$age_pseudo[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt])
#<=GW20  >GW20 
#   501    235

501/(440+1310)
#[1] 0.2862857
235/(752+4083)
#[1] 0.04860393 



116/1192
#[1] 0.09731544
221/5393



DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(4)), group.by="Clone_Index_filt")

ggplot(ls_prog_glia@meta.data, aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill")


ggplot(subset(ls_prog_glia@meta.data, Clone_size_corrected >=2), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill")


table(ls_prog_glia@meta.data$seurat_clusters[ls_prog_glia@meta.data$Clone_Index_filt %in% "IndexE"])
#0   1   2   3   4   5   6   7   8   9  10 
#553 322 155  59 271 140 159 163 102  81  32
table(ls_prog_glia@meta.data$seurat_clusters[ls_prog_glia@meta.data$Clone_Index_filt %in% "Index3"])
#  0   1   2   3   4   5   6   7   8   9  10 
#360 220 593 488 116 245  92  66  86  83  15 

(553+322+163)/(553+322+163+360+220+66)
#[1] 0.6163895

(553+322+163)/(553+322+163+360+220+66)
#[1] 0.6163895

271/(271+116)
#[1] 0.7002584

table(ls_in_v2@meta.data$subclust_idents_paper[rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell)])
#IN IPCs   IN_CGE IN_local   IN_MGE    IN_OB 
#   715       10     2043       24       10 
(715+2043)/(715+10+2043+24+10)
#0.9842969

DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
image = DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% in_containing_clones]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
ggsave('umap_all_in_containing_clones.png', image, width=10, height=8)

DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% dorsal_in_shared_clones_filt]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
image = DimPlot(cortex_midgestation, cells.highlight = rownames(cortex_midgestation@meta.data)[cortex_midgestation@meta.data$Clone_IDs %in% dorsal_in_shared_clones_filt]) +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30, family="Helvetica"), legend.position = "none")
ggsave('umap_dorsal_in_shared_clones.png', image, width=10, height=8)


dim(ls_synthesis_v3_harmony_v3_clustermarkers_unfilt)
ls_synthesis_v3_harmony_v3_100_clustermarkers = as.data.frame(ls_synthesis_v3_harmony_v3_clustermarkers %>% group_by(cluster) %>% top_n(n = 100, wt = avg_log2FC))
write.csv(ls_synthesis_v3_harmony_v3_100_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v3_100_clustermarkers_seurat_clusters.csv',row.names=F)

dim(ls_synthesis_v3_harmony_v4.markers)
ls_synthesis_v3_harmony_v4_100_clustermarkers = as.data.frame(ls_synthesis_v3_harmony_v4.markers %>% group_by(cluster) %>% top_n(n = 100, wt = avg_log2FC))
write.csv(ls_synthesis_v3_harmony_v4_100_clustermarkers,'/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_synthesis_v3_harmony_v4_100_clustermarkers_final_celltypes.csv',row.names=F)


length(ex_in_shared_clones_filt)



######## ------------------------------------------------------------------------ ########
######## code for upset plots and multicellular clone analysis ########
######## ------------------------------------------------------------------------ ########



ls_meta_orig = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_v3_metadata.csv', row.names = 1)
compiled_clust_df_idents_orig = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/compiled_clust_df_idents.csv', row.names=1)

## some notes:
## in the original dataset, there are 26,548 cells that have 1 barcode UMI
## when filtering to barcodes for clones with size >=3, there are 298 cells that have 1 barcode UMI


multicell_metadata_orig = ls_meta_orig[ls_meta_orig$Clone_size_corrected>=3,]
multicell_metadata_orig = multicell_metadata_orig[!is.na(multicell_metadata_orig$Clone_IDs),]
head(multicell_metadata_orig)
dim(multicell_metadata_orig)
#[1] 14030    34
length(unique(multicell_metadata_orig$Clone_IDs_filt))
#[1] 3258

## using this
filt_df = ls_meta_orig
head(filt_df)
filt_df = filt_df[filt_df$Barcode_UMI_Count > 1 & ls_meta_orig$read_counts > 10,]
filt_df = filt_df[!is.na(filt_df$Barcode_UMI_Count),]
dim(filt_df)
#[1] 66219    35

unique_clones_2umi = unique(filt_df$Clone_IDs)
length(unique_clones_2umi)

unique_clones_2umi[48]
filt_df[filt_df$Clone_IDs %in% mc,]
dim(filt_df[filt_df$Clone_IDs %in% mc,])[1]

dim(filt_df)
filt_df = filt_df[,1:34] #drop the last column
head(filt_df)
filt_df$Clone_size_corr_2umi = NA

for (mc in unique_clones_2umi) {
  Clone_size_corrected_2umi = dim(filt_df[filt_df$Clone_IDs %in% mc,])[1]
  filt_df[filt_df$Clone_IDs %in% mc,'Clone_size_corr_2umi'] = Clone_size_corrected_2umi
}
table(filt_df$Clone_size_corrected)
#1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    20    21 
#43985  8505  4695  2805  1814  1316   745   476   444   257   306   261    78    69    75    78   119    36    19    40    21 
#24    25    26 
#24    25    26
table(filt_df$Clone_size_corr_2umi)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    20    21 
#44492  8254  4605  2740  1795  1272   749   464   441   250   319   228    91    70    75    64   119    36    19    40    21 
#24    25    26 
#24    25    26

unique_mc_clones_3cell = unique(filt_df$Clone_IDs[filt_df$Clone_size_corr_2umi >= 3])
length(unique_mc_clones_3cell)
#[1] 3115
unique_mc_clones_2cell = unique(filt_df$Clone_IDs[filt_df$Clone_size_corr_2umi >= 2])
length(unique_mc_clones_2cell)
#[1] 7242

unique_mc_clones_3cell[592]
clone = unique_mc_clones_3cell[592]
filt_df[filt_df$Clone_IDs %in% clone,]
filt_df$Clone_size_corrected[filt_df$Clone_IDs %in% clone][1]
table(filt_df$broad_celltype[filt_df$Clone_IDs %in% clone])

table(filt_df$broad_celltype)
#   EX  GLIA    IN  OLIG 
#42710  9368 10620  2820
table(filt_df$subclust_idents_paper)
#      Astrocytes         Dividing               EX          EX IPCs          IN IPCs           IN_CGE         IN_local 
#           885             7459            33257             7728              956             1842             3996 
#IN_MGE           IN_UNK Oligodendrocytes             OPCs      Radial glia              unk 
#1584              667              504             1842             4156             1343 
table(filt_df[filt_df$subclust_idents_paper %in% 'unk',]$broad_celltype)
#GLIA 
#1309 

table(filt_df[filt_df$subclust_idents_paper %in% 'unk',]$Clone_size_corr_2umi)
#  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  24 
#681 207 145  91  52  35  19  23  18   5  19  12   2   6   7   7   3   1   2   4   2   2 
table(filt_df[filt_df$subclust_idents_paper %in% 'IN_UNK',]$Clone_size_corr_2umi)
#  1   2   3   4   5   6   7   8   9  10  11  12  13  16  17  18 
#489  63  32  16  15  10  15   3   7   4   1   3   1   2   5   1 


## going to remove IN_UNK and and unk from the dataset, they're likely to be excitatory doublets based on nCount_RNA and gene expression

dim(filt_df[!(filt_df$subclust_idents_paper %in% c('unk', 'IN_unk')),])
#[1] 64876    35
filt_df = filt_df[!(filt_df$subclust_idents_paper %in% c('unk', 'IN_UNK')),]

for (mc in unique_clones_2umi) {
  Clone_size_corrected_2umi = dim(filt_df[filt_df$Clone_IDs %in% mc,])[1]
  filt_df[filt_df$Clone_IDs %in% mc,'Clone_size_corr_2umi'] = Clone_size_corrected_2umi
}
table(filt_df$Clone_size_corrected)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    20    21 
#42824  8234  4516  2702  1739  1270   711   449   419   250   288   242    75    63    68    69   111    34    17    36    19 
#24    25    26 
#22    25    26
table(filt_df$Clone_size_corr_2umi)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    22    25 
#43583  8054  4365  2616  1700  1182   714   448   360   270   286   108   104    84    75    64    68    36    19    22    25 
#26 
#26 

unique_mc_clones_3cell = unique(filt_df$Clone_IDs[filt_df$Clone_size_corr_2umi >= 3])
length(unique_mc_clones_3cell)
#[1] 2939
unique_mc_clones_2cell = unique(filt_df$Clone_IDs[filt_df$Clone_size_corr_2umi >= 2])
length(unique_mc_clones_2cell)
#[1] 7029

unique_mc_clones_3cell[592]
clone = unique_mc_clones_3cell[592]
filt_df[filt_df$Clone_IDs %in% clone,]
filt_df$Clone_size_corrected[filt_df$Clone_IDs %in% clone][1]
table(filt_df$broad_celltype[filt_df$Clone_IDs %in% clone])

table(filt_df$broad_celltype)
#   EX  GLIA    IN  OLIG 
#42710  8059 10620  2820
table(filt_df$subclust_idents_paper)
#Astrocytes         Dividing               EX          EX IPCs          IN IPCs           IN_CGE         IN_local 
#       885             7459            33257             7728              956             1842             3996 
#IN_MGE Oligodendrocytes             OPCs      Radial glia 
#  1584              504             1842             4156 
table(filt_df$subclust_idents_v2)
#ASTRO                      EX                EX_early      G2M_phase_div_glia                  IN_CGE 
# 885                   27395                    5862                    1935                    1842 
#IN_local                  IN_MGE                  IPC_EX              IPC_EX_DIV                  IPC_IN 
#    3996                    1584                    7728                    1725                     956 
#IPC_IN_DIV                    OLIG                     OPC                 OPC_DIV                      RG 
#     2242                     504                    1842                     474                    1308 
#RG (putative oRG)       RG (putative tRG)        S_phase_div_glia      Transitioning IPCs Transitioning OPCs/IPCs 
#             613                     580                    1083                     488                    1167 

unique(filt_df$subclust_idents_v2)
[1]                  ""      ""                     
[5] ""              ""                  ""                ""       
[9] ""                  ""                                       ""                
[13] ""                   ""                                    ""      
[17] ""                      ""                         
#v1 of broad trajectories: EX, IN, PROG, ASTRO, OLIG
filt_df$broad_trajectories = NA
filt_df$broad_trajectories[filt_df$subclust_idents_v2 %in% c('EX', 'EX_early', "IPC_EX_DIV", "IPC_EX")] = "EX"
filt_df$broad_trajectories[filt_df$subclust_idents_v2 %in% c("IN_CGE", 'IPC_IN_DIV', "IPC_IN", "IN_MGE", "IN_local")] = "IN"
filt_df$broad_trajectories[filt_df$subclust_idents_v2 %in% c("RG (putative oRG)", 'RG (putative tRG)', "G2M_phase_div_glia", "S_phase_div_glia", "Transitioning OPCs/IPCs", "Transitioning IPCs")] = "PROG"
filt_df$broad_trajectories[filt_df$subclust_idents_v2 %in% c("OPC_DIV", 'OPC', "OLIG")] = "OLIG"
filt_df$broad_trajectories[filt_df$subclust_idents_v2 %in% c("ASTRO", 'RG')] = "ASTRO"
table(filt_df$broad_trajectories)
#ASTRO    EX    IN  OLIG  PROG 
#2193 42710 10620  2820  5866
table(filt_df[filt_df$Clone_size_corr_2umi > 2,'broad_trajectories'])
#ASTRO    EX    IN  OLIG  PROG 
#168  5056  3496  1343  2509 
table(filt_df[filt_df$Clone_size_corr_2umi > 1,'broad_trajectories'])
#ASTRO    EX    IN  OLIG  PROG 
#  576  9360  5002  1926  3762 

#v2 of broad trajectories: EX, IN, PROG, GLIA(astro+committed oligo)
filt_df$broad_trajectories_v2 = NA
filt_df$broad_trajectories_v2[filt_df$subclust_idents_v2 %in% c('EX', 'EX_early', "IPC_EX_DIV", "IPC_EX")] = "EX"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_v2 %in% c("IN_CGE", 'IPC_IN_DIV', "IPC_IN", "IN_MGE", "IN_local")] = "IN"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_v2 %in% c("RG (putative oRG)", 'RG (putative tRG)', "G2M_phase_div_glia", "S_phase_div_glia", "Transitioning OPCs/IPCs", "Transitioning IPCs")] = "PROG"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_v2 %in% c("OPC_DIV", 'OPC', "OLIG", "ASTRO", 'RG')] = "GLIA"
table(filt_df$broad_trajectories_v2)
#   EX  GLIA    IN  PROG 
#42710  5013 10620  5866 
table(filt_df[filt_df$Clone_size_corr_2umi > 2,'broad_trajectories_v2'])
#  EX GLIA   IN PROG 
#5056 1511 3496 2509 
table(filt_df[filt_df$Clone_size_corr_2umi > 1,'broad_trajectories_v2'])
#  EX GLIA   IN PROG 
#9360 2502 5002 3762

#going in to get final glial trajectories based on the glial subclustering analysis
ls_glia_v2_harmony_meta_orig = read.csv('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/ls_glia_v2_harmony_metadata.csv', row.names = 1)
filt_df$CBC = rownames(filt_df)
ls_glia_v2_harmony_meta_orig$CBC = rownames(ls_glia_v2_harmony_meta_orig)
head(filt_df)
head(ls_glia_v2_harmony_meta_orig)
head(left_join(filt_df, ls_glia_v2_harmony_meta_orig[,c("CBC", "glia_lineage_trajectory")]))
filt_df = left_join(filt_df, ls_glia_v2_harmony_meta_orig[,c("CBC", "glia_lineage_trajectory")])
sum(is.na(filt_df$glia_lineage_trajectory))
#[1] 56163
table(filt_df$glia_lineage_trajectory)

filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('OLIG')] = "OLIG"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('PROG')] = "PROG"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('ASTRO')] = "ASTRO"
table(filt_df$broad_trajectories)
#ASTRO    EX    IN  OLIG  PROG 
# 2193 43654 10620  4483  3259

filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('OLIG')] = "GLIA"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('PROG')] = "PROG"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('ASTRO')] = "GLIA"
table(filt_df$broad_trajectories_v2)
#   EX  GLIA    IN  PROG 
#43654  6676 10620  3259 


filt_df$Clone_size_corrected[filt_df$Clone_IDs %in% clone][1]
table(filt_df$broad_trajectories[filt_df$Clone_IDs %in% clone])
which(compiled_clust_df_v2$unique_mc_clones_3cell %in% clone)

## guide for future reference to understand nomenclature of compiled_clust_df variants:
## compiled_clust_df_v2 uses the 2UMI BC cutoff, has cutoff of 3 multicell clones, and uses broad_trajectories (ASTRO and OLIG are separate)

## compiled_clust_df_v2
compiled_clust_df_v2 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v2$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$IN = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$IN_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$PROG_pct = rep(0,length(unique_mc_clones_3cell))

for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corrected[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v2$Clone_IDs %in% clone)
  compiled_clust_df_v2[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v2[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v2[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v2[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v2[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v2[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v2[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v2[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v2[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v2[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
  
}
compiled_clust_df_v2

## noticed a very small error in the ls_div cluster that was causing some dividing INs to be called as dividing EX....
## had to go back and fix it
## pulling the subclust_idents_v2 metadata column from the updated ls_synthesis_v3 here rather than rebuilding all of filt_df
head(rownames(filt_df))
rownames(filt_df) = filt_df$CBC
ls_meta_revised = ls_synthesis_v3_harmony_v3@meta.data
ls_meta_revised = ls_meta_revised[rownames(ls_meta_revised) %in% rownames(filt_df),]
length(rownames(ls_meta_revised))
sum(rownames(ls_meta_revised) == rownames(filt_df))
table(filt_df$subclust_idents_v2)
#ASTRO                      EX                EX_early      G2M_phase_div_glia                  IN_CGE 
#885                   27395                    5862                    1935                    1842 
#IN_local                  IN_MGE                  IPC_EX              IPC_EX_DIV                  IPC_IN 
#3996                    1584                    7728                    1725                     956 
#IPC_IN_DIV                    OLIG                     OPC                 OPC_DIV                      RG 
#2242                     504                    1842                     474                    1308 
#RG (putative oRG)       RG (putative tRG)        S_phase_div_glia      Transitioning IPCs Transitioning OPCs/IPCs 
#613                     580                    1083                     488                    1167
filt_df$subclust_idents_final = ls_meta_revised$subclust_idents_final
table(filt_df$subclust_idents_final)
sum(is.na(filt_df$subclust_idents_final))
unk_cells_to_fix = rownames(filt_df)[is.na(filt_df$subclust_idents_final)]
table(filt_df$subclust_idents_v2[rownames(filt_df) %in% unk_cells_to_fix])
#IPC_EX        IPC_IN_DIV              OLIG               OPC RG (putative oRG) 
#14               239                 3                28                13
table(filt_df$seurat_clusters[rownames(filt_df) %in% unk_cells_to_fix])
#4   6   8   9  10  13 
#2  41 237   1   1  15

#clust 8 --> IPC_IN_DIV
#clust 10 --> IN_MGE

table(filt_df$clust_idents[rownames(filt_df) %in% unk_cells_to_fix])
#IN_CGE IN_local   IN_MGE      OPC       RG   RG_unk 
#1      237        1        2       41       15

filt_df[rownames(filt_df) %in% unk_cells_to_fix & filt_df$subclust_idents_v2 %in% 'IPC_IN_DIV',]
filt_df$subclust_idents_final[rownames(filt_df) %in% unk_cells_to_fix & filt_df$subclust_idents_v2 %in% 'IPC_IN_DIV'] = "IPC_IN_DIV"
filt_df$subclust_idents_final[rownames(filt_df) %in% unk_cells_to_fix & filt_df$subclust_idents_v2 %in% 'OPC'] = "OPC"
filt_df$subclust_idents_final[rownames(filt_df) %in% unk_cells_to_fix & filt_df$subclust_idents_v2 %in% 'RG (putative oRG)'] = "oRG"
filt_df$subclust_idents_final[rownames(filt_df) %in% unk_cells_to_fix & filt_df$subclust_idents_v2 %in% 'RG (putative oRG)'] = "oRG"


#Astrocytes Dividing (G2M phase)   Dividing (S phase)                   EX              EX IPCs           EX_IPC_DIV 
#GLIA_DIV              IN IPCs               IN_CGE             IN_local               IN_MGE                IN_OB 
#IPC_IN_DIV     Oligodendrocytes              OPC_DIV                 OPCs                  oRG                   RG 
#Transitioning IPCs   Transitioning OPCs                  tRG 

#v1 of broad trajectories: EX, IN, PROG, ASTRO, OLIG
filt_df$broad_trajectories = NA
filt_df$broad_trajectories[filt_df$subclust_idents_final %in% c('EX', 'EX IPCs', "EX_IPC_DIV")] = "EX"
filt_df$broad_trajectories[filt_df$subclust_idents_final %in% c("IN IPCs", 'IN_CGE', "IN_local", "IN_MGE", "IN_OB", "IPC_IN_DIV")] = "IN"
filt_df$broad_trajectories[filt_df$subclust_idents_final %in% c("oRG", "tRG", "Dividing (G2M phase)", 'Dividing (S phase)', "GLIA_DIV", "S_phase_div_glia", "Transitioning IPCs", "Transitioning OPCs")] = "PROG"
filt_df$broad_trajectories[filt_df$subclust_idents_final %in% c("OPCs", 'OPC_DIV', "Oligodendrocytes")] = "OLIG"
filt_df$broad_trajectories[filt_df$subclust_idents_final %in% c("Astrocytes", 'RG')] = "ASTRO"
sum(is.na(filt_df$subclust_idents_final))
dim(filt_df)
dim(filt_df[!is.na(filt_df$subclust_idents_final),])
filt_df = filt_df[!is.na(filt_df$subclust_idents_final),]
table(filt_df$broad_trajectories)
#ASTRO    EX    IN  OLIG  PROG 
#2193 42047 10682  3281  5989
table(filt_df[filt_df$Clone_size_corr_2umi > 2,'broad_trajectories'])
#ASTRO    EX    IN  OLIG  PROG 
#  168  4785  3507  1517  2591
table(filt_df[filt_df$Clone_size_corr_2umi > 1,'broad_trajectories'])
#ASTRO    EX    IN  OLIG  PROG 
#576  8930  5036  2214  3861

#v2 of broad trajectories: EX, IN, PROG, GLIA(astro+committed oligo)
filt_df$broad_trajectories_v2 = NA
filt_df$broad_trajectories_v2[filt_df$subclust_idents_final %in% c('EX', 'EX IPCs', "EX_IPC_DIV", "Transitioning IPCs")] = "EX"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_final %in% c("IN IPCs", 'IN_CGE', "IN_local", "IN_MGE", "IN_OB", "IPC_IN_DIV")] = "IN"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_final %in% c("oRG", "tRG", "Dividing (G2M phase)", 'Dividing (S phase)', "GLIA_DIV", "S_phase_div_glia")] = "PROG"
filt_df$broad_trajectories_v2[filt_df$subclust_idents_final %in% c("OPC_DIV", 'OPC', "OLIG", "ASTRO", 'RG', "Transitioning OPCs")] = "GLIA"
table(filt_df$broad_trajectories_v2)
#   EX  GLIA    IN  PROG 
#43187  5989 10682  4334 
table(filt_df[filt_df$Clone_size_corr_2umi > 2,'broad_trajectories_v2'])
#  EX GLIA   IN PROG 
#5186 1951 3507 1924
table(filt_df[filt_df$Clone_size_corr_2umi > 1,'broad_trajectories_v2'])
#  EX GLIA   IN PROG 
#9606 3158 5036 2817

filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('OLIG')] = "OLIG"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('PROG')] = "PROG"
filt_df$broad_trajectories[filt_df$glia_lineage_trajectory %in% c('ASTRO')] = "ASTRO"
table(filt_df$broad_trajectories)
#ASTRO    EX    IN  OLIG  PROG 
#2193 42991 10682  4944  3382 

filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('OLIG')] = "GLIA"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('PROG')] = "PROG"
filt_df$broad_trajectories_v2[filt_df$glia_lineage_trajectory %in% c('ASTRO')] = "GLIA"
table(filt_df$broad_trajectories_v2)
#   EX  GLIA    IN  PROG 
#43455  6673 10682  3382


## compiled_clust_df_v2
compiled_clust_df_v2 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v2$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$IN = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$IN_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v2$PROG_pct = rep(0,length(unique_mc_clones_3cell))

for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v2$Clone_IDs %in% clone)
  compiled_clust_df_v2[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v2[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v2[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v2[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v2[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v2[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v2[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v2[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v2[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v2[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
}
compiled_clust_df_v2
head(compiled_clust_df_v2)
tail(compiled_clust_df_v2)
paste(colnames(compiled_clust_df_idents[1,2:6] %>% select(where(~ any(. != 0)))), collapse="&")

filt_df[filt_df$Clone_IDs %in% c('LS36_Clone_451'),] ##revisit this later

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v2)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v2[r,2:6] %>% select(where(~ any(. != 0)))), collapse="&"))
}
library(UpSetR)
compiled_clust_df_v2$upset = upset_notations
table(compiled_clust_df_v2$upset)

upset(fromExpression(table(compiled_clust_df_v2$upset)),
      order.by = "freq",
      decreasing = T)



## compiled_clust_df_v3 -- uses 2cell clones, but still uses EX/I/ASTRO/OLIG/PROG
compiled_clust_df_v3 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v3$EX = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$IN = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$ASTRO = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$OLIG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$PROG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$EX_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$IN_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$ASTRO_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$OLIG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v3$PROG_pct = rep(0,length(unique_mc_clones_2cell))

for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v3$Clone_IDs %in% clone)
  compiled_clust_df_v3[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v3[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v3[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v3[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v3[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v3[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v3[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v3[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v3[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v3[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
}
compiled_clust_df_v3
head(compiled_clust_df_v3)
tail(compiled_clust_df_v3)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v3)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v3[r,2:6] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v3$upset = upset_notations
table(compiled_clust_df_v3$upset)

upset(fromExpression(table(compiled_clust_df_v3$upset)),
      order.by = "freq",
      decreasing = T)

## compiled_clust_df_v4 -- broad_trajectories_v2 (OLIG, OPC, and ASTRO combined as GLIA)
compiled_clust_df_v4 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v4$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$IN = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$GLIA = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$IN_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$GLIA_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v4$PROG_pct = rep(0,length(unique_mc_clones_3cell))

for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v2[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v4$Clone_IDs %in% clone)
  compiled_clust_df_v4[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v4[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v4[clone_row,"GLIA"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])
  compiled_clust_df_v4[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v4[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v4[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v4[clone_row,"GLIA_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])/clone_size
  compiled_clust_df_v4[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
}
compiled_clust_df_v4
head(compiled_clust_df_v4)
tail(compiled_clust_df_v4)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v4)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v4[r,2:5] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v4$upset = upset_notations
table(compiled_clust_df_v4$upset)

upset(fromExpression(table(compiled_clust_df_v4$upset)),
      order.by = "freq",
      decreasing = T)

## compiled_clust_df_v5 -- broad_trajectories_v2 (OLIG, OPC, and ASTRO combined as GLIA), 2 cell clonse
compiled_clust_df_v5 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v5$EX = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$IN = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$GLIA = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$PROG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$EX_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$IN_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$GLIA_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v5$PROG_pct = rep(0,length(unique_mc_clones_2cell))

for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v2[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v5$Clone_IDs %in% clone)
  compiled_clust_df_v5[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v5[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v5[clone_row,"GLIA"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])
  compiled_clust_df_v5[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v5[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v5[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v5[clone_row,"GLIA_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "GLIA"])/clone_size
  compiled_clust_df_v5[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
}
compiled_clust_df_v5
head(compiled_clust_df_v5)
tail(compiled_clust_df_v5)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v5)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v5[r,2:5] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v5$upset = upset_notations
table(compiled_clust_df_v5$upset)

upset(fromExpression(table(compiled_clust_df_v5$upset)),
      order.by = "freq",
      decreasing = T)


table(filt_df$subclust_idents_final)


filt_df$broad_trajectories_v3 = NA
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c('EX')] = "EX"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("EX_IPC_DIV", "EX IPCs", "Transitioning IPCs")] = "EX_IPC"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("IN_CGE")] = "IN_CGE"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("IN_MGE")] = "IN_MGE"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("IN_local")] = "IN_local"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("IN_OB")] = "IN_OB"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c('IPC_IN_DIV', "IN IPCs")] = "IN_IPC"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("oRG", 'tRG', "Dividing (G2M phase)", "Dividing (S phase)", "GLIA_DIV")] = "RG"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("OPC_DIV", 'OPC', "OPCs", "Oligodendrocytes", "Transitioning OPCs")] = "OLIG"
filt_df$broad_trajectories_v3[filt_df$subclust_idents_final %in% c("ASTRO", 'RG', 'Astrocytes')] = "ASTRO"
table(filt_df$broad_trajectories_v3)
#   ASTRO       EX   EX_IPC   IN_CGE   IN_IPC IN_local   IN_MGE    IN_OB     OLIG       RG 
#    2193    34772     7951     1588     3260     3964     1584      286     4260     4334
sum(is.na(filt_df$broad_trajectories_v3))
#0
filt_df$broad_trajectories_v3[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX_IPC"
filt_df$broad_trajectories_v3[filt_df$glia_lineage_trajectory %in% c('OLIG')] = "OLIG"
filt_df$broad_trajectories_v3[filt_df$glia_lineage_trajectory %in% c('PROG')] = "RG"
filt_df$broad_trajectories_v3[filt_df$glia_lineage_trajectory %in% c('ASTRO')] = "ASTRO"
table(filt_df$broad_trajectories_v3)
#ASTRO       EX   EX_IPC   IN_CGE   IN_IPC IN_local   IN_MGE    IN_OB     OLIG       RG 
#2193    34772     8219     1588     3260     3964     1584      286     4944     3382 

compiled_clust_df_v6 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v6$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$EX_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$RG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$EX_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$IN_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v6$RG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v3[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v6$Clone_IDs %in% clone)
  compiled_clust_df_v6[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v6[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v6[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v6[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v6[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v6[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v6[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v6[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v6[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v6[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v6[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v6[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v6[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v6[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v6[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v6[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v6[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v6[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v6[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v6[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
}
compiled_clust_df_v6
head(compiled_clust_df_v6)
tail(compiled_clust_df_v6)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v6)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v6[r,2:11] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v6$upset = upset_notations
table(compiled_clust_df_v6$upset)

upset(fromExpression(table(compiled_clust_df_v6$upset)),
      nsets=10,
      order.by = "freq",
      decreasing = T)

upset(fromExpression(table(compiled_clust_df_v6$upset)),
      nsets=10,
      nintersects = 50,
      order.by = "freq",
      decreasing = T)



## v7 also uses the broad_trajectories_v3, but has 2cell clones
compiled_clust_df_v7 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v7$EX = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$EX_IPC = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_local = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_CGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_MGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_OB = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_IPC = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$ASTRO = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$OLIG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$RG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$EX_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$EX_IPC_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_local_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_CGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_MGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_OB_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$IN_IPC_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$ASTRO_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$OLIG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v7$RG_pct = rep(0,length(unique_mc_clones_2cell))


for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v3[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v7$Clone_IDs %in% clone)
  compiled_clust_df_v7[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v7[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v7[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v7[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v7[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v7[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v7[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v7[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v7[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v7[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v7[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v7[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v7[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v7[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v7[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v7[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v7[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v7[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v7[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v7[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
}
compiled_clust_df_v7
head(compiled_clust_df_v7)
tail(compiled_clust_df_v7)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v7)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v7[r,2:11] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v7$upset = upset_notations
table(compiled_clust_df_v7$upset)

upset(fromExpression(table(compiled_clust_df_v7$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


## broad_trajectories_v4 breaks out tRG and oRG, and leaves other unknown or cycling progenitors as "PROG"

filt_df$broad_trajectories_v4 = NA
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c('EX')] = "EX"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("EX_IPC_DIV", "EX IPCs", "Transitioning IPCs")] = "EX_IPC"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("IN_CGE")] = "IN_CGE"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("IN_MGE")] = "IN_MGE"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("IN_local")] = "IN_local"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("IN_OB")] = "IN_OB"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c('IPC_IN_DIV', "IN IPCs")] = "IN_IPC"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("oRG")] = "oRG"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c('tRG')] = "tRG"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("Dividing (G2M phase)", "Dividing (S phase)", "GLIA_DIV", 'RG')] = "PROG"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("OPC_DIV", 'OPC', "OPCs", "Oligodendrocytes", "Transitioning OPCs")] = "OLIG"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c("ASTRO", 'Astrocytes')] = "ASTRO"
table(filt_df$broad_trajectories_v4)
#   ASTRO       EX   EX_IPC   IN_CGE   IN_IPC IN_local   IN_MGE    IN_OB     OLIG      oRG     PROG      tRG 
#     885    34772     7951     1588     3260     3964     1584      286     4260      613     4449      580
sum(is.na(filt_df$broad_trajectories_v4))

filt_df$broad_trajectories_v4[filt_df$glia_lineage_trajectory %in% c('EX')] = "EX_IPC"
filt_df$broad_trajectories_v4[filt_df$subclust_idents_final %in% c('Dividing (G2M phase)', 'Dividing (S phase)') & filt_df$glia_lineage_trajectory %in% c('OLIG')] = "OLIG"
table(filt_df$broad_trajectories_v4)
# ASTRO       EX   EX_IPC   IN_CGE   IN_IPC IN_local   IN_MGE    IN_OB     OLIG      oRG     PROG      tRG 
#   885    34772     8219     1588     3260     3964     1584      286     4944      613     3497      580

## v8 uses broad_trajectories_v4 and has 3cell clones
compiled_clust_df_v8 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v8$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$EX_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$EX_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$IN_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$PROG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v8$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v4[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v8$Clone_IDs %in% clone)
  compiled_clust_df_v8[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v8[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v8[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v8[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v8[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v8[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v8[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v8[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v8[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v8[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v8[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v8[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v8[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v8[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v8[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v8[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v8[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v8[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v8[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v8[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v8[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v8[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
  compiled_clust_df_v8[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v8[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v8
head(compiled_clust_df_v8)
tail(compiled_clust_df_v8)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v8)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v8[r,2:13] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v8$upset = upset_notations
table(compiled_clust_df_v8$upset)

upset(fromExpression(table(compiled_clust_df_v8$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

## v9 uses broad_trajectories_v4 and has 3cell clones
compiled_clust_df_v9 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v9$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$EX_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$EX_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$IN_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$PROG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v9$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v4[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v9$Clone_IDs %in% clone)
  compiled_clust_df_v9[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v9[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v9[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v9[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v9[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v9[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v9[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v9[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v9[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v9[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v9[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v9[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v9[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v9[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v9[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v9[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v9[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
  compiled_clust_df_v9[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v9[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v9
head(compiled_clust_df_v9)
tail(compiled_clust_df_v9)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v9)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v9[r,2:13] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v9$upset = upset_notations
table(compiled_clust_df_v9$upset)

upset(fromExpression(table(compiled_clust_df_v9$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

## v9 uses broad_trajectories_v4 and has 2cell clones
compiled_clust_df_v9 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v9$EX = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$EX_IPC = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_local = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_CGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_MGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_OB = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_IPC = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$ASTRO = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$OLIG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$PROG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$tRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$oRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$EX_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$EX_IPC_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_local_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_CGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_MGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_OB_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$IN_IPC_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$ASTRO_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$OLIG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$PROG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$tRG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v9$oRG_pct = rep(0,length(unique_mc_clones_2cell))


for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$broad_trajectories_v4[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v9$Clone_IDs %in% clone)
  compiled_clust_df_v9[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v9[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v9[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v9[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v9[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v9[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v9[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v9[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v9[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v9[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v9[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v9[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v9[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v9[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v9[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v9[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v9[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v9[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
  compiled_clust_df_v9[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v9[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v9
head(compiled_clust_df_v9)
tail(compiled_clust_df_v9)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v9)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v9[r,2:13] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v9$upset = upset_notations
table(compiled_clust_df_v9$upset)

upset(fromExpression(table(compiled_clust_df_v9$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

## if we randomly shuffle the celltypes, what do the identities look like?
filt_df$shuffled_trajectories_v4 = sample(filt_df$broad_trajectories_v4)
table(filt_df$broad_trajectories_v4)
table(filt_df$shuffled_trajectories_v4)

## v10 uses shuffled_trajectories_v4 and has 3cell clones
compiled_clust_df_v10 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v10$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$EX_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_IPC = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$PROG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$EX_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$IN_IPC_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$PROG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v10$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df$Clone_size_corr_2umi[filt_df$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df$shuffled_trajectories_v4[filt_df$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v10$Clone_IDs %in% clone)
  compiled_clust_df_v10[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v10[clone_row,"EX_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])
  compiled_clust_df_v10[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v10[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v10[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v10[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v10[clone_row,"IN_IPC"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])
  compiled_clust_df_v10[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v10[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v10[clone_row,"PROG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])
  compiled_clust_df_v10[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v10[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v10[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v10[clone_row,"EX_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPC"])/clone_size
  compiled_clust_df_v10[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v10[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v10[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v10[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v10[clone_row,"IN_IPC_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPC"])/clone_size
  compiled_clust_df_v10[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v10[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v10[clone_row,"PROG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "PROG"])/clone_size
  compiled_clust_df_v10[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v10[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v10
head(compiled_clust_df_v10)
tail(compiled_clust_df_v10)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v10)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v10[r,2:13] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v10$upset = upset_notations
table(compiled_clust_df_v10$upset)

upset(fromExpression(table(compiled_clust_df_v10$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


## can we add sample and age to these
dim(compiled_clust_df_v8)
#[1] 2939   26
dim(distinct(filt_df[,c('Clone_IDs', 'orig.ident', 'sample_age', 'age_pseudo')]))
head(left_join(compiled_clust_df_v8, distinct(filt_df[,c('Clone_IDs', 'orig.ident', 'sample_age', 'age_pseudo')])))
compiled_clust_df_v8 = left_join(compiled_clust_df_v8, distinct(filt_df[,c('Clone_IDs', 'orig.ident', 'sample_age', 'age_pseudo')]))
dim(compiled_clust_df_v8)
head(compiled_clust_df_v8)
upset(fromExpression(table(subset(compiled_clust_df_v8, age_pseudo %in% c('<=GW20'))$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(subset(compiled_clust_df_v8, age_pseudo %in% c('>GW20'))$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

compiled_clust_df_v9 = left_join(compiled_clust_df_v9, distinct(filt_df[,c('Clone_IDs', 'orig.ident', 'sample_age', 'age_pseudo')]))
dim(compiled_clust_df_v9)
head(compiled_clust_df_v9)
upset(fromExpression(table(subset(compiled_clust_df_v9, age_pseudo %in% c('<=GW20'))$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(subset(compiled_clust_df_v9, age_pseudo %in% c('>GW20'))$upset)),
      nsets=10,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


clones_of_interest = compiled_clust_df_v9$Clone_IDs[compiled_clust_df_v9$upset %in% 'EX&IN_local']
filt_df[filt_df$Clone_IDs %in% clones_of_interest,]
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clones_of_interest], raster=F)


clones_of_interest = compiled_clust_df_v9$Clone_IDs[compiled_clust_df_v9$upset %in% c('EX&IN_IPC', 'EX&IN_local')]
filt_df[filt_df$Clone_IDs %in% clones_of_interest,]
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% clones_of_interest], raster=F)




ggplot(filt_df, aes(x=orig.ident, y=read_counts_filt)) +
  geom_violin() +
  ylim(c(0,10000))
ggplot(filt_df, aes(x=orig.ident, y=Barcode_UMI_Count_filt)) +
  geom_violin() +
  ylim(c(0,100))
ggplot(filt_df, aes(x=subclust_idents_final, y=Barcode_UMI_Count_filt)) +
  geom_violin() +
  ylim(c(0,100))



##### ------------------------------------------------------------------------------------ #########
ls_meta_copy = ls_synthesis_v3_harmony_v3@meta.data
colnames(filt_df)
ls_meta_copy$CBC = rownames(ls_meta_copy)
ls_meta_copy = ls_meta_copy[rownames(ls_meta_copy) %in% rownames(filt_df),]

dim(filt_df)
dim(ls_meta_copy)
dim(left_join(filt_df, ls_meta_copy[,c("CBC", "subclust_idents_definitive")]))
filt_df_v2 = left_join(filt_df, ls_meta_copy[,c("CBC", "subclust_idents_definitive")])
sum(filt_df_v2$CBC == filt_df$CBC)
rownames(filt_df_v2) = filt_df_v2$CBC

table(filt_df_v2$Clone_size_corr_2umi[filt_df_v2$subclust_idents_definitive %in% "MG"])
table(filt_df_v2$Clone_size_corr_2umi[filt_df_v2$subclust_idents_definitive %in% "MG"])
filt_df_v2[filt_df_v2$subclust_idents_definitive %in% "MG" & filt_df_v2$Clone_size_corr_2umi >=2,]
filt_df_v2[filt_df_v2$subclust_idents_definitive %in% "MG" & filt_df_v2$Clone_size_corr_2umi >=2,]
filt_df_v2[filt_df_v2$Clone_IDs_filt %in% c('LS21_Clone_30', 'LS21_Clone_1631', 'LS23_Clone_990', 'LS36_Clone_518'),]

table(filt_df_v2$subclust_idents_definitive)
#Astrocytes               EX          EX IPCs          IN IPCs           IN_CGE         IN_local           IN_MGE 
#527            33146             9076             3833             1403             3963             1579 
#IN_OB               MG Oligodendrocytes             OPCs              oRG               RG              tRG 
#386               40              486             2331              390             6052              410

filt_df_v2$broad_trajectories_v5 = NA
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('Astrocytes')] = 'ASTRO'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('EX')] = 'EX'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('EX IPCs')] = 'EX_IPCs'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('IN IPCs')] = 'IN_IPCs'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('IN_CGE')] = 'IN_CGE'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('IN_local')] = 'IN_local'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('IN_MGE')] = 'IN_MGE'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('IN_OB')] = 'IN_OB'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('OPCs', 'Oligodendrocytes')] = 'OLIG'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('oRG')] = 'oRG'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('RG')] = 'RG'
filt_df_v2$broad_trajectories_v5[filt_df_v2$subclust_idents_definitive %in% c('tRG')] = 'tRG'

## v11 uses subclust_idents_definitive and has 3cell clones
compiled_clust_df_v11 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v11$EX = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$EX_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$ASTRO = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$OLIG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$RG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$EX_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$EX_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$ASTRO_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$OLIG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$RG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v11$Clone_IDs %in% clone)
  compiled_clust_df_v11[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v11[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v11[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v11[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v11[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v11[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v11[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v11[clone_row,"ASTRO"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])
  compiled_clust_df_v11[clone_row,"OLIG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])
  compiled_clust_df_v11[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v11[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v11[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v11[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v11[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v11[clone_row,"ASTRO_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ASTRO"])/clone_size
  compiled_clust_df_v11[clone_row,"OLIG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OLIG"])/clone_size
  compiled_clust_df_v11[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v11[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v11[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v11
head(compiled_clust_df_v11)
tail(compiled_clust_df_v11)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v11)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v11[r,2:13] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v11$upset = upset_notations
table(compiled_clust_df_v11$upset)

upset(fromExpression(table(compiled_clust_df_v11$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


filt_df_v2$ex_in_presence = NA

ex_in_shared_clones = compiled_clust_df_v11$Clone_IDs[compiled_clust_df_v11$upset %in% c('EX_IPCs&IN_IPCs', 'EX_IPCs&IN_IPCs&RG',
                                                                                        'EX_IPCs&IN_local', 'EX_IPCs&IN_local&IN_IPCs',
                                                                                        'EX_IPCs&IN_local&IN_IPCs&RG', 'EX_IPCs&IN_local&IN_IPCs&RG&tRG',
                                                                                        'EX&EX_IPCs&IN_IPCs', 'EX&EX_IPCs&IN_IPCs&RG', 'EX&EX_IPCs&IN_local',
                                                                                        'EX&EX_IPCs&IN_local&IN_IPCs','EX&EX_IPCs&IN_local&IN_IPCs&RG','EX&EX_IPCs&IN_local&RG',
                                                                                        'EX&IN_IPCs','EX&IN_IPCs&RG','EX&IN_local','EX&IN_local&IN_IPCs','EX&IN_local&IN_IPCs&RG ',
                                                                                        'EX&IN_local&RG ')]
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], raster=F)
DimPlot(ls_in_v3, cells.highlight = rownames(ls_in_v3@meta.data)[ls_in_v3@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], raster=F)
DimPlot(ls_prog, cells.highlight = rownames(ls_prog@meta.data)[ls_prog@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], raster=F)
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], raster=F)
#bkmrk1

table(compiled_clust_df_v11$upset)

dorsal_in_shared_clones = compiled_clust_df_v11$Clone_IDs[compiled_clust_df_v11$upset %in% c('EX_IPCs&IN_IPCs', 'EX_IPCs&IN_IPCs&RG',
                                                                                           'EX_IPCs&IN_local', 'EX_IPCs&IN_local&IN_IPCs',
                                                                                           'EX_IPCs&IN_local&IN_IPCs&RG', 'EX_IPCs&IN_local&IN_IPCs&RG&tRG',
                                                                                           'EX&EX_IPCs&IN_IPCs', 'EX&EX_IPCs&IN_IPCs&RG', 'EX&EX_IPCs&IN_local',
                                                                                           'EX&EX_IPCs&IN_local&IN_IPCs','EX&EX_IPCs&IN_local&IN_IPCs&RG','EX&EX_IPCs&IN_local&RG',
                                                                                           'EX&IN_IPCs','EX&IN_IPCs&RG','EX&IN_local','EX&IN_local&IN_IPCs','EX&IN_local&IN_IPCs&RG ',
                                                                                           'EX&IN_local&RG ', 'IN_IPCs&ASTRO', 'IN_IPCs&ASTRO&OLIG',
                                                                                           'IN_IPCs&ASTRO&OLIG&RG', 'IN_IPCs&ASTRO&OLIG&RG&oRG', 'IN_IPCs&ASTRO&RG',
                                                                                           'IN_IPCs&OLIG&oRG','IN_IPCs&OLIG&RG','IN_IPCs&OLIG&RG&oRG','IN_IPCs&OLIG&RG&tRG',
                                                                                           'IN_local&IN_IPCs&RG','IN_local&OLIG&oRG','IN_local&OLIG&RG','IN_local&RG')]
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones], raster=F)

FeaturePlot(ls_prog, features=c("CROCCP2", "TMEM14B", "NOTCH2NLA", "ARHGAP11A"))
FeaturePlot(ls_prog_div, features=c("CROCCP2", "TMEM14B", "NOTCH2NLA", "ARHGAP11A"))
FeaturePlot(ls_prog_div, features=c("MKI67", "PCNA", "NOTCH2NLA", "ARHGAP11A"))
DimPlot(ls_prog_div, group.by="subclust_idents_definitive")

DimPlot(ls_prog, group.by="subclust_idents_definitive")
DimPlot(ls_prog, group.by="subclust_idents_definitive")

DimPlot(ls_prog_glia, group.by="subclust_idents_definitive")
FeaturePlot(ls_prog_glia, features=c("HOPX", "VIM", "OLIG2", "CRYAB"))
FeaturePlot(ls_prog_glia, features=c("HES1", "MT3", "HOPX", "PTPRZ1"))
DimPlot(ls_prog_glia, group.by="Clone_Index_filt")
DimPlot(ls_prog_glia, group.by="age_pseudo")
FeaturePlot(ls_prog_glia, features=c("CRYAB"), split.by="Clone_Index_filt")
FeaturePlot(ls_prog_glia, features=c("CRYAB"), split.by="age_pseudo")
FeaturePlot(ls_prog_glia, features=c("MT3"), split.by="Clone_Index_filt")
FeaturePlot(ls_prog_glia, features=c("MT3"), split.by="age_pseudo")

#save.image("/media/chang/HDD-8/mgkeefe/231210_local_sticr_synthesis.RData")



## revisiting this now that some things have been slightly changed again....
ls_meta_copy = ls_synthesis_v3_harmony_v3@meta.data
colnames(filt_df)
ls_meta_copy$CBC = rownames(ls_meta_copy)
ls_meta_copy = ls_meta_copy[rownames(ls_meta_copy) %in% rownames(filt_df),]

dim(filt_df)
dim(ls_meta_copy)
dim(left_join(filt_df, ls_meta_copy[,c("CBC", "subclust_idents_definitive")]))
filt_df_v2 = left_join(filt_df, ls_meta_copy[,c("CBC", "subclust_idents_definitive")])
sum(filt_df_v2$CBC == filt_df$CBC)
rownames(filt_df_v2) = filt_df_v2$CBC

table(filt_df_v2$subclust_idents_definitive)
#Astrocytes              ENs          EX_IPCs           IN_CGE          IN_IPCs         IN_local           IN_MGE 
#527            33146             9076             1403             3833             3963             1579 
#IN_OB Oligodendrocytes             OPCs              oRG               RG              tRG 
#386              486             2331              390             6052              450 

filt_df_v2$broad_trajectories_v5 = filt_df_v2$subclust_idents_definitive

## v11 uses subclust_idents_definitive and has 3cell clones
compiled_clust_df_v11 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v11$ENs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$EX_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$Astrocytes = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$OPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$Oligodendrocytes = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$RG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$ENs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$EX_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$IN_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$Astrocytes_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$OPCS_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$RG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v11$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v11$Clone_IDs %in% clone)
  compiled_clust_df_v11[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v11[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v11[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v11[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v11[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v11[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v11[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v11[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v11[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v11[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v11[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v11[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v11[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v11[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v11[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v11[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v11[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v11[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v11[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v11[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v11[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v11[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v11
head(compiled_clust_df_v11)
tail(compiled_clust_df_v11)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v11)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v11[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v11$upset = upset_notations
table(compiled_clust_df_v11$upset)

upset(fromExpression(table(compiled_clust_df_v11$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v11)
head(filt_df_v2)
dim(left_join(compiled_clust_df_v11, filt_df_v2[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v11 = left_join(compiled_clust_df_v11, filt_df_v2[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v11$upset[compiled_clust_df_v11$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v11$upset[compiled_clust_df_v11$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


## v12 uses subclust_idents_definitive and has 2cell clones
compiled_clust_df_v12 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v12$ENs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$EX_IPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_local = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_CGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_MGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_OB = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_IPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$Astrocytes = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$OPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$Oligodendrocytes = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$RG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$tRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$oRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$ENs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$EX_IPCs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_local_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_CGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_MGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_OB_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$IN_IPCs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$Astrocytes_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$OPCS_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$RG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$tRG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v12$oRG_pct = rep(0,length(unique_mc_clones_2cell))


for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v12$Clone_IDs %in% clone)
  compiled_clust_df_v12[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v12[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v12[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v12[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v12[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v12[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v12[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v12[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v12[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v12[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v12[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v12[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v12[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v12[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v12[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v12[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v12[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v12[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v12[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v12[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v12[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v12[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v12[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v12[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v12[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v12[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v12
head(compiled_clust_df_v12)
tail(compiled_clust_df_v12)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v12)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v12[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v12$upset = upset_notations
table(compiled_clust_df_v12$upset)

upset(fromExpression(table(compiled_clust_df_v12$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v12)
head(filt_df_v2)
dim(left_join(compiled_clust_df_v12, filt_df_v2[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v12 = left_join(compiled_clust_df_v12, filt_df_v2[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v12$upset[compiled_clust_df_v12$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v12$upset[compiled_clust_df_v12$sample_age>20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)

head(filt_df_v2)
table(filt_df_v2$Barcode_UMI_Count_filt)
table(filt_df_v2$Clone_size_corr_2umi)


## going to implement a 3 UMI cutoff to have higher certainty of barcode calls
filt_df_v3 = filt_df_v2[filt_df_v2$Barcode_UMI_Count > 2 & filt_df_v2$read_counts > 50,]
filt_df_v3 = filt_df_v3[!is.na(filt_df_v3$Barcode_UMI_Count),]
dim(filt_df_v3)
#[1] 58695    45

unique_clones_3umi = unique(filt_df_v3$Clone_IDs)
length(unique_clones_3umi)
#[1] 46186

mc = unique_clones_3umi[48]
filt_df_v3[filt_df_v3$Clone_IDs %in% mc,]
dim(filt_df_v3[filt_df_v3$Clone_IDs %in% mc,])[1]

dim(filt_df_v3)
head(filt_df_v3)
filt_df_v3$Clone_size_corr_3umi = NA

for (mc in unique_clones_3umi) {
  Clone_size_corr_3umi = dim(filt_df_v3[filt_df_v3$Clone_IDs %in% mc,])[1]
  filt_df_v3[filt_df_v3$Clone_IDs %in% mc,'Clone_size_corr_3umi'] = Clone_size_corr_3umi
}
table(filt_df_v3$Clone_size_corr_3umi)
#    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    22    25 
#39770  7388  4110  2416  1560  1062   588   464   261   320   187   132   130    42    45    80    34    18    19    44    25
table(filt_df_v3$Clone_size_corr_2umi)
#.   1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19    22    25 
#39298  7463  4140  2476  1609  1124   680   428   343   263   269   105   103    75    72    61    66    32    19    22    22 
#26 
#25

table(filt_df_v3$broad_trajectories_v5)
table(filt_df_v3$subclust_idents_definitive)

## v13 uses subclust_idents_definitive and has 3cell clones (3UMI cutoff now!)
compiled_clust_df_v13 = data.frame(Clone_IDs=unique_mc_clones_3cell)
compiled_clust_df_v13$ENs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$EX_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_local = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_CGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_MGE = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_OB = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_IPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$Astrocytes = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$OPCs = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$Oligodendrocytes = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$RG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$tRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$oRG = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$ENs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$EX_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_local_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_CGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_MGE_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_OB_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$IN_IPCs_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$Astrocytes_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$OPCS_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$RG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$tRG_pct = rep(0,length(unique_mc_clones_3cell))
compiled_clust_df_v13$oRG_pct = rep(0,length(unique_mc_clones_3cell))


for(clone in unique_mc_clones_3cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v13$Clone_IDs %in% clone)
  compiled_clust_df_v13[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v13[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v13[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v13[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v13[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v13[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v13[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v13[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v13[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v13[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v13[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v13[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v13[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v13[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v13[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v13[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v13[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v13[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v13[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v13
head(compiled_clust_df_v13)
tail(compiled_clust_df_v13)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v13)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v13[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v13$upset = upset_notations
table(compiled_clust_df_v13$upset)

upset(fromExpression(table(compiled_clust_df_v13$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v13)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v13, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v13 = left_join(compiled_clust_df_v13, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


## v14 uses subclust_idents_definitive and has 2cell clones (3UMI cutoff now!)
compiled_clust_df_v14 = data.frame(Clone_IDs=unique_mc_clones_2cell)
compiled_clust_df_v14$ENs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$EX_IPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_local = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_CGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_MGE = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_OB = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_IPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$Astrocytes = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$OPCs = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$Oligodendrocytes = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$RG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$tRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$oRG = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$ENs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$EX_IPCs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_local_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_CGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_MGE_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_OB_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$IN_IPCs_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$Astrocytes_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$OPCS_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$RG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$tRG_pct = rep(0,length(unique_mc_clones_2cell))
compiled_clust_df_v14$oRG_pct = rep(0,length(unique_mc_clones_2cell))


for(clone in unique_mc_clones_2cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v14$Clone_IDs %in% clone)
  compiled_clust_df_v14[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v14[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v14[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v14[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v14[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v14[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v14[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v14[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v14[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v14[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v14[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v14[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v14[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v14[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v14[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v14[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v14[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v14[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v14[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v14
head(compiled_clust_df_v14)
tail(compiled_clust_df_v14)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v14)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v14[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v14$upset = upset_notations
table(compiled_clust_df_v14$upset)

upset(fromExpression(table(compiled_clust_df_v14$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v14)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v14, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v14 = left_join(compiled_clust_df_v14, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


pdf('upset_all_samples_2cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_<=GW20_samples_2cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_>GW20_samples_2cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()

pdf('upset_all_samples_3cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_<=GW20_samples_3cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_>GW20_samples_3cell_clones.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()

table(compiled_clust_df_v14$upset)
compiled_clust_df_v14$ex_count = compiled_clust_df_v14$ENs + compiled_clust_df_v14$EX_IPCs
compiled_clust_df_v14$in_count = compiled_clust_df_v14$IN_IPCs + compiled_clust_df_v14$IN_local + compiled_clust_df_v14$IN_CGE + compiled_clust_df_v14$IN_MGE + compiled_clust_df_v14$IN_OB
compiled_clust_df_v14$ex_in_shared = NA
compiled_clust_df_v14$ex_in_shared[compiled_clust_df_v14$ex_count > 0 & compiled_clust_df_v14$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v14$ex_in_shared))
tail(compiled_clust_df_v14)
head(compiled_clust_df_v14)
compiled_clust_df_v14$rg_count = compiled_clust_df_v14$RG + compiled_clust_df_v14$tRG + compiled_clust_df_v14$oRG
compiled_clust_df_v14$dorsal_in_shared = NA
compiled_clust_df_v14$dorsal_in_shared[compiled_clust_df_v14$ex_count+compiled_clust_df_v14$rg_count > 0 & compiled_clust_df_v14$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$dorsal_in_shared %in% "dorsal_in_shared"]


DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])


table(compiled_clust_df_v13$upset)
compiled_clust_df_v13$ex_count = compiled_clust_df_v13$ENs + compiled_clust_df_v13$EX_IPCs
compiled_clust_df_v13$in_count = compiled_clust_df_v13$IN_IPCs + compiled_clust_df_v13$IN_local + compiled_clust_df_v13$IN_CGE + compiled_clust_df_v13$IN_MGE + compiled_clust_df_v13$IN_OB
compiled_clust_df_v13$ex_in_shared = NA
compiled_clust_df_v13$ex_in_shared[compiled_clust_df_v13$ex_count > 0 & compiled_clust_df_v13$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v13$ex_in_shared))
tail(compiled_clust_df_v13)
head(compiled_clust_df_v13)
compiled_clust_df_v13$rg_count = compiled_clust_df_v13$RG + compiled_clust_df_v13$tRG + compiled_clust_df_v13$oRG
compiled_clust_df_v13$dorsal_in_shared = NA
compiled_clust_df_v13$dorsal_in_shared[compiled_clust_df_v13$ex_count+compiled_clust_df_v13$rg_count > 0 & compiled_clust_df_v13$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])
DimPlot(ls_in_v3, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3])

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])


compiled_clust_df_v13[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared", ]
table(compiled_clust_df_v13$IN_MGE[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"])
compiled_clust_df_v13[compiled_clust_df_v13$IN_MGE>0 & compiled_clust_df_v13$ex_in_shared %in% c("ex_in_shared"),]

ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS21_Clone_30',]
filt_df_v3[filt_df_v3$Clone_IDs %in% 'LS21_Clone_30',]
ls_in_v2@meta.data[ls_in_v2@meta.data$Clone_IDs_filt %in% 'LS21_Clone_30',]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS35_Clone_189',] ##this is the clone that has a shared MGE with EX.... nothing I can really do about it
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS36_Clone_140',]

#plot shared clones and in multicell clones on the umaps
ex_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"]
dorsal_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_3cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_3cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_ex_in_shared_clones_3cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="full_umap_ex_in_shared_clones_3cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_3cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_3cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_dorsal_in_shared_clones_3cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="full_umap_dorsal_in_shared_clones_3cells_3umis.svg", plot=image, width=12, height=8)

in_multicell_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$in_count > 0]
DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_3cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_3cells_3umis.svg", plot=image, width=12, height=8)


#repeat for 2cell clones (compiled_clust_df_v14)
ex_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$ex_in_shared %in% "ex_in_shared"]
dorsal_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_ex_in_shared_clones_2cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="full_umap_ex_in_shared_clones_2cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_dorsal_in_shared_clones_2cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="full_umap_dorsal_in_shared_clones_2cells_3umis.svg", plot=image, width=12, height=8)

in_multicell_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$in_count > 0]
DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_2cells_3umis.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_2cells_3umis.svg", plot=image, width=12, height=8)






upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T, scale.intersections = "log10", show.numbers = F)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T, scale.intersections = "log10", show.numbers = F)




## the code was wrong for making compiled_clust_df_v13 and _v14, and they are still using the 2umi clone lists
## going to re-do that now and re-run some of the code blocks just to see what 3umi changes

unique_mc_clones_3umi_3cell = unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 3])
length(unique_mc_clones_3umi_3cell)
#[1] 2722
unique_mc_clones_3umi_2cell = unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_2umi >= 2])
length(unique_mc_clones_3umi_2cell)
#[1] 6888


## v13 uses subclust_idents_definitive and has 3cell clones (3UMI cutoff now!)
compiled_clust_df_v13 = data.frame(Clone_IDs=unique_mc_clones_3umi_3cell)
compiled_clust_df_v13$ENs = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$EX_IPCs = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_local = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_CGE = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_MGE = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_OB = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_IPCs = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$Astrocytes = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$OPCs = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$Oligodendrocytes = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$RG = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$tRG = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$oRG = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$ENs_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$EX_IPCs_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_local_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_CGE_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_MGE_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_OB_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$IN_IPCs_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$Astrocytes_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$OPCS_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$RG_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$tRG_pct = rep(0,length(unique_mc_clones_3umi_3cell))
compiled_clust_df_v13$oRG_pct = rep(0,length(unique_mc_clones_3umi_3cell))


for(clone in unique_mc_clones_3umi_3cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v13$Clone_IDs %in% clone)
  compiled_clust_df_v13[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v13[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v13[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v13[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v13[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v13[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v13[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v13[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v13[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v13[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v13[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v13[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v13[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v13[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v13[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v13[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v13[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v13[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v13[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v13[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v13[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v13
head(compiled_clust_df_v13)
tail(compiled_clust_df_v13)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v13)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v13[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
library(UpSetR)
compiled_clust_df_v13$upset = upset_notations
table(compiled_clust_df_v13$upset)

upset(fromExpression(table(compiled_clust_df_v13$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v13)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v13, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v13 = left_join(compiled_clust_df_v13, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


## v14 uses subclust_idents_definitive and has 2cell clones (3UMI cutoff now!)
compiled_clust_df_v14 = data.frame(Clone_IDs=unique_mc_clones_3umi_2cell)
compiled_clust_df_v14$ENs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$EX_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_local = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_CGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_MGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_OB = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$Astrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$OPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$Oligodendrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$RG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$tRG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$oRG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$ENs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$EX_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_local_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_CGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_MGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_OB_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$IN_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$Astrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$OPCS_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$RG_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$tRG_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v14$oRG_pct = rep(0,length(unique_mc_clones_3umi_2cell))


for(clone in unique_mc_clones_3umi_2cell) {
  clone_size = filt_df_v2$Clone_size_corr_2umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v2$broad_trajectories_v5[filt_df_v2$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v14$Clone_IDs %in% clone)
  compiled_clust_df_v14[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v14[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v14[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v14[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v14[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v14[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v14[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v14[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v14[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v14[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v14[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v14[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v14[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v14[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v14[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v14[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v14[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v14[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v14[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v14[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v14[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v14
head(compiled_clust_df_v14)
tail(compiled_clust_df_v14)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v14)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v14[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v14$upset = upset_notations
table(compiled_clust_df_v14$upset)

upset(fromExpression(table(compiled_clust_df_v14$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v14)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v14, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first"))
compiled_clust_df_v14 = left_join(compiled_clust_df_v14, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


pdf('upset_all_samples_2cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_<=GW20_samples_2cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_>GW20_samples_2cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()

pdf('upset_all_samples_3cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_<=GW20_samples_3cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_>GW20_samples_3cell_clones_3umi_fixed.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v13$upset[compiled_clust_df_v13$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()

table(compiled_clust_df_v14$upset)
compiled_clust_df_v14$ex_count = compiled_clust_df_v14$ENs + compiled_clust_df_v14$EX_IPCs
compiled_clust_df_v14$in_count = compiled_clust_df_v14$IN_IPCs + compiled_clust_df_v14$IN_local + compiled_clust_df_v14$IN_CGE + compiled_clust_df_v14$IN_MGE + compiled_clust_df_v14$IN_OB
compiled_clust_df_v14$ex_in_shared = NA
compiled_clust_df_v14$ex_in_shared[compiled_clust_df_v14$ex_count > 0 & compiled_clust_df_v14$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v14$ex_in_shared))
tail(compiled_clust_df_v14)
head(compiled_clust_df_v14)
compiled_clust_df_v14$rg_count = compiled_clust_df_v14$RG + compiled_clust_df_v14$tRG + compiled_clust_df_v14$oRG
compiled_clust_df_v14$dorsal_in_shared = NA
compiled_clust_df_v14$dorsal_in_shared[compiled_clust_df_v14$ex_count+compiled_clust_df_v14$rg_count > 0 & compiled_clust_df_v14$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$dorsal_in_shared %in% "dorsal_in_shared"]


DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])


table(compiled_clust_df_v13$upset)
compiled_clust_df_v13$ex_count = compiled_clust_df_v13$ENs + compiled_clust_df_v13$EX_IPCs
compiled_clust_df_v13$in_count = compiled_clust_df_v13$IN_IPCs + compiled_clust_df_v13$IN_local + compiled_clust_df_v13$IN_CGE + compiled_clust_df_v13$IN_MGE + compiled_clust_df_v13$IN_OB
compiled_clust_df_v13$ex_in_shared = NA
compiled_clust_df_v13$ex_in_shared[compiled_clust_df_v13$ex_count > 0 & compiled_clust_df_v13$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v13$ex_in_shared))
tail(compiled_clust_df_v13)
head(compiled_clust_df_v13)
compiled_clust_df_v13$rg_count = compiled_clust_df_v13$RG + compiled_clust_df_v13$tRG + compiled_clust_df_v13$oRG
compiled_clust_df_v13$dorsal_in_shared = NA
compiled_clust_df_v13$dorsal_in_shared[compiled_clust_df_v13$ex_count+compiled_clust_df_v13$rg_count > 0 & compiled_clust_df_v13$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])
DimPlot(ls_in_v3, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3])

DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3])


compiled_clust_df_v13[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared", ]
table(compiled_clust_df_v13$IN_MGE[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"])
compiled_clust_df_v13[compiled_clust_df_v13$IN_MGE>0 & compiled_clust_df_v13$ex_in_shared %in% c("ex_in_shared"),]

ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS21_Clone_30',]
filt_df_v3[filt_df_v3$Clone_IDs %in% 'LS21_Clone_30',]
ls_in_v2@meta.data[ls_in_v2@meta.data$Clone_IDs_filt %in% 'LS21_Clone_30',]
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS35_Clone_189',] ##this is the clone that has a shared MGE with EX.... nothing I can really do about it
ls_synthesis_v3_harmony_v3@meta.data[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% 'LS36_Clone_140',]

#plot shared clones and in multicell clones on the umaps
ex_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$ex_in_shared %in% "ex_in_shared"]
dorsal_in_shared_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_3cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_3cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_ex_in_shared_clones_3cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="full_umap_ex_in_shared_clones_3cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_3cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_3cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_dorsal_in_shared_clones_3cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="full_umap_dorsal_in_shared_clones_3cells_3umis_fixed.svg", plot=image, width=12, height=8)

in_multicell_clones = compiled_clust_df_v13$Clone_IDs[compiled_clust_df_v13$in_count > 0]
DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 3)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_3cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_3cells_3umis_fixed.svg", plot=image, width=12, height=8)


#repeat for 2cell clones (compiled_clust_df_v14)
ex_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$ex_in_shared %in% "ex_in_shared"]
dorsal_in_shared_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$dorsal_in_shared %in% "dorsal_in_shared"]

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% ex_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('EX and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_ex_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="full_umap_ex_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones & ls_synthesis_v3_harmony_v4@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#f27527") +
  ggtitle('Dorsal and IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="full_umap_dorsal_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="full_umap_dorsal_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

in_multicell_clones = compiled_clust_df_v14$Clone_IDs[compiled_clust_df_v14$in_count > 0]
DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones & ls_in_v2@meta.data$Barcode_UMI_Count_filt >=3], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)


## want to look at clone size in addition to clone comp... not sure the best way to do it
table(compiled_clust_df_v14$rg_count)
colnames(compiled_clust_df_v14)
dim(compiled_clust_df_v14)
dim(left_join(compiled_clust_df_v14, filt_df_v3[,c('Clone_IDs', 'Clone_size_corr_3umi')], multiple="first"))
compiled_clust_df_v14 = left_join(compiled_clust_df_v14, filt_df_v3[,c('Clone_IDs', 'Clone_size_corr_3umi')], multiple="first")
compiled_clust_df_v14 = left_join(compiled_clust_df_v14, filt_df_v3[,c('Clone_IDs', 'Clone_size_corr_2umi')], multiple="first")
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$Clone_size_corr_3umi>=5])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$Clone_size_corr_2umi>=4])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$Clone_size_corr_2umi>=3])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$Clone_size_corr_2umi>=2])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
hist(compiled_clust_df_v14$Clone_size_corr_2umi[compiled_clust_df_v14$upset %in% "EX_IPCs"])

sum(compiled_clust_df_v14$Clone_size_corr_2umi>=4)

compiled_clust_df_v14 = left_join(compiled_clust_df_v14, filt_df_v3[,c('Clone_IDs', 'Clone_Index')], multiple="first")
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$tRG>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$oRG>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$RG>=1 & compiled_clust_df_v14$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$RG>=1 & compiled_clust_df_v14$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)

upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$OPCs>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$OPCs>=1 & compiled_clust_df_v14$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v14$upset[compiled_clust_df_v14$OPCs>=1 & compiled_clust_df_v14$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)

colnames(compiled_clust_df_v14)
table(compiled_clust_df_v14$dorsal_in_shared)
#dorsal_in_shared 
#880
dim(compiled_clust_df_v14)
#[1] 6888
sum(compiled_clust_df_v14$IN_local + compiled_clust_df_v14$IN_IPCs >= 1)
#[1] 2224 
sum(compiled_clust_df_v14$IN_local + compiled_clust_df_v14$IN_IPCs + compiled_clust_df_v14$IN_CGE + compiled_clust_df_v14$IN_MGE + compiled_clust_df_v14$IN_OB>= 1)
#[1] 2256

#out of 6,888 clones with at least 2 cells (2UMI filt), there are 2,224 that have an IN_local or IN_IPC and 880 of those have a dorsal cell type (RG or EX)



## somehow still managed to mess up the 3umi filter...... actually fixing it changes total 2cell clones from 6800 to 6400. just going to look
length(unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2]))
unique_mc_clones_3umi_2cell = unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2])
## v15 uses subclust_idents_definitive and has 2cell clones (3UMI cutoff now!)
compiled_clust_df_v15 = data.frame(Clone_IDs=unique_mc_clones_3umi_2cell)
compiled_clust_df_v15$ENs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$EX_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_local = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_CGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_MGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_OB = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$Astrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$OPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$Oligodendrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$RG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$tRG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$oRG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$ENs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$EX_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_local_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_CGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_MGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_OB_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$IN_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$Astrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$OPCS_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$RG_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$tRG_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v15$oRG_pct = rep(0,length(unique_mc_clones_3umi_2cell))


colnames(filt_df_v3)
table(filt_df_v3$broad_trajectories_v5)
table(filt_df_v3$subclust_idents_definitive)

for(clone in unique_mc_clones_3umi_2cell) {
  clone_size = filt_df_v3$Clone_size_corr_3umi[filt_df_v2$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v3$broad_trajectories_v5[filt_df_v3$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v15$Clone_IDs %in% clone)
  compiled_clust_df_v15[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v15[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v15[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v15[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v15[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v15[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v15[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v15[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v15[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v15[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v15[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v15[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v15[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v15[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v15[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v15[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v15[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v15[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v15[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v15[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v15[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v15[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v15[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v15[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v15[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v15[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v15
head(compiled_clust_df_v15)
tail(compiled_clust_df_v15)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v15)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v15[r,2:14] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v15$upset = upset_notations
table(compiled_clust_df_v15$upset)

upset(fromExpression(table(compiled_clust_df_v15$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v15)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v15, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first"))
compiled_clust_df_v15 = left_join(compiled_clust_df_v15, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)


pdf('upset_all_samples_2cell_clones_3umi_fixed_real.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v15$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_<=GW20_samples_2cell_clones_3umi_fixed_real.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$sample_age<=20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()
pdf('upset_>GW20_samples_2cell_clones_3umi_fixed_real.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$sample_age>20])),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)
dev.off()

compiled_clust_df_v15 = left_join(compiled_clust_df_v15, filt_df_v3[,c('Clone_IDs', 'age_pseudo')], multiple="first")

table(compiled_clust_df_v15$Clone_Index)
#Index3 IndexE 
#2477   3939 
table(compiled_clust_df_v15$age_pseudo)
#<=GW20  >GW20 
#4238   2178

upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$Clone_size_corr_3umi>=5])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$Clone_size_corr_3umi>=4])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$Clone_size_corr_3umi>=3])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$Clone_size_corr_3umi>=2])),
      nsets=13,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
hist(compiled_clust_df_v15$Clone_size_corr_3umi[compiled_clust_df_v15$upset %in% "EX_IPCs"])

upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$tRG>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$oRG>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE" & compiled_clust_df_v15$sample_age <= 20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "Index3" & compiled_clust_df_v15$sample_age <= 20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE" & compiled_clust_df_v15$sample_age > 20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$RG>=1 & compiled_clust_df_v15$Clone_Index %in% "Index3" & compiled_clust_df_v15$sample_age > 20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)


upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$OPCs>=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$OPCs>=1 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$OPCs>=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)

table(compiled_clust_df_v15$Clone_Index[compiled_clust_df_v15$sample_age <= 20])
#Index3 IndexE 
#943   3295
sum(compiled_clust_df_v15$OPCs[compiled_clust_df_v15$Clone_Index %in% "Index3"] >=1)
#[1] 633
633/943
#[1] 0.6712619
sum(compiled_clust_df_v15$OPCs[compiled_clust_df_v15$Clone_Index %in% "IndexE"] >=1)
#[1] 384
384/3295
#[1] 0.1165402

sum(compiled_clust_df_v15$RG + compiled_clust_df_v15$tRG + compiled_clust_df_v15$oRG + compiled_clust_df_v15$EX_IPCs + compiled_clust_df_v15$ENs >= 1)
#[1] 4923
sum(compiled_clust_df_v15$IN_local + compiled_clust_df_v15$IN_CGE + compiled_clust_df_v15$IN_MGE + compiled_clust_df_v15$IN_OB + compiled_clust_df_v15$IN_IPCs >= 1)
#[1] 2079
sum(compiled_clust_df_v15$RG + compiled_clust_df_v15$tRG + compiled_clust_df_v15$oRG + compiled_clust_df_v15$EX_IPCs + compiled_clust_df_v15$ENs >= 1 &
      compiled_clust_df_v15$IN_local + compiled_clust_df_v15$IN_CGE + compiled_clust_df_v15$IN_MGE + compiled_clust_df_v15$IN_OB + compiled_clust_df_v15$IN_IPCs >= 1)
#[1] 820
compiled_clust_df_v15$dorsal_in_shared_clone = NA
compiled_clust_df_v15$dorsal_in_shared_clone[compiled_clust_df_v15$RG + compiled_clust_df_v15$tRG + compiled_clust_df_v15$oRG + compiled_clust_df_v15$EX_IPCs + compiled_clust_df_v15$ENs >= 1 &
                                     compiled_clust_df_v15$IN_local + compiled_clust_df_v15$IN_CGE + compiled_clust_df_v15$IN_MGE + compiled_clust_df_v15$IN_OB + compiled_clust_df_v15$IN_IPCs >= 1] = "dorsal"
table(compiled_clust_df_v15$dorsal_in_shared_clone)
compiled_clust_df_v15$dorsal_clone = NA
#dorsal 
#820
compiled_clust_df_v15$dorsal_clone[compiled_clust_df_v15$RG + compiled_clust_df_v15$tRG + compiled_clust_df_v15$oRG + compiled_clust_df_v15$EX_IPCs + compiled_clust_df_v15$ENs >= 1] = "dorsal"
table(compiled_clust_df_v15$dorsal_clone)
#dorsal 
#4923

upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$IN_CGE + compiled_clust_df_v15$IN_MGE + compiled_clust_df_v15$IN_OB >=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)

DimPlot(ls_prog_olig, cells.highlight = rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$Clone_IDs_filt %in% compiled_clust_df_v15$Clone_IDs[compiled_clust_df_v15$dorsal_clone %in% "dorsal"]]) + ggtitle('OPCs and oligos -- Dorsal clones')
DimPlot(ls_prog_olig, cells.highlight = rownames(ls_prog_olig@meta.data)[ls_prog_olig@meta.data$Clone_IDs_filt %in% subset(compiled_clust_df_v15$Clone_IDs, !(compiled_clust_df_v15$dorsal_clone %in% "dorsal"))]) + ggtitle('OPCs and oligos --Non-dorsal clones')


upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$sample_age <=20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$sample_age >20])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$sample_age <=20 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$sample_age <=20 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)

table(compiled_clust_df_v15$Clone_Index[compiled_clust_df_v15$sample_age <= 20])
#Index3 IndexE 
#943   3295
sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age <= 20] >= 1)
#[1] 2445
sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age <= 20 & compiled_clust_df_v15$Clone_Index %in% "IndexE"] >= 1)
#[1] 2084
2084/3295
#[1] 0.6324734
sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age <= 20 & compiled_clust_df_v15$Clone_Index %in% "Index3"] >= 1)
#[1] 361
361/943
#[1] 0.3828208

## can I make a dataframe that has the count of clones that contain a given cell type, including age and index metadata
dim(filt_df_v3)
table(filt_df_v3$Barcode_UMI_Count) #confirm this is filtered to 3UMIs
table(filt_df_v3$Clone_size_corr_3umi) #does include clone size == 1

filt_df_v3_multicell = subset(filt_df_v3, Clone_size_corr_3umi>=2)
dim(filt_df_v3_multicell)
table(filt_df_v3_multicell$Clone_size_corr_3umi)
#   2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   22   25 
#7388 4110 2416 1560 1062  588  464  261  320  187  132  130   42   45   80   34   18   19   44   25 
length(unique(filt_df_v3_multicell$Clone_IDs))
#[1] 6416
dim(compiled_clust_df_v15)
#[1] 6416   38
table(filt_df_v3_multicell$Clone_Index)
#Index3 IndexE 
#7185  11740 
table(filt_df_v3_multicell$age_pseudo)
#<=GW20  >GW20 
#12691   6234
ggplot(filt_df_v3_multicell, aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, all samples") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, sample_age <=20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, <=GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
image = ggplot(subset(filt_df_v3_multicell, sample_age <=20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, <=GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggsave(file="cell_types_by_gz_barplot_young_samples.svg", plot=image, width=10, height=5)
ggplot(subset(filt_df_v3_multicell, sample_age >20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, >GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
image = ggplot(subset(filt_df_v3_multicell, sample_age >20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, >GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggsave(file="cell_types_by_gz_barplot_old_samples.svg", plot=image, width=10, height=5)


ggplot(subset(filt_df_v3_multicell, Clone_size_corr_3umi >=5), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in large clones, >=5") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")

ggplot(subset(filt_df_v3_multicell, Clone_size_corr_3umi >=5), aes(x=subclust_idents_definitive)) +
  geom_bar() +
  #geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in large clones, >=5") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")

ggplot(filt_df_v3_multicell, aes(x=subclust_idents_definitive)) +
  geom_bar() +
  #geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  #scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("All cell types in multicellular clones") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")


##chord diagram for IPC clones to show increased expansion in VZ?
library(circlize)

subclust_names = rownames(table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive))
length(subclust_names)
chord_matrix = matrix(rep(0, length(subclust_names)*length(subclust_names)), ncol=length(subclust_names))
dim(chord_matrix)
rownames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
colnames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")

dim(filt_df_v3_multicell)
for (clust_id in "EX_IPCs") {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)

for (clust_id in "EX_IPCs") {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)

for (clust_id in c("EX_IPCs", "OPCs")) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)

for (clust_id in c("EX_IPCs", "OPCs")) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)

for (clust_id in c("RG", "EX_IPCs", "OPCs")) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)

## chords might be doomed.... what about Sankey
library(ggsankey)

clones_with_ipcs = compiled_clust_df_v15$Clone_IDs[compiled_clust_df_v15$EX_IPCs>=1]
length(clones_with_ipcs)

sankey_vz_ipcs = filt_df_v3_multicell[filt_df_v3_multicell$Clone_IDs %in% clones_with_ipcs,] %>% make_long(Clone_Index, subclust_idents_definitive)
ggplot(sankey_vz_ipcs, aes(x = x, next_x = next_x, node = node, next_node = next_node,fill = factor(node), label=node)) +
  geom_sankey(flow.alpha = 0.5, node.color = "black", show.legend = TRUE) +
  geom_sankey_label(Size = 3, 
                    color = "black", 
                    fill = "white", ) +
  theme_bw() +
  labs(fill='Nodes')

## what about just zeroing out half of the matrix for the chord diagram to get unidirectional relationships?
subclust_names = rownames(table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive))
length(subclust_names)
chord_matrix = matrix(rep(0, length(subclust_names)*length(subclust_names)), ncol=length(subclust_names))
dim(chord_matrix)
rownames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
colnames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, transparency=0.5)

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  clust_interactions_temp = table(factor(filt_df_v3_multicell$subclust_idents_definitive[filt_df_v3_multicell$Clone_IDs %in% clone_list_temp], levels = c(
    "RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
  ))
  chord_matrix[rownames(chord_matrix) %in% clust_id,] = as.numeric(clust_interactions_temp)
}
chord_matrix
chordDiagram(chord_matrix, transparency=0.5)
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, transparency=0.5)

## would it help if this was instead more binarized like the upset plots?
subclust_names = rownames(table(ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive))
length(subclust_names)
chord_matrix = matrix(rep(0, length(subclust_names)*length(subclust_names)), ncol=length(subclust_names))
dim(chord_matrix)
rownames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")
colnames(chord_matrix) = c("RG","oRG","tRG","Astrocytes","OPCs","Oligodendrocytes","EX_IPCs","ENs","IN_IPCs","IN_local","IN_CGE","IN_OB","IN_MGE")

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, transparency=0.5)

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, transparency=0.5)


for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, transparency=0.5)

col_mat = rand_color(length(chord_matrix_lower), transparency = 0.5)
dim(col_mat) = dim(chord_matrix_lower)  # to make sure it is a matrix
chordDiagram(chord_matrix_lower, col = col_mat)
grid.col = col_mat[1,]
grid.col
chordDiagram(chord_matrix_lower, grid.col = grid.col, col = col_mat)
chordDiagram(chord_matrix_lower, grid.col = grid.col)

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3"]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)

for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age > 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age > 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)

sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age<=20 & compiled_clust_df_v15$Clone_Index %in% "IndexE"] >=1)

sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age<=20 & compiled_clust_df_v15$Clone_Index %in% "IndexE"] >1)
sum(compiled_clust_df_v15$EX_IPCs[compiled_clust_df_v15$sample_age<=20 & compiled_clust_df_v15$Clone_Index %in% "Index3"] >1)

207/361
1351/2084

sum(compiled_clust_df_v15$ENs[compiled_clust_df_v15$EX_IPCs>=1 & compiled_clust_df_v15$sample_age<=20 & compiled_clust_df_v15$Clone_Index %in% "IndexE"] >=1)

## altering the code here to not count every single self-interaction, only those where there are actually multiples of the celltype
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age > 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age > 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)

## one last try, altering the code so that self-interactions are only when the clone is only that specific cell type
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      col_pos = which(colnames(compiled_clust_df_v15) %in% clust_name)
      all_col_pos = seq(2,14)
      other_col_pos = all_col_pos[!all_col_pos %in% col_pos]
      unipotent_list_temp = apply(compiled_clust_df_v15[compiled_clust_df_v15$Clone_Index %in% "IndexE" & compiled_clust_df_v15$sample_age <= 20,other_col_pos], 1, sum)
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(unipotent_list_temp == 0) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      col_pos = which(colnames(compiled_clust_df_v15) %in% clust_name)
      all_col_pos = seq(2,14)
      other_col_pos = all_col_pos[!all_col_pos %in% col_pos]
      unipotent_list_temp = apply(compiled_clust_df_v15[compiled_clust_df_v15$Clone_Index %in% "Index3" & compiled_clust_df_v15$sample_age <= 20,other_col_pos], 1, sum)
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(unipotent_list_temp == 0) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_lower = lower.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_lower
chordDiagram(chord_matrix_lower, grid.col = grid.col)


## trying to look at clone size distribution
clone_size_dist_df = data.frame(matrix(0, ncol=length(subclust_names), nrow=length(unique(compiled_clust_df_v15$Clone_size_corr_3umi))))
colnames(clone_size_dist_df) = subclust_names
rownames(clone_size_dist_df) = unique(sort(compiled_clust_df_v15$Clone_size_corr_3umi))
clone_size_dist_df

for (clust_id in c(subclust_names)) {
  clone_size_dist_df[,clust_id] = table(factor(compiled_clust_df_v15$Clone_size_corr_3umi[compiled_clust_df_v15[,clust_id] >= 1], levels = unique(sort(compiled_clust_df_v15$Clone_size_corr_3umi))))
}
clone_size_dist_df

clone_size_dist_df_pct = clone_size_dist_df
for (clust_id in c(subclust_names)) {
  clone_size_dist_df_pct[,clust_id] = table(factor(compiled_clust_df_v15$Clone_size_corr_3umi[compiled_clust_df_v15[,clust_id] >= 1], levels = unique(sort(compiled_clust_df_v15$Clone_size_corr_3umi))))/sum(compiled_clust_df_v15[,clust_id] >= 1)
}
clone_size_dist_df_pct

library(tidyr)
clone_size_dist_df_pct_pivot = pivot_longer(clone_size_dist_df_pct, c(1:ncol(clone_size_dist_df_pct)))
tail(clone_size_dist_df_pct_pivot)
seq(1, length(unique(compiled_clust_df_v15$Clone_size_corr_3umi))*length(subclust_names), by=length(subclust_names))
clone_size_dist_df_pct_pivot$clone_size = 2
clone_size_dist_df_pct_pivot
c=1
for (nr in seq(1, length(unique(compiled_clust_df_v15$Clone_size_corr_3umi))*length(subclust_names), by=length(subclust_names))){
  clone_size_dist_df_pct_pivot$clone_size[nr:(nr+length(subclust_names)-1)] = unique(sort(compiled_clust_df_v15$Clone_size_corr_3umi))[c]
  c=c+1
}
library(ggridges)



clone_size_dist_df
clone_sizes_full_length = data.frame(nrow=clone_size_dist_df[1,1], ncol=2)
colnames(clone_sizes_full_length) = c("cell_type", "clone_size")
clone_sizes_full_length$cell_type="Astrocyte"
clone_sizes_full_length$clone_size=2
clone_sizes_full_length = data.frame(c("del"), c(1))
colnames(clone_sizes_full_length) = c("cell_type", "clone_size")
for (clust_id in subclust_names){
  for (nr in c(1:nrow(clone_size_dist_df))){
    clone_sizes_full_length = rbind(clone_sizes_full_length, data.frame("cell_type" = rep(clust_id, max(1,clone_size_dist_df[nr,clust_id])), "clone_size" = rep(rownames(clone_size_dist_df)[nr], max(1,clone_size_dist_df[nr,clust_id]))))
  }
}
clone_sizes_full_length = clone_sizes_full_length[2:nrow(clone_sizes_full_length),]
dim(clone_sizes_full_length)
ggplot(clone_sizes_full_length, aes(x=as.numeric(clone_size), y=cell_type)) +
  geom_density_ridges_gradient()
#this works, but ends up with weird long tails because everything has a value of 1 at the long tails. need to just not include those

clone_sizes_full_length = data.frame(c("del"), c(1))
colnames(clone_sizes_full_length) = c("cell_type", "clone_size")
for (clust_id in subclust_names){
  for (nr in c(1:nrow(clone_size_dist_df))){
    if(clone_size_dist_df[nr,clust_id] != 0){
      clone_sizes_full_length = rbind(clone_sizes_full_length, data.frame("cell_type" = rep(clust_id, max(1,clone_size_dist_df[nr,clust_id])), "clone_size" = rep(rownames(clone_size_dist_df)[nr], max(1,clone_size_dist_df[nr,clust_id]))))
    } else{
    }
  }
}
clone_sizes_full_length = clone_sizes_full_length[2:nrow(clone_sizes_full_length),]
dim(clone_sizes_full_length)
ggplot(clone_sizes_full_length, aes(x=as.numeric(clone_size), y=cell_type)) +
  geom_density_ridges_gradient() 
ggplot(clone_sizes_full_length, aes(x=as.numeric(clone_size), y=cell_type)) +
  geom_density_ridges_gradient() +
  xlim(0,10)

ggplot(subset(clone_sizes_full_length, cell_type %in% c('RG', 'oRG', 'tRG', 'EX_IPCs', 'IN_IPCs', 'OPCs')), aes(x=as.numeric(clone_size), y=cell_type)) +
  geom_density_ridges_gradient() +
  xlim(0,20)

ggplot(subset(clone_sizes_full_length, cell_type %in% c('RG', 'oRG', 'tRG', 'EX_IPCs', 'IN_IPCs', 'OPCs')), aes(x=as.numeric(clone_size), color=cell_type)) +
  geom_histogram(aes(y = stat(count / sum(count))), position="dodge") +
  xlim(0,10)

head(clone_size_dist_df_pct_pivot)
ggplot(subset(clone_size_dist_df_pct_pivot, name %in% c('RG', 'oRG', 'tRG', 'EX_IPCs', 'IN_IPCs', 'OPCs')), aes(x=clone_size, y=value, color=name)) +
  geom_line(position = "dodge") +
  xlim(0,20)




## working on an idea: want to see the relative percentage of renewal divisions vs. other
## for example, for EX_IPCs, curious how many clones are just EX_IPCs, how many are EX_IPC/EN, and how many are EX_IPC/other
## valid?
## can use the most recent iteration of the chord matrix, which has unipotent clones on the diagonal
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      col_pos = which(colnames(compiled_clust_df_v15) %in% clust_name)
      all_col_pos = seq(2,14)
      other_col_pos = all_col_pos[!all_col_pos %in% col_pos]
      unipotent_list_temp = apply(compiled_clust_df_v15[compiled_clust_df_v15$Clone_Index %in% "IndexE" & compiled_clust_df_v15$sample_age <= 20,other_col_pos], 1, sum)
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(unipotent_list_temp == 0) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix
#EX_IPC only: 926
#EX_IPC and EN: 630
#EX_IPC and other: 784

#IN_IPC only: 40
#IN_IPC and IN: 145
#IN_IPC and other: 529

#OPCs only: 16
#OPCs and Oligos: 4
#OPCs and other: 379
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive %in% clust_id & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v15[compiled_clust_df_v15$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v15) %in% clust_name] >=1)
    } else{
      col_pos = which(colnames(compiled_clust_df_v15) %in% clust_name)
      all_col_pos = seq(2,14)
      other_col_pos = all_col_pos[!all_col_pos %in% col_pos]
      unipotent_list_temp = apply(compiled_clust_df_v15[compiled_clust_df_v15$Clone_Index %in% "Index3" & compiled_clust_df_v15$sample_age <= 20,other_col_pos], 1, sum)
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(unipotent_list_temp == 0) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix
#EX_IPC only: 129
#EX_IPC and EN: 122
#EX_IPC and other: 159

#IN_IPC only: 19
#IN_IPC and IN: 52
#IN_IPC and other: 230

#OPCs only: 24
#OPCs and Oligos: 6
#OPCs and other: 214

prog_potency_simplified_df = data.frame("counts" = c(926,630,784,40,145,529,16,4,379,129,122,159,19,52,230,24,6,214), 
           "cell_type" = rep(c(rep("EX_IPC",3),rep("IN_IPC",3),rep("OPC",3)),2), 
           "div_type" = rep(c("renewal", "mature", "other"),6),
           "GZ" = c(rep("VZ",9),rep("OSVZ",9)),
           "age" = rep("<=GW20", 18))
prog_potency_simplified_df$cell_type_gz_comb = paste(prog_potency_simplified_df$cell_type, prog_potency_simplified_df$GZ)
prog_potency_simplified_df
ggplot(prog_potency_simplified_df, aes(x=div_type, fill=cell_type_gz_comb, y=counts))+
  geom_bar(stat="identity", position = "dodge")

prog_potency_simplified_df_pct = data.frame("counts" = c(926/2340,630/2340,784/2340,
                                                         40/714,145/714,529/714,
                                                         16/399,4/399,379/399,
                                                         129/410,122/410,159/410,
                                                         19/301,52/301,230/301,
                                                         24/244,6/244,214/244), 
                                            "cell_type" = rep(c(rep("EX_IPC",3),rep("IN_IPC",3),rep("OPC",3)),2), 
                                            "div_type" = rep(c("renewal", "mature", "other"),6),
                                            "GZ" = c(rep("VZ",9),rep("OSVZ",9)),
                                            "age" = rep("<=GW20", 18))
prog_potency_simplified_df_pct$cell_type_gz_comb = paste(prog_potency_simplified_df$cell_type, prog_potency_simplified_df$GZ)
ggplot(prog_potency_simplified_df_pct, aes(x=div_type, fill=cell_type_gz_comb, y=counts))+
  geom_bar(stat="identity", position = "dodge")

ggplot(subset(prog_potency_simplified_df_pct, cell_type %in% "EX_IPC"), aes(x=div_type, fill=GZ, y=counts))+
  geom_bar(stat="identity", position = "dodge")

ggplot(subset(prog_potency_simplified_df_pct, cell_type %in% "EX_IPC" & div_type %in% c('mature', 'renewal')), aes(x=GZ, fill=div_type, y=counts))+
  geom_bar(stat="identity", position = "dodge")

ggplot(subset(prog_potency_simplified_df_pct, cell_type %in% "IN_IPC" & div_type %in% c('mature', 'renewal')), aes(x=GZ, fill=div_type, y=counts))+
  geom_bar(stat="identity", position = "dodge")

ggplot(subset(prog_potency_simplified_df_pct, cell_type %in% "OPC" & div_type %in% c('mature', 'renewal')), aes(x=GZ, fill=div_type, y=counts))+
  geom_bar(stat="identity", position = "dodge")
(926/2340)/(630/2340)
(129/410)/(122/410)

## on second thought, the upset plots already do this, just not in a binned way...
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$sample_age <=20 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$EX_IPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$OPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$OPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$IN_IPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v15$upset[compiled_clust_df_v15$IN_IPCs >=1 & compiled_clust_df_v15$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)


in_ipc_en_clone_list = compiled_clust_df_v15$Clone_IDs[compiled_clust_df_v15$IN_IPCs>=1 & compiled_clust_df_v15$ENs>=1]
DimPlot(ls_synthesis_v3_harmony_v3, cells.highlight = rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% in_ipc_en_clone_list], raster=F)


## 1/10 figure generation...
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240110_figs')
#first: remake upset plots combining tRG and oRG into just RG
#yet another new metadata column........
ls_synthesis_v3_harmony_v4 = AddMetaData(ls_synthesis_v3_harmony_v4, ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive, col.name = "subclust_idents_upset_final")
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final == 'tRG'] = 'RG'
ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final[ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final == 'oRG'] = 'RG'
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_definitive)
table(ls_synthesis_v3_harmony_v4@meta.data$subclust_idents_upset_final)

filt_df_v3$subclust_idents_upset_final = filt_df_v3$subclust_idents_definitive
filt_df_v3$subclust_idents_upset_final[filt_df_v3$subclust_idents_upset_final == 'tRG'] = 'RG'
filt_df_v3$subclust_idents_upset_final[filt_df_v3$subclust_idents_upset_final == 'oRG'] = 'RG'

length(unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2]))
unique_mc_clones_3umi_2cell = unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2])
## v16 uses subclust_idents_upset_final and has 2cell clones (3UMI cutoff now!)
compiled_clust_df_v16 = data.frame(Clone_IDs=unique_mc_clones_3umi_2cell)
compiled_clust_df_v16$ENs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$EX_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_local = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_CGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_MGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_OB = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$Astrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$OPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$Oligodendrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$RG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$ENs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$EX_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_local_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_CGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_MGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_OB_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$IN_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$Astrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$OPCS_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v16$RG_pct = rep(0,length(unique_mc_clones_3umi_2cell))

for(clone in unique_mc_clones_3umi_2cell) {
  clone_size = filt_df_v3$Clone_size_corr_3umi[filt_df_v3$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v3$broad_trajectories_v5[filt_df_v3$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v16$Clone_IDs %in% clone)
  compiled_clust_df_v16[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v16[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v16[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v16[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v16[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v16[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v16[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v16[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v16[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v16[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v16[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v16[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v16[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v16[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v16[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v16[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v16[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v16[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v16[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v16[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v16[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v16[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v16[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v16[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v16[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v16[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v16
head(compiled_clust_df_v16)
tail(compiled_clust_df_v16)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v16)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v16[r,2:12] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v16$upset = upset_notations
table(compiled_clust_df_v16$upset)

upset(fromExpression(table(compiled_clust_df_v16$upset)),
      nsets=12,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v16)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v16, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first"))
compiled_clust_df_v16 = left_join(compiled_clust_df_v16, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first")

upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age>20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5)) #text scale order: c(intersection size title, intersection size tick labels, set size title, set size tick labels, set names, numbers above bars)

pdf('upset_all_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset)),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age>20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

in_multicell_clones = compiled_clust_df_v16[]

compiled_clust_df_v16$ex_count = compiled_clust_df_v16$ENs + compiled_clust_df_v16$EX_IPCs
compiled_clust_df_v16$in_count = compiled_clust_df_v16$IN_IPCs + compiled_clust_df_v16$IN_local + compiled_clust_df_v16$IN_CGE + compiled_clust_df_v16$IN_MGE + compiled_clust_df_v16$IN_OB
compiled_clust_df_v16$ex_in_shared = NA
compiled_clust_df_v16$ex_in_shared[compiled_clust_df_v16$ex_count > 0 & compiled_clust_df_v16$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v16$ex_in_shared))
#[1] 6082
compiled_clust_df_v16$rg_count = compiled_clust_df_v16$RG
compiled_clust_df_v16$dorsal_in_shared = NA
compiled_clust_df_v16$dorsal_in_shared[compiled_clust_df_v16$ex_count+compiled_clust_df_v16$rg_count > 0 & compiled_clust_df_v16$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$dorsal_in_shared %in% "dorsal_in_shared"]
sum(is.na(compiled_clust_df_v16$dorsal_in_shared))
#[1] 5631
in_multicell_clones = compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$in_count >=1]
DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], cols.highlight = "#f27527") +
  ggtitle('INs in EX/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones], cols.highlight = "#f27527") +
  ggtitle('INs in EX/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

cge_ex_clones = compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$IN_CGE >=1 & compiled_clust_df_v16$ENs >=1]
DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight=2, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% cge_ex_clones], cols.highlight = "#f27527", raster=F)
filt_df_v3[filt_df_v3$Clone_IDs %in% cge_ex_clones,]

dim(filt_df_v3_multicell)
table(filt_df_v3_multicell$subclust_idents_definitive)
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)], raster=F)
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3)], raster=F)

DimPlot(ls_synthesis_v3_harmony_v4, group.by="Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=0.5, na.value = "lightgray") +
  ggtitle("GZ origin") +
  labs(color = "GZ labeled") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
image = DimPlot(ls_synthesis_v3_harmony_v4, group.by="Clone_Index_filt", cols = c(osvz_col, vz_col), pt.size=0.5, na.value = "lightgray") +
  ggtitle("GZ origin") +
  labs(color = "GZ labeled") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=30))
ggsave(file="full_umap_index_labeled_all_cells.png", plot=image, width=12, height=8)
ggsave(file="full_umap_index_labeled_all_cells.svg", plot=image, width=12, height=8)


## going to try to re-do this idea of looking at likelihood of IPCs to self-renew vs. include an IN
## per Tom, want to also break this out by samples and see if it still holds true
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$sample_age <=20 & compiled_clust_df_v16$Clone_Index %in% "Index3"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
colnames(compiled_clust_df_v16)
table(compiled_clust_df_v16$orig.ident)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW16_1220_S1", "GW16_1220_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW16_1220_S1", "GW16_1220_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW19_0221_S1","GW19_0221_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW19_0221_S1","GW19_0221_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% "GW20_0308_PFC_2S"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% "GW20_0308_PFC_2S"])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "IndexE" & compiled_clust_df_v16$orig.ident %in% c("GW24_0221_1S")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$EX_IPCs >=1 & compiled_clust_df_v16$Clone_Index %in% "Index3" & compiled_clust_df_v16$orig.ident %in% c("GW24_0221_1S")])),
      nsets=10,
      nintersects=20,
      order.by = "freq",
      decreasing = T)
table(compiled_clust_df_v16$orig.ident)
compiled_clust_df_v16$orig.ident.merge = NA
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW24_0221_1S')] = 'GW24'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW21_0502_1S')] = 'GW21'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 'GW23'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 'GW22'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW16_1220_S1', 'GW16_1220_S2')] = 'GW16'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 'GW19'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2')] = 'GW20_V1'
compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$orig.ident %in% c('GW20_0308_PFC_2S')] = 'GW20_PFC'
table(compiled_clust_df_v16$orig.ident.merge)
#GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#954      252      665     2367       13      757      962      446
active_sample = "GW16"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW19"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW20_PFC"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW20_V1"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
#GW16     GW19 GW20_PFC  GW20_V1     GW21     GW22     GW23     GW24 
#954      252      665     2367       13      757      962      446
active_sample = "GW21"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW22"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW23"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
active_sample = "GW24"
pdf(paste('upset_',active_sample,'_2cell_clones.pdf',sep=''), width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$orig.ident.merge %in% c(active_sample)])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      mainbar.y.label = active_sample,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
#following through with Tom's idea of striking clone types that are only present in one sample
all_clone_types = unique(compiled_clust_df_v16$upset)
all_clone_types_sample_counts = data.frame("clone_types" = all_clone_types, "sample_counts" = 0)
all_clone_types_sample_counts
for (ct in all_clone_types){
  all_clone_types_sample_counts$sample_counts[all_clone_types_sample_counts$clone_types %in% ct] = length(table(compiled_clust_df_v16$orig.ident.merge[compiled_clust_df_v16$upset %in% ct]))
}
table(all_clone_types_sample_counts$sample_counts)
#1  2  3  4  5  6  7  8 
#29 15 14  7  6  7 12  6
all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts == 1]
single_sample_clone_types = all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts == 1]
multi_sample_clone_types = all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts > 1]

upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age <= 20 & compiled_clust_df_v16$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=40,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age > 20 & compiled_clust_df_v16$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=40,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
## going to use these, since these are even higher confidence
pdf('upset_all_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=40,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age <= 20 & compiled_clust_df_v16$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=40,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_samples_2cell_clones_3umi_fixed_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v16$upset[compiled_clust_df_v16$sample_age > 20 & compiled_clust_df_v16$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=40,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
## can also apply this logic to the UMAP
length(compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$upset %in% single_sample_clone_types])
#[1] 46
single_sample_clone_type_clone_ids = compiled_clust_df_v16$Clone_IDs[compiled_clust_df_v16$upset %in% single_sample_clone_types]
length(ex_in_shared_clones)
#[1] 334
sum(ex_in_shared_clones %in% single_sample_clone_type_clone_ids)
#[1] 19
DimPlot(ls_in_v2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% single_sample_clone_type_clone_ids])

ex_in_shared_clones_filt = ex_in_shared_clones[!ex_in_shared_clones %in% single_sample_clone_type_clone_ids]
length(ex_in_shared_clones_filt)
#[1] 315
length(in_multicell_clones)
#[1] 2079
in_multicell_clones_filt = in_multicell_clones[!in_multicell_clones %in% single_sample_clone_type_clone_ids]
length(in_multicell_clones_filt)
#[1] 2037
length(dorsal_in_shared_clones)
#[1] 785
dorsal_in_shared_clones_filt = dorsal_in_shared_clones[!dorsal_in_shared_clones %in% single_sample_clone_type_clone_ids]
length(dorsal_in_shared_clones_filt)
#[1] 757

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones_filt], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% in_multicell_clones_filt], cols.highlight = "#c44e33") +
  ggtitle('INs in multicellular clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_in_multicell_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in EX/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% ex_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in EX/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_ex_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)

DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
image = DimPlot(ls_in_v2, sizes.highlight=2, cells.highlight = rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$Clone_IDs_filt %in% dorsal_in_shared_clones_filt], cols.highlight = "#f27527") +
  ggtitle('INs in Dorsal/IN shared clones (Clone size >= 2)') +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        legend.position = "none", plot.title = element_text(size=25, hjust=0.5))
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.png", plot=image, width=12, height=8)
ggsave(file="in_umap_dorsal_in_shared_clones_2cells_3umis_fixed.svg", plot=image, width=12, height=8)



## 1/11
## have to redo some stuff after reassigning Astrocyte, RG, and OPC identities
filt_df_v3$subclust_idents_definitive_final = ls_synthesis_v3_harmony_v3@meta.data$subclust_idents_definitive_final[rownames(ls_synthesis_v3_harmony_v3@meta.data) %in% rownames(filt_df_v3)]
table(filt_df_v3$subclust_idents_definitive)
table(filt_df_v3$subclust_idents_definitive_final)
filt_df_v3$subclust_idents_upset_definitive_final = filt_df_v3$subclust_idents_definitive_final
filt_df_v3$subclust_idents_upset_definitive_final[filt_df_v3$subclust_idents_upset_definitive_final %in% c("aRG", "bRG")] = "RG"
filt_df_v3$subclust_idents_upset_definitive_final[filt_df_v3$subclust_idents_upset_definitive_final %in% c("EX IPCs")] = "EX_IPCs"
filt_df_v3$subclust_idents_upset_definitive_final[filt_df_v3$subclust_idents_upset_definitive_final %in% c("IN IPCs")] = "IN_IPCs"
table(filt_df_v3$subclust_idents_upset_definitive_final)
table(filt_df_v3$subclust_idents_upset_final)

##redoing upset plot generation
length(unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2]))
unique_mc_clones_3umi_2cell = unique(filt_df_v3$Clone_IDs[filt_df_v3$Clone_size_corr_3umi >= 2])
## v18 uses subclust_idents_upset_definitive_final and has 2cell clones (3UMI cutoff now!)
compiled_clust_df_v18 = data.frame(Clone_IDs=unique_mc_clones_3umi_2cell)
compiled_clust_df_v18$ENs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$EX_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_local = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_CGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_MGE = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_OB = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_IPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$Astrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$OPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$Oligodendrocytes = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$RG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$ENs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$EX_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_local_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_CGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_MGE_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_OB_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$IN_IPCs_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$Astrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$Oligodendrocytes_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$OPCS_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v18$RG_pct = rep(0,length(unique_mc_clones_3umi_2cell))

for(clone in unique_mc_clones_3umi_2cell) {
  clone_size = filt_df_v3$Clone_size_corr_3umi[filt_df_v3$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v3$subclust_idents_upset_definitive_final[filt_df_v3$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v18$Clone_IDs %in% clone)
  compiled_clust_df_v18[clone_row,"ENs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])
  compiled_clust_df_v18[clone_row,"EX_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])
  compiled_clust_df_v18[clone_row,"IN_local"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])
  compiled_clust_df_v18[clone_row,"IN_CGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])
  compiled_clust_df_v18[clone_row,"IN_MGE"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])
  compiled_clust_df_v18[clone_row,"IN_OB"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])
  compiled_clust_df_v18[clone_row,"IN_IPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])
  compiled_clust_df_v18[clone_row,"Astrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])
  compiled_clust_df_v18[clone_row,"Oligodendrocytes"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])
  compiled_clust_df_v18[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v18[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v18[clone_row,"oRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])
  compiled_clust_df_v18[clone_row,"tRG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])
  compiled_clust_df_v18[clone_row,"ENs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "ENs"])/clone_size
  compiled_clust_df_v18[clone_row,"EX_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX_IPCs"])/clone_size
  compiled_clust_df_v18[clone_row,"IN_local_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_local"])/clone_size
  compiled_clust_df_v18[clone_row,"IN_CGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_CGE"])/clone_size
  compiled_clust_df_v18[clone_row,"IN_MGE_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_MGE"])/clone_size
  compiled_clust_df_v18[clone_row,"IN_OB_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_OB"])/clone_size
  compiled_clust_df_v18[clone_row,"IN_IPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN_IPCs"])/clone_size
  compiled_clust_df_v18[clone_row,"Astrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Astrocytes"])/clone_size
  compiled_clust_df_v18[clone_row,"Oligodendrocytes_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Oligodendrocytes"])/clone_size
  compiled_clust_df_v18[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v18[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
  compiled_clust_df_v18[clone_row,"tRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "tRG"])/clone_size
  compiled_clust_df_v18[clone_row,"oRG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "oRG"])/clone_size
}
compiled_clust_df_v18
head(compiled_clust_df_v18)
tail(compiled_clust_df_v18)

upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v18)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v18[r,2:12] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v18$upset = upset_notations
table(compiled_clust_df_v18$upset)

upset(fromExpression(table(compiled_clust_df_v18$upset)),
      nsets=12,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

dim(compiled_clust_df_v18)
head(filt_df_v3)
dim(left_join(compiled_clust_df_v18, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first"))
compiled_clust_df_v18 = left_join(compiled_clust_df_v18, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], multiple="first")
dim(left_join(compiled_clust_df_v18, filt_df_v3[,c('Clone_IDs', 'age_pseudo')], multiple="first"))
compiled_clust_df_v18 = left_join(compiled_clust_df_v18, filt_df_v3[,c('Clone_IDs', 'age_pseudo')], multiple="first")


upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age>20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5)) #text scale order: c(intersection size title, intersection size tick labels, set size title, set size tick labels, set names, numbers above bars)

pdf('upset_all_samples_2cell_clones_3umi_fixed_definitive_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset)),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_samples_2cell_clones_3umi_fixed_definitive_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age<=20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_samples_2cell_clones_3umi_fixed_definitive_final.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age>20])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

## some numbers for the text:
length(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age<=20])
#[1] 4238
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 2445
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#[1] 0.5769231 
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 1288
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#[1] 0.3039169
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 1596
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#[1] 0.3765927
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 3917
3917/4238
#[1] 0.9242567
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 3289
3289/4238
#[1] 0.7760736
sum(compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 732
sum(compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#0.172723
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$Oligodendrocytes[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 471
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age<=20] + compiled_clust_df_v18$Oligodendrocytes[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#[1] 0.1111373
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 399
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1)/4238
#[1] 0.09414818
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 233
233/399
#[1] 0.5839599
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age<=20] +compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 218
218/399
#[1] 0.5463659
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$IN_CGE[compiled_clust_df_v18$sample_age<=20] +compiled_clust_df_v18$IN_MGE[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 1
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age<=20] +compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age<=20] >=1)
#124
124/399
#[1] 0.3107769
sum(compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 62
sum(compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age<=20] >=1)/6416
#[1] 0.009663342
sum(compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age<=20] >=1 & compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age<=20] >=1)
#[1] 11
11/62
#[1] 0.1774194


length(compiled_clust_df_v18$upset[compiled_clust_df_v18$sample_age>20])
#[1] 2178
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 137
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.06290174 
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 735
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.3374656
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 86
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.03948577
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 904
904/2178
#[1] 0.4150597
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 210
210/2178
#[1] 0.09641873
sum(compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 1332
sum(compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age>20] >=1)/2178
#0.6115702
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$Oligodendrocytes[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 858
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$Oligodendrocytes[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.3939394
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 757
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.3475666
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 342
342/757
#[1] 0.4517834
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$IN_CGE[compiled_clust_df_v18$sample_age>20] +compiled_clust_df_v18$IN_MGE[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 2
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age>20] +compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 318
318/757
#[1] 0.4200793
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$IN_IPCs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$IN_local[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$RG[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 93
93/757
#[1] 0.1228534
sum(compiled_clust_df_v18$OPCs[compiled_clust_df_v18$sample_age>20] >=1 & compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age>20]  >=1)
#[1] 26
26/757
#[1] 0.0343461
sum(compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 80
sum(compiled_clust_df_v18$Astrocytes[compiled_clust_df_v18$sample_age>20] >=1)/2178
#[1] 0.03673095
26/80
#[1] 0.325
(26+11)/(62+80)
#[1] 0.2605634

sum(compiled_clust_df_v18$IN_IPCs >=1)
#[1] 1535
sum(compiled_clust_df_v18$IN_IPCs >=1 & compiled_clust_df_v18$IN_local >=1)
#[1] 591
591/1535
#[1] 0.3850163
sum(compiled_clust_df_v18$IN_IPCs >=1 & compiled_clust_df_v18$RG >=1)
#[1] 495
495/1535
#[1] 0.3224756
sum(compiled_clust_df_v18$IN_IPCs >=1 & compiled_clust_df_v18$EX_IPCs + compiled_clust_df_v18$ENs >=1)
#[1] 271
271/1535
#[1] 0.1765472
sum(compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local >=1 & compiled_clust_df_v18$EX_IPCs + compiled_clust_df_v18$ENs >=1)
#[1] 331
sum(compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local >=1 & compiled_clust_df_v18$EX_IPCs + compiled_clust_df_v18$ENs >=1)/sum(compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local >=1)
#[1] 0.1603682

sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20] + compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20] >=1)
#[1] 210
210/2178
#[1] 0.09641873
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20 & compiled_clust_df_v18$Clone_Index %in% "IndexE"] + compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20 & compiled_clust_df_v18$Clone_Index %in% "IndexE"] >=1)
#[1] 109
109/210
#[1] 0.5190476
sum(compiled_clust_df_v18$ENs[compiled_clust_df_v18$sample_age>20 & compiled_clust_df_v18$Clone_Index %in% "Index3"] + compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$sample_age>20 & compiled_clust_df_v18$Clone_Index %in% "Index3"] >=1)
#[1] 101
101/210
#[1] 0.4809524
table(compiled_clust_df_v18$Clone_Index[compiled_clust_df_v18$sample_age>20])
#Index3 IndexE 
#1534    644
109/644
#[1] 0.1692547 ## pct of VZ clones with excitatory lineage cells (>GW20)
101/1534
#[1] 0.06584094 ## pct of OSVZ clones with excitatory lineage cells (>GW20)

sum(compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)])
#[1] 135
sum(compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)] & compiled_clust_df_v18$EX_IPCs + compiled_clust_df_v18$ENs >=1)
#[1] 38
38/135
#[1] 0.2814815

dim(ls_in_v2)
#[1] 27581 12118
sum(rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell))
#[1] 2802
sum(rownames(ls_in_v2@meta.data) %in% rownames(filt_df_v3_multicell))/length(rownames(ls_in_v2@meta.data))
#[1] 0.2312263
length(rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_MGE", "IN_CGE")])
#[1] 4990
sum(rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_MGE", "IN_CGE")] %in% rownames(filt_df_v3_multicell))
#34
sum(rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_MGE", "IN_CGE")] %in% rownames(filt_df_v3_multicell))/length(rownames(ls_in_v2@meta.data)[ls_in_v2@meta.data$subclust_idents_paper %in% c("IN_MGE", "IN_CGE")] %in% rownames(filt_df_v3_multicell))
#[1] 0.006813627

table(ls_div@meta.data$subclust_idents)/sum(table(ls_div@meta.data$subclust_idents))

## how many clones are multipotent
library(stringr)
table(str_count(compiled_clust_df_v18$upset, "&"))
#0    1    2    3    4    5 
#3169 2661  474   84   26    2
table(str_count(compiled_clust_df_v18$upset, "&"))/6416
#0            1            2            3            4            5 
#0.4939214464 0.4147443890 0.0738778055 0.0130922693 0.0040523691 0.0003117207


upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$IN_IPCs >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))


table(compiled_clust_df_v18$Clone_size_corr_3umi)/6416
#2            3            4            5            6            7            8            9           10           11 
#0.5757481297 0.2135286783 0.0941396509 0.0486284289 0.0275872818 0.0130922693 0.0090399002 0.0045199501 0.0049875312 0.0026496259 
#12           13           14           15           16           17           18           19           22           25 
#0.0017144638 0.0015586035 0.0004675810 0.0004675810 0.0007793017 0.0003117207 0.0001558603 0.0001558603 0.0003117207 0.0001558603

in_multicell_clones = compiled_clust_df_v18[]

compiled_clust_df_v18$ex_count = compiled_clust_df_v18$ENs + compiled_clust_df_v18$EX_IPCs
compiled_clust_df_v18$in_count = compiled_clust_df_v18$IN_IPCs + compiled_clust_df_v18$IN_local + compiled_clust_df_v18$IN_CGE + compiled_clust_df_v18$IN_MGE + compiled_clust_df_v18$IN_OB
compiled_clust_df_v18$ex_in_shared = NA
compiled_clust_df_v18$ex_in_shared[compiled_clust_df_v18$ex_count > 0 & compiled_clust_df_v18$in_count > 0] = "ex_in_shared"
ex_in_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$ex_in_shared %in% "ex_in_shared"]
sum(is.na(compiled_clust_df_v18$ex_in_shared))
#[1] 6082
compiled_clust_df_v18$rg_count = compiled_clust_df_v18$RG
compiled_clust_df_v18$astro_count = compiled_clust_df_v18$Astrocytes
compiled_clust_df_v18$dorsal_in_shared = NA
compiled_clust_df_v18$dorsal_in_shared[compiled_clust_df_v18$ex_count+compiled_clust_df_v18$rg_count+compiled_clust_df_v18$astro_count > 0 & compiled_clust_df_v18$in_count > 0] = "dorsal_in_shared"
dorsal_in_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$dorsal_in_shared %in% "dorsal_in_shared"]
sum(is.na(compiled_clust_df_v18$dorsal_in_shared))
#[1] 5631

colnames(filt_df_v3)
dim(filt_df_v3_multicell)
table(filt_df_v3$subclust_idents_definitive_final)
filt_df_v3_multicell$subclust_idents_definitive_final = filt_df_v3$subclust_idents_definitive_final[rownames(filt_df_v3) %in% rownames(filt_df_v3_multicell)]
arg_clones = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive_final %in% "aRG"])
length(arg_clones)
brg_clones = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_definitive_final %in% "bRG"])
length(brg_clones)
DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% arg_clones], raster=F) +
  ggtitle("Cells in aRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% arg_clones], raster=F) +
  ggtitle("Cells in aRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_aRG_clones.svg', image, width=12, height=8)
ggsave('ls_full_umap_aRG_clones.png', image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% brg_clones], raster=F) +
  ggtitle("Cells in bRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v3@meta.data)[ls_synthesis_v3_harmony_v3@meta.data$Clone_IDs_filt %in% brg_clones], raster=F) +
  ggtitle("Cells in bRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_bRG_clones.svg', image, width=12, height=8)
ggsave('ls_full_umap_bRG_clones.png', image, width=12, height=8)

## restricting to >=GW20 doesn't really help, and there are basically no cells anyway
DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% arg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20"], raster=F) +
  ggtitle("Cells in aRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% brg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20"], raster=F) +
  ggtitle("Cells in bRG-containing clones") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")

## looking at just VZ and OSVZ derived also doesn't look that different by whole cell numbers, but likely would be different by proportion of all cells generated
DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_size_corrected >=2], raster=F) +
  ggtitle("VZ-derived cells, >GW20") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#0786cf", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_size_corrected >=2], raster=F) +
  ggtitle("VZ-derived cells, >GW20") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_vz_derived_cells_>GW20.svg', image, width=12, height=8)
ggsave('ls_full_umap_vz_derived_cells_>GW20.png', image, width=12, height=8)

DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
          rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_size_corrected >=2], raster=F) +
  ggtitle("OSVZ-derived cells, >GW20") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
image = DimPlot(ls_synthesis_v3_harmony_v4, sizes.highlight = 3, cols.highlight = "#aa00e3", cells.highlight = 
                  rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index_filt %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20" & ls_synthesis_v3_harmony_v4@meta.data$Clone_size_corrected >=2], raster=F) +
  ggtitle("OSVZ-derived cells, >GW20") +
  theme(axis.line = element_blank(), axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(size=25, hjust=0.5), legend.position = "none")
ggsave('ls_full_umap_osvz_derived_cells_>GW20.svg', image, width=12, height=8)
ggsave('ls_full_umap_osvz_derived_cells_>GW20.png', image, width=12, height=8)




ggplot(filt_df_v3_multicell, aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, all samples") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, sample_age <=20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, <=GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, sample_age >20), aes(fill=Clone_Index, x=subclust_idents_definitive, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, >GW20") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")

colnames(filt_df_v3_multicell)

table(filt_df_v3_multicell$orig.ident)
colnames(filt_df_v3_multicell)
filt_df_v3_multicell$subclust_idents_upset_definitive_final = filt_df_v3_multicell$subclust_idents_definitive_final
filt_df_v3_multicell$subclust_idents_upset_definitive_final[filt_df_v3_multicell$subclust_idents_upset_definitive_final %in% 'aRG'] = "RG"
filt_df_v3_multicell$subclust_idents_upset_definitive_final[filt_df_v3_multicell$subclust_idents_upset_definitive_final %in% 'bRG'] = "RG"

ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW16_1220_S1", "GW16_1220_S2")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW16") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW19_0221_S1", "GW19_0221_S2")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW19") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW20_0308_PFC_2S")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW20_PFC") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW20_PFC") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW20_0308_V1_S1", "GW20_0308_V1_S2")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW20_V1") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW22_1118_S1", "GW22_1118_S2", "GW22_1118_S3")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW22") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW23_0910_S1", "GW23_0910_S2", "GW23_0910_S4")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW23") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(filt_df_v3_multicell, orig.ident %in% c("GW24_0221_1S")), aes(fill=Clone_Index, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types in multicell clones, GW24") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")

table(ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge)
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW16") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW16") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW19") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW19") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW20_PFC") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW20_PFC") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW20_V1") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW20_V1") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW22") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW22") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW23") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW23") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")
ggplot(subset(ls_synthesis_v3_harmony_v4@meta.data, orig.ident.merge %in% c("GW24") & Clone_Index_filt %in% c("IndexE", "Index3")), aes(fill=Clone_Index_filt, x=subclust_idents_upset_definitive_final, group=Clone_Index)) +
  geom_bar(position = "dodge", aes(y=after_stat(prop))) +
  scale_fill_manual(values=c(osvz_col,vz_col))+
  ggtitle("Percentage of cell types, GW24") +
  theme_bw() +
  xlab('Cell type') +
  ylab('Percentage of labeled cells') +
  theme(plot.title = element_text(hjust = 0.5, size=30), axis.text.x = element_text(angle = 45, hjust=1, size=20), axis.title.x = element_blank(),
        legend.title = element_text(size=25), legend.text = element_text(size=20), axis.text.y = element_text(size=20), axis.title.y = element_text(size=25))+
  labs(fill="GZ labeled", y="Proportion of cells")


## chord diagrams by age to replace Figure 1 Sankey(?)
## altering the code here to not count every single self-interaction, only those where there are actually multiples of the celltype
col_mat = rand_color(length(chord_matrix_lower), transparency = 0.5)
dim(col_mat) = dim(chord_matrix_lower)  # to make sure it is a matrix
grid.col = col_mat[1,]
chordDiagram(chord_matrix_lower, grid.col = grid.col)

pdf('chord_diagram_<=GW20.pdf', width=8, height=8)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_definitive_final %in% clust_id & filt_df_v3_multicell$sample_age <= 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v18[compiled_clust_df_v18$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v18) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v18[compiled_clust_df_v18$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v18) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_upper = upper.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_upper
chordDiagram(chord_matrix_upper, grid.col = grid.col)
dev.off()

pdf('chord_diagram_>GW20.pdf', width=8, height=8)
for (clust_id in c(subclust_names)) {
  clone_list_temp = filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_definitive_final %in% clust_id & filt_df_v3_multicell$sample_age > 20]
  clone_list_temp = clone_list_temp[!is.na(clone_list_temp)]
  for (clust_name in subclust_names){
    if (clust_id != clust_name){
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v18[compiled_clust_df_v18$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v18) %in% clust_name] >=1)
    } else{
      chord_matrix[rownames(chord_matrix) %in% clust_id, colnames(chord_matrix) %in% clust_name] = sum(compiled_clust_df_v18[compiled_clust_df_v18$Clone_IDs %in% clone_list_temp, colnames(compiled_clust_df_v18) %in% clust_name] >1) #for self-interactions, only count it if there are multiple of the given cell type
    }
  }
}
chord_matrix_upper = upper.tri(chord_matrix, diag=T) * chord_matrix
chord_matrix_upper
chordDiagram(chord_matrix_upper, grid.col = grid.col)
dev.off()



upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$RG >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5)
pdf('upset_vz_rg_all_ages_2cell_3umi.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$RG >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

vz_rg_exipc_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "EX_IPCs&RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]
vz_rg_only_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]
vz_rg_opc_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "OPCs&RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]
vz_rg_inipc_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "IN_IPCs&RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]
vz_rg_opc_inipc_shared_clones =  compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "IN_IPCs&OPCs&RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]
vz_rg_astro_shared_clones = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% "Astrocytes&RG" & compiled_clust_df_v18$Clone_Index %in% "IndexE"]


DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_only_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_exipc_shared_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_opc_shared_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_inipc_shared_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_opc_inipc_shared_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% vz_rg_astro_shared_clones])
DimPlot(ls_prog_glia, group.by="subclust_idents_definitive_final")

#subclust 0, VZ:
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5)
#subclust 1, VZ:
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(1)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5)
#subclust 4, VZ:
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5)
#subclust 7, VZ:
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5)

#subclust 4, VZ:
pdf('upset_vz_trg_subclust4.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

pdf('upset_all_trg_subclust4.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

pdf('upset_osvz_trg_subclust4.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "Index3" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

pdf('upset_osvz_brg_subclust0_1_7.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "Index3" & compiled_clust_df_v18$upset %in% multi_sample_clone_types & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1,7)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

pdf('upset_vz_brg_subclust0_1_7.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$upset %in% multi_sample_clone_types & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1,7)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

pdf('upset_all_brg_subclust0_1_7.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$upset %in% multi_sample_clone_types & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1,7)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()


DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_Index %in% "IndexE" & ls_prog_glia@meta.data$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]])
table(ls_prog_glia@meta.data$Barcode_UMI_Count_filt[ls_prog_glia@meta.data$Clone_Index %in% "IndexE" & ls_prog_glia@meta.data$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]])
table(ls_prog_glia@meta.data$Clone_size_corrected[ls_prog_glia@meta.data$Clone_Index %in% "IndexE" & ls_prog_glia@meta.data$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]])

#subclust 7, VZ
sc7_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc7_clones])
#subclust 4, VZ
sc4_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc4_clones])
#subclust 4, all
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc4_clones])
#subclust 1, VZ
sc1_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(1)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(1)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc1_clones])
#subclust 0, VZ
sc0_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc0_clones])


sc0_1_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0,1)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc0_1_clones])

sc4_7_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4,7)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4,7)]))]
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc4_7_clones])

sc0_1_7_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0, 1,7)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0, 1,7)]))]
sc0_1_7_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0, 1,7)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0, 1,7)]))]

length(compiled_clust_df_v18$ %in% rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$seurat_clusters %in% c(0)])
sum(rownames(compiled_clust_df_v18) %in% rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$seurat_clusters %in% c(0)])

length(rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$seurat_clusters %in% c(0)])

head(rownames(compiled_clust_df_v18))


rg_clones = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG"])

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc4_clones])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc4_clones])

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc0_1_7_clones])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sc0_1_7_clones])

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones])

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% "<=GW20"])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% "<=GW20"])

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "IndexE" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20"])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_Index %in% "Index3" & ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% rg_clones & ls_synthesis_v3_harmony_v4@meta.data$age_pseudo %in% ">GW20"])


rg_clones_vz = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "IndexE"])
rg_clones_osvz = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "Index3"])
rg_clones_vz_young = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$age_pseudo %in% "<=GW20"])
rg_clones_vz_old = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$age_pseudo %in% ">GW20"])
rg_clones_osvz_young = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$age_pseudo %in% "<=GW20"])
rg_clones_osvz_old = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$subclust_idents_upset_final %in% "RG" & filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$age_pseudo %in% ">GW20"])

length(rg_clones_vz)
#[1] 1163
length(rg_clones_vz_young)
#[1] 958
length(rg_clones_vz_old)
#[1] 205
length(rg_clones_osvz)
#[1] 974
length(rg_clones_osvz_young)
#[1] 364
length(rg_clones_osvz_old)
#[1] 610

DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sample(rg_clones_vz_young, length(rg_clones_osvz_young))])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sample(rg_clones_osvz_old, length(rg_clones_vz_old))])

clones_vz_old = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$Clone_Index %in% "IndexE" & filt_df_v3_multicell$age_pseudo %in% ">GW20"])
clones_osvz_old = unique(filt_df_v3_multicell$Clone_IDs[filt_df_v3_multicell$Clone_Index %in% "Index3" & filt_df_v3_multicell$age_pseudo %in% ">GW20"])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_vz_old])
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sample(clones_osvz_old, length(clones_vz_old))])


DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = list(rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_vz_old],
                                                           rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% sample(clones_osvz_old, length(clones_vz_old))]),
        cols.highlight = c(osvz_col, vz_col))
DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = list(rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_vz_old],
                                                           rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_osvz_old]),
        cols.highlight = c(osvz_col, "#00C3C6"), na.value = "lightgray", sizes.highlight = 3, ) +
  ggtitle("") +
  labs(color="GZ labeled")+
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20), legend.title = element_text(size=25))
image = DimPlot(ls_synthesis_v3_harmony_v4, cells.highlight = list(rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_vz_old],
                                                                   rownames(ls_synthesis_v3_harmony_v4@meta.data)[ls_synthesis_v3_harmony_v4@meta.data$Clone_IDs_filt %in% clones_osvz_old]),
                cols.highlight = c(osvz_col, "#00C3C6"), na.value = "lightgray", sizes.highlight = 3) +
  ggtitle("") +
  labs(color="GZ labeled")+
  theme(axis.line =element_blank(), legend.position="right", axis.ticks = element_blank(), axis.text = element_blank(), axis.title = element_blank(),
        plot.title = element_text(hjust=0.5, size=30), legend.text = element_text(size=20), legend.title = element_text(size=25))
ggsave("full_umap_>GW20_clones_by_gz.png", plot = image, width=10, height=8)
ggsave("full_umap_>GW20_clones_by_gz.svg", plot = image, width=10, height=8)




sample(rg_clones_vz_young, length(rg_clones_osvz_young))

FeaturePlot(ls_synthesis_v3_harmony_v4, features=c("NEUROD2"), order=T)

table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])
sum(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]]))
#135
table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])/135
##bkmrk
trg_clones = unique(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)])
trg_clones = trg_clones[!is.na(trg_clones)]

sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$Clone_IDs %in% trg_clones] >=1)
#[1] 135
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$Clone_IDs %in% trg_clones] >1)
#[1] 90
sum(compiled_clust_df_v18$RG[compiled_clust_df_v18$Clone_IDs %in% trg_clones] >1)/135
#[1] 0.6666667
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$Clone_IDs %in% trg_clones] >=1)
#[1] 33
sum(compiled_clust_df_v18$EX_IPCs[compiled_clust_df_v18$Clone_IDs %in% trg_clones] >=1)/135
#[1] 0.2444444
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_IDs %in% ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T)
48/135
#[1] 0.3555556

## 1/16 
sc0_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(0)]))]
sc1_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(1)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(1)]))]
sc4_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(4)]))]
sc7_clones = ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)][(!is.na(ls_prog_glia@meta.data$Clone_IDs_filt[ls_prog_glia@meta.data$seurat_clusters %in% c(7)]))]

sc4_clones
sc4_clones[sc4_clones %in% filt_df_v3_multicell$Clone_IDs]
DimPlot(ls_prog_glia)
FeaturePlot(ls_prog_glia, features=c("ANGPTL4", "TIMP3", "ITGB4", "S100A11"))
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc4_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_1_7_clones])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc4_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "IndexE"])
DimPlot(ls_prog_glia, cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_1_7_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "IndexE"])


DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc4_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "IndexE"], sizes.highlight = 3, cols.highlight = vz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc4_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "Index3"], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_1_7_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "IndexE"], sizes.highlight = 3, cols.highlight = vz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_1_7_clones & ls_prog_glia@meta.data$Clone_Index_filt %in% "Index3"], sizes.highlight = 3, cols.highlight = osvz_col)

DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc1_clones], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_clones], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc0_clones & ls_prog_glia@meta.data$age_pseudo %in% ">GW20"], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc1_clones & ls_prog_glia@meta.data$age_pseudo %in% ">GW20"], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc7_clones & ls_prog_glia@meta.data$age_pseudo %in% ">GW20"], sizes.highlight = 3, cols.highlight = osvz_col)
DimPlot(subset(ls_prog_glia, seurat_clusters %in% c(3,6)), cells.highlight = rownames(ls_prog_glia@meta.data)[ls_prog_glia@meta.data$Clone_IDs_filt %in% sc4_clones & ls_prog_glia@meta.data$age_pseudo %in% ">GW20"], sizes.highlight = 3, cols.highlight = osvz_col)


ggplot(subset(ls_prog_glia@meta.data, Clone_IDs_filt %in% sc4_clones & Clone_Index_filt %in% "IndexE"), aes(x=seurat_clusters)) +
  geom_bar()
ggplot(subset(ls_prog_glia@meta.data, Clone_IDs_filt %in% sc0_1_7_clones), aes(x=seurat_clusters)) +
  geom_bar()

table(ls_prog_glia@meta.data$Clone_Index_filt)
#Index3 IndexE 
#2364   2037
table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$age_pseudo %in% "<=GW20"])
#Index3 IndexE 
#911   1667
table(ls_prog_glia@meta.data$Clone_Index_filt[ls_prog_glia@meta.data$age_pseudo %in% ">GW20"])
#Index3 IndexE 
#1453    370
DimPlot(subset(ls_prog_glia, age_pseudo %in% "<=GW20"), group.by="Clone_Index_filt", pt.size=1)

ggplot(subset(ls_prog_glia@meta.data, age_pseudo %in% c("<=GW20") & Clone_Index_filt %in% c('IndexE', 'Index3')), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_fill_manual = c(vz_col, osvz_col)
ggplot(subset(ls_prog_glia@meta.data, age_pseudo %in% c("<=GW20") & Clone_Index_filt %in% c('IndexE', 'Index3')), aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_fill_manual = c(vz_col, osvz_col)

ggplot(tempdf, aes(x=seurat_clusters, fill=Clone_Index_filt)) +
  geom_bar(position="fill") +
  scale_fill_manual = c(vz_col, osvz_col)




dim(ls_prog_glia@meta.data[ls_prog_glia@meta.data$age_pseudo %in% c("<=GW20") & ls_prog_glia@meta.data$Clone_Index_filt %in% c('IndexE', 'Index3'),])
tempdf =ls_prog_glia@meta.data[ls_prog_glia@meta.data$age_pseudo %in% c("<=GW20") & ls_prog_glia@meta.data$Clone_Index_filt %in% c('IndexE', 'Index3'),]

colnames(ls_prog_glia@meta.data)

table(ls_synthesis_v3_harmony_v4@meta.data$orig.ident.merge[rownames(ls_synthesis_v3_harmony_v4@meta.data) %in% rownames(filt_df_v3_multicell)])
dim(filt_df_v3_multicell)

library(stringr)
table(str_count(compiled_clust_df_v18$upset, "&"))
sum(table(str_count(compiled_clust_df_v18$upset, "&")))
table(str_count(compiled_clust_df_v18$upset, "&"))/sum(table(str_count(compiled_clust_df_v18$upset, "&")))
#           0            1            2            3            4            5 
#0.4939214464 0.4147443890 0.0738778055 0.0130922693 0.0040523691 0.0003117207
table(str_count(compiled_clust_df_v18$upset[compiled_clust_df_v18$age_pseudo %in% "<=GW20"], "&"))/sum(table(str_count(compiled_clust_df_v18$upset[compiled_clust_df_v18$age_pseudo %in% "<=GW20"], "&")))
#           0            1            2            3            4            5 
#0.5243039169 0.3857951864 0.0667767815 0.0169891458 0.0056630486 0.0004719207
table(str_count(compiled_clust_df_v18$upset[compiled_clust_df_v18$age_pseudo %in% ">GW20"], "&"))/sum(table(str_count(compiled_clust_df_v18$upset[compiled_clust_df_v18$age_pseudo %in% ">GW20"], "&")))
#           0            1            2            3            4 
#0.4348025712 0.4710743802 0.0876951331 0.0055096419 0.0009182736 

0.0738778055+ 0.0130922693+ 0.0040523691 +0.0003117207
0.0667767815+ 0.0169891458 +0.0056630486+ 0.0004719207
0.0876951331+ 0.0055096419+ 0.0009182736 



## 1/23/24
## making a bunch of extra upset plots to be used for whatever
setwd('/media/chang/HDD-11/mgkeefe/R_work_in_progress_cajal/local_STICR_synthesis_figs/240123_figs')
pdf('upset_all_IndexE.pdf', width=14, height=8)

# striking clone types that are only present in one sample
compiled_clust_df_v18$orig.ident.merge = NA
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW24_0221_1S')] = 'GW24'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW21_0502_1S')] = 'GW21'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW23_0910_S1', 'GW23_0910_S2', 'GW23_0910_S4')] = 'GW23'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW22_1118_S1', 'GW22_1118_S2', 'GW22_1118_S3')] = 'GW22'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW16_1220_S1', 'GW16_1220_S2')] = 'GW16'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW19_0221_S1', 'GW19_0221_S2')] = 'GW19'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW20_0308_V1_S1', 'GW20_0308_V1_S2')] = 'GW20_V1'
compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$orig.ident %in% c('GW20_0308_PFC_2S')] = 'GW20_PFC'
table(compiled_clust_df_v18$orig.ident.merge)

all_clone_types = unique(compiled_clust_df_v18$upset)
all_clone_types_sample_counts = data.frame("clone_types" = all_clone_types, "sample_counts" = 0)
all_clone_types_sample_counts
for (ct in all_clone_types){
  all_clone_types_sample_counts$sample_counts[all_clone_types_sample_counts$clone_types %in% ct] = length(table(compiled_clust_df_v18$orig.ident.merge[compiled_clust_df_v18$upset %in% ct]))
}
table(all_clone_types_sample_counts$sample_counts)
#1  2  3  4  5  6  7  8 
#30 18 10 10  8  7 11  6
all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts == 1]
single_sample_clone_types = all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts == 1]
multi_sample_clone_types = all_clone_types_sample_counts$clone_types[all_clone_types_sample_counts$sample_counts > 1]
single_sample_clone_type_clone_ids = compiled_clust_df_v18$Clone_IDs[compiled_clust_df_v18$upset %in% single_sample_clone_types]


upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages Index E",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_Index3.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "Index3"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages Index 3",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<GW20_IndexE.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$age_pseudo %in% "<=GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, Index E",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_IndexE.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "IndexE" & compiled_clust_df_v18$age_pseudo %in% ">GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, Index E",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<GW20_Index3.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "Index3" & compiled_clust_df_v18$age_pseudo %in% "<=GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, Index 3",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_IndexE.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Clone_Index %in% "Index3" & compiled_clust_df_v18$age_pseudo %in% ">GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, Index 3",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$OPCs >= 1 & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, OPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$OPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% "<=GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, OPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$OPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% ">GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, OPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_astros.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Astrocytes >= 1 & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, Astrocytes",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_astros.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Astrocytes >= 1 & compiled_clust_df_v18$age_pseudo %in% "<=GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, Astrocytes",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_astros.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$Astrocytes >= 1 & compiled_clust_df_v18$age_pseudo %in% ">GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, Astrocytes",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_ipcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$EX_IPCs >= 1 & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, EX_IPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_ipcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$EX_IPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% "<=GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, EX_IPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_ipcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$EX_IPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% ">GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, EX_IPCs",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$RG >= 1 & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$RG >= 1 & compiled_clust_df_v18$age_pseudo %in% "<=GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$RG >= 1 & compiled_clust_df_v18$age_pseudo %in% ">GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_all_in_ipc.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$IN_IPCs >= 1 & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_<=GW20_in_ipc.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$IN_IPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% "<=GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_>GW20_in_ipc.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$IN_IPCs >= 1 & compiled_clust_df_v18$age_pseudo %in% ">GW20" & compiled_clust_df_v18$upset %in% multi_sample_clone_types])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, RG",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()



## upset plots by sample
#pdf('', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$orig.ident.merge %in% c("GW24")])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
upset(fromExpression(table(compiled_clust_df_v18$upset[compiled_clust_df_v18$orig.ident.merge %in% c("GW22") & compiled_clust_df_v18$Clone_Index %in% c("Index3")])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5,
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
#dev.off()


compiled_clust_df_v18_sample = compiled_clust_df_v18[compiled_clust_df_v18$orig.ident.merge %in% c("GW21"),]
library(stringr)
#GW16
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3              
#0.56079665 0.39203354 0.04192872 0.00524109 
0.04192872+0.00524109
#[1] 0.04716981

#GW19
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3              
#0.599206349 0.373015873 0.023809524 0.003968254
0.023809524+0.003968254
#[1] 0.02777778

#GW20_PFC
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3          4    
#0.409022556 0.421052632 0.135338346 0.028571429 0.006015038
0.135338346 +0.028571429 +0.006015038
#[1] 0.1699248

#GW20_V1
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3          4                5
#0.5340092945 0.3747359527 0.0621039290 0.0202788340 0.0080270384 0.0008449514
0.0621039290+ 0.0202788340 +0.0080270384 +0.0008449514
#[1] 0.09125475

#GW22
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3            4            
#0.376486129 0.496697490 0.117569353 0.006605020 0.002642008 
0.117569353+0.006605020+0.002642008
#[1] 0.1268164

#GW23
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0            1            2            3                   
#0.468814969 0.449064449 0.079002079 0.003118503 
0.079002079+ 0.003118503 
#[1] 0.08212058

#GW24
table(str_count(compiled_clust_df_v18_sample$upset, "&"))/sum(table(str_count(compiled_clust_df_v18_sample$upset, "&")))
#0          1          2          3 
#0.45515695 0.47757848 0.05829596 0.00896861 
0.05829596+0.00896861 
#[1] 0.06726457




## going back one more time for another characterization of broad celltypes...
## paper frequently discusses combined relationships as the following: IN lineage, EX lineage, Glia, OPCs, Radial glia
filt_df_v3$broad_trajectories_v6 = NA
table(filt_df_v3$subclust_idents_upset_definitive_final)
filt_df_v3$broad_trajectories_v6[filt_df_v3$subclust_idents_upset_definitive_final %in% c("Astrocytes", "Oligodendrocytes")] = "Glia"
filt_df_v3$broad_trajectories_v6[filt_df_v3$subclust_idents_upset_definitive_final %in% c("ENs", "EX_IPCs")] = "EX"
filt_df_v3$broad_trajectories_v6[filt_df_v3$subclust_idents_upset_definitive_final %in% c("IN_CGE", "IN_IPCs", "IN_local", "IN_MGE", "IN_OB")] = "IN"
filt_df_v3$broad_trajectories_v6[filt_df_v3$subclust_idents_upset_definitive_final %in% c("OPCs")] = "OPCs"
filt_df_v3$broad_trajectories_v6[filt_df_v3$subclust_idents_upset_definitive_final %in% c("RG")] = "RG"
table(filt_df_v3$broad_trajectories_v6)

compiled_clust_df_v19 = data.frame(Clone_IDs=unique_mc_clones_3umi_2cell)
compiled_clust_df_v19$EX = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$IN = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$Glia = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$OPCs = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$RG = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$EX_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$IN_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$Glia_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$OPCS_pct = rep(0,length(unique_mc_clones_3umi_2cell))
compiled_clust_df_v19$RG_pct = rep(0,length(unique_mc_clones_3umi_2cell))

for(clone in unique_mc_clones_3umi_2cell) {
  clone_size = filt_df_v3$Clone_size_corr_3umi[filt_df_v3$Clone_IDs %in% clone][1]
  clone_clusts = table(filt_df_v3$broad_trajectories_v6[filt_df_v3$Clone_IDs %in% clone])
  clone_row = which(compiled_clust_df_v19$Clone_IDs %in% clone)
  compiled_clust_df_v19[clone_row,"EX"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])
  compiled_clust_df_v19[clone_row,"IN"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])
  compiled_clust_df_v19[clone_row,"Glia"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Glia"])
  compiled_clust_df_v19[clone_row,"OPCs"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])
  compiled_clust_df_v19[clone_row,"RG"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])
  compiled_clust_df_v19[clone_row,"EX_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "EX"])/clone_size
  compiled_clust_df_v19[clone_row,"IN_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "IN"])/clone_size
  compiled_clust_df_v19[clone_row,"Glia_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "Glia"])/clone_size
  compiled_clust_df_v19[clone_row,"OPCs_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "OPCs"])/clone_size
  compiled_clust_df_v19[clone_row,"RG_pct"] = sum(clone_clusts[dimnames(clone_clusts)[[1]] %in% "RG"])/clone_size
}


upset_notations = c()
for(r in 1:nrow(compiled_clust_df_v19)) {
  upset_notations=c(upset_notations, paste(colnames(compiled_clust_df_v19[r,2:6] %>% select(where(~ any(. != 0)))), collapse="&"))
}
#library(UpSetR)
compiled_clust_df_v19$upset = upset_notations
table(compiled_clust_df_v19$upset)

upset(fromExpression(table(compiled_clust_df_v19$upset)),
      nsets=13,
      nintersects=50,
      order.by = "freq",
      decreasing = T)

compiled_clust_df_v19 = left_join(compiled_clust_df_v19, filt_df_v3[,c('orig.ident' ,'Clone_IDs', 'sample_age', 'age_pseudo', 'Clone_Index', 'Clone_size_corr_3umi', 'trg_org_containing', 'vz_osvz_prog_containing')], 
                                  multiple="first")

pdf('upset_broad_trajectories_all_cells.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset)),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_<=GW20.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% "<=GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_>GW20.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% ">GW20"])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_all_ages_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$OPCs >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, OPCs, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_<=GW20_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% "<=GW20" & compiled_clust_df_v19$OPCs >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, OPCs, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_>GW20_opcs.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% ">GW20" & compiled_clust_df_v19$OPCs >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, OPCs, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_all_ages_glia.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$Glia >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, Glia, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_<=GW20_glia.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% "<=GW20" & compiled_clust_df_v19$Glia >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, Glia, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_>GW20_glia.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% ">GW20" & compiled_clust_df_v19$Glia >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, Glia, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_all_ages_ins.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$IN >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, IN, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_<=GW20_ins.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% "<=GW20" & compiled_clust_df_v19$IN >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, IN, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_>GW20_ins.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% ">GW20" & compiled_clust_df_v19$IN >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, IN, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_all_ages_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$RG >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "All ages, RG, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_<=GW20_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% "<=GW20" & compiled_clust_df_v19$RG >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = "<=GW20, RG, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()
pdf('upset_broad_trajectories_>GW20_rg.pdf', width=14, height=8)
upset(fromExpression(table(compiled_clust_df_v19$upset[compiled_clust_df_v19$age_pseudo %in% ">GW20" & compiled_clust_df_v19$RG >=1])),
      nsets=13,
      nintersects=30,
      order.by = "freq",
      decreasing = T,
      point.size=5, line.size=1.5, mainbar.y.label = ">GW20, RG, broad",
      text.scale = c(4,3,3,3,3.5,0), mb.ratio = c(0.5, 0.5))
dev.off()

## clone size histograms
table(filt_df_v3$Clone_size_corr_3umi)
table(filt_df_v3_multicell$Clone_size_corr_3umi)
table(compiled_clust_df_v18$Clone_size_corr_3umi)
ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_all_cells.svg", image, width=10, height=8)

ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,350) +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,350) +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_all_cells_large_clones_only.svg", image, width=10, height=8)

ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(compiled_clust_df_v18, aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- all cells") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_all_cells_log_scale.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_<=GW20_all_cells.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,200) +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,200) +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_<=GW20_large_clones_only.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% "<=GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- <=GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_<=GW20_log_scale.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_>GW20_all_cells.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,200) +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  ylab("Number of clones") +
  xlab("Clone size") +
  xlim(2,25) +
  ylim(0,200) +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_>GW20_large_clones_only.svg", image, width=10, height=8)

ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
image = ggplot(subset(compiled_clust_df_v18, age_pseudo %in% ">GW20"), aes(x=Clone_size_corr_3umi)) +
  geom_bar() +
  scale_y_log10() +
  ylab("Number of clones") +
  xlab("Clone size") +
  #xlim(2,25) +
  #ylim(0,1500) +
  ggtitle("Clone sizes -- >GW20") +
  theme_bw() +
  theme(axis.text.y = element_text(size=20), axis.text.x = element_text(size=20), axis.title = element_text(size=25), 
        plot.title = element_text(size=30, hjust=0.5),
        axis.line.x.bottom=element_line(size=1), axis.line.y.left=element_line(size=1), axis.ticks=element_line(size=1))
ggsave("histogram_clone_size_>GW20_log_scale.svg", image, width=10, height=8)

